---
title: "`EWCE`: `E`xpression `W`eighted `C`elltype `E`nrichment"
author: "<h4>Alan Murphy, Brian Schilder, and Nathan Skene</h4>"
date: "<h4>`r format( Sys.Date(), '%b-%d-%Y')`</h4>"
bibliography: inst/cit/EWCE.bib
csl: inst/cit/nature.csl
output: rmarkdown::github_document
vignette: >
  %\VignetteIndexEntry{README}
  %\VignetteEngine{knitr::rmarkdown}
  %\usepackage[utf8]{inputenc}    
---

```{r, echo=FALSE}
knitr::opts_chunk$set(tidy         = FALSE, 
                      message      = FALSE) 
```

<!-- badges: start -->
<!-- badger::badge_last_commit()  -->
<!-- badger::badge_license() -->
<!-- badger::badge_codecov() -->
[![R build status](https://github.com/bschilder/EWCE/workflows/R-CMD-check-bioc/badge.svg)](https://github.com/bschilder/EWCE/actions) 
[![BioC status](http://www.bioconductor.org/shields/build/release/bioc/EWCE.svg)](https://bioconductor.org/checkResults/release/bioc-LATEST/EWCE)
[![platforms](http://www.bioconductor.org/images/shields/availability/all.svg)](http://www.bioconductor.org/packages/devel/data/experiment/html/ewce.html#archives)
[![](https://img.shields.io/badge/doi-10.18129/B9.bioc.EWCE%20-green.svg)](https://doi.org/10.18129/B9.bioc.EWCE )
[![](https://img.shields.io/github/last-commit/bschilder/EWCE.svg)](https://github.com/bschilder/EWCE/commits/master)
[![](https://codecov.io/gh/bschilder/EWCE/branch/master/graph/badge.svg)](https://codecov.io/gh/bschilder/EWCE)
[![License: GPL-3](https://img.shields.io/badge/license-GPL--3-blue.svg)](https://cran.r-project.org/web/licenses/GPL-3)
[![](https://img.shields.io/badge/download-2145/total-blue.svg)](https://bioconductor.org/packages/stats/bioc/EWCE)
<!-- badges: end -->

<img height="200" src="https://github.com/bschilder/EWCE/raw/DelayedArray/inst/hex/EWCE.png">

# Introduction

The `EWCE` R package is designed to facilitate expression weighted cell type
enrichment analysis as described in our *Frontiers in Neuroscience* 
paper [@skene_2016]. `EWCE` can be applied to any gene list. 
 
Using `EWCE` essentially involves two steps:

1. Prepare a single-cell reference; i.e. CellTypeDataset (CTD). Alternatively, 
you can use one of the pre-generated CTDs we provide via the package `ewceData`
(which comes with `EWCE`).  
2. Run cell type enrichment on a user-provided gene list.

# Installation

`EWCE>=###` requires [`R>=4.1`](https://www.r-project.org/) 
and `Bioconductor>=1.4`. 
To install *EWCE* on Bioconductor run:

```{r, eval=FALSE}
if (!require("BiocManager")){install.packages("BiocManager")}

BiocManager::install(version = "devel")
BiocManager::install("EWCE") 
```

# [Documentation site](https://nathanskene.github.io/EWCE)

# Vignettes

## [EWCE: Getting started](https://nathanskene.github.io/EWCE/articles/EWCE.html)

A minimal example to get started with running *EWCE*.

## [EWCE: Extended examples](https://nathanskene.github.io/EWCE/articles/extended.html)

Example *EWCE* enrichment tests with in-depth explanations of each step. 

## [EWCE: Creating CellTypeDatasets](https://nathanskene.github.io/EWCE/articles/create_CTD.html)

Instructions on how to create new make new CellTypeDataset references
to use with *EWCE*.

## [EWCE: Conditional examples](https://nathanskene.github.io/EWCE/articles/conditional.html)

Examples and explanations of conditional cell type enrichment tests 
(e.g. controlling for a dominant cell type signal) .

## [EWCE: Transcriptome examples](https://nathanskene.github.io/EWCE/articles/transcriptomes.html)

Additional applications of *EWCE* to transcriptomic studies.


# Docker container

[Docker](https://www.docker.com/) containers with 
the latest version of `EWCE` are regularly pushed to [Dockerhub](https://hub.docker.com/).
If you already have Docker installed,
you can load up a working copy using the following commands.  

Note, that you will need to replace the initial directory path 
with a location on your computer that you wish to be able to
access from within the docker image.

```
docker pull nathanskene/ewce
docker run --name=ewce -e PASSWORD=ewcedocker -p 8790:8790 -d -v /User/$USER:/var/ewce nathanskene/ewce:latest
docker exec -ti ewce R
```

# Troubleshooting

If you have any problems, please do submit an [issue here on GitHub](https://github.com/NathanSkene/EWCE/issues) with a reproducible example.


# Citation

If you use EWCE, please cite:

> [Skene, et al. Identification of Vulnerable Cell 
Types in Major Brain Disorders Using Single Cell Transcriptomes 
and Expression Weighted Cell Type Enrichment. Front. Neurosci, 2016.](https://www.frontiersin.org/articles/10.3389/fnins.2016.00016/full)

If you use the cortex/hippocampus single-cell data associated 
`EWCE`/`ewceData` this package then please cite the following:

> [Zeisel, et al. Cell types in the mouse cortex and
hippocampus revealed by single-cell RNA-seq. Science, 2015.
](http://www.sciencemag.org/content/early/2015/02/18/science.aaa1934.abstract)


# Session Info

<details>

```{r Session Info}
utils::sessionInfo()
```

</details>


# References

