---
title: "`EWCE`: `E`xpression `W`eighted `C`elltype `E`nrichment"
author: "<h4>Authors: Alan Murphy, Brian Schilder, and Nathan Skene</h4>"
date: "<h4>README updated: `r format( Sys.Date(), '%b-%d-%Y')`</h4>"
bibliography: inst/cit/EWCE.bib
csl: inst/cit/nature.csl
output: 
  github_document
---

<!-- To modify Package/Title/Description/Authors fields, edit the DESCRIPTION file -->

```{r setup, echo=FALSE, include=FALSE}
pkg <- read.dcf("DESCRIPTION", fields = "Package")[1]
title <- read.dcf("DESCRIPTION", fields = "Title")[1]
description <- read.dcf("DESCRIPTION", fields = "Description")[1]
URL <- read.dcf('DESCRIPTION', fields = 'URL')[1]
owner <- tolower(strsplit(URL,"/")[[1]][4])
```

`r badger::badge_github_version(color = 'black')` 
`r badger::badge_bioc_release(color='green')`
`r badger::badge_github_actions(action = 'R-CMD-check-bioc')`
`r badger::badge_github_actions(action = 'DockerHub')`
`r badger::badge_last_commit()`
`r badger::badge_codecov()` 
`r badger::badge_bioc_download(by = 'total', color='green')`
`r badger::badge_license()` 

<img height="200" src="https://github.com/NathanSkene/EWCE/blob/bschilder_dev/inst/hex/hex.png?raw=true">

## Introduction

The *EWCE* R package is designed to facilitate expression weighted cell type
enrichment analysis as described in our *Frontiers in Neuroscience* 
paper [@skene_2016]. *EWCE* can be applied to any gene list. 
 
Using *EWCE* essentially involves two steps:

1. Prepare a single-cell reference; i.e. CellTypeDataset (CTD). Alternatively, 
you can use one of the pre-generated CTDs we provide via the package `ewceData`
(which comes with *EWCE*).  
2. Run cell type enrichment on a user-provided gene list.

## Installation

*EWCE* requires [`R>=4.1`](https://www.r-project.org/) 
and `Bioconductor>=3.14`. 
To install *EWCE* on Bioconductor run:

```{r, eval=FALSE}
if (!require("BiocManager")){install.packages("BiocManager")}

BiocManager::install("EWCE") 
```

## Documentation 

### [Website](https://`r owner`.github.io/`r pkg`) 
### [Getting started](https://`r owner`.github.io/`r pkg`/articles/`r pkg`)   

Includes:  

- A minimal example to get started with running *EWCE*. 
- How to install and use the dedicated *EWCE* Docker container usage. [Docker](https://www.docker.com/) containers with the latest version of `EWCE` 
are regularly pushed to 
[Dockerhub](https://hub.docker.com/repository/docker/neurogenomicslab/ewce).  

### [Extended examples](https://`r owner`.github.io/`r pkg`/articles/extended.html)

Additional tutorials of various *EWCE* features, including how to:  

- Run cell-type enrichment tests 
- Create a CellTypeDataset
- Merge two single-cell datasets 
- Run conditional cell-type enrichment tests
- Apply to transcriptomic data
 
## Updates

Major upgrades to *EWCE* were made in version 1.3.1.
Please see the [NEWS page](https://nathanskene.github.io/EWCE/articles/news/index.html) for more details. 

## Troubleshooting

If you have any problems, please do submit an [Issue here on GitHub](https://github.com/nathanskene/EWCE/issues) with a reproducible example.

## Citation

If you use ``r pkg``, please cite: 

<!-- Modify this my editing the file: inst/CITATION  -->
> `r citation(pkg)$textVersion`

If you use the cortex/hippocampus single-cell data associated 
*EWCE*/*ewceData* this package then please cite the following:

> [Zeisel, et al. Cell types in the mouse cortex and
hippocampus revealed by single-cell RNA-seq. Science, 2015.
](http://www.sciencemag.org/content/early/2015/02/18/science.aaa1934.abstract)

<hr> 

## Contact
 
### [Neurogenomics Lab](https://www.neurogenomics.co.uk/)

UK Dementia Research Institute  
Department of Brain Sciences  
Faculty of Medicine  
Imperial College London   
[GitHub](https://github.com/neurogenomics)  
[DockerHub](https://hub.docker.com/orgs/neurogenomicslab)  

## References

