<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • EWCE</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/NathanSkene/EWCE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/NathanSkene/EWCE/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="1.11.5" id="ewce-1115">EWCE 1.11.5<a class="anchor" aria-label="anchor" href="#ewce-1115"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-11-5">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-11-5"></a></h3>
<ul><li>
<code>check_bootstrap_args</code>:
<ul><li>annotation level for checking cell type names was previous hardcoded to level 1, now updated to match user input for annotation level.</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.11.4" id="ewce-1114">EWCE 1.11.4<a class="anchor" aria-label="anchor" href="#ewce-1114"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-11-4">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-11-4"></a></h3>
<ul><li>
<code>drop_uninformative_genes</code>:
<ul><li>Will now catch cases where expression matrix is a dataframe and convert to a matrix</li>
<li>This was causing weird errors, see issue 92: <a href="https://github.com/NathanSkene/EWCE/issues/92" class="external-link uri">https://github.com/NathanSkene/EWCE/issues/92</a>
</li>
<li>Fix made in <code><a href="../reference/check_sce.html">check_sce()</a></code> function.</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.11.2" id="ewce-1112">EWCE 1.11.2<a class="anchor" aria-label="anchor" href="#ewce-1112"></a></h2>
<div class="section level3">
<h3 id="new-features-1-11-2">New features<a class="anchor" aria-label="anchor" href="#new-features-1-11-2"></a></h3>
<ul><li>
<code>bootstrap_enrichment_test</code>:
<ul><li>New args: <code>standardise_sct_data=</code>, <code>standardise_hits=</code>: let users have more control over data standardisation steps.</li>
<li>
<code>check_ewce_genelist_inputs</code>: updated accordingly.</li>
<li>New arg: <code>store_gene_data</code> to avoid hitting memory limits.</li>
<li>Modify <em>test-bootstrap_enrichment_test_2.R</em> to use test new args.</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.11.1" id="ewce-1111">EWCE 1.11.1<a class="anchor" aria-label="anchor" href="#ewce-1111"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-11-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-11-1"></a></h3>
<ul><li>
<code><a href="../reference/ewce_plot.html">ewce_plot()</a></code> - Dendrogram not reordering cell types in plot
<ul><li>see <a href="https://github.com/NathanSkene/EWCE/issues/88" class="external-link">issue</a>
</li>
<li>Occurs when ctd does not contain the plotting info</li>
<li>Fixed now and unit test added.</li>
<li>Note that cell type order on the x-axis is based on hierarchical clustering for both plots if <code>make_dendro = TRUE</code> for <code><a href="../reference/ewce_plot.html">ewce_plot()</a></code>.</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.9.3" id="ewce-193">EWCE 1.9.3<a class="anchor" aria-label="anchor" href="#ewce-193"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-9-3">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-9-3"></a></h3>
<ul><li>
<code>generate_bootstrap_plots</code>
<ul><li>Use stored <code>gene_data</code> object whenever possible.</li>
<li>Only show filtered celltypes.</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.9.2" id="ewce-192">EWCE 1.9.2<a class="anchor" aria-label="anchor" href="#ewce-192"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-9-2">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-9-2"></a></h3>
<ul><li>
<code>generate_bootstrap_plots</code>
<ul><li>Missing available parameters for <code><a href="../reference/check_ewce_genelist_inputs.html">check_ewce_genelist_inputs()</a></code> call.</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.9.1" id="ewce-191">EWCE 1.9.1<a class="anchor" aria-label="anchor" href="#ewce-191"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-9-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-9-1"></a></h3>
<ul><li>
<code>drop_uninformative_genes</code>
<ul><li>Hash out DGE options. Somehow these got re-exposed to users in Bioc&gt;=3.16.</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.4" id="ewce-174">EWCE 1.7.4<a class="anchor" aria-label="anchor" href="#ewce-174"></a></h2>
<div class="section level3">
<h3 id="new-features-1-7-4">New features<a class="anchor" aria-label="anchor" href="#new-features-1-7-4"></a></h3>
<ul><li>
<code>check_species</code>:
<ul><li>Added new arg <code>sctSpecies_origin_default</code>
</li>
</ul></li>
<li>
<code>standardise_ctd</code>:
<ul><li>Can now specify <code>sctSpecies_origin</code>, which will be added to the metadata.</li>
</ul></li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-7-4">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-7-4"></a></h3>
<ul><li>
<code>standardise_ctd</code>:
<ul><li>When “annot” slot is not provided in the original CTD, a new one will be created from matrix columns names instead of assigning <code>NULL</code>.</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.3" id="ewce-173">EWCE 1.7.3<a class="anchor" aria-label="anchor" href="#ewce-173"></a></h2>
<div class="section level3">
<h3 id="new-features-1-7-3">New features<a class="anchor" aria-label="anchor" href="#new-features-1-7-3"></a></h3>
<ul><li>
<code>check_ewce_genelist_inputs</code>/<code>bootstrap_enrichment_test</code>
<ul><li>New arg: <code>sctSpecies_origin</code> lets users clarify that their data originally came from mouse even when it is currently formatted as human orthologs. This is necessary for creating the appropriate background gene lists.</li>
</ul></li>
<li>Remove <code>grDevices</code> as dep entirely.</li>
<li>
<code>fix_celltype_names</code>
<ul><li>Add new arg <code>make_unique</code> to make this function easily usable for vectors where the same celltype appears multiple times.</li>
</ul></li>
<li>
<code>bootstrap_enrichment_test</code>
<ul><li>Return gene-level scores based on adaptation of code from <code>generate_bootstrap_plots</code>. now stored as a list element named <code>gene_data</code> in <code>data.table</code> format.</li>
</ul></li>
<li>
<code>generate_bootstrap_plots</code>
<ul><li>Revamp wrap code into reusable subfunctions.</li>
<li>Avoid resampling random genes when the gene data is stored in the bootstrap results as <code>gene_data</code>. It will also tell you which of these options it’s using.</li>
<li>Save with <code>ggsave</code> instead if <code>grDevices</code>.</li>
<li>Facet by celltype instead of generating tons of separate plots.</li>
<li>Let users decide cutoff threshold with new arg <code>adj_pval_thresh</code>
</li>
<li>Now returns a named list with the plots themselves (“plot”) and the paths to where they’re saved (“paths”) rather than just a higher-level directory path in which users had to search for the right files (and didn’t ever have access to the ggplot2 objects).</li>
<li>Show significance with barplot fill/color instead of asterices. Much easier to see now.</li>
<li>Change <code>savePath</code> arg to the more accurate <code>save_dir</code>. Expose appending <em>BootstrapPlots</em> to the user within the argument.</li>
</ul></li>
<li>
<code>generate_bootstrap_plots_for_transcriptome</code>
<ul><li>Change <code>savePath</code> arg to the more accurate <code>save_dir</code>. Expose appending <em>BootstrapPlots</em> to the user within the argument.</li>
<li>Save with <code>ggsave</code> instead if <code>grDevices</code>.</li>
</ul></li>
<li>Standardise <code>hits</code> + <code>hitGenes</code> arg all to <code>hits</code>.</li>
<li>Update hex:
<ul><li>Off load large source image from DALLE to Releases instead of including it within the package.</li>
</ul></li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-7-3">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-7-3"></a></h3>
<ul><li>
<code>drop_uninformative_genes</code> / <code>generate_celltype_data</code>
<ul><li>Pass <code>verbose</code> arg to matrix formatting functions.</li>
</ul></li>
<li>
<code>generate_controlled_bootstrap_geneset</code>
<ul><li>Removed <code>combinedGenes</code> arg as it was not being used anywhere within.</li>
</ul></li>
<li>
<code>check_args_for_bootstrap_plot_generation</code>
<ul><li>Removed unused args: <code>ttSpecies</code>, <code>sctSpecies</code>
</li>
</ul></li>
<li>
<em>test-bootstrap_enrichment_test_2.R</em>
<ul><li>“monkey_ctd” tests seems to be running more smoothly than before (not just getting NAs). This might have to with <code>orthogene</code> databases improving.</li>
<li>Reassuringly, “godzilla” tests still fail as expected :)</li>
</ul></li>
<li>Add <em>tess/testthat/Rplots.pdf</em> to <em>.gitignore</em>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.1" id="ewce-171">EWCE 1.7.1<a class="anchor" aria-label="anchor" href="#ewce-171"></a></h2>
<div class="section level3">
<h3 id="new-features-1-7-1">New features<a class="anchor" aria-label="anchor" href="#new-features-1-7-1"></a></h3>
<ul><li>Use <code>rworkflows</code> GHA.
<ul><li>Add <code><a href="https://rdrr.io/pkg/rworkflows/man/use_badges.html" class="external-link">rworkflows::use_badges</a></code> to <em>README.Rmd</em>.</li>
<li>Remove <em>Dockerfile</em> (no longer necessary).</li>
<li>Make all 3 platforms (Linux, Mac, Windows) use Bioc dev, as <code>ewceData (&gt;=1.7.1)</code> is now required, due to a fix made only in the development version of <code>rtracklayer</code>.</li>
</ul></li>
<li>Remove <code>cowplot</code> dependency.</li>
<li>Replace all <code>%&gt;%</code> with <code>|&gt;</code>
</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-7-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-7-1"></a></h3>
<ul><li>
<code>calc_quantiles</code>:
<ul><li>This function was only used in <code>filter_variance_quantiles</code>
</li>
<li>Compare <code><a href="https://rdrr.io/r/stats/ecdf.html" class="external-link">stats::ecdf</a></code> vs. <code><a href="https://dplyr.tidyverse.org/reference/ntile.html" class="external-link">dplyr::ntile</a></code> methods.</li>
<li>Remove from <code>EWCE</code> as it’s not longer used anywhere.</li>
</ul></li>
<li>
<code>bin_columns_into_quantiles</code>:
<ul><li>Rename arg <code>matrixIn</code> –&gt; <code>vec</code> to reflect what the function actually does.</li>
</ul></li>
<li>
<code>filter_variance_quantiles</code>:
<ul><li>Change to use <code>bin_columns_into_quantiles</code> instead of <code>calc_quantiles</code> to be consistent with how quantiles are handled in the rest of <code>EWCE</code>.</li>
<li>Updated tests in <em>test-get_celltype_table.r</em> to reflect that the number of genes filtered is unaffected by the normalization procedure (when quantiles are computed with <code><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">stats::quantile</a></code>).</li>
</ul></li>
<li>
<code>ewce_plot</code>:
<ul><li>Celltypes were producing NAs because the names in the results were not always standardized in the same way as the CTD. Now this is done internally.</li>
<li>Celltypes were not ordered factors, meaning the dendrogram didn’t line up correctly.</li>
<li>Switched from <code>gridArrange</code>/<code>cowplot</code> to <code>patchwork</code>.</li>
<li>Added dedicated unit tests file: <em>test-ewce_plot.r</em>
</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.1" id="ewce-171-1">EWCE 1.7.1<a class="anchor" aria-label="anchor" href="#ewce-171-1"></a></h2>
<div class="section level3">
<h3 id="new-features-1-7-1-1">New features<a class="anchor" aria-label="anchor" href="#new-features-1-7-1-1"></a></h3>
<ul><li>Offline runs enabled with functions using reference datasets (from <code>ewceData</code>). These functions have the parameter <code>localhub</code> added to control this.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.8" id="ewce-158">EWCE 1.5.8<a class="anchor" aria-label="anchor" href="#ewce-158"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-5-8">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-5-8"></a></h3>
<ul><li>GHA fix.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.7" id="ewce-157">EWCE 1.5.7<a class="anchor" aria-label="anchor" href="#ewce-157"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-5-7">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-5-7"></a></h3>
<ul><li>orthogene dependency has been replacing user entered background gene list with one generated from all known genes when species across gene lists and reference dataset are the same. This has now been fixed.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.5" id="ewce-155">EWCE 1.5.5<a class="anchor" aria-label="anchor" href="#ewce-155"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-5-5">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-5-5"></a></h3>
<ul><li>
<code>standardise_ctd</code>:
<ul><li>Always force “specificity_quantiles” to be one of the matrices in each level.</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.4" id="ewce-154">EWCE 1.5.4<a class="anchor" aria-label="anchor" href="#ewce-154"></a></h2>
<div class="section level3">
<h3 id="new-features-1-5-4">New features<a class="anchor" aria-label="anchor" href="#new-features-1-5-4"></a></h3>
<ul><li>
<code>filter_ctd_genes</code>
<ul><li>Now exported.</li>
<li>Can handle standardized CTD format.</li>
</ul></li>
<li>
<code>get_ctd_matrix_names</code>: New function to get a list of all data matrices in CTD.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-5-4">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-5-4"></a></h3>
<ul><li>
<code>check_ewce_genelist_inputs</code>:
<ul><li>User reported potential bug in code:<a href="https://github.com/NathanSkene/EWCE/issues/71" class="external-link uri">https://github.com/NathanSkene/EWCE/issues/71</a>
</li>
<li>Fixed by removing conditional and instead always filtering out genes not present in CTD/SCT.</li>
</ul></li>
<li>
<code>standardise_ctd</code>:
<ul><li>Add <code><a href="../reference/check_species.html">check_species()</a></code>
</li>
<li>Ensure all matrices become sparse when <code>as_sparse=TRUE</code>.</li>
<li>Generalize to matrices of any name.</li>
</ul></li>
<li>
<code>fix_celltype_names</code>:
<ul><li>Ensure all celltype names are unique after standardization.</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.3" id="ewce-153">EWCE 1.5.3<a class="anchor" aria-label="anchor" href="#ewce-153"></a></h2>
<div class="section level3">
<h3 id="new-features-1-5-3">New features<a class="anchor" aria-label="anchor" href="#new-features-1-5-3"></a></h3>
<ul><li>
<code>genelistSpecies</code> now passed to <code>prepare_genesize_control_network</code> in <code>bootstrap_enrichment_test</code> meaning gene list species will be inferred from user input.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.2" id="ewce-152">EWCE 1.5.2<a class="anchor" aria-label="anchor" href="#ewce-152"></a></h2>
<div class="section level3">
<h3 id="new-features-1-5-2">New features<a class="anchor" aria-label="anchor" href="#new-features-1-5-2"></a></h3>
<ul><li>
<code>drop_uninformative_genes</code>:
<ul><li>Expose new args: <code>dge_method</code>, <code>dge_test</code>, <code>min_variance_decile</code>
</li>
</ul></li>
<li>
<code>merged_ctd</code>: Actually merge the CTDs into one when <code>as_SCE=FALSE</code>.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-5-2">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-5-2"></a></h3>
<ul><li>Remove hard-coded file path separators (e.g. <code>sprintf("%s/MRK_List2.rpt", tempdir())</code>) to be more compatible with Windows.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.1" id="ewce-151">EWCE 1.5.1<a class="anchor" aria-label="anchor" href="#ewce-151"></a></h2>
<div class="section level3">
<h3 id="new-features-1-5-1">New features<a class="anchor" aria-label="anchor" href="#new-features-1-5-1"></a></h3>
<ul><li>Made substantial updates to <code>orthogene</code>, so going through and making sure everything still works / is able to take advantage of new features (e.g. separation of <code>non121_strategy</code> and <code>agg_func</code> args, many:many mapping):
<ul><li>
<code>filter_nonorthologs</code>: Pass up args from <code><a href="https://rdrr.io/pkg/orthogene/man/convert_orthologs.html" class="external-link">orthogene::convert_orthologs</a></code>.</li>
<li>
<code>generate_celltype_data</code>: <a href="https://github.com/inheritDotParams" class="external-link">@inheritDotParams</a></li>
</ul></li>
<li>Update GHA.</li>
<li>Bump to R (&gt;= 4.2) now that we’re developing on Bioc 3.16.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-5-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-5-1"></a></h3>
<ul><li>Avoid downloading large “MRK_List2.rpt” file any more than is necessary for testing.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.3" id="ewce-133">EWCE 1.3.3<a class="anchor" aria-label="anchor" href="#ewce-133"></a></h2>
<div class="section level3">
<h3 id="new-features-1-3-3">New features<a class="anchor" aria-label="anchor" href="#new-features-1-3-3"></a></h3>
<ul><li>
<code>method</code> argument from <code><a href="https://rdrr.io/pkg/orthogene/man/create_background.html" class="external-link">orthogene::create_background</a></code> and <code><a href="https://rdrr.io/pkg/orthogene/man/convert_orthologs.html" class="external-link">orthogene::convert_orthologs</a></code> is now passed up as an argument to <code>EWCE</code> functions to give users more control. “homologene” chosen as default for all functions. “homologene” has fewer species than “orthogene” but doesnt need to import data from the web. It also has more 1:1 mouse:human orthologs.</li>
<li>Include notes on mismatches between GitHub documentation and current Bioc release version.</li>
<li>Allow <code>bin_specificity_into_quantiles</code> to set specificity matrix name produced.</li>
<li>Merge GHA workflow yamls into one.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-3-3">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-3-3"></a></h3>
<ul><li>Add <code>try({})</code> and <code>error=TRUE</code> to avoid <em>“polygon edge not found”</em> error in vignettes.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.1" id="ewce-131">EWCE 1.3.1<a class="anchor" aria-label="anchor" href="#ewce-131"></a></h2>
<div class="section level3">
<h3 id="new-features-1-3-1">New features<a class="anchor" aria-label="anchor" href="#new-features-1-3-1"></a></h3>
<ul><li>Major changes: Pull Request from <em>bschilder_dev</em> branch.</li>
<li>All functions can now use lists and CellTypeDatasets (CTD) from any species and convert them to a common species (human by default) via <code>orthogene</code>.<br></li>
<li>Automated CTD standardisation via <code>standardise_ctd</code>.<br></li>
<li>Can handle (sparse) matrices.<br></li>
<li>Can create CTD from very large datasets using <code>DelayedArray</code> object class.<br></li>
<li>All functions automatically create appropriate gene backgrounds given species.<br></li>
<li>More modular, simplified vignettes.<br></li>
<li>Additional gene pre-filtering options (DESeq2, MAST, variance quantiles).<br></li>
<li>New/improved plotting functions (e.g. <code>plot_ctd</code>).<br></li>
<li>Added example bootstrapping enrichment results as <em>extdata</em> to speed up examples (documented in <em>data.R</em>). Accessed via <code><a href="../reference/example_bootstrap_results.html">EWCE::example_bootstrap_results()</a></code>.</li>
<li>Replaced GHA workflow with check-bioc to automatically: run R-CMD checks, run <code>BiocCheck</code>, and rebuild/deploy pkgdown site.</li>
<li>Parallelised functions:
<ul><li>
<code>drop_uninformative_genes</code><br></li>
<li>
<code>generate_celltype-data</code><br></li>
<li>
<code>bootstrap_enrichment_test</code><br></li>
</ul></li>
<li>Added tests (multiple functions tests per file to reduce number of times <code>ewceData</code> files have to be downloaded):
<ul><li><code>test-DelayedArray</code></li>
<li><code>test-merge_sce</code></li>
<li><code>test-get_celltype_table</code></li>
<li><code>test-list_species</code></li>
<li><code>test-run_DGE</code></li>
<li><code>test-check_percent_hits</code></li>
</ul></li>
<li>Added function <code><a href="../reference/is_32bit.html">is_32bit()</a></code> to all tests to ensure they don’t get run twice on Windows OS.<br></li>
<li>Added GitHub Actions workflows:
<ul><li>
<code>check-bioc-docker.yml</code>: Runs CRAN/Bioc checks, rebuilds and pushes <code>pkgdown</code> website, runs and uploads test coverage report,</li>
<li>
<code>dockerhub.yml</code>: Builds Bioconductor Docker container with <code>EWCE</code> installed, runs CRAN checks and (if checks are successful) pushes container to <a href="https://hub.docker.com/repository/docker/neurogenomicslab/ewce" class="external-link"><em>neurogenomicslab</em> DockerHub</a>.<br></li>
</ul></li>
<li>Removed <code>docs</code> folder, as the documentation website comes from the <em>gh-pages</em> branch now, and is automatically built by GHA workflow after each push to <em>main</em> branch.</li>
<li>Added new exported function <code>fix_celltype_names</code> to help with standardising celltype names in alignment with <code>standardise_ctd</code>.</li>
<li>
<code>generate_bootstrap_plots_for_transcriptome</code>: Now supports any species (not just mouse or human).
<ul><li>Converts CTD and DGE table (<code>tt</code>) into <code>output_species</code> gene symbols.</li>
<li>Automatically generates appropriate gene background.<br></li>
<li>Faster due to now having the option to only generate certain plot types.<br></li>
</ul></li>
<li>Provide precomputed results from <code>ewce_expression_data</code> via new <code>example_transcriptome_results</code> function.<br></li>
<li>Reduced build runtime and oversized vignettes by not evaluating certain code chunks.
<ul><li>Prevent <em>extended</em> vignette from running entirely.</li>
</ul></li>
<li>
<code>@return</code> documentation for internal functions.<br></li>
<li>Added more installation checks to GHA.</li>
<li>Fixed inconsistent naming of unit test files: <code>test_</code> ==&gt; <code>test-</code>
</li>
<li>Removed DGE args in <code>drop_uninformative_genes</code> for now until we run benchmarking to see how each affects the <code>EWCE</code> results.<br></li>
<li>Make <code>bootstrap_plots</code> function internal.</li>
<li>Add report on how <code>orthogene</code> improve within- and across-species gene mappings in <em>extended</em> vignette.</li>
<li>Record extra info in <code>standardise_ctd</code> output:
<ul><li>“species”: both <code>input_species</code> and <code>output_species</code>
</li>
<li>“versions”: of <code>EWCE</code>, <code>orthogene</code>, and <code>homologene</code>
</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.0" id="ewce-100">EWCE 1.0.0<a class="anchor" aria-label="anchor" href="#ewce-100"></a></h2>
<div class="section level3">
<h3 id="new-features-1-0-0">New features<a class="anchor" aria-label="anchor" href="#new-features-1-0-0"></a></h3>
<ul><li>EWCE v1.0 on Bioconductor replaces the defunct <a href="https://bioconductor.riken.jp/packages/3.5/bioc/html/EWCE.html" class="external-link">EWCE v1.3.0</a> available on Bioconductor v3.5.</li>
<li>EWCE has been rendered scalable to the analysis of large datasets</li>
<li>
<code><a href="../reference/drop_uninformative_genes.html">drop_uninformative_genes()</a></code> has been expanded to allow the utilisation of differential expression approaches</li>
<li>EWCE can now handle SingleCellExperiment (SCE) objects or other Ranged SummarizedExperiment (SE) data types and as input as well as the original format, described as a single cell transcriptome (SCT) object.</li>
</ul><p>Deprecated &amp; Defunct</p>
<ul><li>The following functions have been renamed to use underscore in compliance with Bioconductor nomenclature:
<ul><li>
<code>check.ewce.genelist.inputs</code><br></li>
<li>
<code>cell.list.dist</code><br></li>
<li>
<code>bootstrap.enrichment.test</code><br></li>
<li>
<code>bin.specificity.into.quantiles</code><br></li>
<li>
<code>bin.columns.into.quantiles</code><br></li>
<li>
<code>add.res.to.merging.list</code><br></li>
<li>
<code>prepare.genesize.control.network</code><br></li>
<li>
<code>prep.dendro</code><br></li>
<li>
<code>get.celltype.table</code><br></li>
<li>
<code>calculate.specificity.for.level</code><br></li>
<li>
<code>calculate.meanexp.for.level</code><br></li>
<li>
<code>generate.celltype.data</code><br></li>
<li>
<code>generate.bootstrap.plots</code><br></li>
<li>
<code>generate.bootstrap.plots.for.transcriptome</code><br></li>
<li>
<code>fix.bad.mgi.symbols</code><br></li>
<li>
<code>fix.bad.hgnc.symbols</code><br></li>
<li>
<code>filter.genes.without.1to1.homolog</code><br></li>
<li>
<code>ewce.plot</code><br></li>
<li>
<code>cells.in.ctd</code><br></li>
<li><code>drop.uninformative.genes</code></li>
</ul></li>
</ul></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer></div>

  


  

  </body></html>

