---
title: "Extended examples"
author: "<h4>Alan Murphy, Brian Schilder, and Nathan Skene</h4>"
date: "<h4>`r format( Sys.Date(), '%b-%d-%Y')`</h4>"
output:
  BiocStyle::html_document:
vignette: >
  %\VignetteIndexEntry{Extended examples}
  %\VignetteEngine{knitr::rmarkdown}
  %\usepackage[utf8]{inputenc}     
---

# Setup  

```{r setup, include=TRUE}
library(EWCE)
set.seed(1234)
```

# Introduction

In the following vignette, we provide more a more in-depth version of the 
examples provided in the [Getting started vignette](https://nathanskene.github.io/EWCE/articles/EWCE.html).  

# Prepare input data

## CellTypeDataset

For this example we use a subset of the genes from the merged dataset 
generated in the [creating CTD vignette](https://nathanskene.github.io/EWCE/articles/creating_CTD.html),
which is accessed using `ewceData::ctd()`. 

### CTD levels 

Each level of a CTD corresponds to increasingly refined cell-type/-subtype annotations. For example, in the CTD  `ewceData::ctd()` level 1 includes the cell-type "interneurons", while level 2 breaks these this group into 16 different
interneuron subtypes ("Int...").

```{r, fig.width = 7, fig.height = 4} 
## Load merged cortex and hypothalamus dataset generated by Karolinska institute
ctd <- ewceData::ctd() # i.e. ctd_MergedKI

plt <- EWCE::plot_ctd(ctd = ctd,
                      level = 1,
                      genes = c("Apoe","Gfap","Gapdh"),
                      metric = "mean_exp")
```


## Gene lists

For the first demonstration of EWCE we will test for whether genes that are genetically associated with Alzheimer's disease are enriched in any particular cell type. 



This example gene list is stored within the `ewceData` package:

```{r}
hits <- ewceData::example_genelist()
print(hits)
```


### Gene formats and species 

All gene IDs are assumed by the package to be provided in gene symbol format (rather than Ensembl/Entrez). Symbols can be provided as any 
species-specific gene symbols supported by the package `orthogene`,
though the `genelistSpecies` argument will need to be set
appropriately. 

Likewise, the single-cell dataset can be from any species, 
but the `sctSpecies` argument must be set accordingly.  

The example gene list here stores the human genes associated with human disease, and hence are HGNC symbols.

The next step is to determine the most suitable background set. 
This can be user-supplied, but by default the background is all 1:1
ortholog genes shared by `genelistSpecies` and `sctSpecies` 
that are also present in `sct_data`.

## Setting analysis parameters

We now need to set the parameters for the analysis.
For a publishable analysis we would want to generate over 10,000 random 
lists and determine their expression levels, but for computational speed 
let us only use `reps=100`. We want to analyse level 1 
annotations so set level to 1.

```{r }
# Use 100 bootstrap lists for speed, for publishable analysis use >=10000
reps <- 100 
# Use level 1 annotations (i.e. Interneurons)
annotLevel <- 1 
```

# Run cell type enrichment tests

## Default tests

We have now loaded the SCT data, prepared the gene lists and set the parameters. We run the model as follows.

**Note**: We set the seed at the top of this vignette to ensure reproducibility in the bootstrap sampling function.

### Parallelisation

You can now speed up the bootstrapping process by parallelising across multiple cores with the parameter `no_cores` (`=1` by default).

```{r }
# Bootstrap significance test, no control for transcript length and GC content 
full_results <- EWCE::bootstrap_enrichment_test(sct_data = ctd,
                                                sctSpecies = "mouse",
                                                genelistSpecies = "human",
                                                hits = hits, 
                                                reps = reps,
                                                annotLevel = annotLevel)
```

A note on both the background and target gene lists, 
other common gene list objects can be used as inputs such as
`BiocSet::BiocSet` and `GSEABase::GeneSet`. 
Below is an example of how to format each for the target gene list (`hits`):

```
if(!"BiocSet" %in% rownames(installed.packages())) {
  BiocManager::install("BiocSet")
}
if(!"GSEABase" %in% rownames(installed.packages())) {
  BiocManager::install("GSEABase")
}

library(BiocSet)
library(GSEABase)

# Save both approaches as hits which will be passed to bootstrap_enrichment_test
genes <- c("Apoe","Inpp5d","Cd2ap","Nme8",
          "Cass4","Mef2c","Zcwpw1","Bin1",
          "Clu","Celf1","Abca7","Slc24a4",
          "Ptk2b","Picalm","Fermt2","Sorl1")

#BiocSet::BiocSet, BiocSet_target contains the gene list target
BiocSet_target <- BiocSet::BiocSet(set1 = genes) 
hits <- unlist(BiocSet::es_element(BiocSet_target))

#GSEABase::GeneSet, GeneSet_target contains the gene list target
GeneSet_target <- GSEABase::GeneSet(genes)
hits <- GSEABase::geneIds(GeneSet_target) 
```

The main table of results is stored in `full_results$results`. 
We can see the most significant results using:

```{r }
knitr::kable(full_results$results)
```


### Plot results 

The results can be visualised using another function, which shows for each cell type, the number of standard deviations from the mean the level of expression was found to be in the target gene list, relative to the bootstrapped mean:

```{r, fig.width = 7, fig.height = 4}
plot_list <- EWCE::ewce_plot(total_res = full_results$results,
                             mtc_method ="BH",
                             ctd = ctd) # optional (unless you want dendrogram)
print(plot_list$plain)
```

For publications it can be useful to plot a dendrogram alongside the plot. This can be done by including the cell type data as an additional argument. The dendrogram should automatically align with the graph ticks (thanks to Robert Gordon-Smith, Mres Molecular and Cellular Biosciences, Imperial College London for this solution):

```{r }
print(plot_list$withDendro)
```

If you want to view the characteristics of enrichment for each gene within the list then the `generate_bootstrap_plots` function should be used. This saves the plots into the BootstrapPlots folder. This takes the results of a bootstrapping analysis so as to only generate plots for significant enrichments. The `listFileName` argument is used to give the generated graphs a particular file name. The `savePath` argument is used here to save the files to a temporary directory, this can be updated to your preferred location. The file path where it was saved is returned so the temporary directory
can be located if used.

```
bt_plot_location <- 
  generate_bootstrap_plots(sct_data=ctd,hits=mouse.hits,
                          bg=mouse.bg,reps=reps,annotLevel=1,
                          full_results=full_results,
                          listFileName="VignetteGraphs",
                          savePath = tempdir())
```
    
## Control for transcript length and GC-content

When analysing genes found through genetic association studies it is important to consider biases which might be introduced as a result of transcript length and GC-content. The package can control for these by selecting the bootstrap lists such that the *i^th^* gene in the random list has properties similar to the*i^th^* gene in the target list. To enable the algorithm to do this it needs to be passed the gene lists as HGNC symbols rather than MGI.

The bootstrapping function then takes different arguments:

```{r, results="hide"} 
# Bootstrap significance test controlling for transcript length and GC content
cont_results <- EWCE::bootstrap_enrichment_test(sct_data = ctd,
                                                hits = hits,
                                                sctSpecies = "mouse",
                                                genelistSpecies = "human", 
                                                reps = reps,
                                                annotLevel = 1,
                                                geneSizeControl = TRUE)
```

### Plot results 

We plot these results using `ewce_plot`:

```{r, fig.width = 7, fig.height = 4}
plot_list <- EWCE::ewce_plot(total_res = cont_results$results,
                             mtc_method = "BH")
print(plot_list$plain)
```

This shows that the controlled method generates enrichments that are generally comparable to the standard method.


## Test different CTD levels

Both the analyses shown above were run on level 1 annotations. It is possible to test on the level 2 cell type level annotations by changing one of the arguments.

```R
# Bootstrap significance test controlling for transcript length and GC content
cont_results <- EWCE::bootstrap_enrichment_test(sct_data = ctd,
                                                hits = hits, 
                                                sctSpecies = "mouse",
                                                genelistSpecies = "human",
                                                reps = reps,
                                                annotLevel = 2,
                                                geneSizeControl = TRUE)
```
### Plot results

```R
plot_list <- EWCE::ewce_plot(total_res = cont_results$results,
                             mtc_method = "BH")
print(plot_list$plain)
```

With the subcell type analysis each microglial subtype was enriched and correspondingly we see here that the microglial cell type is enriched.


# Plot results from multiple sets of enrichment results

It is often useful to plot results from multiple gene list analyses together.
The `ewce_plot` function allows multiple enrichment analyses to be performed together. To achieve this the results data frames are just appended onto each other, with an additional `list` column added detailing which analysis 
they relate to.

To demonstrate this we need to first generate a second analysis so let us sample thirty random genes, and run the bootstrapping analysis on it.

```R
human.bg <- ewceData::mouse_to_human_homologs()$HGNC.symbol
gene.list.2 <- sample(human.bg,size = 30)
second_results <- EWCE::bootstrap_enrichment_test(sct_data = ctd,
                                                  sctSpecies = "mouse",
                                                  hits = gene.list.2, 
                                                  reps = reps,
                                                  annotLevel = 1)
full_res2 = data.frame(full_results$results,
                       list="Alzheimers")
rando_res2 = data.frame(second_results$results,
                        list="Random")
merged_results = rbind(full_res2, rando_res2)
```

### Plot results

Here we apply multiple-testing correction (BH = Benjamini-Hochberg) to guard against positive results due to chance.

```R
plot_list <- EWCE::ewce_plot(total_res = merged_results,
                             mtc_method = "BH")
print(plot_list$plain)
```

As expected, the second randomly generated gene list shows no significant enrichment results.


# Session Info

<details>

```{r Session Info}
utils::sessionInfo()
```

</details>
<hr>

# References

