---
title: "Applications to transcriptomes"
author: "<h4>Alan Murphy, Brian Schilder, and Nathan Skene</h4>"
date: "<h4>`r format( Sys.Date(), '%b-%d-%Y')`</h4>"
bibliography: ../inst/cit/EWCE.bib
csl: ../inst/cit/nature.csl
output:
  BiocStyle::html_document:
vignette: >
  %\VignetteIndexEntry{Applications to transcriptomes}
  %\VignetteEngine{knitr::rmarkdown}
  %\usepackage[utf8]{inputenc}     
---

# Setup  

```{r setup, include=TRUE}
library(EWCE) 
set.seed(1234)
```

# Application to transcriptomic data

## Prepare data 

```{r}
ctd <- ewceData::ctd()
tt_alzh <- ewceData::tt_alzh()

## NOTE: rep=100 for demo purposes only. 
## Use >=10,000 for publication-quality results.
reps <- 100 
```

## Analysing single transcriptome study

For the prior analyses the gene lists were not associated with any numeric 
values or directionality. 
The methodology for extending this form of analysis to transcriptomic studies simply involves thresholding the most upregulated and downregulated genes.

To demonstrate this we have an example dataset `tt_alzh`. This data frame was generated using `limma` from a set of post-mortem tissue samples from 
Brodmann area 46 which were described in a paper by the Haroutunian lab[@haroutunian2009transcriptional]. 

The first step is to load the data, obtain the MGI ids, sort the rows by t-statistic and then select the most up/down-regulated genes. 
The package then has a function `ewce_expression_data` which thresholds and selects the gene sets, and calls the EWCE function. 

Below we show the function call using the default settings, but if desired different threshold values can be used, or alternative columns used to sort the table.

```{r, results="hide"}
# ewce_expression_data calls bootstrap_enrichment_test so  
tt_results <- EWCE:: ewce_expression_data(sct_data = ctd,
                                          tt = tt_alzh,
                                          annotLevel = 1,
                                          ttSpecies = "human",
                                          sctSpecies = "mouse")
```

### Plot results 

The results of this analysis can be plotted using the `ewce_plot` function.

```{r fig.width = 7, fig.height = 4}
plot_list <- EWCE::ewce_plot(tt_results$joint_results)
print(plot_list$plain)
```

As was reported in our paper, neuronal genes are found to be downregulated while glial genes are upregulated.



# Generating bootstrap plots for transcriptomes

A common request is to explain which differentially expressed genes
are associated with a cell type...

```R
full_result_path <- EWCE::generate_bootstrap_plots_for_transcriptome(
  sct_data = ctd,
  tt = tt_alzh,
  annotLevel = 1,
  full_results = tt_results,
  listFileName = "examples",
  reps = reps,
  ttSpecies = "human",
  sctSpecies = "mouse",
  sig_only = FALSE,
  savePath = tempdir())
```


# Merging multiple transcriptome studies

Where multiple transcriptomic studies have been performed with the same purpose, i.e. seeking differential expression in dlPFC of post-mortem schizophrenics, it is common to want to determine whether they exhibit any shared signal. EWCE can be used to merge the results of multiple studies.

To demonstrate this we use a two further Alzheimer's transcriptome datasets coming from Brodmann areas 36 and 44: these area stored in `tt_alzh_BA36` and `tt_alzh_BA44`. The first step is to run EWCE on each of these individually and store the output into one list.


## Load data

```{r}
tt_alzh_BA36 <- ewceData::tt_alzh_BA36()
tt_alzh_BA44 <- ewceData::tt_alzh_BA44() 
```

## Run EWCE analysis

```{r results="hide"} 
tt_results_36 <- EWCE::ewce_expression_data(sct_data = ctd,
                                             tt = tt_alzh_BA36,
                                             annotLevel = 1,
                                             ttSpecies = "human",
                                             sctSpecies = "mouse")
tt_results_44 <- EWCE::ewce_expression_data(sct_data = ctd,
                                             tt = tt_alzh_BA44,
                                             annotLevel = 1,
                                             ttSpecies = "human",
                                             sctSpecies = "mouse")

# Fill a list with the results
results <- EWCE::add_res_to_merging_list(tt_results)
results <- EWCE::add_res_to_merging_list(tt_results_36,results)
results <- EWCE::add_res_to_merging_list(tt_results_44,results)

# Perform the merged analysis
# For publication reps should be higher
merged_res <- EWCE::merged_ewce(results = results,
                                reps = 10) 
print(merged_res)
```

The results can then be plotted as normal using the `ewce_plot` function.

```{r fig.width = 7, fig.height = 4}
plot_list <- EWCE::ewce_plot(merged_res)
print(plot_list$plain)
```

The merged results from all three Alzheimer's brain regions are found to be remarkably similar, as was reported in our paper.  

# Session Info

<details>

```{r Session Info}
utils::sessionInfo()
```

</details>


# References

