<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Bootstrap cell type enrichment test — bootstrap_enrichment_test • EWCE</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Bootstrap cell type enrichment test — bootstrap_enrichment_test"><meta property="og:description" content="bootstrap_enrichment_test takes a genelist and a single cell type
transcriptome dataset and determines the probability of enrichment and fold
changes for each cell type."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/NathanSkene/EWCE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Bootstrap cell type enrichment test</h1>
    <small class="dont-index">Source: <a href="https://github.com/NathanSkene/EWCE/blob/HEAD/R/bootstrap_enrichment_test.R" class="external-link"><code>R/bootstrap_enrichment_test.R</code></a></small>
    <div class="hidden name"><code>bootstrap_enrichment_test.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>bootstrap_enrichment_test</code> takes a genelist and a single cell type
transcriptome dataset and determines the probability of enrichment and fold
changes for each cell type.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">bootstrap_enrichment_test</span><span class="op">(</span>
  sct_data <span class="op">=</span> <span class="cn">NULL</span>,
  hits <span class="op">=</span> <span class="cn">NULL</span>,
  bg <span class="op">=</span> <span class="cn">NULL</span>,
  genelistSpecies <span class="op">=</span> <span class="cn">NULL</span>,
  sctSpecies <span class="op">=</span> <span class="cn">NULL</span>,
  output_species <span class="op">=</span> <span class="st">"human"</span>,
  method <span class="op">=</span> <span class="st">"homologene"</span>,
  reps <span class="op">=</span> <span class="fl">100</span>,
  no_cores <span class="op">=</span> <span class="fl">1</span>,
  annotLevel <span class="op">=</span> <span class="fl">1</span>,
  geneSizeControl <span class="op">=</span> <span class="cn">FALSE</span>,
  controlledCT <span class="op">=</span> <span class="cn">NULL</span>,
  mtc_method <span class="op">=</span> <span class="st">"BH"</span>,
  sort_results <span class="op">=</span> <span class="cn">TRUE</span>,
  verbose <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>sct_data</dt>
<dd><p>List generated using <a href="generate_celltype_data.html">generate_celltype_data</a>.</p></dd>


<dt>hits</dt>
<dd><p>List of gene symbols containing the target gene list.
Will automatically be converted to human gene symbols
if <code>geneSizeControl=TRUE</code>.</p></dd>


<dt>bg</dt>
<dd><p>List of gene symbols containing the background gene list
(including hit genes). If <code>bg=NULL</code>,
 an appropriate gene background will be created automatically.</p></dd>


<dt>genelistSpecies</dt>
<dd><p>Species that <code>hits</code> genes came from
(no longer limited to just "mouse" and "human").
See <a href="list_species.html">list_species</a> for all available species.</p></dd>


<dt>sctSpecies</dt>
<dd><p>Species that <code>sct_data</code> came from
(no longer limited to just "mouse" and "human").
See <a href="list_species.html">list_species</a> for all available species.</p></dd>


<dt>output_species</dt>
<dd><p>Species to convert <code>sct_data</code> and <code>hits</code> to
(Default: "human").
See <a href="list_species.html">list_species</a> for all available species.</p></dd>


<dt>method</dt>
<dd><p>R package to use for gene mapping:</p><ul><li><p><code>"gprofiler"</code> : Slower but more species and genes.</p></li>
<li><p><code>"homologene"</code> : Faster but fewer species and genes.</p></li>
<li><p><code>"babelgene"</code> : Faster but fewer species and genes.
Also gives consensus scores for each gene mapping based on a
 several different data sources.</p></li>
</ul></dd>


<dt>reps</dt>
<dd><p>Number of random gene lists to generate (<em>Default: 100</em>,
but should be &gt;=10,000 for publication-quality results).</p></dd>


<dt>no_cores</dt>
<dd><p>Number of cores to parallelise
bootstrapping <code>reps</code> over.</p></dd>


<dt>annotLevel</dt>
<dd><p>An integer indicating which level of <code>sct_data</code> to
analyse (<em>Default: 1</em>).</p></dd>


<dt>geneSizeControl</dt>
<dd><p>Whether you want to control for
GC content and transcript length. Recommended if the gene list originates
from genetic studies (<em>Default: FALSE</em>).
If set to <code>TRUE</code>, then <code>hits</code> must be from humans.</p></dd>


<dt>controlledCT</dt>
<dd><p>[Optional] If not NULL, and instead is the name of a
cell type, then the bootstrapping controls for expression within that
cell type.</p></dd>


<dt>mtc_method</dt>
<dd><p>Multiple-testing correction method
(passed to <a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a>).</p></dd>


<dt>sort_results</dt>
<dd><p>Sort enrichment results from
smallest to largest p-values.</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A list containing three data frames:</p><ul><li><p><code>results</code>: dataframe in which each row gives the statistics
  (p-value, fold change and number of standard deviations from the mean)
  associated with the enrichment of the stated cell type in the gene list</p></li>
<li><p><code>hit.cells</code>: vector containing the summed proportion of
  expression in each cell type for the target list</p></li>
<li><p><code>bootstrap_data</code>: matrix in which each row represents the
  summed proportion of expression in each cell type for one of the
  random lists</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co"># Load the single cell data</span></span>
<span class="r-in"><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?ewceData and browseVignettes('ewceData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span class="co"># Set the parameters for the analysis</span></span>
<span class="r-in"><span class="co"># Use 3 bootstrap lists for speed, for publishable analysis use &gt;=10,000</span></span>
<span class="r-in"><span class="va">reps</span> <span class="op">&lt;-</span> <span class="fl">3</span></span>
<span class="r-in"><span class="co"># Load gene list from Alzheimer's disease GWAS</span></span>
<span class="r-in"><span class="va">example_genelist</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/example_genelist.html" class="external-link">example_genelist</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?ewceData and browseVignettes('ewceData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Bootstrap significance test, no control for transcript length or GC content</span></span>
<span class="r-in"><span class="va">full_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu">bootstrap_enrichment_test</span><span class="op">(</span></span>
<span class="r-in">    sct_data <span class="op">=</span> <span class="va">ctd</span>,</span>
<span class="r-in">    hits <span class="op">=</span> <span class="va">example_genelist</span>,</span>
<span class="r-in">    reps <span class="op">=</span> <span class="va">reps</span>,</span>
<span class="r-in">    annotLevel <span class="op">=</span> <span class="fl">1</span>,</span>
<span class="r-in">    sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span class="r-in">    genelistSpecies <span class="op">=</span> <span class="st">"human"</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 core(s) assigned as workers (1 reserved).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Generating gene background for mouse x human ==&gt; human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 21,207 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 21,207 genes from mouse.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.frame format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from Gene.Symbol.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 21,207 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting mouse ==&gt; human orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17,355 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17,355 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 131 genes that have multiple input_gene per ortholog_gene (many:1).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 498 genes that have multiple ortholog_gene per input_gene (1:many).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    4,725 / 21,207 (22%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    16,482 / 21,207 (78%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,482 intersect background genes used.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising CellTypeDataset</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to sparse matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to sparse matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking gene list inputs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising sct_data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting gene list input to standardised human genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running without gene size control.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17 hit genes remain after filtering.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing summed proportions.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for enrichment in 7 cell types...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sorting results by p-value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing BH-corrected q-values.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2 significant cell type enrichment results @ q&lt;0.05 : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>               CellType annotLevel p fold_change sd_from_mean q</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 astrocytes_ependymal          1 0    1.491187     4.058933 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2            microglia          1 0    2.051216     2.670103 0</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

