<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Drop uninformative genes — drop_uninformative_genes • EWCE</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Drop uninformative genes — drop_uninformative_genes"><meta property="og:description" content="drop_uninformative_genes drops uninformative genes in order to reduce
compute time and noise in subsequent steps. It achieves this through several
steps, each of which are optional:
Drop non-1:1 orthologs:Removes genes that don't have 1:1 orthologs
with the output_species (&quot;human&quot; by default).
Drop non-varying genes:Removes genes that don't vary across cells
based on variance deciles.
Drop non-differentially expressed genes (DEGs):
Removes genes that are not significantly differentially
expressed across cell-types (multiple DEG methods available).

"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.11.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/NathanSkene/EWCE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Drop uninformative genes</h1>
    <small class="dont-index">Source: <a href="https://github.com/NathanSkene/EWCE/blob/HEAD/R/drop_uninformative_genes.r" class="external-link"><code>R/drop_uninformative_genes.r</code></a></small>
    <div class="hidden name"><code>drop_uninformative_genes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>drop_uninformative_genes</code> drops uninformative genes in order to reduce
compute time and noise in subsequent steps. It achieves this through several
steps, each of which are optional:</p><ul><li><p>Drop non-1:1 orthologs:<br>Removes genes that don't have 1:1 orthologs
with the <code>output_species</code> ("human" by default).</p></li>
<li><p>Drop non-varying genes:<br>Removes genes that don't vary across cells
based on variance deciles.</p></li>
<li><p>Drop non-differentially expressed genes (DEGs):<br>
Removes genes that are not significantly differentially
expressed across cell-types (multiple DEG methods available).</p></li>
</ul></div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">drop_uninformative_genes</span><span class="op">(</span></span>
<span>  <span class="va">exp</span>,</span>
<span>  <span class="va">level2annot</span>,</span>
<span>  mtc_method <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>  adj_pval_thresh <span class="op">=</span> <span class="fl">1e-05</span>,</span>
<span>  convert_orths <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  input_species <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  output_species <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  non121_strategy <span class="op">=</span> <span class="st">"drop_both_species"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"homologene"</span>,</span>
<span>  as_sparse <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  as_DelayedArray <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  return_sce <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  no_cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>exp</dt>
<dd><p>Expression matrix with gene names as rownames.</p></dd>


<dt>level2annot</dt>
<dd><p>Array of cell types, with each sequentially corresponding
a column in the expression matrix.</p></dd>


<dt>mtc_method</dt>
<dd><p>Multiple-testing correction method used by DGE step.
See <a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a> for more details.</p></dd>


<dt>adj_pval_thresh</dt>
<dd><p>Minimum differential expression significance
that a gene must demonstrate across <code>level2annot</code> (i.e. cell types).</p></dd>


<dt>convert_orths</dt>
<dd><p>If <code>input_species!=output_species</code> and
<code>convert_orths=TRUE</code>, will drop genes without
1:1 <code>output_species</code> orthologs and then convert <code>exp</code> gene names
to those of <code>output_species</code>.</p></dd>


<dt>input_species</dt>
<dd><p>Which species the gene names in <code>exp</code> come from.
See <a href="list_species.html">list_species</a> for all available species.</p></dd>


<dt>output_species</dt>
<dd><p>Which species' genes names to convert <code>exp</code> to.
See <a href="list_species.html">list_species</a> for all available species.</p></dd>


<dt>non121_strategy</dt>
<dd><p>How to handle genes that don't have
1:1 mappings between <code>input_species</code>:<code>output_species</code>.
Options include:<br></p><ul><li><p><code>"drop_both_species" or "dbs" or 1</code> : <br>
Drop genes that have duplicate
mappings in either the <code>input_species</code> or <code>output_species</code> <br>
(<em>DEFAULT</em>).</p></li>
<li><p><code>"drop_input_species" or "dis" or 2</code> : <br>
Only drop genes that have duplicate
mappings in the <code>input_species</code>.</p></li>
<li><p><code>"drop_output_species" or "dos" or 3</code> : <br>
Only drop genes that have duplicate
mappings in the <code>output_species</code>.</p></li>
<li><p><code>"keep_both_species" or "kbs" or 4</code> : <br>
Keep all genes regardless of whether
they have duplicate mappings in either species.</p></li>
<li><p><code>"keep_popular" or "kp" or 5</code> : <br>
Return only the most "popular" interspecies ortholog mappings.
 This procedure tends to yield a greater number of returned genes
 but at the cost of many of them not being true biological 1:1 orthologs.</p></li>
<li><p><code>"sum","mean","median","min" or "max"</code> : <br>
 When <code>gene_df</code> is a matrix and <code>gene_output="rownames"</code>,
  these options will aggregate many-to-one gene mappings
  (<code>input_species</code>-to-<code>output_species</code>)
  after dropping any duplicate genes in the <code>output_species</code>.</p></li>
</ul></dd>


<dt>method</dt>
<dd><p>R package to use for gene mapping:</p><ul><li><p><code>"gprofiler"</code> : Slower but more species and genes.</p></li>
<li><p><code>"homologene"</code> : Faster but fewer species and genes.</p></li>
<li><p><code>"babelgene"</code> : Faster but fewer species and genes.
Also gives consensus scores for each gene mapping based on a
 several different data sources.</p></li>
</ul></dd>


<dt>as_sparse</dt>
<dd><p>Convert <code>exp</code> to sparse matrix.</p></dd>


<dt>as_DelayedArray</dt>
<dd><p>Convert <code>exp</code> to <code>DelayedArray</code>
for scalable processing.</p></dd>


<dt>return_sce</dt>
<dd><p>Whether to return the filtered results
as an expression matrix or a <span class="pkg">SingleCellExperiment</span>.</p></dd>


<dt>no_cores</dt>
<dd><p>Number of cores to parallelise across.
Set to <code>NULL</code> to automatically optimise.</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.
#' @inheritParams orthogene::convert_orthologs</p></dd>


<dt>...</dt>
<dd><p>Arguments passed on to <code><a href="https://rdrr.io/pkg/orthogene/man/convert_orthologs.html" class="external-link">orthogene::convert_orthologs</a></code></p><dl><dt><code>gene_df</code></dt>
<dd><p>Data object containing the genes
(see <code>gene_input</code> for options on how
the genes can be stored within the object).<br>
Can be one of the following formats:<br></p><ul><li><p><code>matrix</code> : <br>A sparse or dense matrix.</p></li>
<li><p><code>data.frame</code> : <br>A <code>data.frame</code>,
 <code>data.table</code>. or <code>tibble</code>.</p></li>
<li><p>codelist : <br>A <code>list</code> or character <code>vector</code>.</p></li>
</ul><p>Genes, transcripts, proteins, SNPs, or genomic ranges
 can be provided in any format
(HGNC, Ensembl, RefSeq, UniProt, etc.) and will be
automatically converted to gene symbols unless
specified otherwise with the <code>...</code> arguments.<br><em>Note</em>: If you set <code>method="homologene"</code>, you
must either supply genes in gene symbol format (e.g. "Sox2")
 OR set <code>standardise_genes=TRUE</code>.</p></dd>

    <dt><code>gene_input</code></dt>
<dd><p>Which aspect of <code>gene_df</code> to
get gene names from:<br></p><ul><li><p><code>"rownames"</code> : <br>From row names of data.frame/matrix.</p></li>
<li><p><code>"colnames"</code> : <br>From column names of data.frame/matrix.</p></li>
<li><p><code>&lt;column name&gt;</code> : <br>From a column in <code>gene_df</code>,
 e.g. <code>"gene_names"</code>.</p></li>
</ul></dd>

    <dt><code>gene_output</code></dt>
<dd><p>How to return genes.
Options include:<br></p><ul><li><p><code>"rownames"</code> : <br>As row names of <code>gene_df</code>.</p></li>
<li><p><code>"colnames"</code> : <br>As column names of <code>gene_df</code>.</p></li>
<li><p><code>"columns"</code> : <br>As new columns "input_gene", "ortholog_gene"
(and "input_gene_standard" if <code>standardise_genes=TRUE</code>)
in <code>gene_df</code>.</p></li>
<li><p><code>"dict"</code> : <br>As a dictionary (named list) where the names
are input_gene and the values are ortholog_gene.</p></li>
<li><p><code>"dict_rev"</code> : <br>As a reversed dictionary (named list)
where the names are ortholog_gene and the values are input_gene.</p></li>
</ul></dd>

    <dt><code>standardise_genes</code></dt>
<dd><p>If <code>TRUE</code> AND
<code>gene_output="columns"</code>, a new column "input_gene_standard"
will be added to <code>gene_df</code> containing standardised HGNC symbols
identified by <a href="https://rdrr.io/pkg/gprofiler2/man/gorth.html" class="external-link">gorth</a>.</p></dd>

    <dt><code>drop_nonorths</code></dt>
<dd><p>Drop genes that don't have an ortholog
in the <code>output_species</code>.</p></dd>

    <dt><code>agg_fun</code></dt>
<dd><p>Aggregation function passed to 
 <a href="https://rdrr.io/pkg/orthogene/man/aggregate_mapped_genes.html" class="external-link">aggregate_mapped_genes</a>. 
Set to <code>NULL</code> to skip aggregation step (default).</p></dd>

    <dt><code>mthreshold</code></dt>
<dd><p>Maximum number of ortholog names per gene to show.
Passed to <a href="https://rdrr.io/pkg/gprofiler2/man/gorth.html" class="external-link">gorth</a>.
Only used when <code>method="gprofiler"</code> (<em>DEFAULT : </em><code>Inf</code>).</p></dd>

    <dt><code>sort_rows</code></dt>
<dd><p>Sort <code>gene_df</code> rows alphanumerically.</p></dd>

    <dt><code>gene_map</code></dt>
<dd><p>A <a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a> that maps the current gene names
to new gene names. 
This function's behaviour will adapt to different situations as follows:</p><ul><li><p><code>gene_map=&lt;data.frame&gt;</code> :<br> When a data.frame containing the
gene key:value columns 
(specified by <code>input_col</code> and <code>output_col</code>, respectively)
is provided, this will be used to perform aggregation/expansion.</p></li>
<li><p><code>gene_map=NULL</code> and <code>input_species!=output_species</code> :<br>
A <code>gene_map</code> is automatically generated by
 <a href="https://rdrr.io/pkg/orthogene/man/map_orthologs.html" class="external-link">map_orthologs</a> to perform inter-species 
 gene aggregation/expansion.</p></li>
<li><p><code>gene_map=NULL</code> and <code>input_species==output_species</code> :<br>
A <code>gene_map</code> is automatically generated by
 <a href="https://rdrr.io/pkg/orthogene/man/map_genes.html" class="external-link">map_genes</a> to perform within-species 
 gene gene symbol standardization and aggregation/expansion.</p></li>
</ul></dd>

    <dt><code>input_col</code></dt>
<dd><p>Column name within <code>gene_map</code> with gene names matching 
the row names of <code>X</code>.</p></dd>

    <dt><code>output_col</code></dt>
<dd><p>Column name within <code>gene_map</code> with gene names
that you wish you map the row names of <code>X</code> onto.</p></dd>

  
</dl></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>exp Expression matrix with gene names as row names.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">cortex_mrna</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/cortex_mrna.html" class="external-link">cortex_mrna</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?ewceData and browseVignettes('ewceData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span><span class="co"># Use only a subset of genes to keep the example quick</span></span></span>
<span class="r-in"><span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span> <span class="op">&lt;-</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">300</span>, <span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Convert orthologs at the same time</span></span></span>
<span class="r-in"><span><span class="va">exp2_orth</span> <span class="op">&lt;-</span> <span class="fu">drop_uninformative_genes</span><span class="op">(</span></span></span>
<span class="r-in"><span>    exp <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span>,</span></span>
<span class="r-in"><span>    level2annot <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level2class</span>,</span></span>
<span class="r-in"><span>    input_species <span class="op">=</span> <span class="st">"mouse"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Check 300Check 3005</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning: sctSpecies_origin not provided. Setting to 'mouse' by default.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 core(s) assigned as workers (1 reserved).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to sparse matrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for non-expressed genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for cells with no expressed genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> DGE:: Limma...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3 / 300 genes dropped @ DGE adj_pval_thresh &lt; 1e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.09651661 secs</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

