<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Extract a matrix from a CellTypeDataset — extract_matrix • EWCE</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Extract a matrix from a CellTypeDataset — extract_matrix"><meta property="og:description" content="Extracts a particular matrix (e.g., mean_exp, specificity)
from a CellTypeDataset object."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.3.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/NathanSkene/EWCE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Extract a matrix from a CellTypeDataset</h1>
    <small class="dont-index">Source: <a href="https://github.com/NathanSkene/EWCE/blob/HEAD/R/extract_matrix.R" class="external-link"><code>R/extract_matrix.R</code></a></small>
    <div class="hidden name"><code>extract_matrix.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Extracts a particular matrix (e.g., mean_exp, specificity)
from a CellTypeDataset object.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">extract_matrix</span><span class="op">(</span>
  <span class="va">ctd</span>,
  <span class="va">dataset</span>,
  level <span class="op">=</span> <span class="fl">1</span>,
  input_species <span class="op">=</span> <span class="cn">NULL</span>,
  output_species <span class="op">=</span> <span class="st">"human"</span>,
  metric <span class="op">=</span> <span class="st">"specificity"</span>,
  non121_strategy <span class="op">=</span> <span class="st">"drop_both_species"</span>,
  method <span class="op">=</span> <span class="st">"homologene"</span>,
  numberOfBins <span class="op">=</span> <span class="fl">40</span>,
  remove_unlabeled_clusters <span class="op">=</span> <span class="cn">FALSE</span>,
  force_new_quantiles <span class="op">=</span> <span class="cn">FALSE</span>,
  as_sparse <span class="op">=</span> <span class="cn">TRUE</span>,
  as_DelayedArray <span class="op">=</span> <span class="cn">FALSE</span>,
  rename_columns <span class="op">=</span> <span class="cn">TRUE</span>,
  make_columns_unique <span class="op">=</span> <span class="cn">FALSE</span>,
  verbose <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>ctd</dt>
<dd><p>Input CellTypeData.</p></dd>
<dt>dataset</dt>
<dd><p>CellTypeData. name.</p></dd>
<dt>level</dt>
<dd><p>CTD level to extract from.</p></dd>
<dt>input_species</dt>
<dd><p>Which species the gene names in <code>exp</code> come from.
See <a href="list_species.html">list_species</a> for all available species.</p></dd>
<dt>output_species</dt>
<dd><p>Which species' genes names to convert <code>exp</code> to.
See <a href="list_species.html">list_species</a> for all available species.</p></dd>
<dt>metric</dt>
<dd><p>Name of the matrix to extract.</p></dd>
<dt>non121_strategy</dt>
<dd><p>How to handle genes that don't have
1:1 mappings between <code>input_species</code>:<code>output_species</code>.
Options include:<br></p><ul><li><p><code>"drop_both_species" or "dbs" or 1</code> : <br>
Drop genes that have duplicate
mappings in either the <code>input_species</code> or <code>output_species</code> <br>
(<em>DEFAULT</em>).</p></li>
<li><p><code>"drop_input_species" or "dis" or 2</code> : <br>
Only drop genes that have duplicate
mappings in the <code>input_species</code>.</p></li>
<li><p><code>"drop_output_species" or "dos" or 3</code> : <br>
Only drop genes that have duplicate
mappings in the <code>output_species</code>.</p></li>
<li><p><code>"keep_both_species" or "kbs" or 4</code> : <br>
Keep all genes regardless of whether
they have duplicate mappings in either species.</p></li>
<li><p><code>"keep_popular" or "kp" or 5</code> : <br>
Return only the most "popular" interspecies ortholog mappings.
 This procedure tends to yield a greater number of returned genes
 but at the cost of many of them not being true biological 1:1 orthologs.</p></li>
<li><p><code>"sum","mean","median","min" or "max"</code> : <br>
 When <code>gene_df</code> is a matrix and <code>gene_output="rownames"</code>,
  these options will aggregate many-to-one gene mappings
  (<code>input_species</code>-to-<code>output_species</code>)
  after dropping any duplicate genes in the <code>output_species</code>.</p></li>
</ul></dd>
<dt>method</dt>
<dd><p>R package to to use for gene mapping:</p><ul><li><p><code>"gprofiler"</code> : Slower but more species and genes.</p></li>
<li><p><code>"homologene"</code> : Faster but fewer species and genes.</p></li>
<li><p><code>"babelgene"</code> : Faster but fewer species and genes.
Also gives</p></li>
</ul><p>(slower but more species and genes) or
<code>"homologene"</code> (faster but fewer species and genes).</p></dd>
<dt>numberOfBins</dt>
<dd><p>Number of non-zero quantile bins.</p></dd>
<dt>remove_unlabeled_clusters</dt>
<dd><p>Remove any samples that have
numeric column names.</p></dd>
<dt>force_new_quantiles</dt>
<dd><p>By default, quantile computation is
skipped if they have already been computed.
Set <code>=TRUE</code> to override this and generate new quantiles.</p></dd>
<dt>as_sparse</dt>
<dd><p>Convert to sparse matrix.</p></dd>
<dt>as_DelayedArray</dt>
<dd><p>Convert to <code>DelayedArray</code>.</p></dd>
<dt>rename_columns</dt>
<dd><p>Remove <code>replace_chars</code> from column names.</p></dd>
<dt>make_columns_unique</dt>
<dd><p>Rename each columns with the prefix
<code>dataset.species.celltype</code>.</p></dd>
<dt>verbose</dt>
<dd><p>Print messages.
Set <code>verbose=2</code> if you want to print all messages
 from internal functions as well.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>(specificity) matrix.</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

