<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Generate bootstrap plots — generate_bootstrap_plots • EWCE</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Generate bootstrap plots — generate_bootstrap_plots"><meta property="og:description" content="generate_bootstrap_plots takes a gene list and a single cell type
transcriptome dataset and generates plots which show how the expression of
the genes in the list compares to those in randomly generated gene lists."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.14.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/NathanSkene/EWCE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate bootstrap plots</h1>
    <small class="dont-index">Source: <a href="https://github.com/NathanSkene/EWCE/blob/RELEASE_3_20/R/generate_bootstrap_plots.r" class="external-link"><code>R/generate_bootstrap_plots.r</code></a></small>
    <div class="hidden name"><code>generate_bootstrap_plots.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>generate_bootstrap_plots</code> takes a gene list and a single cell type
transcriptome dataset and generates plots which show how the expression of
the genes in the list compares to those in randomly generated gene lists.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">generate_bootstrap_plots</span><span class="op">(</span></span>
<span>  sct_data <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  hits <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  bg <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  genelistSpecies <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sctSpecies <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  output_species <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"homologene"</span>,</span>
<span>  reps <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  annotLevel <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  geneSizeControl <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  full_results <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  listFileName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"_level"</span>, <span class="va">annotLevel</span><span class="op">)</span>,</span>
<span>  adj_pval_thresh <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  facets <span class="op">=</span> <span class="st">"CellType"</span>,</span>
<span>  scales <span class="op">=</span> <span class="st">"free_x"</span>,</span>
<span>  save_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"BootstrapPlots"</span><span class="op">)</span>,</span>
<span>  show_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-sct-data">sct_data<a class="anchor" aria-label="anchor" href="#arg-sct-data"></a></dt>
<dd><p>List generated using <a href="generate_celltype_data.html">generate_celltype_data</a>.</p></dd>


<dt id="arg-hits">hits<a class="anchor" aria-label="anchor" href="#arg-hits"></a></dt>
<dd><p>List of gene symbols containing the target gene list.
Will automatically be converted to human gene symbols
if <code>geneSizeControl=TRUE</code>.</p></dd>


<dt id="arg-bg">bg<a class="anchor" aria-label="anchor" href="#arg-bg"></a></dt>
<dd><p>List of gene symbols containing the background gene list
(including hit genes). If <code>bg=NULL</code>,
 an appropriate gene background will be created automatically.</p></dd>


<dt id="arg-genelistspecies">genelistSpecies<a class="anchor" aria-label="anchor" href="#arg-genelistspecies"></a></dt>
<dd><p>Species that <code>hits</code> genes came from
(no longer limited to just "mouse" and "human").
See <a href="list_species.html">list_species</a> for all available species.</p></dd>


<dt id="arg-sctspecies">sctSpecies<a class="anchor" aria-label="anchor" href="#arg-sctspecies"></a></dt>
<dd><p>Species that <code>sct_data</code> is currently formatted as
(no longer limited to just "mouse" and "human").
See <a href="list_species.html">list_species</a> for all available species.</p></dd>


<dt id="arg-output-species">output_species<a class="anchor" aria-label="anchor" href="#arg-output-species"></a></dt>
<dd><p>Species to convert <code>sct_data</code> and <code>hits</code> to
(Default: "human").
See <a href="list_species.html">list_species</a> for all available species.</p></dd>


<dt id="arg-method">method<a class="anchor" aria-label="anchor" href="#arg-method"></a></dt>
<dd><p>R package to use for gene mapping:</p><ul><li><p><code>"gprofiler"</code> : Slower but more species and genes.</p></li>
<li><p><code>"homologene"</code> : Faster but fewer species and genes.</p></li>
<li><p><code>"babelgene"</code> : Faster but fewer species and genes.
Also gives consensus scores for each gene mapping based on a
 several different data sources.</p></li>
</ul></dd>


<dt id="arg-reps">reps<a class="anchor" aria-label="anchor" href="#arg-reps"></a></dt>
<dd><p>Number of random gene lists to generate (<em>Default: 100</em>,
but should be &gt;=10,000 for publication-quality results).</p></dd>


<dt id="arg-annotlevel">annotLevel<a class="anchor" aria-label="anchor" href="#arg-annotlevel"></a></dt>
<dd><p>An integer indicating which level of <code>sct_data</code> to
analyse (<em>Default: 1</em>).</p></dd>


<dt id="arg-genesizecontrol">geneSizeControl<a class="anchor" aria-label="anchor" href="#arg-genesizecontrol"></a></dt>
<dd><p>Whether you want to control for
GC content and transcript length. Recommended if the gene list originates
from genetic studies (<em>Default: FALSE</em>).
If set to <code>TRUE</code>, then <code>hits</code> must be from humans.</p></dd>


<dt id="arg-full-results">full_results<a class="anchor" aria-label="anchor" href="#arg-full-results"></a></dt>
<dd><p>The full output of
<a href="bootstrap_enrichment_test.html">bootstrap_enrichment_test</a> for the same gene list.</p></dd>


<dt id="arg-listfilename">listFileName<a class="anchor" aria-label="anchor" href="#arg-listfilename"></a></dt>
<dd><p>String used as the root for files saved using this
function.</p></dd>


<dt id="arg-adj-pval-thresh">adj_pval_thresh<a class="anchor" aria-label="anchor" href="#arg-adj-pval-thresh"></a></dt>
<dd><p>Adjusted p-value threshold of celltypes to include
in plots.</p></dd>


<dt id="arg-facets">facets<a class="anchor" aria-label="anchor" href="#arg-facets"></a></dt>
<dd><p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated" class="external-link"><img src="figures/lifecycle-deprecated.svg" alt="[Deprecated]"></a> Please use <code>rows</code>
and <code>cols</code> instead.</p></dd>


<dt id="arg-scales">scales<a class="anchor" aria-label="anchor" href="#arg-scales"></a></dt>
<dd><p>Are scales shared across all facets (the default,
<code>"fixed"</code>), or do they vary across rows (<code>"free_x"</code>),
columns (<code>"free_y"</code>), or both rows and columns (<code>"free"</code>)?</p></dd>


<dt id="arg-save-dir">save_dir<a class="anchor" aria-label="anchor" href="#arg-save-dir"></a></dt>
<dd><p>Directory where the BootstrapPlots folder should be saved,
default is a temp directory.</p></dd>


<dt id="arg-show-plot">show_plot<a class="anchor" aria-label="anchor" href="#arg-show-plot"></a></dt>
<dd><p>Print the plot.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Saves a set of pdf files containing graphs and returns the file where
they are saved. These will be saved with the file name adjusted using the
value of <code>listFileName</code>. The files are saved into the
'BootstrapPlot' folder.
Files start with one of the following:</p><ul><li><p><code>qqplot_noText</code>: sorts the gene list according to how enriched
  it is in the relevant cell type. Plots the value in the target list against
  the mean value in the bootstrapped lists.</p></li>
<li><p><code>qqplot_wtGSym</code>: as above but labels the gene symbols for the
  highest expressed genes.</p></li>
<li><p><code>bootDists</code>: rather than just showing the mean of the
  bootstrapped lists, a boxplot shows the distribution of values</p></li>
<li><p><code>bootDists_LOG</code>: shows the bootstrapped distributions with the
  y-axis shown on a log scale</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Load the single cell data</span></span></span>
<span class="r-in"><span><span class="va">sct_data</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?ewceData and browseVignettes('ewceData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Set the parameters for the analysis</span></span></span>
<span class="r-in"><span><span class="co">## Use 5 bootstrap lists for speed, for publishable analysis use &gt;10000</span></span></span>
<span class="r-in"><span><span class="va">reps</span> <span class="op">&lt;-</span> <span class="fl">5</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Load the gene list and get human orthologs</span></span></span>
<span class="r-in"><span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/example_genelist.html" class="external-link">example_genelist</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?ewceData and browseVignettes('ewceData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Bootstrap significance test,</span></span></span>
<span class="r-in"><span><span class="co">##  no control for transcript length or GC content</span></span></span>
<span class="r-in"><span><span class="co">## Use pre-computed results to speed up example</span></span></span>
<span class="r-in"><span><span class="va">full_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="example_bootstrap_results.html">example_bootstrap_results</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading precomputed example bootstrap results.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### Skip this for example purposes</span></span></span>
<span class="r-in"><span><span class="co"># full_results &lt;- EWCE::bootstrap_enrichment_test(</span></span></span>
<span class="r-in"><span><span class="co">#    sct_data = sct_data,</span></span></span>
<span class="r-in"><span><span class="co">#    hits = hits,</span></span></span>
<span class="r-in"><span><span class="co">#    reps = reps,</span></span></span>
<span class="r-in"><span><span class="co">#    annotLevel = 1,</span></span></span>
<span class="r-in"><span><span class="co">#    sctSpecies = "mouse",</span></span></span>
<span class="r-in"><span><span class="co">#    genelistSpecies = "human"</span></span></span>
<span class="r-in"><span><span class="co"># )</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">output</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu">generate_bootstrap_plots</span><span class="op">(</span></span></span>
<span class="r-in"><span>    sct_data <span class="op">=</span> <span class="va">sct_data</span>,</span></span>
<span class="r-in"><span>    hits <span class="op">=</span> <span class="va">hits</span>,</span></span>
<span class="r-in"><span>    reps <span class="op">=</span> <span class="va">reps</span>,</span></span>
<span class="r-in"><span>    full_results <span class="op">=</span> <span class="va">full_results</span>,</span></span>
<span class="r-in"><span>    sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,</span></span>
<span class="r-in"><span>    genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,</span></span>
<span class="r-in"><span>    annotLevel <span class="op">=</span> <span class="fl">1</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning: sctSpecies_origin not provided. Setting to 'mouse' by default.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Generating gene background for mouse x human ==&gt; human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering ortholog reports.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 21,207 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 21,207 genes from mouse.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.frame format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from Gene.Symbol.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 21,207 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting mouse ==&gt; human orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17,355 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17,355 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 131 genes that have multiple input_gene per ortholog_gene (many:1).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 498 genes that have multiple ortholog_gene per input_gene (1:many).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    4,725 / 21,207 (22%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    16,482 / 21,207 (78%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering ortholog reports.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,482 intersect background genes used.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising sct_data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Aligning celltype names with standardise_ctd format.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking gene list inputs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 celltype(s) remain @ &lt;= 0.05</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Resampling random genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing gene scores.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Generating bootstrap plot for 1 celltype(s).</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The `facets` argument of `facet_grid()` is deprecated as of ggplot2 2.2.0.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Please use the `rows` argument instead.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">EWCE</span> package.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span>   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/NathanSkene/EWCE/issues&gt;</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving plot --&gt; /tmp/RtmpTwJkJv/BootstrapPlots/qqplot_noText_____level1.pdf</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving plot --&gt; /tmp/RtmpTwJkJv/BootstrapPlots/qqplot_wtgene_____level1.pdf</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving plot --&gt; /tmp/RtmpTwJkJv/BootstrapPlots/bootDists_____level1.pdf</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving plot --&gt; /tmp/RtmpTwJkJv/BootstrapPlots/bootDists_LOG_____level1.pdf</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $plot1</span>
<span class="r-plt img"><img src="generate_bootstrap_plots-1.png" alt="" width="700" height="433"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $plot2</span>
<span class="r-plt img"><img src="generate_bootstrap_plots-2.png" alt="" width="700" height="433"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $plot3</span>
<span class="r-plt img"><img src="generate_bootstrap_plots-3.png" alt="" width="700" height="433"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $plot4</span>
<span class="r-plt img"><img src="generate_bootstrap_plots-4.png" alt="" width="700" height="433"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

