<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Generate bootstrap plots — generate_bootstrap_plots_for_transcriptome • EWCE</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Generate bootstrap plots — generate_bootstrap_plots_for_transcriptome" />
<meta property="og:description" content="generate_bootstrap_plots_for_transcriptome takes a genelist and a
single cell type transcriptome dataset and generates plots which show how
the expression of the genes in the list compares to those in randomly
generated gene lists" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/NathanSkene/EWCE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate bootstrap plots</h1>
    <small class="dont-index">Source: <a href='https://github.com/NathanSkene/EWCE/blob/master/R/generate_bootstrap_plots_for_transcriptome.r'><code>R/generate_bootstrap_plots_for_transcriptome.r</code></a></small>
    <div class="hidden name"><code>generate_bootstrap_plots_for_transcriptome.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>generate_bootstrap_plots_for_transcriptome</code> takes a genelist and a
single cell type transcriptome dataset and generates plots which show how
the expression of the genes in the list compares to those in randomly
generated gene lists</p>
    </div>

    <pre class="usage"><span class='fu'>generate_bootstrap_plots_for_transcriptome</span><span class='op'>(</span>
  <span class='va'>sct_data</span>,
  <span class='va'>tt</span>,
  thresh <span class='op'>=</span> <span class='fl'>250</span>,
  annotLevel <span class='op'>=</span> <span class='fl'>1</span>,
  <span class='va'>reps</span>,
  full_results <span class='op'>=</span> <span class='cn'>NA</span>,
  listFileName <span class='op'>=</span> <span class='st'>""</span>,
  showGNameThresh <span class='op'>=</span> <span class='fl'>25</span>,
  ttSpecies <span class='op'>=</span> <span class='st'>"mouse"</span>,
  sctSpecies <span class='op'>=</span> <span class='st'>"mouse"</span>,
  sortBy <span class='op'>=</span> <span class='st'>"t"</span>,
  onlySignif <span class='op'>=</span> <span class='cn'>TRUE</span>,
  savePath <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>sct_data</th>
      <td><p>List generated using <code><a href='generate_celltype_data.html'>generate_celltype_data</a></code></p></td>
    </tr>
    <tr>
      <th>tt</th>
      <td><p>Differential expression table. Can be output of limma::topTable
function. Minimum requirement is that one column stores a metric of
increased/decreased expression (i.e. log fold change, t-statistic for
differential expression etc) and another contains either HGNC or MGI symbols.</p></td>
    </tr>
    <tr>
      <th>thresh</th>
      <td><p>The number of up- and down- regulated genes to be included in
each analysis. Dafault=250</p></td>
    </tr>
    <tr>
      <th>annotLevel</th>
      <td><p>an integer indicating which level of the annotation to
analyse. Default = 1.</p></td>
    </tr>
    <tr>
      <th>reps</th>
      <td><p>Number of random gene lists to generate (default=100 but should
be over 10000 for publication quality results)</p></td>
    </tr>
    <tr>
      <th>full_results</th>
      <td><p>The full output of
<code><a href='bootstrap_enrichment_test.html'>bootstrap_enrichment_test</a></code> for the same genelist</p></td>
    </tr>
    <tr>
      <th>listFileName</th>
      <td><p>String used as the root for files saved using
this function</p></td>
    </tr>
    <tr>
      <th>showGNameThresh</th>
      <td><p>Integer. If a gene has over X percent of it's
expression proportion in a celltype, then list the gene name</p></td>
    </tr>
    <tr>
      <th>ttSpecies</th>
      <td><p>Either 'mouse' or 'human' depending on which species the
differential expression table was generated from</p></td>
    </tr>
    <tr>
      <th>sctSpecies</th>
      <td><p>Either 'mouse' or 'human' depending on which species the
single cell data was generated from</p></td>
    </tr>
    <tr>
      <th>sortBy</th>
      <td><p>Column name of metric in tt which should be used to sort up-
from down- regulated genes. Default="t"</p></td>
    </tr>
    <tr>
      <th>onlySignif</th>
      <td><p>Should plots only be generated for cells which have
significant changes?</p></td>
    </tr>
    <tr>
      <th>savePath</th>
      <td><p>Directory where the BootstrapPlots folder should be saved, 
default is a temp directory</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Saves a set of pdf files containing graphs and returns the file where 
they are saved. These will be saved with the filename adjusted using the 
value of listFileName. The files are saved into the 'BootstrapPlot' folder. 
Files start with one of the following:</p><ul>
<li><p><code>qqplot_noText</code>: sorts the gene list according to how enriched
  it is in the relevant celltype. Plots the value in the target list against
  the mean value in the bootstrapped lists.</p></li>
<li><p><code>qqplot_wtGSym</code>: as above but labels the gene symbols for the
  highest expressed genes.</p></li>
<li><p><code>bootDists</code>: rather than just showing the mean of the
  bootstrapped lists, a boxplot shows the distribution of values</p></li>
<li><p><code>bootDists_LOG</code>: shows the bootstrapped distributions with the
  y-axis shown on a log scale</p></li>
</ul>


    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># \donttest{</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/neurogenomics/ewceData'>ewceData</a></span><span class='op'>)</span>
<span class='co'># Load the single cell data</span>
<span class='va'>ctd</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/ewceData/man/ctd.html'>ctd</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'>
<span class='co'># Set the parameters for the analysis</span>
<span class='co'># Use 3 bootstrap lists for speed, for publishable analysis use &gt;10000</span>
<span class='va'>reps</span> <span class='op'>&lt;-</span> <span class='fl'>3</span>
<span class='va'>annotLevel</span> <span class='op'>&lt;-</span> <span class='fl'>1</span> <span class='co'># &lt;- Use cell level annotations (i.e. Interneurons)</span>
<span class='co'># Use 5 up/down regulated genes (thresh) for speed, default is 250</span>
<span class='va'>thresh</span> <span class='op'>=</span> <span class='fl'>5</span>

<span class='co'># Load the top table</span>
<span class='va'>tt_alzh</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/ewceData/man/tt_alzh.html'>tt_alzh</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'>
<span class='va'>tt_results</span> <span class='op'>&lt;-</span> <span class='fu'><a href='ewce_expression_data.html'>ewce_expression_data</a></span><span class='op'>(</span>
    sct_data <span class='op'>=</span> <span class='va'>ctd</span>, tt <span class='op'>=</span> <span class='va'>tt_alzh</span>, annotLevel <span class='op'>=</span> <span class='fl'>1</span>,thresh <span class='op'>=</span> <span class='va'>thresh</span>,
    reps <span class='op'>=</span> <span class='va'>reps</span>, ttSpecies <span class='op'>=</span> <span class='st'>"human"</span>, sctSpecies <span class='op'>=</span> <span class='st'>"mouse"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='output co'>#&gt; <span class='message'>Tbc1d2b, Cxcr4, Nrip2, Sox12, Selplg</span></div><div class='output co'>#&gt; <span class='message'>astrocytes_ependymal</span></div><div class='output co'>#&gt; <span class='message'>1</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>endothelial-mural</span></div><div class='output co'>#&gt; <span class='message'>0.333333333333333</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>interneurons</span></div><div class='output co'>#&gt; <span class='message'>1</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>microglia</span></div><div class='output co'>#&gt; <span class='message'>0</span></div><div class='output co'>#&gt; <span class='message'>Fold enrichment: 2.61769742353418</span></div><div class='output co'>#&gt; <span class='message'>Standard deviations from mean: 4.97571470589427</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>oligodendrocytes</span></div><div class='output co'>#&gt; <span class='message'>1</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>pyramidal CA1</span></div><div class='output co'>#&gt; <span class='message'>1</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>pyramidal SS</span></div><div class='output co'>#&gt; <span class='message'>1</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='output co'>#&gt; <span class='message'>Cirbp, Camk1g, Nat10, Sema3a, Sssca1</span></div><div class='output co'>#&gt; <span class='message'>astrocytes_ependymal</span></div><div class='output co'>#&gt; <span class='message'>1</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>endothelial-mural</span></div><div class='output co'>#&gt; <span class='message'>1</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>interneurons</span></div><div class='output co'>#&gt; <span class='message'>0</span></div><div class='output co'>#&gt; <span class='message'>Fold enrichment: 1.14872287560453</span></div><div class='output co'>#&gt; <span class='message'>Standard deviations from mean: 1.13513736354519</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>microglia</span></div><div class='output co'>#&gt; <span class='message'>0.333333333333333</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>oligodendrocytes</span></div><div class='output co'>#&gt; <span class='message'>1</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>pyramidal CA1</span></div><div class='output co'>#&gt; <span class='message'>0.333333333333333</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>pyramidal SS</span></div><div class='output co'>#&gt; <span class='message'>0</span></div><div class='output co'>#&gt; <span class='message'>Fold enrichment: 1.75923544360529</span></div><div class='output co'>#&gt; <span class='message'>Standard deviations from mean: 5.3458534400276</span></div><div class='output co'>#&gt; </div><div class='input'>
<span class='co'># Bootstrap significance test, no control for transcript length or GC content</span>
<span class='va'>full_results</span> <span class='op'>&lt;-</span> <span class='fu'>generate_bootstrap_plots_for_transcriptome</span><span class='op'>(</span>
    sct_data <span class='op'>=</span> <span class='va'>ctd</span>, tt <span class='op'>=</span> <span class='va'>tt_alzh</span>, thresh <span class='op'>=</span> <span class='va'>thresh</span>, annotLevel <span class='op'>=</span> <span class='fl'>1</span>,
    full_results <span class='op'>=</span> <span class='va'>tt_results</span>, listFileName <span class='op'>=</span> <span class='st'>"examples"</span>, reps <span class='op'>=</span> <span class='va'>reps</span>,
    ttSpecies <span class='op'>=</span> <span class='st'>"human"</span>, sctSpecies <span class='op'>=</span> <span class='st'>"mouse"</span>, savePath<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Duplicated aesthetics after name standardisation: colour</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Duplicated aesthetics after name standardisation: colour</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Duplicated aesthetics after name standardisation: colour</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Transformation introduced infinite values in continuous y-axis</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 1 rows containing non-finite values (stat_boxplot).</span></div><div class='input'><span class='co'># }</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Alan Murphy, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


