<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Multiple EWCE results from multiple studies — merged_ewce • EWCE</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Multiple EWCE results from multiple studies — merged_ewce"><meta property="og:description" content="merged_ewce combines enrichment results from multiple studies
targetting the same scientific problem"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.12.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/NathanSkene/EWCE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Multiple EWCE results from multiple studies</h1>
    <small class="dont-index">Source: <a href="https://github.com/NathanSkene/EWCE/blob/HEAD/R/merged_ewce.r" class="external-link"><code>R/merged_ewce.r</code></a></small>
    <div class="hidden name"><code>merged_ewce.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>merged_ewce</code> combines enrichment results from multiple studies
targetting the same scientific problem</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">merged_ewce</span><span class="op">(</span><span class="va">results</span>, reps <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>results</dt>
<dd><p>a list of EWCE results generated using
<a href="add_res_to_merging_list.html">add_res_to_merging_list</a>.</p></dd>


<dt>reps</dt>
<dd><p>Number of random gene lists to generate (Default=100 but should
be &gt;=10,000 for publication-quality results).</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>dataframe in which each row gives the statistics (p-value, fold
change and number of standard deviations from the mean) associated with the
enrichment of the stated cell type in the gene list.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Load the single cell data</span></span></span>
<span class="r-in"><span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?ewceData and browseVignettes('ewceData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Use 3 bootstrap lists for speed, for publishable analysis use &gt;10000</span></span></span>
<span class="r-in"><span><span class="va">reps</span> <span class="op">&lt;-</span> <span class="fl">3</span></span></span>
<span class="r-in"><span><span class="co"># Use 5 up/down regulated genes (thresh) for speed, default is 250</span></span></span>
<span class="r-in"><span><span class="va">thresh</span> <span class="op">&lt;-</span> <span class="fl">5</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Load the data</span></span></span>
<span class="r-in"><span><span class="va">tt_alzh_BA36</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/tt_alzh_BA36.html" class="external-link">tt_alzh_BA36</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?ewceData and browseVignettes('ewceData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span><span class="va">tt_alzh_BA44</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/tt_alzh_BA44.html" class="external-link">tt_alzh_BA44</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?ewceData and browseVignettes('ewceData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Run EWCE analysis</span></span></span>
<span class="r-in"><span><span class="va">tt_results_36</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="ewce_expression_data.html">ewce_expression_data</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    sct_data <span class="op">=</span> <span class="va">ctd</span>,</span></span>
<span class="r-in"><span>    tt <span class="op">=</span> <span class="va">tt_alzh_BA36</span>,</span></span>
<span class="r-in"><span>    thresh <span class="op">=</span> <span class="va">thresh</span>,</span></span>
<span class="r-in"><span>    annotLevel <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>    reps <span class="op">=</span> <span class="va">reps</span>,</span></span>
<span class="r-in"><span>    ttSpecies <span class="op">=</span> <span class="st">"human"</span>,</span></span>
<span class="r-in"><span>    sctSpecies <span class="op">=</span> <span class="st">"mouse"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning: genelistSpecies not provided. Setting to 'human' by default.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning: sctSpecies_origin not provided. Setting to 'mouse' by default.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning: sctSpecies_origin not provided. Setting to 'mouse' by default.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> character format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to data.frame</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 15,259 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting mouse ==&gt; human orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 13,416 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 13,416 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 46 genes that have multiple input_gene per ortholog_gene (many:1).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 56 genes that have multiple ortholog_gene per input_gene (1:many).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning gene_map as dictionary</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    2,016 / 15,259 (13%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    13,243 / 15,259 (87%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Generating gene background for mouse x human ==&gt; human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering ortholog reports.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 21,207 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 21,207 genes from mouse.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.frame format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from Gene.Symbol.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 21,207 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting mouse ==&gt; human orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17,355 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17,355 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 131 genes that have multiple input_gene per ortholog_gene (many:1).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 498 genes that have multiple ortholog_gene per input_gene (1:many).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    4,725 / 21,207 (22%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    16,482 / 21,207 (78%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering ortholog reports.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,482 intersect background genes used.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning 19,129 unique genes from entire human genome.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using intersect between background gene lists: 16,482 genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising sct_data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using 1st column of tt as gene column: HGNC.symbol</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 core(s) assigned as workers (3 reserved).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising CellTypeDataset</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking gene list inputs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running without gene size control.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 6 hit gene(s) remain after filtering.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing gene scores.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using previously sampled genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing gene counts.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for enrichment in 7 cell types...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sorting results by p-value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing BH-corrected q-values.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2 significant cell type enrichment results @ q&lt;0.05 : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>        CellType annotLevel p fold_change sd_from_mean q</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 pyramidal_CA1          1 0    1.411635     2.789677 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2  pyramidal_SS          1 0    1.192536     1.785915 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 core(s) assigned as workers (3 reserved).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising CellTypeDataset</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking gene list inputs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running without gene size control.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 5 hit gene(s) remain after filtering.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing gene scores.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using previously sampled genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing gene counts.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for enrichment in 7 cell types...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sorting results by p-value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing BH-corrected q-values.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3 significant cell type enrichment results @ q&lt;0.05 : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>               CellType annotLevel p fold_change sd_from_mean q</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1            microglia          1 0    2.841819     6.454695 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2         pyramidal_SS          1 0    1.455683     4.158702 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3 astrocytes_ependymal          1 0    1.530297     1.227513 0</span>
<span class="r-in"><span><span class="va">tt_results_44</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="ewce_expression_data.html">ewce_expression_data</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    sct_data <span class="op">=</span> <span class="va">ctd</span>,</span></span>
<span class="r-in"><span>    tt <span class="op">=</span> <span class="va">tt_alzh_BA44</span>,</span></span>
<span class="r-in"><span>    thresh <span class="op">=</span> <span class="va">thresh</span>,</span></span>
<span class="r-in"><span>    annotLevel <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>    reps <span class="op">=</span> <span class="va">reps</span>,</span></span>
<span class="r-in"><span>    ttSpecies <span class="op">=</span> <span class="st">"human"</span>,</span></span>
<span class="r-in"><span>    sctSpecies <span class="op">=</span> <span class="st">"mouse"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning: genelistSpecies not provided. Setting to 'human' by default.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning: sctSpecies_origin not provided. Setting to 'mouse' by default.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning: sctSpecies_origin not provided. Setting to 'mouse' by default.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> character format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting to data.frame</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 15,259 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting mouse ==&gt; human orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 13,416 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 13,416 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 46 genes that have multiple input_gene per ortholog_gene (many:1).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 56 genes that have multiple ortholog_gene per input_gene (1:many).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning gene_map as dictionary</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    2,016 / 15,259 (13%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    13,243 / 15,259 (87%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Generating gene background for mouse x human ==&gt; human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering ortholog reports.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 21,207 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 21,207 genes from mouse.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.frame format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from Gene.Symbol.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 21,207 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting mouse ==&gt; human orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17,355 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 17,355 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 131 genes that have multiple input_gene per ortholog_gene (many:1).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 498 genes that have multiple ortholog_gene per input_gene (1:many).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    4,725 / 21,207 (22%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    16,482 / 21,207 (78%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering ortholog reports.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,482 intersect background genes used.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning 19,129 unique genes from entire human genome.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using intersect between background gene lists: 16,482 genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising sct_data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using 1st column of tt as gene column: HGNC.symbol</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 core(s) assigned as workers (3 reserved).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising CellTypeDataset</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking gene list inputs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running without gene size control.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 6 hit gene(s) remain after filtering.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing gene scores.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using previously sampled genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing gene counts.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for enrichment in 7 cell types...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sorting results by p-value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing BH-corrected q-values.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3 significant cell type enrichment results @ q&lt;0.05 : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>               CellType annotLevel p fold_change sd_from_mean q</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1    endothelial_mural          1 0    2.002147     5.407831 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2     oligodendrocytes          1 0    2.010357     4.419611 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3 astrocytes_ependymal          1 0    1.156367     3.094029 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 core(s) assigned as workers (3 reserved).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising CellTypeDataset</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking gene list inputs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running without gene size control.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 5 hit gene(s) remain after filtering.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing gene scores.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using previously sampled genes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing gene counts.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Testing for enrichment in 7 cell types...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sorting results by p-value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Computing BH-corrected q-values.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2 significant cell type enrichment results @ q&lt;0.05 : </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>               CellType annotLevel p fold_change sd_from_mean q</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 astrocytes_ependymal          1 0    1.803066     2.350458 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2        pyramidal_CA1          1 0    1.493630     1.528110 0</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Fill a list with the results</span></span></span>
<span class="r-in"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="add_res_to_merging_list.html">add_res_to_merging_list</a></span><span class="op">(</span><span class="va">tt_results_36</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="add_res_to_merging_list.html">add_res_to_merging_list</a></span><span class="op">(</span><span class="va">tt_results_44</span>, <span class="va">results</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Perform the merged analysis</span></span></span>
<span class="r-in"><span><span class="co"># For publication reps should be higher</span></span></span>
<span class="r-in"><span><span class="va">merged_res</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu">merged_ewce</span><span class="op">(</span></span></span>
<span class="r-in"><span>    results <span class="op">=</span> <span class="va">results</span>,</span></span>
<span class="r-in"><span>    reps <span class="op">=</span> <span class="fl">2</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">merged_res</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                   CellType       p        fc sd_from_mean</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> astrocytes_ependymal  astrocytes_ependymal 0.66130 0.8866746   -0.5885042</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> endothelial_mural        endothelial_mural 0.33110 1.1171572    0.8496895</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> interneurons                  interneurons 0.78115 0.7658062   -0.9465746</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> microglia                        microglia 0.66555 0.8459999   -0.6892282</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> oligodendrocytes          oligodendrocytes 0.00000 1.7426915    2.4818484</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pyramidal_CA1                pyramidal_CA1 0.44395 1.0142639    0.2497222</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pyramidal_SS                  pyramidal_SS 1.00000 0.9174883   -1.4337207</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> astrocytes_ependymal1 astrocytes_ependymal 0.00000 1.6844598    3.1154348</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> endothelial_mural1       endothelial_mural 1.00000 0.5559731   -1.8899465</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> interneurons1                 interneurons 0.77590 0.7063486   -0.8681162</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> microglia1                       microglia 0.00000 1.4252970    1.4699207</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> oligodendrocytes1         oligodendrocytes 0.22070 1.1261607    0.5880097</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pyramidal_CA11               pyramidal_CA1 0.11095 1.1953337    1.6270440</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pyramidal_SS1                 pyramidal_SS 0.00000 1.2607602    3.0064648</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       Direction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> astrocytes_ependymal         Up</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> endothelial_mural            Up</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> interneurons                 Up</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> microglia                    Up</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> oligodendrocytes             Up</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pyramidal_CA1                Up</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pyramidal_SS                 Up</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> astrocytes_ependymal1      Down</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> endothelial_mural1         Down</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> interneurons1              Down</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> microglia1                 Down</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> oligodendrocytes1          Down</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pyramidal_CA11             Down</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pyramidal_SS1              Down</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer></div>

  


  

  </body></html>

