<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Creating CellTypeDatasets • EWCE</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Creating CellTypeDatasets">
<meta property="og:description" content="EWCE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/conditional.html">Conditional enrichment examples</a>
    </li>
    <li>
      <a href="../articles/create_CTD.html">Creating CellTypeDatasets</a>
    </li>
    <li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
    <li>
      <a href="../articles/transcriptomes.html">Applications to transcriptomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/NathanSkene/EWCE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="create_CTD_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Creating CellTypeDatasets</h1>
                        <h4 class="author"><h4>
Alan Murphy, Brian Schilder, and Nathan Skene
</h4></h4>
            
            <h4 class="date"><h4>
Oct-22-2021
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/NathanSkene/EWCE/blob/master/vignettes/create_CTD.Rmd"><code>vignettes/create_CTD.Rmd</code></a></small>
      <div class="hidden name"><code>create_CTD.Rmd</code></div>

    </div>

    
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><code>EWCE</code> was originally designed to work with the single-cell cortical transcriptome data from the <a href="http://linnarssonlab.org/">Linnarsson Lab</a><span class="citation"><sup>1</sup></span> (available <a href="http://linnarssonlab.org/cortex/">here</a>).</p>
<p>Using this package it is possible to read in any single-cell transcriptome data, provided that you have a cell by gene expression matrix (with each cell as a separate column) and a separate annotation dataframe, with a row for each cell.</p>
<p>The <code>EWCE</code> process involves testing for whether the genes in a target list have higher levels of expression in a given cell type than can reasonably be expected by chance. The probability distribution for this is estimated by randomly generating gene lists of equal length from a set of background genes.</p>
<p><code>EWCE</code> can be applied to any gene list. In the original paper we reported its application to genetic and transcriptomic datasets..<span class="citation"><sup>2</sup></span></p>
<p>Note that throughout this vignette we use the terms ‘cell type’ and ‘cell subtype’ to refer to two levels of annotation of what a cell type is. This is described in further detail in our paper<span class="citation"><sup>2</sup></span>, but relates to the two levels of annotation provided in the Linnarsson dataset.<span class="citation"><sup>1</sup></span> In this dataset a cell is described as having a cell type (i.e. ‘Interneuron’) and subcell type (i.e. ‘Int11’ a.k.a Interneuron type 11).</p>
</div>
<div id="setup" class="section level1">
<h1 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/NathanSkene/EWCE">EWCE</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span></code></pre></div>
</div>
<div id="creating-a-celltypedataset" class="section level1">
<h1 class="hasAnchor">
<a href="#creating-a-celltypedataset" class="anchor"></a>Creating a CellTypeDataset</h1>
<p>Here we provide an example of converting a scRNA-seq dataset (<a href="https://www.science.org/doi/10.1126/science.aaa1934">Zeisel 2015</a>) into a CellTypeDataset (CTD) so it can used with <em>EWCE</em>.</p>
<div id="loading-datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-datasets" class="anchor"></a>Loading datasets</h2>
<p>The first step for all analyses is to load the single-cell transcriptome (SCT) data. For the purposes of this example we will use the dataset described in: &gt; <a href="https://www.science.org/doi/10.1126/science.aaa1934">Cell types in the mouse cortex and hippocampus revealed by single-cell RNA-seq</a>, Science, 2015</p>
<p>This is stored and can be loaded as follows from <code>ewceData</code> package.</p>
<p><em>EWCE</em> can generate CTD from two different data input types: 1. A gene x cell expression matrix (can be dense matrix, sparse matrix, data.frame, or DelayedArray).<br>
2. A <code>SingleCellExperiment</code> (SCE) or <code>SummarizedExperiment</code> object.</p>
<p>To check the data, we can quickly plot the distribution of expression of a given gene across all the cell types.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># example datasets held in ewceData package</span>
<span class="va">cortex_mrna</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/cortex_mrna.html">cortex_mrna</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">gene</span> <span class="op">&lt;-</span> <span class="st">"Necab1"</span>
<span class="va">cellExpDist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>Expression<span class="op">=</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">[</span><span class="va">gene</span>,<span class="op">]</span>,
                          Celltype<span class="op">=</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span><span class="op">[</span>
                            <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span>,<span class="op">]</span><span class="op">$</span><span class="va">level1class</span><span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">cellExpDist</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Celltype</span>, y<span class="op">=</span><span class="va">Expression</span><span class="op">)</span>, outlier.alpha <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="create_CTD_files/figure-html/unnamed-chunk-1-1.png" width="700"></p>
<p>To note for the analysis of the <code>cortex_mrna</code> dataset, I will proved the alternative for an SCE object under each segment of code if there is any difference.</p>
</div>
<div id="converting-single-cell-formats" class="section level2">
<h2 class="hasAnchor">
<a href="#converting-single-cell-formats" class="anchor"></a>Converting single-cell formats</h2>
<p>If you would like to try running the functions on an SCE object instead, we can convert <code>cortex_mrna</code> with the package <a href="https://github.com/bschilder/scKirby"><code>scKirby</code></a>. <code>scKirby</code> automatically infers the input file type and converts it to SCE format by default.</p>
<pre><code>if (!"SingleCellExperiment" %in% rownames(installed.packages())){BiocManager::install("SingleCellExperiment")}
library(SingleCellExperiment) 
if (!"scKirby" %in% rownames(installed.packages())){remotes::install_github("bschilder/scKirby")}
library(scKirby)

# Make SCE object
cortex_mrna &lt;- ewceData::cortex_mrna()
cortex_mrna_sce &lt;- scKirby::ingest_data(cortex_mrna, save_output = FALSE)

# Now to plot the SCE dataset:
gene="Necab1"
cellExpDist_sce = data.frame(e=assays(cortex_mrna_sce)[[1]][gene,],
                             l1=colData(cortex_mrna_sce)[
                               colnames(assays(cortex_mrna_sce)[[1]]),
                             ]$level1class)
                             
ggplot(cellExpDist_sce) + geom_boxplot(aes(x=l1,y=e)) + xlab("Cell type") + 
  ylab("Unique Molecule Count") + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) </code></pre>
<div id="correcting-gene-symbols" class="section level3">
<h3 class="hasAnchor">
<a href="#correcting-gene-symbols" class="anchor"></a>Correcting gene symbols</h3>
<p>It is very common for publicly available transcriptome datasets to use incorrect gene symbols (often some gene names will have been mangled by opening in Excel).</p>
<p>A function is provided to correct these where out of date aliases were used and give a warning if appears possible that excel has mangled some of the gene names. This function can be used with a downloaded reference file like the <a href="%22http://www.informatics.jax.org/downloads/reports/MRK_List2.rpt%22">MRK_List2 file</a> from MGI which lists known synonyms for MGI gene symbols. If no file path is passed, there is a loaded reference dataset of the MRK_List2 which will be used: <code><a href="https://rdrr.io/pkg/ewceData/man/mgi_synonym_data.html">mgi_synonym_data()</a></code>. We recommend running this function on all input datasets.</p>
<pre><code>cortex_mrna$exp = fix_bad_mgi_symbols(cortex_mrna$exp)</code></pre>
<p>For the SCE version, pass the whole SCE object</p>
<pre><code>#Note that data in SCE object must be in 'counts' assay
#The fix_bad_hgnc_symbols function can similarly take an SCE object as an input
cortex_mrna_sce = fix_bad_mgi_symbols(cortex_mrna_sce)</code></pre>
</div>
</div>
<div id="drop-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#drop-genes" class="anchor"></a>Drop genes</h2>
<p>The vignette takes a while to go through if you use all genes. So let’s restrict to 1000 random genes.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">nKeep</span> <span class="op">&lt;-</span> <span class="fl">1000</span>
<span class="va">must_keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span> 
<span class="va">keep_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">must_keep</span>,<span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span>,<span class="fl">997</span><span class="op">)</span><span class="op">)</span>
<span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span> <span class="op">&lt;-</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">[</span><span class="va">keep_genes</span>,<span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 1000 3005</code></pre>
<pre><code>#SCE
nKeep = 1000
must_keep = c("Apoe","Gfap","Gapdh") 
keep_genes = c(must_keep,sample(names(rowRanges(cortex_mrna_sce)),997))
cortex_mrna_sce = cortex_mrna_sce[keep_genes,]</code></pre>
</div>
<div id="normalization" class="section level2">
<h2 class="hasAnchor">
<a href="#normalization" class="anchor"></a>Normalization</h2>
<p>A different number of reads is found across each cell. We suggest using <a href="https://cran.r-project.org/web/packages/sctransform/index.html"><code>scTransform</code></a> to normalise for differences due to cell size, then linearly scale. Note that this might be slow.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp_scT_normed</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/sct_normalize.html">sct_normalize</a></span><span class="op">(</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span> </code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%</code></pre>
<pre><code>## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached</code></pre>
<pre><code>## 
  |                                                                            
  |===================================                                   |  50%</code></pre>
<pre><code>## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached

## Warning in theta.ml(y = y, mu = fit$fitted): iteration limit reached</code></pre>
<pre><code>## 
  |                                                                            
  |======================================================================| 100%
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |======================================================================| 100%
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># SCE</span>
<span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span><span class="op">$</span><span class="va">normcounts</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/sct_normalize.html">sct_normalize</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span></code></pre></div>
<p>Note that this is optional (and was not used in the original EWCE publication) so by all means ignore this <code>scTransform</code> step.</p>
</div>
<div id="calculate-specificity-matrices" class="section level2">
<h2 class="hasAnchor">
<a href="#calculate-specificity-matrices" class="anchor"></a>Calculate specificity matrices</h2>
<p>The next steps are as follows:</p>
<ol style="list-style-type: decimal">
<li>
<code><a href="../reference/drop_uninformative_genes.html">drop_uninformative_genes()</a></code>: Drop genes which do not show significant evidence of varying between level 2 cell types (based on ANOVA). Optionally, you can also drop all genes which do not have 1:1 mouse:human orthologues (<strong>Note</strong>: this used to a separate step but has since been integrated directly into <code><a href="../reference/drop_uninformative_genes.html">drop_uninformative_genes()</a></code> via the arguments <code>drop_nonhuman_genes=</code> and <code>input_species=</code>).</li>
<li>Calculate cell type averages and specificity for each gene.</li>
</ol>
<p>The last step is only necessary if you plan to compare to human data, i.e. gene sets resulting from human genetics.</p>
<p>Rather than returning the data directly, the functions save the calculated data to a file and the file name is returned.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Generate cell type data for just the cortex/hippocampus data  </span>
<span class="va">exp_CortexOnly_DROPPED</span>  <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/drop_uninformative_genes.html">drop_uninformative_genes</a></span><span class="op">(</span>
  exp <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp_scT_normed</span>, 
  input_species <span class="op">=</span> <span class="st">"mouse"</span>,
  output_species <span class="op">=</span> <span class="st">"human"</span>,
  level2annot <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span> </code></pre></div>
<pre><code>## Time difference of 0.2545247 secs</code></pre>
</div>
<div id="generate-celltypedataset" class="section level2">
<h2 class="hasAnchor">
<a href="#generate-celltypedataset" class="anchor"></a>Generate CellTypeDataset</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">annotLevels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>level1class<span class="op">=</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level1class</span>,
                    level2class<span class="op">=</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span>

<span class="va">fNames_CortexOnly</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/generate_celltype_data.html">generate_celltype_data</a></span><span class="op">(</span>
  exp <span class="op">=</span> <span class="va">exp_CortexOnly_DROPPED</span>,
  annotLevels <span class="op">=</span> <span class="va">annotLevels</span>,
  groupName <span class="op">=</span> <span class="st">"kiCortexOnly"</span>, 
  savePath <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> 

<span class="va">ctd_CortexOnly</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/load_rdata.html">load_rdata</a></span><span class="op">(</span><span class="va">fNames_CortexOnly</span><span class="op">)</span></code></pre></div>
<pre><code>#SCE
# Generate cell type data for just the cortex/hippocampus data
exp_CortexOnly_DROPPED &lt;- drop_uninformative_genes(exp=cortex_mrna_sce,
                                                   drop_nonhuman_genes = T,
                                                   input_species = "mouse",
                                                   level2annot=cortex_mrna_sce$level2class)

annotLevels = list(level1class=cortex_mrna_sce$level1class,
                   level2class=cortex_mrna_sce$level2class)

fNames_CortexOnly &lt;- generate_celltype_data(exp=exp_CortexOnly_DROPPED,
                                           annotLevels=annotLevels,
                                           groupName="kiCortexOnly",
                                           savePath=tempdir()) </code></pre>
<p>To note on the SCE approach, these are the only differences for handling the analysis of an SCE object. All downstream analysis on <code>cortex_mrna</code> would be the same. The same approach can be used for other datasets in the vignette if converted such as <code>hypothalamus_mrna</code> or your own SCE object.</p>
</div>
</div>
<div id="merging-two-single-cell-datasets" class="section level1">
<h1 class="hasAnchor">
<a href="#merging-two-single-cell-datasets" class="anchor"></a>Merging two single-cell datasets</h1>
<p>Often it is useful to merge two single-cell datasets. For instance, there are separate files for the cortex and hypothalamus datasets generated by the Karolinska institute. A subset of the resulting merged dataset is loaded using <code><a href="https://rdrr.io/pkg/ewceData/man/ctd.html">ctd()</a></code> but it is worth understanding the approach (outlined below) if you want to repeat this for other single-cell datasets.</p>
<p>The dataset is available from the <a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE74nnn/GSE74672/suppl/GSE74672_expressed_mols_with_classes.xlsx.gz">GEO</a> but is also available in the <code>ewceData</code> package: <code><a href="https://rdrr.io/pkg/ewceData/man/hypothalamus_mrna.html">hypothalamus_mrna()</a></code>. See the <code>ewceData</code> for details of how this file was preprocessed after first being downloaded from GEO, unzipped and read into R.</p>
<p>We can now merge the hypothalamus dataset with the cortex dataset and then calculate specificity:</p>
<div id="load-hypothalamus-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#load-hypothalamus-dataset" class="anchor"></a>Load hypothalamus dataset</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hypothalamus_mrna</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/hypothalamus_mrna.html">hypothalamus_mrna</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div id="fix-bad-mgi-symbols" class="section level2">
<h2 class="hasAnchor">
<a href="#fix-bad-mgi-symbols" class="anchor"></a>Fix bad MGI symbols</h2>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hypothalamus_mrna</span><span class="op">$</span><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/fix_bad_mgi_symbols.html">fix_bad_mgi_symbols</a></span><span class="op">(</span>exp <span class="op">=</span> <span class="va">hypothalamus_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span>
<span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/fix_bad_mgi_symbols.html">fix_bad_mgi_symbols</a></span><span class="op">(</span>exp <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span></code></pre></div>
</div>
<div id="merge-ctd" class="section level2">
<h2 class="hasAnchor">
<a href="#merge-ctd" class="anchor"></a>Merge CTD</h2>
<p>Merge the CTDs - again this can be run with SCE or other ranged SE objects as input.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">merged_KI</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/merge_two_expfiles.html">merge_two_expfiles</a></span><span class="op">(</span>exp1 <span class="op">=</span> <span class="va">hypothalamus_mrna</span><span class="op">$</span><span class="va">exp</span>, 
                                      exp2 <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span>,
                                      annot1 <span class="op">=</span> <span class="va">hypothalamus_mrna</span><span class="op">$</span><span class="va">annot</span>, 
                                      annot2 <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span>,
                                      name1 <span class="op">=</span> <span class="st">"Hypothalamus (KI)"</span>, 
                                      name2 <span class="op">=</span> <span class="st">"Cortex/Hippo (KI)"</span><span class="op">)</span></code></pre></div>
</div>
<div id="drop-uninformative-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#drop-uninformative-genes" class="anchor"></a>Drop uninformative genes</h2>
<p>Drop genes which don’t vary significantly between cell types.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp_merged_DROPPED</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/drop_uninformative_genes.html">drop_uninformative_genes</a></span><span class="op">(</span>
  exp <span class="op">=</span> <span class="va">merged_KI</span><span class="op">$</span><span class="va">exp</span>, 
  drop_nonhuman_genes <span class="op">=</span> <span class="cn">TRUE</span>, 
  input_species <span class="op">=</span> <span class="st">"mouse"</span>,
  level2annot <span class="op">=</span> <span class="va">merged_KI</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span></code></pre></div>
<pre><code>## Time difference of 0.6415126 secs</code></pre>
</div>
<div id="generate-celltypedataset-1" class="section level2">
<h2 class="hasAnchor">
<a href="#generate-celltypedataset-1" class="anchor"></a>Generate CellTypeDataset</h2>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">annotLevels</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>level1class<span class="op">=</span><span class="va">merged_KI</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level1class</span>,
                   level2class<span class="op">=</span><span class="va">merged_KI</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span>

<span class="co"># Update file path to where you want the cell type data files to save on disk</span>
<span class="va">fNames_MergedKI</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/generate_celltype_data.html">generate_celltype_data</a></span><span class="op">(</span>exp <span class="op">=</span> <span class="va">exp_merged_DROPPED</span>,
                                                annotLevels <span class="op">=</span> <span class="va">annotLevels</span>,
                                                groupName <span class="op">=</span> <span class="st">"MergedKI"</span>,
                                                savePath <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> 
<span class="va">ctd_MergedKI</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_rdata.html">load_rdata</a></span><span class="op">(</span><span class="va">fNames_MergedKI</span><span class="op">)</span> </code></pre></div>
</div>
<div id="understanding-specificity-matrices" class="section level2">
<h2 class="hasAnchor">
<a href="#understanding-specificity-matrices" class="anchor"></a>Understanding specificity matrices</h2>
<p>While not required for further analyses it helps to understand what the outputs of this function are.</p>
<p>Note firstly, that it is a list such that <code>ctd[[1]]</code> contains data relating to level 1 annotations and <code>ctd[[2]]</code> relates to level 2 annotations.</p>
<p>Using the <code>ggplot2</code> package to visualise the data, let us examine the expression of a few genes. If you have not already done so you will need to first install the ggplot2 package with <code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages("ggplot2")</a></code>.</p>
<p>For this example we use a subset of the genes from the merged dataset generated above, which is accessed using <code><a href="https://rdrr.io/pkg/ewceData/man/ctd.html">ewceData::ctd()</a></code>. We recommend that you use the code above to regenerate this though and drop the first command from the below section.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Load merged cortex and hypothalamus dataset generated by Karolinska institute</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html">ctd</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># i.e. ctd_MergedKI</span>

<span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_ctd.html">plot_ctd</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,
                      level <span class="op">=</span> <span class="fl">1</span>,
                      genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,
                      metric <span class="op">=</span> <span class="st">"mean_exp"</span><span class="op">)</span></code></pre></div>
<p><img src="create_CTD_files/figure-html/unnamed-chunk-11-1.png" width="672"></p>
<p>This graph shows the average expression of three genes: <em>Apoe, Gfap</em> and <em>Gapdh</em>. While there are substantial differences in which cell types express these genes, the dominant effect seen here is the overall expression level of the data. For the purposes of this analysis though, we are not interested in overall expression level and only wish to know about the proportion of a genes expression which is found in a particular cell type. We can study this instead using the following code which examines the data frame <code>ctd[[1]]$specificity</code>:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_ctd.html">plot_ctd</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,
                      level <span class="op">=</span> <span class="fl">1</span>,
                      genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,
                      metric <span class="op">=</span> <span class="st">"specificity"</span><span class="op">)</span></code></pre></div>
<p><img src="create_CTD_files/figure-html/unnamed-chunk-12-1.png" width="672"></p>
<p>We can now see in this graph that <em>Gfap</em> is the most specific to a cell type (Type 1 Astrocytes) of any of those three genes, with over 60% of its expression found in that cell type.</p>
<p>It can also be seen that the majority of expression of Gapdh is in neurons but because their are a greater number of neuronal subtypes, the total expression proportion appears lower. We can examine expression across level 2 cell type level annotations by looking at <code>ctd[[2]]$specificity</code>:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_ctd.html">plot_ctd</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,
                      level <span class="op">=</span> <span class="fl">2</span>,
                      genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,
                      metric <span class="op">=</span> <span class="st">"specificity"</span><span class="op">)</span></code></pre></div>
<p><img src="create_CTD_files/figure-html/unnamed-chunk-13-1.png" width="672"></p>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<details><div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.1 (2021-08-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] ewceData_1.1.0      ExperimentHub_2.1.4 AnnotationHub_3.1.6
##  [4] BiocFileCache_2.1.1 dbplyr_2.1.1        BiocGenerics_0.39.2
##  [7] ggplot2_3.3.5       EWCE_2.0.0          RNOmni_1.0.0       
## [10] BiocStyle_2.21.4   
## 
## loaded via a namespace (and not attached):
##   [1] readxl_1.3.1                  backports_1.2.1              
##   [3] systemfonts_1.0.3             plyr_1.8.6                   
##   [5] lazyeval_0.2.2                orthogene_0.99.8             
##   [7] BiocParallel_1.27.17          listenv_0.8.0                
##   [9] GenomeInfoDb_1.29.10          digest_0.6.28                
##  [11] htmltools_0.5.2               fansi_0.5.0                  
##  [13] magrittr_2.0.1                memoise_2.0.0                
##  [15] openxlsx_4.2.4                limma_3.49.4                 
##  [17] globals_0.14.0                Biostrings_2.61.2            
##  [19] matrixStats_0.61.0            pkgdown_1.6.1                
##  [21] colorspace_2.0-2              blob_1.2.2                   
##  [23] rappdirs_0.3.3                textshaping_0.3.6            
##  [25] haven_2.4.3                   xfun_0.27                    
##  [27] dplyr_1.0.7                   crayon_1.4.1                 
##  [29] RCurl_1.98-1.5                jsonlite_1.7.2               
##  [31] glue_1.4.2                    gtable_0.3.0                 
##  [33] zlibbioc_1.39.0               XVector_0.33.0               
##  [35] HGNChelper_0.8.1              DelayedArray_0.19.4          
##  [37] car_3.0-11                    future.apply_1.8.1           
##  [39] SingleCellExperiment_1.15.2   abind_1.4-5                  
##  [41] scales_1.1.1                  DBI_1.1.1                    
##  [43] rstatix_0.7.0                 Rcpp_1.0.7                   
##  [45] viridisLite_0.4.0             xtable_1.8-4                 
##  [47] foreign_0.8-81                bit_4.0.4                    
##  [49] stats4_4.1.1                  htmlwidgets_1.5.4            
##  [51] httr_1.4.2                    ellipsis_0.3.2               
##  [53] farver_2.1.0                  pkgconfig_2.0.3              
##  [55] sass_0.4.0                    utf8_1.2.2                   
##  [57] labeling_0.4.2                tidyselect_1.1.1             
##  [59] rlang_0.4.12                  reshape2_1.4.4               
##  [61] later_1.3.0                   AnnotationDbi_1.55.2         
##  [63] munsell_0.5.0                 BiocVersion_3.14.0           
##  [65] cellranger_1.1.0              tools_4.1.1                  
##  [67] cachem_1.0.6                  generics_0.1.0               
##  [69] RSQLite_2.2.8                 broom_0.7.9                  
##  [71] ggdendro_0.1.22               evaluate_0.14                
##  [73] stringr_1.4.0                 fastmap_1.1.0                
##  [75] yaml_2.2.1                    ragg_1.1.3                   
##  [77] babelgene_21.4                knitr_1.36                   
##  [79] bit64_4.0.5                   fs_1.5.0                     
##  [81] zip_2.2.0                     purrr_0.3.4                  
##  [83] KEGGREST_1.33.0               gprofiler2_0.2.1             
##  [85] future_1.22.1                 mime_0.12                    
##  [87] compiler_4.1.1                plotly_4.10.0                
##  [89] filelock_1.0.2                curl_4.3.2                   
##  [91] png_0.1-7                     interactiveDisplayBase_1.31.2
##  [93] ggsignif_0.6.3                tibble_3.1.5                 
##  [95] bslib_0.3.1                   homologene_1.4.68.19.3.27    
##  [97] stringi_1.7.5                 highr_0.9                    
##  [99] desc_1.4.0                    forcats_0.5.1                
## [101] lattice_0.20-45               Matrix_1.3-4                 
## [103] vctrs_0.3.8                   pillar_1.6.4                 
## [105] lifecycle_1.0.1               BiocManager_1.30.16          
## [107] jquerylib_0.1.4               data.table_1.14.2            
## [109] bitops_1.0-7                  httpuv_1.6.3                 
## [111] patchwork_1.1.1               GenomicRanges_1.45.0         
## [113] R6_2.5.1                      bookdown_0.24                
## [115] promises_1.2.0.1              gridExtra_2.3                
## [117] rio_0.5.27                    IRanges_2.27.2               
## [119] parallelly_1.28.1             codetools_0.2-18             
## [121] MASS_7.3-54                   assertthat_0.2.1             
## [123] SummarizedExperiment_1.23.5   rprojroot_2.0.2              
## [125] withr_2.4.2                   sctransform_0.3.2            
## [127] S4Vectors_0.31.5              GenomeInfoDbData_1.2.7       
## [129] parallel_4.1.1                hms_1.1.1                    
## [131] grid_4.1.1                    tidyr_1.1.4                  
## [133] rmarkdown_2.11                MatrixGenerics_1.5.4         
## [135] carData_3.0-4                 ggpubr_0.4.0                 
## [137] Biobase_2.53.0                shiny_1.7.1</code></pre>
</details>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-zeisel2015cell" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Zeisel, A. <em>et al.</em> Cell types in the mouse cortex and hippocampus revealed by single-cell RNA-seq. <em>Science</em> <strong>347,</strong> 1138–1142 (2015).</div>
</div>
<div id="ref-skene_2016" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Skene, N. &amp; Grant, S. Identification of vulnerable cell types in major brain disorders using single cell transcriptomes and expression weighted cell type enrichment. <em>Frontiers in Neuroscience</em> (2016). doi:<a href="https://doi.org/10.3389/fnins.2016.00016">10.3389/fnins.2016.00016</a>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
