<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Extended examples • EWCE</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Extended examples">
<meta property="og:description" content="EWCE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/conditional.html">Conditional enrichment examples</a>
    </li>
    <li>
      <a href="../articles/create_CTD.html">Creating CellTypeDatasets</a>
    </li>
    <li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
    <li>
      <a href="../articles/transcriptomes.html">Applications to transcriptomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/NathanSkene/EWCE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="extended_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Extended examples</h1>
                        <h4 class="author"><h4>
Alan Murphy, Brian Schilder, and Nathan Skene
</h4></h4>
            
            <h4 class="date"><h4>
Oct-22-2021
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/NathanSkene/EWCE/blob/master/vignettes/extended.Rmd"><code>vignettes/extended.Rmd</code></a></small>
      <div class="hidden name"><code>extended.Rmd</code></div>

    </div>

    
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="setup" class="section level1">
<h1 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/NathanSkene/EWCE">EWCE</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span></code></pre></div>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>In the following vignette, we provide more a more in-depth version of the examples provided in the <a href="https://nathanskene.github.io/EWCE/articles/EWCE.html">Getting started vignette</a>.</p>
</div>
<div id="prepare-input-data" class="section level1">
<h1 class="hasAnchor">
<a href="#prepare-input-data" class="anchor"></a>Prepare input data</h1>
<div id="celltypedataset" class="section level2">
<h2 class="hasAnchor">
<a href="#celltypedataset" class="anchor"></a>CellTypeDataset</h2>
<p>For this example we use a subset of the genes from the merged dataset generated in the <a href="https://nathanskene.github.io/EWCE/articles/creating_CTD.html">creating CTD vignette</a>, which is accessed using <code><a href="https://rdrr.io/pkg/ewceData/man/ctd.html">ewceData::ctd()</a></code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Load merged cortex and hypothalamus dataset generated by Karolinska institute</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html">ctd</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># i.e. ctd_MergedKI</span>

<span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_ctd.html">plot_ctd</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,
                      level <span class="op">=</span> <span class="fl">1</span>,
                      genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,
                      metric <span class="op">=</span> <span class="st">"mean_exp"</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-1-1.png" width="672"></p>
</div>
<div id="gene-lists" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-lists" class="anchor"></a>Gene lists</h2>
<p>For the first demonstration of EWCE we will test for whether genes that are genetically associated with Alzheimer’s disease are enriched in any particular cell type.</p>
<p>This example gene list is stored within the <code>ewceData</code> package:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/example_genelist.html">example_genelist</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "APOE"     "BIN1"     "CLU"      "ABCA7"    "CR1"      "PICALM"  
##  [7] "MS4A6A"   "CD33"     "MS4A4E"   "CD2AP"    "EOGA1"    "INPP5D"  
## [13] "MEF2C"    "HLA-DRB5" "ZCWPW1"   "NME8"     "PTK2B"    "CELF1"   
## [19] "SORL1"    "FERMT2"   "SLC24A4"  "CASS4"</code></pre>
<p>All gene IDs are assumed by the package to be provided in gene symbol format (rather than Ensembl/Entrez). Symbols can be provided as any species-specific gene symbols supported by the package <code>orthogene</code>, though the <code>genelistSpecies</code> argument will need to be set appropriately.</p>
<p>Likewise, the single-cell dataset can be from any species, but the <code>sctSpecies</code> argument must be set accordingly.</p>
<p>The example gene list here stores the human genes associated with human disease, and hence are HGNC symbols.</p>
<p>The next step is to determine the most suitable background set. This can be user-supplied, but by default the background is all 1:1 ortholog genes shared by <code>genelistSpecies</code> and <code>sctSpecies</code> that are also present in <code>sct_data</code>.</p>
</div>
<div id="setting-analysis-parameters" class="section level2">
<h2 class="hasAnchor">
<a href="#setting-analysis-parameters" class="anchor"></a>Setting analysis parameters</h2>
<p>We now need to set the parameters for the analysis. For a publishable analysis we would want to generate over 10,000 random lists and determine their expression levels, but for computational speed let us only use <code>reps=100</code>. We want to analyse level 1 annotations so set level to 1.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Use 100 bootstrap lists for speed, for publishable analysis use &gt;=10000</span>
<span class="va">reps</span> <span class="op">&lt;-</span> <span class="fl">100</span> 
<span class="co"># Use level 1 annotations (i.e. Interneurons)</span>
<span class="va">annotLevel</span> <span class="op">&lt;-</span> <span class="fl">1</span> </code></pre></div>
</div>
</div>
<div id="run-cell-type-enrichment-tests" class="section level1">
<h1 class="hasAnchor">
<a href="#run-cell-type-enrichment-tests" class="anchor"></a>Run cell type enrichment tests</h1>
<div id="default-tests" class="section level2">
<h2 class="hasAnchor">
<a href="#default-tests" class="anchor"></a>Default tests</h2>
<p>We have now loaded the SCT data, prepared the gene lists and set the parameters. We run the model as follows, Note we set the seed to ensure reproducibility in the bootstrap sampling function:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Bootstrap significance test, no control for transcript length and GC content </span>
<span class="va">full_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,
                                                sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,
                                                genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,
                                                hits <span class="op">=</span> <span class="va">hits</span>, 
                                                reps <span class="op">=</span> <span class="va">reps</span>,
                                                annotLevel <span class="op">=</span> <span class="va">annotLevel</span><span class="op">)</span></code></pre></div>
<p>A note on both the background and target gene lists, other common gene list objects can be used as inputs such as <code>BiocSet::BiocSet</code> and <code>GSEABase::GeneSet</code>. Below is an example of how to format each for the target gene list (<code>hits</code>):</p>
<pre><code>if(!"BiocSet" %in% rownames(installed.packages())) {
  BiocManager::install("BiocSet")
}
if(!"GSEABase" %in% rownames(installed.packages())) {
  BiocManager::install("GSEABase")
}

library(BiocSet)
library(GSEABase)

# Save both approaches as hits which will be passed to bootstrap_enrichment_test
genes &lt;- c("Apoe","Inpp5d","Cd2ap","Nme8",
          "Cass4","Mef2c","Zcwpw1","Bin1",
          "Clu","Celf1","Abca7","Slc24a4",
          "Ptk2b","Picalm","Fermt2","Sorl1")

#BiocSet::BiocSet, BiocSet_target contains the gene list target
BiocSet_target &lt;- BiocSet::BiocSet(set1 = genes) 
hits &lt;- unlist(BiocSet::es_element(BiocSet_target))

#GSEABase::GeneSet, GeneSet_target contains the gene list target
GeneSet_target &lt;- GSEABase::GeneSet(genes)
hits &lt;- GSEABase::geneIds(GeneSet_target) </code></pre>
<p>The main table of results is stored in <code>full_results$results</code>. We can see the most significant results using:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">full_results</span><span class="op">$</span><span class="va">results</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">CellType</th>
<th align="right">annotLevel</th>
<th align="right">p</th>
<th align="right">fold_change</th>
<th align="right">sd_from_mean</th>
<th align="right">q</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">microglia</td>
<td align="left">microglia</td>
<td align="right">1</td>
<td align="right">0.00</td>
<td align="right">2.0037539</td>
<td align="right">3.8229690</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">astrocytes_ependymal</td>
<td align="left">astrocytes_ependymal</td>
<td align="right">1</td>
<td align="right">0.11</td>
<td align="right">1.3594176</td>
<td align="right">1.4291523</td>
<td align="right">0.385</td>
</tr>
<tr class="odd">
<td align="left">oligodendrocytes</td>
<td align="left">oligodendrocytes</td>
<td align="right">1</td>
<td align="right">0.78</td>
<td align="right">0.7903958</td>
<td align="right">-0.8909301</td>
<td align="right">1.000</td>
</tr>
<tr class="even">
<td align="left">endothelial-mural</td>
<td align="left">endothelial-mural</td>
<td align="right">1</td>
<td align="right">0.83</td>
<td align="right">0.7587306</td>
<td align="right">-0.9521828</td>
<td align="right">1.000</td>
</tr>
<tr class="odd">
<td align="left">pyramidal SS</td>
<td align="left">pyramidal SS</td>
<td align="right">1</td>
<td align="right">0.84</td>
<td align="right">0.8338200</td>
<td align="right">-0.9271986</td>
<td align="right">1.000</td>
</tr>
<tr class="even">
<td align="left">pyramidal CA1</td>
<td align="left">pyramidal CA1</td>
<td align="right">1</td>
<td align="right">0.90</td>
<td align="right">0.7882024</td>
<td align="right">-1.1989117</td>
<td align="right">1.000</td>
</tr>
<tr class="odd">
<td align="left">interneurons</td>
<td align="left">interneurons</td>
<td align="right">1</td>
<td align="right">1.00</td>
<td align="right">0.3868205</td>
<td align="right">-3.1123590</td>
<td align="right">1.000</td>
</tr>
</tbody>
</table>
<div id="plot-results" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-results" class="anchor"></a>Plot results</h3>
<p>The results can be visualised using another function, which shows for each cell type, the number of standard deviations from the mean the level of expression was found to be in the target gene list, relative to the bootstrapped mean:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">full_results</span><span class="op">$</span><span class="va">results</span>,
                             mtc_method <span class="op">=</span><span class="st">"BH"</span>,
                             ctd <span class="op">=</span> <span class="va">ctd</span><span class="op">)</span> <span class="co"># optional (unless you want dendrogram)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-6-1.png" width="672"></p>
<p>For publications it can be useful to plot a dendrogram alongside the plot. This can be done by including the cell type data as an additional argument. The dendrogram should automatically align with the graph ticks (thanks to Robert Gordon-Smith, Mres Molecular and Cellular Biosciences, Imperial College London for this solution):</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">withDendro</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>If you want to view the characteristics of enrichment for each gene within the list then the <code>generate_bootstrap_plots</code> function should be used. This saves the plots into the BootstrapPlots folder. This takes the results of a bootstrapping analysis so as to only generate plots for significant enrichments. The <code>listFileName</code> argument is used to give the generated graphs a particular file name. The <code>savePath</code> argument is used here to save the files to a temporary directory, this can be updated to your preferred location. The file path where it was saved is returned so the temporary directory can be located if used.</p>
<pre><code>bt_plot_location &lt;- 
  generate_bootstrap_plots(sct_data=ctd,hits=mouse.hits,
                          bg=mouse.bg,reps=reps,annotLevel=1,
                          full_results=full_results,
                          listFileName="VignetteGraphs",
                          savePath = tempdir())</code></pre>
</div>
</div>
<div id="control-for-transcript-length-and-gc-content" class="section level2">
<h2 class="hasAnchor">
<a href="#control-for-transcript-length-and-gc-content" class="anchor"></a>Control for transcript length and GC-content</h2>
<p>When analysing genes found through genetic association studies it is important to consider biases which might be introduced as a result of transcript length and GC-content. The package can control for these by selecting the bootstrap lists such that the <em>i<sup>th</sup></em> gene in the random list has properties similar to the<em>i<sup>th</sup></em> gene in the target list. To enable the algorithm to do this it needs to be passed the gene lists as HGNC symbols rather than MGI.</p>
<p>The bootstrapping function then takes different arguments:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Bootstrap significance test controlling for transcript length and GC content</span>
<span class="va">cont_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,
                                                hits <span class="op">=</span> <span class="va">hits</span>,
                                                sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,
                                                genelistSpecies <span class="op">=</span> <span class="st">"human"</span>, 
                                                reps <span class="op">=</span> <span class="va">reps</span>,
                                                annotLevel <span class="op">=</span> <span class="fl">1</span>,
                                                geneSizeControl <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div id="plot-results-1" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-results-1" class="anchor"></a>Plot results</h3>
<p>We plot these results using <code>ewce_plot</code>:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">cont_results</span><span class="op">$</span><span class="va">results</span>,
                             mtc_method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-9-1.png" width="672"></p>
<p>This shows that the controlled method generates enrichments that are generally comparable to the standard method.</p>
</div>
</div>
<div id="test-different-ctd-levels" class="section level2">
<h2 class="hasAnchor">
<a href="#test-different-ctd-levels" class="anchor"></a>Test different CTD levels</h2>
<p>Both the analyses shown above were run on level 1 annotations. It is possible to test on the level 2 cell type level annotations by changing one of the arguments.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Bootstrap significance test controlling for transcript length and GC content</span>
<span class="va">cont_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,
                                                hits <span class="op">=</span> <span class="va">hits</span>, 
                                                sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,
                                                genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,
                                                reps <span class="op">=</span> <span class="va">reps</span>,
                                                annotLevel <span class="op">=</span> <span class="fl">2</span>,
                                                geneSizeControl <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div id="plot-results-2" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-results-2" class="anchor"></a>Plot results</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">cont_results</span><span class="op">$</span><span class="va">results</span>,
                             mtc_method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>With the subcell type analysis each microglial subtype was enriched and correspondingly we see here that the microglial cell type is enriched.</p>
</div>
</div>
</div>
<div id="plot-results-from-multiple-sets-of-enrichment-results" class="section level1">
<h1 class="hasAnchor">
<a href="#plot-results-from-multiple-sets-of-enrichment-results" class="anchor"></a>Plot results from multiple sets of enrichment results</h1>
<p>It is often useful to plot results from multiple gene list analyses together. The <code>ewce_plot</code> function allows multiple enrichment analyses to be performed together. To achieve this the results data frames are just appended onto each other, with an additional <code>list</code> column added detailing which analysis they relate to.</p>
<p>To demonstrate this we need to first generate a second analysis so let us sample thirty random genes, and run the bootstrapping analysis on it.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">human.bg</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/mouse_to_human_homologs.html">mouse_to_human_homologs</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">HGNC.symbol</span>
<span class="va">gene.list.2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="va">human.bg</span>,size <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>
<span class="va">second_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,
                                                  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,
                                                  hits <span class="op">=</span> <span class="va">gene.list.2</span>, 
                                                  reps <span class="op">=</span> <span class="va">reps</span>,
                                                  annotLevel <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="va">full_res2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">full_results</span><span class="op">$</span><span class="va">results</span>,
                       list<span class="op">=</span><span class="st">"Alzheimers"</span><span class="op">)</span>
<span class="va">rando_res2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">second_results</span><span class="op">$</span><span class="va">results</span>,
                        list<span class="op">=</span><span class="st">"Random"</span><span class="op">)</span>
<span class="va">merged_results</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">full_res2</span>, <span class="va">rando_res2</span><span class="op">)</span></code></pre></div>
<div id="plot-results-3" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-results-3" class="anchor"></a>Plot results</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">merged_results</span>,
                             mtc_method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-13-1.png" width="672"></p>
<p>As expected, the second randomly generated gene list shows no significant enrichment results.</p>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<details><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.1 (2021-08-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] ewceData_1.1.0      ExperimentHub_2.1.4 AnnotationHub_3.1.6
## [4] BiocFileCache_2.1.1 dbplyr_2.1.1        BiocGenerics_0.39.2
## [7] EWCE_2.0.0          RNOmni_1.0.0        BiocStyle_2.21.4   
## 
## loaded via a namespace (and not attached):
##   [1] readxl_1.3.1                  backports_1.2.1              
##   [3] systemfonts_1.0.3             plyr_1.8.6                   
##   [5] lazyeval_0.2.2                orthogene_0.99.8             
##   [7] listenv_0.8.0                 GenomeInfoDb_1.29.10         
##   [9] ggplot2_3.3.5                 digest_0.6.28                
##  [11] htmltools_0.5.2               fansi_0.5.0                  
##  [13] magrittr_2.0.1                memoise_2.0.0                
##  [15] openxlsx_4.2.4                limma_3.49.4                 
##  [17] globals_0.14.0                Biostrings_2.61.2            
##  [19] matrixStats_0.61.0            pkgdown_1.6.1                
##  [21] colorspace_2.0-2              blob_1.2.2                   
##  [23] rappdirs_0.3.3                textshaping_0.3.6            
##  [25] haven_2.4.3                   xfun_0.27                    
##  [27] dplyr_1.0.7                   crayon_1.4.1                 
##  [29] RCurl_1.98-1.5                jsonlite_1.7.2               
##  [31] glue_1.4.2                    gtable_0.3.0                 
##  [33] zlibbioc_1.39.0               XVector_0.33.0               
##  [35] HGNChelper_0.8.1              DelayedArray_0.19.4          
##  [37] car_3.0-11                    future.apply_1.8.1           
##  [39] SingleCellExperiment_1.15.2   abind_1.4-5                  
##  [41] scales_1.1.1                  DBI_1.1.1                    
##  [43] rstatix_0.7.0                 Rcpp_1.0.7                   
##  [45] viridisLite_0.4.0             xtable_1.8-4                 
##  [47] foreign_0.8-81                bit_4.0.4                    
##  [49] stats4_4.1.1                  htmlwidgets_1.5.4            
##  [51] httr_1.4.2                    ellipsis_0.3.2               
##  [53] farver_2.1.0                  pkgconfig_2.0.3              
##  [55] sass_0.4.0                    utf8_1.2.2                   
##  [57] labeling_0.4.2                tidyselect_1.1.1             
##  [59] rlang_0.4.12                  reshape2_1.4.4               
##  [61] later_1.3.0                   AnnotationDbi_1.55.2         
##  [63] munsell_0.5.0                 BiocVersion_3.14.0           
##  [65] cellranger_1.1.0              tools_4.1.1                  
##  [67] cachem_1.0.6                  generics_0.1.0               
##  [69] RSQLite_2.2.8                 broom_0.7.9                  
##  [71] evaluate_0.14                 stringr_1.4.0                
##  [73] fastmap_1.1.0                 yaml_2.2.1                   
##  [75] ragg_1.1.3                    babelgene_21.4               
##  [77] knitr_1.36                    bit64_4.0.5                  
##  [79] fs_1.5.0                      zip_2.2.0                    
##  [81] purrr_0.3.4                   KEGGREST_1.33.0              
##  [83] gprofiler2_0.2.1              future_1.22.1                
##  [85] mime_0.12                     compiler_4.1.1               
##  [87] plotly_4.10.0                 filelock_1.0.2               
##  [89] curl_4.3.2                    png_0.1-7                    
##  [91] interactiveDisplayBase_1.31.2 ggsignif_0.6.3               
##  [93] tibble_3.1.5                  bslib_0.3.1                  
##  [95] homologene_1.4.68.19.3.27     stringi_1.7.5                
##  [97] highr_0.9                     desc_1.4.0                   
##  [99] forcats_0.5.1                 lattice_0.20-45              
## [101] Matrix_1.3-4                  vctrs_0.3.8                  
## [103] pillar_1.6.4                  lifecycle_1.0.1              
## [105] BiocManager_1.30.16           jquerylib_0.1.4              
## [107] cowplot_1.1.1                 data.table_1.14.2            
## [109] bitops_1.0-7                  httpuv_1.6.3                 
## [111] patchwork_1.1.1               GenomicRanges_1.45.0         
## [113] R6_2.5.1                      bookdown_0.24                
## [115] promises_1.2.0.1              gridExtra_2.3                
## [117] rio_0.5.27                    IRanges_2.27.2               
## [119] parallelly_1.28.1             codetools_0.2-18             
## [121] MASS_7.3-54                   assertthat_0.2.1             
## [123] SummarizedExperiment_1.23.5   rprojroot_2.0.2              
## [125] withr_2.4.2                   sctransform_0.3.2            
## [127] S4Vectors_0.31.5              GenomeInfoDbData_1.2.7       
## [129] parallel_4.1.1                hms_1.1.1                    
## [131] grid_4.1.1                    tidyr_1.1.4                  
## [133] rmarkdown_2.11                MatrixGenerics_1.5.4         
## [135] carData_3.0-4                 ggpubr_0.4.0                 
## [137] Biobase_2.53.0                shiny_1.7.1</code></pre>
</details>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
