<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Applications to transcriptomes • EWCE</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Applications to transcriptomes">
<meta property="og:description" content="EWCE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/conditional.html">Conditional enrichment examples</a>
    </li>
    <li>
      <a href="../articles/create_CTD.html">Creating CellTypeDatasets</a>
    </li>
    <li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
    <li>
      <a href="../articles/transcriptomes.html">Applications to transcriptomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/NathanSkene/EWCE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="transcriptomes_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Applications to transcriptomes</h1>
                        <h4 class="author"><h4>
Alan Murphy, Brian Schilder, and Nathan Skene
</h4></h4>
            
            <h4 class="date"><h4>
Oct-22-2021
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/NathanSkene/EWCE/blob/master/vignettes/transcriptomes.Rmd"><code>vignettes/transcriptomes.Rmd</code></a></small>
      <div class="hidden name"><code>transcriptomes.Rmd</code></div>

    </div>

    
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="setup" class="section level1">
<h1 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/NathanSkene/EWCE">EWCE</a></span><span class="op">)</span> 
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span></code></pre></div>
</div>
<div id="application-to-transcriptomic-data" class="section level1">
<h1 class="hasAnchor">
<a href="#application-to-transcriptomic-data" class="anchor"></a>Application to transcriptomic data</h1>
<div id="prepare-data" class="section level2">
<h2 class="hasAnchor">
<a href="#prepare-data" class="anchor"></a>Prepare data</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html">ctd</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">tt_alzh</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/tt_alzh.html">tt_alzh</a></span><span class="op">(</span><span class="op">)</span>

<span class="co">## NOTE: rep=100 for demo purposes only. </span>
<span class="co">## Use &gt;=10,000 for publication-quality results.</span>
<span class="va">reps</span> <span class="op">&lt;-</span> <span class="fl">100</span> </code></pre></div>
</div>
<div id="analysing-single-transcriptome-study" class="section level2">
<h2 class="hasAnchor">
<a href="#analysing-single-transcriptome-study" class="anchor"></a>Analysing single transcriptome study</h2>
<p>For the prior analyses the gene lists were not associated with any numeric values or directionality. The methodology for extending this form of analysis to transcriptomic studies simply involves thresholding the most upregulated and downregulated genes.</p>
<p>To demonstrate this we have an example dataset <code>tt_alzh</code>. This data frame was generated using <code>limma</code> from a set of post-mortem tissue samples from Brodmann area 46 which were described in a paper by the Haroutunian lab<span class="citation"><sup>1</sup></span>.</p>
<p>The first step is to load the data, obtain the MGI ids, sort the rows by t-statistic and then select the most up/down-regulated genes. The package then has a function <code>ewce_expression_data</code> which thresholds and selects the gene sets, and calls the EWCE function.</p>
<p>Below we show the function call using the default settings, but if desired different threshold values can be used, or alternative columns used to sort the table.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># ewce_expression_data calls bootstrap_enrichment_test so  </span>
<span class="va">tt_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span> <span class="fu"><a href="../reference/ewce_expression_data.html">ewce_expression_data</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,
                                          tt <span class="op">=</span> <span class="va">tt_alzh</span>,
                                          annotLevel <span class="op">=</span> <span class="fl">1</span>,
                                          ttSpecies <span class="op">=</span> <span class="st">"human"</span>,
                                          sctSpecies <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span></code></pre></div>
<div id="plot-results" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-results" class="anchor"></a>Plot results</h3>
<p>The results of this analysis can be plotted using the <code>ewce_plot</code> function.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span><span class="va">tt_results</span><span class="op">$</span><span class="va">joint_results</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></code></pre></div>
<p><img src="transcriptomes_files/figure-html/unnamed-chunk-3-1.png" width="672"></p>
<p>As was reported in our paper, neuronal genes are found to be downregulated while glial genes are upregulated.</p>
</div>
</div>
</div>
<div id="generating-bootstrap-plots-for-transcriptomes" class="section level1">
<h1 class="hasAnchor">
<a href="#generating-bootstrap-plots-for-transcriptomes" class="anchor"></a>Generating bootstrap plots for transcriptomes</h1>
<p>A common request is to explain which differentially expressed genes are associated with a cell type…</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">full_result_path</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/generate_bootstrap_plots_for_transcriptome.html">generate_bootstrap_plots_for_transcriptome</a></span><span class="op">(</span>
  sct_data <span class="op">=</span> <span class="va">ctd</span>,
  tt <span class="op">=</span> <span class="va">tt_alzh</span>,
  annotLevel <span class="op">=</span> <span class="fl">1</span>,
  full_results <span class="op">=</span> <span class="va">tt_results</span>,
  listFileName <span class="op">=</span> <span class="st">"examples"</span>,
  reps <span class="op">=</span> <span class="va">reps</span>,
  ttSpecies <span class="op">=</span> <span class="st">"human"</span>,
  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,
  onlySignif <span class="op">=</span> <span class="cn">FALSE</span>,
  savePath <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="merging-multiple-transcriptome-studies" class="section level1">
<h1 class="hasAnchor">
<a href="#merging-multiple-transcriptome-studies" class="anchor"></a>Merging multiple transcriptome studies</h1>
<p>Where multiple transcriptomic studies have been performed with the same purpose, i.e. seeking differential expression in dlPFC of post-mortem schizophrenics, it is common to want to determine whether they exhibit any shared signal. EWCE can be used to merge the results of multiple studies.</p>
<p>To demonstrate this we use a two further Alzheimer’s transcriptome datasets coming from Brodmann areas 36 and 44: these area stored in <code>tt_alzh_BA36</code> and <code>tt_alzh_BA44</code>. The first step is to run EWCE on each of these individually and store the output into one list.</p>
<div id="load-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load data</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tt_alzh_BA36</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/tt_alzh_BA36.html">tt_alzh_BA36</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">tt_alzh_BA44</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/tt_alzh_BA44.html">tt_alzh_BA44</a></span><span class="op">(</span><span class="op">)</span> </code></pre></div>
</div>
<div id="run-ewce-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#run-ewce-analysis" class="anchor"></a>Run EWCE analysis</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tt_results_36</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_expression_data.html">ewce_expression_data</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,
                                             tt <span class="op">=</span> <span class="va">tt_alzh_BA36</span>,
                                             annotLevel <span class="op">=</span> <span class="fl">1</span>,
                                             ttSpecies <span class="op">=</span> <span class="st">"human"</span>,
                                             sctSpecies <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span>
<span class="va">tt_results_44</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_expression_data.html">ewce_expression_data</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,
                                             tt <span class="op">=</span> <span class="va">tt_alzh_BA44</span>,
                                             annotLevel <span class="op">=</span> <span class="fl">1</span>,
                                             ttSpecies <span class="op">=</span> <span class="st">"human"</span>,
                                             sctSpecies <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span>

<span class="co"># Fill a list with the results</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/add_res_to_merging_list.html">add_res_to_merging_list</a></span><span class="op">(</span><span class="va">tt_results</span><span class="op">)</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/add_res_to_merging_list.html">add_res_to_merging_list</a></span><span class="op">(</span><span class="va">tt_results_36</span>,<span class="va">results</span><span class="op">)</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/add_res_to_merging_list.html">add_res_to_merging_list</a></span><span class="op">(</span><span class="va">tt_results_44</span>,<span class="va">results</span><span class="op">)</span>

<span class="co"># Perform the merged analysis</span>
<span class="co"># For publication reps should be higher</span>
<span class="va">merged_res</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/merged_ewce.html">merged_ewce</a></span><span class="op">(</span>results <span class="op">=</span> <span class="va">results</span>,
                                reps <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> 
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">merged_res</span><span class="op">)</span></code></pre></div>
<p>The results can then be plotted as normal using the <code>ewce_plot</code> function.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span><span class="va">merged_res</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></code></pre></div>
<p><img src="transcriptomes_files/figure-html/unnamed-chunk-6-1.png" width="672"></p>
<p>The merged results from all three Alzheimer’s brain regions are found to be remarkably similar, as was reported in our paper.</p>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<details><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.1 (2021-08-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] ewceData_1.1.0      ExperimentHub_2.1.4 AnnotationHub_3.1.6
## [4] BiocFileCache_2.1.1 dbplyr_2.1.1        BiocGenerics_0.39.2
## [7] EWCE_2.0.0          RNOmni_1.0.0        BiocStyle_2.21.4   
## 
## loaded via a namespace (and not attached):
##   [1] readxl_1.3.1                  backports_1.2.1              
##   [3] systemfonts_1.0.3             plyr_1.8.6                   
##   [5] lazyeval_0.2.2                orthogene_0.99.8             
##   [7] listenv_0.8.0                 GenomeInfoDb_1.29.10         
##   [9] ggplot2_3.3.5                 digest_0.6.28                
##  [11] htmltools_0.5.2               fansi_0.5.0                  
##  [13] magrittr_2.0.1                memoise_2.0.0                
##  [15] openxlsx_4.2.4                limma_3.49.4                 
##  [17] globals_0.14.0                Biostrings_2.61.2            
##  [19] matrixStats_0.61.0            pkgdown_1.6.1                
##  [21] colorspace_2.0-2              blob_1.2.2                   
##  [23] rappdirs_0.3.3                textshaping_0.3.6            
##  [25] haven_2.4.3                   xfun_0.27                    
##  [27] dplyr_1.0.7                   crayon_1.4.1                 
##  [29] RCurl_1.98-1.5                jsonlite_1.7.2               
##  [31] glue_1.4.2                    gtable_0.3.0                 
##  [33] zlibbioc_1.39.0               XVector_0.33.0               
##  [35] HGNChelper_0.8.1              DelayedArray_0.19.4          
##  [37] car_3.0-11                    future.apply_1.8.1           
##  [39] SingleCellExperiment_1.15.2   abind_1.4-5                  
##  [41] scales_1.1.1                  DBI_1.1.1                    
##  [43] rstatix_0.7.0                 Rcpp_1.0.7                   
##  [45] viridisLite_0.4.0             xtable_1.8-4                 
##  [47] foreign_0.8-81                bit_4.0.4                    
##  [49] stats4_4.1.1                  htmlwidgets_1.5.4            
##  [51] httr_1.4.2                    ellipsis_0.3.2               
##  [53] farver_2.1.0                  pkgconfig_2.0.3              
##  [55] sass_0.4.0                    utf8_1.2.2                   
##  [57] tidyselect_1.1.1              rlang_0.4.12                 
##  [59] reshape2_1.4.4                later_1.3.0                  
##  [61] AnnotationDbi_1.55.2          munsell_0.5.0                
##  [63] BiocVersion_3.14.0            cellranger_1.1.0             
##  [65] tools_4.1.1                   cachem_1.0.6                 
##  [67] generics_0.1.0                RSQLite_2.2.8                
##  [69] broom_0.7.9                   evaluate_0.14                
##  [71] stringr_1.4.0                 fastmap_1.1.0                
##  [73] yaml_2.2.1                    ragg_1.1.3                   
##  [75] babelgene_21.4                knitr_1.36                   
##  [77] bit64_4.0.5                   fs_1.5.0                     
##  [79] zip_2.2.0                     purrr_0.3.4                  
##  [81] KEGGREST_1.33.0               gprofiler2_0.2.1             
##  [83] future_1.22.1                 mime_0.12                    
##  [85] compiler_4.1.1                plotly_4.10.0                
##  [87] filelock_1.0.2                curl_4.3.2                   
##  [89] png_0.1-7                     interactiveDisplayBase_1.31.2
##  [91] ggsignif_0.6.3                tibble_3.1.5                 
##  [93] bslib_0.3.1                   homologene_1.4.68.19.3.27    
##  [95] stringi_1.7.5                 highr_0.9                    
##  [97] desc_1.4.0                    forcats_0.5.1                
##  [99] lattice_0.20-45               Matrix_1.3-4                 
## [101] vctrs_0.3.8                   pillar_1.6.4                 
## [103] lifecycle_1.0.1               BiocManager_1.30.16          
## [105] jquerylib_0.1.4               cowplot_1.1.1                
## [107] data.table_1.14.2             bitops_1.0-7                 
## [109] httpuv_1.6.3                  patchwork_1.1.1              
## [111] GenomicRanges_1.45.0          R6_2.5.1                     
## [113] bookdown_0.24                 promises_1.2.0.1             
## [115] gridExtra_2.3                 rio_0.5.27                   
## [117] IRanges_2.27.2                parallelly_1.28.1            
## [119] codetools_0.2-18              MASS_7.3-54                  
## [121] assertthat_0.2.1              SummarizedExperiment_1.23.5  
## [123] rprojroot_2.0.2               withr_2.4.2                  
## [125] sctransform_0.3.2             S4Vectors_0.31.5             
## [127] GenomeInfoDbData_1.2.7        parallel_4.1.1               
## [129] hms_1.1.1                     grid_4.1.1                   
## [131] tidyr_1.1.4                   rmarkdown_2.11               
## [133] MatrixGenerics_1.5.4          carData_3.0-4                
## [135] ggpubr_0.4.0                  Biobase_2.53.0               
## [137] shiny_1.7.1</code></pre>
</details>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-haroutunian2009transcriptional" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Haroutunian, V., Katsel, P. &amp; Schmeidler, J. Transcriptional vulnerability of brain regions in alzheimer’s disease and dementia. <em>Neurobiology of aging</em> <strong>30,</strong> 561–573 (2009).</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
