<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Bootstrap cell type enrichment test — bootstrap_enrichment_test • EWCE</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Bootstrap cell type enrichment test — bootstrap_enrichment_test" />
<meta property="og:description" content="bootstrap_enrichment_test takes a genelist and a single cell type
transcriptome dataset and determines the probability of enrichment and fold
changes for each cell type." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/conditional.html">Conditional enrichment examples</a>
    </li>
    <li>
      <a href="../articles/create_CTD.html">Creating CellTypeDatasets</a>
    </li>
    <li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
    <li>
      <a href="../articles/transcriptomes.html">Applications to transcriptomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/NathanSkene/EWCE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Bootstrap cell type enrichment test</h1>
    <small class="dont-index">Source: <a href='https://github.com/NathanSkene/EWCE/blob/master/R/bootstrap_enrichment_test.R'><code>R/bootstrap_enrichment_test.R</code></a></small>
    <div class="hidden name"><code>bootstrap_enrichment_test.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>bootstrap_enrichment_test</code> takes a genelist and a single cell type
transcriptome dataset and determines the probability of enrichment and fold
changes for each cell type.</p>
    </div>

    <pre class="usage"><span class='fu'>bootstrap_enrichment_test</span><span class='op'>(</span>
  sct_data <span class='op'>=</span> <span class='cn'>NULL</span>,
  hits <span class='op'>=</span> <span class='cn'>NULL</span>,
  bg <span class='op'>=</span> <span class='cn'>NULL</span>,
  genelistSpecies <span class='op'>=</span> <span class='cn'>NULL</span>,
  sctSpecies <span class='op'>=</span> <span class='cn'>NULL</span>,
  output_species <span class='op'>=</span> <span class='st'>"human"</span>,
  reps <span class='op'>=</span> <span class='fl'>100</span>,
  annotLevel <span class='op'>=</span> <span class='fl'>1</span>,
  geneSizeControl <span class='op'>=</span> <span class='cn'>FALSE</span>,
  controlledCT <span class='op'>=</span> <span class='cn'>NULL</span>,
  mtc_method <span class='op'>=</span> <span class='st'>"BH"</span>,
  sort_results <span class='op'>=</span> <span class='cn'>TRUE</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>sct_data</th>
      <td><p>List generated using <a href='generate_celltype_data.html'>generate_celltype_data</a>.</p></td>
    </tr>
    <tr>
      <th>hits</th>
      <td><p>List of gene symbols containing the target gene list.
Will automatically be converted to human gene symbols
if <code>geneSizeControl=TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>bg</th>
      <td><p>List of gene symbols containing the background gene list
(including hit genes). If <code>bg=NULL</code>,
 an appropriate gene background will be created automatically.
if <code>geneSizeControl=TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>genelistSpecies</th>
      <td><p>Species that <code>hits</code> genes came from
(no longer limited to just "mouse" and "human").</p></td>
    </tr>
    <tr>
      <th>sctSpecies</th>
      <td><p>Species that <code>sct_data</code> came from
(no longer limited to just "mouse" and "human").</p></td>
    </tr>
    <tr>
      <th>output_species</th>
      <td><p>Species to convert <code>sct_data</code> and <code>hits</code> to
(Default: "human").</p></td>
    </tr>
    <tr>
      <th>reps</th>
      <td><p>Number of random gene lists to generate (<em>Default: 100</em>,
but should be &gt;=10,000 for publication-quality results).</p></td>
    </tr>
    <tr>
      <th>annotLevel</th>
      <td><p>An integer indicating which level of <code>sct_data</code> to
analyse (<em>Default: 1</em>).</p></td>
    </tr>
    <tr>
      <th>geneSizeControl</th>
      <td><p>Whether you want to control for
GC content and transcript length. Recommended if the gene list originates
from genetic studies (<em>Default: FALSE</em>).
 If set to <code>TRUE</code>, then <code>hits</code> must be from humans.
should be used rather than mouse.</p></td>
    </tr>
    <tr>
      <th>controlledCT</th>
      <td><p>[Optional] If not NULL, and instead is the name of a
cell type, then the bootstrapping controls for expression within that
cell type.</p></td>
    </tr>
    <tr>
      <th>mtc_method</th>
      <td><p>Multiple-testing correction method
(passed to <a href='https://rdrr.io/r/stats/p.adjust.html'>p.adjust</a>).</p></td>
    </tr>
    <tr>
      <th>sort_results</th>
      <td><p>Sort enrichment results from
smallest to largest p-values.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Print messages.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A list containing three data frames:</p><ul>
<li><p><code>results</code>: dataframe in which each row gives the statistics
  (p-value, fold change and number of standard deviations from the mean)
  associated with the enrichment of the stated cell type in the gene list</p></li>
<li><p><code>hit.cells</code>: vector containing the summed proportion of
  expression in each cell type for the target list</p></li>
<li><p><code>bootstrap_data</code>: matrix in which each row represents the
  summed proportion of expression in each cell type for one of the
  random lists</p></li>
</ul>


    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Load the single cell data</span>
<span class='va'>ctd</span> <span class='op'>&lt;-</span> <span class='fu'>ewceData</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ewceData/man/ctd.html'>ctd</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'><span class='co'># Set the parameters for the analysis</span>
<span class='co'># Use 3 bootstrap lists for speed, for publishable analysis use &gt;=10,000</span>
<span class='va'>reps</span> <span class='op'>&lt;-</span> <span class='fl'>3</span>
<span class='co'># Load gene list from Alzheimer's disease GWAS</span>
<span class='va'>example_genelist</span> <span class='op'>&lt;-</span> <span class='fu'>ewceData</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ewceData/man/example_genelist.html'>example_genelist</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'>
<span class='co'># Bootstrap significance test, no control for transcript length or GC content</span>
<span class='va'>full_results</span> <span class='op'>&lt;-</span> <span class='fu'>EWCE</span><span class='fu'>::</span><span class='fu'>bootstrap_enrichment_test</span><span class='op'>(</span>
    sct_data <span class='op'>=</span> <span class='va'>ctd</span>,
    hits <span class='op'>=</span> <span class='va'>example_genelist</span>,
    reps <span class='op'>=</span> <span class='va'>reps</span>,
    annotLevel <span class='op'>=</span> <span class='fl'>1</span>,
    sctSpecies <span class='op'>=</span> <span class='st'>"mouse"</span>,
    genelistSpecies <span class='op'>=</span> <span class='st'>"human"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Generating gene background for mouse x human ==&gt; human</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: mouse</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for mouse</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: mmusculus</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: human</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for human</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: mmusculus</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 10090</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 21,207 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 21,207 genes from mmusculus.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 19,129 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 19,129 genes from hsapiens.</span></div><div class='output co'>#&gt; <span class='message'>Preparing gene_df.</span></div><div class='output co'>#&gt; <span class='message'>data.frame format detected.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from Gene.Symbol.</span></div><div class='output co'>#&gt; <span class='message'>21,207 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Converting mmusculus ==&gt; hsapiens orthologs using: homologene</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: mmusculus</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 10090</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Checking for genes without orthologs in hsapiens.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from input_gene.</span></div><div class='output co'>#&gt; <span class='message'>17,355 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from ortholog_gene.</span></div><div class='output co'>#&gt; <span class='message'>17,355 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Checking for genes without 1:1 orthologs.</span></div><div class='output co'>#&gt; <span class='message'>Dropping 131 genes that have multiple input_gene per ortholog_gene.</span></div><div class='output co'>#&gt; <span class='message'>Dropping 498 genes that have multiple ortholog_gene per input_gene.</span></div><div class='output co'>#&gt; <span class='message'>Filtering gene_df with gene_map</span></div><div class='output co'>#&gt; <span class='message'>Adding input_gene col to gene_df.</span></div><div class='output co'>#&gt; <span class='message'>Adding ortholog_gene col to gene_df.</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>=========== REPORT SUMMARY ===========</span></div><div class='output co'>#&gt; <span class='message'>Total genes dropped after convert_orthologs :</span>
#&gt; <span class='message'>   4,725 / 21,207 (22%)</span></div><div class='output co'>#&gt; <span class='message'>Total genes remaining after convert_orthologs :</span>
#&gt; <span class='message'>   16,482 / 21,207 (78%)</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>=========== REPORT SUMMARY ===========</span></div><div class='output co'>#&gt; <span class='message'>16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></div><div class='output co'>#&gt; <span class='message'>16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: human</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for human</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: human</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for human</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 19,129 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 19,129 genes from hsapiens.</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>=========== REPORT SUMMARY ===========</span></div><div class='output co'>#&gt; <span class='message'>19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></div><div class='output co'>#&gt; <span class='message'>19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></div><div class='output co'>#&gt; <span class='message'>16,482 intersect background genes used.</span></div><div class='output co'>#&gt; <span class='message'>Standardising CellTypeDataset</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Checking gene list inputs.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: human</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for human</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 19,129 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 19,129 genes from human.</span></div><div class='output co'>#&gt; <span class='message'>Standardising sct_data.</span></div><div class='output co'>#&gt; <span class='message'>Converting gene list input to standardised human genes.</span></div><div class='output co'>#&gt; <span class='message'>Running without gene size control.</span></div><div class='output co'>#&gt; <span class='message'>17 hit genes remain after filtering.</span></div><div class='output co'>#&gt; <span class='message'>Computing summed proportions.</span></div><div class='output co'>#&gt; <span class='message'>Testing for enrichment in 7 cell types...</span></div><div class='output co'>#&gt; <span class='message'>Sorting results by p-value.</span></div><div class='output co'>#&gt; <span class='message'>Computing BH-corrected q-values.</span></div><div class='output co'>#&gt; <span class='message'>2 significant cell type enrichment results @ q&lt;0.05 : </span></div><div class='output co'>#&gt; <span class='message'>              CellType annotLevel p fold_change sd_from_mean q</span>
#&gt; <span class='message'>1            microglia          1 0    2.041600     1.660623 0</span>
#&gt; <span class='message'>2 astrocytes_ependymal          1 0    1.309126     1.229830 0</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


