<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>check_ewce_genelist_inputs — check_ewce_genelist_inputs • EWCE</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="check_ewce_genelist_inputs — check_ewce_genelist_inputs" />
<meta property="og:description" content="check_ewce_genelist_inputs Is used to check that hits and bg gene
lists passed to EWCE are setup correctly. Checks they are the
appropriate length.
Checks all hits genes are in bg. Checks the species match and if not
reduces to 1:1 orthologs." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/conditional.html">Conditional enrichment examples</a>
    </li>
    <li>
      <a href="../articles/create_CTD.html">Creating CellTypeDatasets</a>
    </li>
    <li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
    <li>
      <a href="../articles/transcriptomes.html">Applications to transcriptomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/NathanSkene/EWCE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>check_ewce_genelist_inputs</h1>
    <small class="dont-index">Source: <a href='https://github.com/NathanSkene/EWCE/blob/master/R/check_ewce_genelist_input.r'><code>R/check_ewce_genelist_input.r</code></a></small>
    <div class="hidden name"><code>check_ewce_genelist_inputs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>check_ewce_genelist_inputs</code> Is used to check that hits and bg gene
lists passed to EWCE are setup correctly. Checks they are the
appropriate length.
Checks all hits genes are in bg. Checks the species match and if not
reduces to 1:1 orthologs.</p>
    </div>

    <pre class="usage"><span class='fu'>check_ewce_genelist_inputs</span><span class='op'>(</span>
  <span class='va'>sct_data</span>,
  <span class='va'>hits</span>,
  bg <span class='op'>=</span> <span class='cn'>NULL</span>,
  genelistSpecies <span class='op'>=</span> <span class='cn'>NULL</span>,
  sctSpecies <span class='op'>=</span> <span class='cn'>NULL</span>,
  output_species <span class='op'>=</span> <span class='st'>"human"</span>,
  geneSizeControl <span class='op'>=</span> <span class='cn'>FALSE</span>,
  standardise <span class='op'>=</span> <span class='cn'>FALSE</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>sct_data</th>
      <td><p>List generated using <a href='generate_celltype_data.html'>generate_celltype_data</a>.</p></td>
    </tr>
    <tr>
      <th>hits</th>
      <td><p>List of gene symbols containing the target gene list.
Will automatically be converted to human gene symbols
if <code>geneSizeControl=TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>bg</th>
      <td><p>List of gene symbols containing the background gene list
(including hit genes). If <code>bg=NULL</code>,
 an appropriate gene background will be created automatically.
if <code>geneSizeControl=TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>genelistSpecies</th>
      <td><p>Species that <code>hits</code> genes came from
(no longer limited to just "mouse" and "human").</p></td>
    </tr>
    <tr>
      <th>sctSpecies</th>
      <td><p>Species that <code>sct_data</code> came from
(no longer limited to just "mouse" and "human").</p></td>
    </tr>
    <tr>
      <th>output_species</th>
      <td><p>Species to convert <code>sct_data</code> and <code>hits</code> to
(Default: "human").</p></td>
    </tr>
    <tr>
      <th>geneSizeControl</th>
      <td><p>Whether you want to control for
GC content and transcript length. Recommended if the gene list originates
from genetic studies (<em>Default: FALSE</em>).
 If set to <code>TRUE</code>, then <code>hits</code> must be from humans.
should be used rather than mouse.</p></td>
    </tr>
    <tr>
      <th>standardise</th>
      <td><p>If <code>input_species==output_species</code>,
should the genes be standardised using <a href='https://rdrr.io/pkg/orthogene/man/map_genes.html'>map_genes</a>?</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Print messages.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A list containing</p><ul>
<li><p><code>hits</code>: Array of MGI/HGNC gene symbols containing the target
  gene list.</p></li>
<li><p><code>bg</code>: Array of MGI/HGNC gene symbols containing the background
  gene list.</p></li>
</ul>


    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>ctd</span> <span class='op'>&lt;-</span> <span class='fu'>ewceData</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ewceData/man/ctd.html'>ctd</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'><span class='va'>example_genelist</span> <span class='op'>&lt;-</span> <span class='fu'>ewceData</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ewceData/man/example_genelist.html'>example_genelist</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'>
<span class='co'># Called from "bootstrap_enrichment_test()" and "generate_bootstrap_plots()"</span>
<span class='va'>checkedLists</span> <span class='op'>&lt;-</span> <span class='fu'>EWCE</span><span class='fu'>::</span><span class='fu'>check_ewce_genelist_inputs</span><span class='op'>(</span>
    sct_data <span class='op'>=</span> <span class='va'>ctd</span>,
    hits <span class='op'>=</span> <span class='va'>example_genelist</span>,
    sctSpecies <span class='op'>=</span> <span class='st'>"mouse"</span>,
    genelistSpecies <span class='op'>=</span> <span class='st'>"human"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Checking gene list inputs.</span></div><div class='output co'>#&gt; <span class='message'>Generating gene background for mouse x human ==&gt; human</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: mouse</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for mouse</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: mmusculus</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: human</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for human</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: mmusculus</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 10090</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 21,207 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 21,207 genes from mmusculus.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 19,129 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 19,129 genes from hsapiens.</span></div><div class='output co'>#&gt; <span class='message'>Preparing gene_df.</span></div><div class='output co'>#&gt; <span class='message'>data.frame format detected.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from Gene.Symbol.</span></div><div class='output co'>#&gt; <span class='message'>21,207 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Converting mmusculus ==&gt; hsapiens orthologs using: homologene</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: mmusculus</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 10090</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Checking for genes without orthologs in hsapiens.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from input_gene.</span></div><div class='output co'>#&gt; <span class='message'>17,355 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from ortholog_gene.</span></div><div class='output co'>#&gt; <span class='message'>17,355 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Checking for genes without 1:1 orthologs.</span></div><div class='output co'>#&gt; <span class='message'>Dropping 131 genes that have multiple input_gene per ortholog_gene.</span></div><div class='output co'>#&gt; <span class='message'>Dropping 498 genes that have multiple ortholog_gene per input_gene.</span></div><div class='output co'>#&gt; <span class='message'>Filtering gene_df with gene_map</span></div><div class='output co'>#&gt; <span class='message'>Adding input_gene col to gene_df.</span></div><div class='output co'>#&gt; <span class='message'>Adding ortholog_gene col to gene_df.</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>=========== REPORT SUMMARY ===========</span></div><div class='output co'>#&gt; <span class='message'>Total genes dropped after convert_orthologs :</span>
#&gt; <span class='message'>   4,725 / 21,207 (22%)</span></div><div class='output co'>#&gt; <span class='message'>Total genes remaining after convert_orthologs :</span>
#&gt; <span class='message'>   16,482 / 21,207 (78%)</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>=========== REPORT SUMMARY ===========</span></div><div class='output co'>#&gt; <span class='message'>16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></div><div class='output co'>#&gt; <span class='message'>16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: human</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for human</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: human</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for human</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 19,129 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 19,129 genes from hsapiens.</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>=========== REPORT SUMMARY ===========</span></div><div class='output co'>#&gt; <span class='message'>19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></div><div class='output co'>#&gt; <span class='message'>19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></div><div class='output co'>#&gt; <span class='message'>16,482 intersect background genes used.</span></div><div class='output co'>#&gt; <span class='message'>Standardising sct_data.</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Converting gene list input to standardised human genes.</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


