<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Celltype controlled geneset enrichment — controlled_geneset_enrichment • EWCE</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Celltype controlled geneset enrichment — controlled_geneset_enrichment" />
<meta property="og:description" content="controlled_geneset_enrichment tests whether a functional gene set is
still enriched in a disease gene set after controlling for the
disease gene set's enrichment in a particular cell type (the 'controlledCT')" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/conditional.html">Conditional enrichment examples</a>
    </li>
    <li>
      <a href="../articles/create_CTD.html">Creating CellTypeDatasets</a>
    </li>
    <li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
    <li>
      <a href="../articles/transcriptomes.html">Applications to transcriptomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/NathanSkene/EWCE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Celltype controlled geneset enrichment</h1>
    <small class="dont-index">Source: <a href='https://github.com/NathanSkene/EWCE/blob/master/R/controlled_geneset_enrichment.r'><code>R/controlled_geneset_enrichment.r</code></a></small>
    <div class="hidden name"><code>controlled_geneset_enrichment.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>controlled_geneset_enrichment</code> tests whether a functional gene set is
still enriched in a disease gene set after controlling for the
disease gene set's enrichment in a particular cell type (the 'controlledCT')</p>
    </div>

    <pre class="usage"><span class='fu'>controlled_geneset_enrichment</span><span class='op'>(</span>
  <span class='va'>disease_genes</span>,
  <span class='va'>functional_genes</span>,
  bg <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>sct_data</span>,
  sctSpecies <span class='op'>=</span> <span class='cn'>NULL</span>,
  output_species <span class='op'>=</span> <span class='st'>"human"</span>,
  disease_genes_species <span class='op'>=</span> <span class='cn'>NULL</span>,
  functional_genes_species <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>annotLevel</span>,
  reps <span class='op'>=</span> <span class='fl'>10000</span>,
  <span class='va'>controlledCT</span>,
  use_intersect <span class='op'>=</span> <span class='cn'>FALSE</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>disease_genes</th>
      <td><p>Array of gene symbols containing the disease gene list.
Does not have to be disease genes. Must be from same species as the single
cell transcriptome dataset.</p></td>
    </tr>
    <tr>
      <th>functional_genes</th>
      <td><p>Array of gene symbols containing the functional gene
list. The enrichment of this gene set within the disease_genes is tested.
Must be from same species as the single cell transcriptome dataset.</p></td>
    </tr>
    <tr>
      <th>bg</th>
      <td><p>List of gene symbols containing the background gene list
(including hit genes). If <code>bg=NULL</code>,
 an appropriate gene background will be created automatically.
if <code>geneSizeControl=TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>sct_data</th>
      <td><p>List generated using <a href='generate_celltype_data.html'>generate_celltype_data</a>.</p></td>
    </tr>
    <tr>
      <th>sctSpecies</th>
      <td><p>Species that <code>sct_data</code> came from
(no longer limited to just "mouse" and "human").</p></td>
    </tr>
    <tr>
      <th>output_species</th>
      <td><p>Species to convert <code>sct_data</code> and <code>hits</code> to
(Default: "human").</p></td>
    </tr>
    <tr>
      <th>disease_genes_species</th>
      <td><p>Species of the
<code>disease_genes</code> gene set.</p></td>
    </tr>
    <tr>
      <th>functional_genes_species</th>
      <td><p>Species of the
<code>functional_genes</code> gene set.</p></td>
    </tr>
    <tr>
      <th>annotLevel</th>
      <td><p>An integer indicating which level of <code>sct_data</code> to
analyse (<em>Default: 1</em>).</p></td>
    </tr>
    <tr>
      <th>reps</th>
      <td><p>Number of random gene lists to generate (<em>Default: 100</em>,
but should be &gt;=10,000 for publication-quality results).</p></td>
    </tr>
    <tr>
      <th>controlledCT</th>
      <td><p>[Optional] If not NULL, and instead is the name of a
cell type, then the bootstrapping controls for expression within that
cell type.</p></td>
    </tr>
    <tr>
      <th>use_intersect</th>
      <td><p>When <code>species1</code> and <code>species2</code> are both
different from <code>output_species</code>, this argument will determine whether 
to use the intersect (<code>TRUE</code>) or union (<code>FALSE</code>) of all genes
from <code>species1</code> and <code>species2</code>.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Print messages.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A list containing three data frames:</p><ul>
<li><p><code>p_controlled</code> The probability that functional_genes are
  enriched in disease_genes while controlling for the level of specificity
  in controlledCT</p></li>
<li><p><code>z_controlled</code> The z-score that functional_genes are enriched
  in disease_genes while controlling for the level of specificity in
  controlledCT</p></li>
<li><p><code>p_uncontrolled</code> The probability that functional_genes are
  enriched in disease_genes WITHOUT controlling for the level of
  specificity in controlledCT</p></li>
<li><p><code>z_uncontrolled</code> The z-score that functional_genes are enriched
  in disease_genes WITHOUT controlling for the level of specificity in
  controlledCT</p></li>
<li><p><code>reps=reps</code></p></li>
<li><p><code>controlledCT</code></p></li>
<li><p><code>actualOverlap=actual</code> The number of genes that overlap between
  functional and disease gene sets</p></li>
</ul>


    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># See the vignette for more detailed explanations</span>
<span class='co'># Gene set enrichment analysis controlling for cell type expression</span>
<span class='co'># set seed for bootstrap reproducibility</span>
<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>12345678</span><span class='op'>)</span>
<span class='co'>## load merged dataset from vignette</span>
<span class='va'>ctd</span> <span class='op'>&lt;-</span> <span class='fu'>ewceData</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ewceData/man/ctd.html'>ctd</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'><span class='va'>schiz_genes</span> <span class='op'>&lt;-</span> <span class='fu'>ewceData</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ewceData/man/schiz_genes.html'>schiz_genes</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'><span class='va'>hpsd_genes</span> <span class='op'>&lt;-</span> <span class='fu'>ewceData</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ewceData/man/hpsd_genes.html'>hpsd_genes</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'><span class='co'># Use 3 bootstrap lists for speed, for publishable analysis use &gt;10000</span>
<span class='va'>reps</span> <span class='op'>&lt;-</span> <span class='fl'>3</span>

<span class='va'>res_hpsd_schiz</span> <span class='op'>&lt;-</span> <span class='fu'>EWCE</span><span class='fu'>::</span><span class='fu'>controlled_geneset_enrichment</span><span class='op'>(</span>
    disease_genes <span class='op'>=</span> <span class='va'>schiz_genes</span>,
    functional_genes <span class='op'>=</span> <span class='va'>hpsd_genes</span>,
    sct_data <span class='op'>=</span> <span class='va'>ctd</span>,
    annotLevel <span class='op'>=</span> <span class='fl'>1</span>,
    reps <span class='op'>=</span> <span class='va'>reps</span>,
    controlledCT <span class='op'>=</span> <span class='st'>"pyramidal CA1"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Warning: genelistSpecies not provided. Setting to 'human' by default.</span></div><div class='output co'>#&gt; <span class='message'>Warning: sctSpecies not provided. Setting to 'mouse' by default.</span></div><div class='output co'>#&gt; <span class='message'>Warning: genelistSpecies not provided. Setting to 'human' by default.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: human</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for human</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 19,129 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 19,129 genes from human.</span></div><div class='output co'>#&gt; <span class='message'>Standardising CellTypeDataset</span></div><div class='output co'>#&gt; <span class='message'>Level: 1</span></div><div class='output co'>#&gt; <span class='message'>Extracting  mean_exp</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Matrix dimensions: 13243 x 7</span></div><div class='output co'>#&gt; <span class='message'>Extracting  specificity</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Matrix dimensions: 13243 x 7</span></div><div class='output co'>#&gt; <span class='message'>Extracting  specificity_quantiles</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Matrix dimensions: 13243 x 7</span></div><div class='output co'>#&gt; <span class='message'>Level: 2</span></div><div class='output co'>#&gt; <span class='message'>Extracting  mean_exp</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Matrix dimensions: 13243 x 48</span></div><div class='output co'>#&gt; <span class='message'>Extracting  specificity</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Matrix dimensions: 13243 x 48</span></div><div class='output co'>#&gt; <span class='message'>Extracting  specificity_quantiles</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Matrix dimensions: 13243 x 48</span></div><div class='output co'>#&gt; <span class='message'>Generating controlled bootstrap gene sets.</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


