<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Drop uninformative genes — drop_uninformative_genes • EWCE</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Drop uninformative genes — drop_uninformative_genes" />
<meta property="og:description" content="drop_uninformative_genes first removes genes
that do not have 1:1 orthologs with humans." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/conditional.html">Conditional enrichment examples</a>
    </li>
    <li>
      <a href="../articles/create_CTD.html">Creating CellTypeDatasets</a>
    </li>
    <li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
    <li>
      <a href="../articles/transcriptomes.html">Applications to transcriptomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/NathanSkene/EWCE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Drop uninformative genes</h1>
    <small class="dont-index">Source: <a href='https://github.com/NathanSkene/EWCE/blob/master/R/drop_uninformative_genes.r'><code>R/drop_uninformative_genes.r</code></a></small>
    <div class="hidden name"><code>drop_uninformative_genes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>drop_uninformative_genes</code> first removes genes
that do not have 1:1 orthologs with humans.</p>
    </div>

    <pre class="usage"><span class='fu'>drop_uninformative_genes</span><span class='op'>(</span>
  <span class='va'>exp</span>,
  <span class='va'>level2annot</span>,
  DGE_method <span class='op'>=</span> <span class='st'>"limma"</span>,
  min_variance_decile <span class='op'>=</span> <span class='cn'>NULL</span>,
  adj_pval_thresh <span class='op'>=</span> <span class='fl'>1e-05</span>,
  convert_orths <span class='op'>=</span> <span class='cn'>FALSE</span>,
  input_species <span class='op'>=</span> <span class='cn'>NULL</span>,
  output_species <span class='op'>=</span> <span class='st'>"human"</span>,
  non121_strategy <span class='op'>=</span> <span class='st'>"drop_both_species"</span>,
  as_sparse <span class='op'>=</span> <span class='cn'>TRUE</span>,
  as_DelayedArray <span class='op'>=</span> <span class='cn'>FALSE</span>,
  return_sce <span class='op'>=</span> <span class='cn'>FALSE</span>,
  no_cores <span class='op'>=</span> <span class='fl'>1</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>exp</th>
      <td><p>Expression matrix with gene names as rownames.</p></td>
    </tr>
    <tr>
      <th>level2annot</th>
      <td><p>Array of cell types, with each sequentially corresponding
a column in the expression matrix</p></td>
    </tr>
    <tr>
      <th>DGE_method</th>
      <td><p>Which method to use for the Differential Gene Expression
(DGE) step.</p></td>
    </tr>
    <tr>
      <th>min_variance_decile</th>
      <td><p>If <code>min_variance_decile!=NULL</code>,
 calculates the variance of the mean gene expression
  across `level2annot` (i.e. cell types),
and then removes any genes that are below <code>min_variance_decile</code>
 (on a 0-1 scale).</p></td>
    </tr>
    <tr>
      <th>adj_pval_thresh</th>
      <td><p>Minimum differential expression significance
that a gene must demonstrate across <code>level2annot</code> (i.e. cell types).</p></td>
    </tr>
    <tr>
      <th>convert_orths</th>
      <td><p>If <code>input_species!=output_species</code> and
<code>convert_orths=TRUE</code>, will drop genes without
1:1 <code>output_species</code> orthologs and then convert <code>exp</code> gene names
to those of <code>output_species</code>.</p></td>
    </tr>
    <tr>
      <th>input_species</th>
      <td><p>Which species the gene names in <code>exp</code> come from.</p></td>
    </tr>
    <tr>
      <th>output_species</th>
      <td><p>Which species' genes names to convert <code>exp</code> to.</p></td>
    </tr>
    <tr>
      <th>non121_strategy</th>
      <td><p>How to handle genes that don't have
1:1 mappings between <code>input_species</code>:<code>output_species</code>.
Options include:<br /></p><ul>
<li><p><code>"drop_both_species" or "dbs" or 1</code> : <br />
Drop genes that have duplicate
mappings in either the <code>input_species</code> or <code>output_species</code> <br />
(<em>DEFAULT</em>).</p></li>
<li><p><code>"drop_input_species" or "dis" or 2</code> : <br />
Only drop genes that have duplicate
mappings in the <code>input_species</code>.</p></li>
<li><p><code>"drop_output_species" or "dos" or 3</code> : <br />
Only drop genes that have duplicate
mappings in the <code>output_species</code>.</p></li>
<li><p><code>"keep_both_species" or "kbs" or 4</code> : <br />
Keep all genes regardless of whether
they have duplicate mappings in either species.</p></li>
<li><p><code>"keep_popular" or "kp" or 5</code> : <br />
Return only the most "popular" interspecies ortholog mappings.
 This procedure tends to yield a greater number of returned genes
 but at the cost of many of them not being true biological 1:1 orthologs.</p></li>
<li><p><code>"sum","mean","median","min" or "max"</code> : <br />
 When <code>gene_df</code> is a matrix and <code>gene_output="rownames"</code>,
  these options will aggregate many-to-one gene mappings
  (<code>input_species</code>-to-<code>output_species</code>)
  after dropping any duplicate genes in the <code>output_species</code>.</p></li>
</ul></td>
    </tr>
    <tr>
      <th>as_sparse</th>
      <td><p>Convert <code>exp</code> to sparse matrix.</p></td>
    </tr>
    <tr>
      <th>as_DelayedArray</th>
      <td><p>Convert <code>exp</code> to <code>DelayedArray</code>
for scalable processing.</p></td>
    </tr>
    <tr>
      <th>return_sce</th>
      <td><p>Whether to return the filtered results
as an expression matrix or a <span class="pkg">SingleCellExperiment</span>.</p></td>
    </tr>
    <tr>
      <th>no_cores</th>
      <td><p>Number of cores to parallelise across.
Set to <code>NULL</code> to automatically optimise.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Print messages.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments to be passed to the selected DGE method.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>exp Expression matrix with gene names as row names.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p><code>drop_uninformative_genes</code> then drops genes from an
SCT expression matrix
if they do not significantly vary between any cell types.
Makes this decision based on use of an ANOVA (implemented with <code>limma</code>).
If the F-statistic for variation amongst type2 annotations
is less than a strict p-threshold, then the gene is dropped.</p>
<p>A very fast alternative to DGE methods is filtering by
<code>min_variance_decile</code>,
which selects only genes with the top variance deciles.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>cortex_mrna</span> <span class='op'>&lt;-</span> <span class='fu'>ewceData</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ewceData/man/cortex_mrna.html'>cortex_mrna</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'><span class='co'># Use only a subset of genes to keep the example quick</span>
<span class='va'>cortex_mrna</span><span class='op'>$</span><span class='va'>exp</span> <span class='op'>&lt;-</span> <span class='va'>cortex_mrna</span><span class='op'>$</span><span class='va'>exp</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>300</span>, <span class='op'>]</span>

<span class='co'>## Convert orthologs at the same time</span>
<span class='va'>exp2_orth</span> <span class='op'>&lt;-</span> <span class='fu'>drop_uninformative_genes</span><span class='op'>(</span>
    exp <span class='op'>=</span> <span class='va'>cortex_mrna</span><span class='op'>$</span><span class='va'>exp</span>,
    level2annot <span class='op'>=</span> <span class='va'>cortex_mrna</span><span class='op'>$</span><span class='va'>annot</span><span class='op'>$</span><span class='va'>level2class</span>,
    input_species <span class='op'>=</span> <span class='st'>"mouse"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Check 300Check 3005</span></div><div class='output co'>#&gt; <span class='message'>+  1  core(s) assigned as workers ( 63  reserved).</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Checking for non-expressed genes.</span></div><div class='output co'>#&gt; <span class='message'>Checking for cells with no expressed genes.</span></div><div class='output co'>#&gt; <span class='message'>DGE:: Limma...</span></div><div class='output co'>#&gt; <span class='message'>3 / 300 genes dropped @ DGE adj_pval_thresh &lt; 1e-05</span></div><div class='output co'>#&gt; Time difference of 0.1156094 secs</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


