<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>fix_bad_mgi_symbols
- Given an expression matrix, wherein the rows are supposed to be MGI
symbols, find those symbols which are not official MGI symbols, then
check in the MGI synonm database for whether they match to a proper MGI
symbol. Where a symbol is found to be an aliases for a gene that is already
in the dataset, the combined reads are summed together. — fix_bad_mgi_symbols • EWCE</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="fix_bad_mgi_symbols
- Given an expression matrix, wherein the rows are supposed to be MGI
symbols, find those symbols which are not official MGI symbols, then
check in the MGI synonm database for whether they match to a proper MGI
symbol. Where a symbol is found to be an aliases for a gene that is already
in the dataset, the combined reads are summed together. — fix_bad_mgi_symbols" />
<meta property="og:description" content="Also checks whether any gene names contain &quot;Sep&quot;, &quot;Mar&quot; or &quot;Feb&quot;.
These should be checked for any suggestion that excel has corrupted the
gene names." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/conditional.html">Conditional enrichment examples</a>
    </li>
    <li>
      <a href="../articles/create_CTD.html">Creating CellTypeDatasets</a>
    </li>
    <li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
    <li>
      <a href="../articles/transcriptomes.html">Applications to transcriptomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/NathanSkene/EWCE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>fix_bad_mgi_symbols
- Given an expression matrix, wherein the rows are supposed to be MGI
symbols, find those symbols which are not official MGI symbols, then
check in the MGI synonm database for whether they match to a proper MGI
symbol. Where a symbol is found to be an aliases for a gene that is already
in the dataset, the combined reads are summed together.</h1>
    <small class="dont-index">Source: <a href='https://github.com/NathanSkene/EWCE/blob/master/R/fix_bad_mgi_symbols.r'><code>R/fix_bad_mgi_symbols.r</code></a></small>
    <div class="hidden name"><code>fix_bad_mgi_symbols.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Also checks whether any gene names contain "Sep", "Mar" or "Feb".
These should be checked for any suggestion that excel has corrupted the
gene names.</p>
    </div>

    <pre class="usage"><span class='fu'>fix_bad_mgi_symbols</span><span class='op'>(</span>
  <span class='va'>exp</span>,
  mrk_file_path <span class='op'>=</span> <span class='cn'>NULL</span>,
  printAllBadSymbols <span class='op'>=</span> <span class='cn'>FALSE</span>,
  as_sparse <span class='op'>=</span> <span class='cn'>TRUE</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>exp</th>
      <td><p>An expression matrix where the rows are MGI symbols, or a
SingleCellExperiment (SCE) or
other Ranged Summarized Experiment (SE) type object.</p></td>
    </tr>
    <tr>
      <th>mrk_file_path</th>
      <td><p>Path to the MRK_List2 file which can be downloaded
from www.informatics.jax.org/downloads/reports/index.html</p></td>
    </tr>
    <tr>
      <th>printAllBadSymbols</th>
      <td><p>Output to console all the bad gene symbols</p></td>
    </tr>
    <tr>
      <th>as_sparse</th>
      <td><p>Convert <code>exp</code> to sparse matrix.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Print messages.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns the expression matrix with the rownames corrected and rows
representing the same gene merged. If no corrections are necessary, input
expression matrix is returned. If a SingleCellExperiment (SCE) or other
Ranged Summarized Experiment (SE) type object was inputted this will be
returned with the corrected expression matrix under counts.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Load the single cell data</span>
<span class='va'>cortex_mrna</span> <span class='op'>&lt;-</span> <span class='fu'>ewceData</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ewceData/man/cortex_mrna.html'>cortex_mrna</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'><span class='co'># take a subset for speed</span>
<span class='va'>cortex_mrna</span><span class='op'>$</span><span class='va'>exp</span> <span class='op'>&lt;-</span> <span class='va'>cortex_mrna</span><span class='op'>$</span><span class='va'>exp</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>50</span>, <span class='fl'>1</span><span class='op'>:</span><span class='fl'>5</span><span class='op'>]</span>
<span class='va'>cortex_mrna</span><span class='op'>$</span><span class='va'>exp</span> <span class='op'>&lt;-</span> <span class='fu'>fix_bad_mgi_symbols</span><span class='op'>(</span><span class='va'>cortex_mrna</span><span class='op'>$</span><span class='va'>exp</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='output co'>#&gt; <span class='message'>5 rows do not have proper MGI symbols</span></div><div class='output co'>#&gt; <span class='message'>2310042E22Rik, BC005764, C130030K03Rik, Stmn1-rs1, Gm9846</span></div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='output co'>#&gt; <span class='message'>0 poorly annotated genes are replicates of existing genes. These are: </span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>3 rows should have been corrected by checking synonyms.</span></div><div class='output co'>#&gt; <span class='message'>2 rows STILL do not have proper MGI symbols.</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


