<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Generate CellTypeData (CTD) file — generate_celltype_data • EWCE</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Generate CellTypeData (CTD) file — generate_celltype_data" />
<meta property="og:description" content="generate_celltype_data takes gene expression data and
cell type annotations and creates CellTypeData (CTD) files which
contain matrices of mean expression and specificity per cell type." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/conditional.html">Conditional enrichment examples</a>
    </li>
    <li>
      <a href="../articles/create_CTD.html">Creating CellTypeDatasets</a>
    </li>
    <li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
    <li>
      <a href="../articles/transcriptomes.html">Applications to transcriptomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/NathanSkene/EWCE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate CellTypeData (CTD) file</h1>
    <small class="dont-index">Source: <a href='https://github.com/NathanSkene/EWCE/blob/master/R/generate_celltype_data.r'><code>R/generate_celltype_data.r</code></a></small>
    <div class="hidden name"><code>generate_celltype_data.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>generate_celltype_data</code> takes gene expression data and
cell type annotations and creates CellTypeData (CTD) files which
contain matrices of mean expression and specificity per cell type.</p>
    </div>

    <pre class="usage"><span class='fu'>generate_celltype_data</span><span class='op'>(</span>
  <span class='va'>exp</span>,
  <span class='va'>annotLevels</span>,
  <span class='va'>groupName</span>,
  no_cores <span class='op'>=</span> <span class='fl'>1</span>,
  savePath <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span>,
  file_prefix <span class='op'>=</span> <span class='st'>"ctd"</span>,
  as_sparse <span class='op'>=</span> <span class='cn'>TRUE</span>,
  as_DelayedArray <span class='op'>=</span> <span class='cn'>FALSE</span>,
  normSpec <span class='op'>=</span> <span class='cn'>FALSE</span>,
  convert_orths <span class='op'>=</span> <span class='cn'>FALSE</span>,
  input_species <span class='op'>=</span> <span class='st'>"mouse"</span>,
  output_species <span class='op'>=</span> <span class='st'>"human"</span>,
  non121_strategy <span class='op'>=</span> <span class='st'>"drop_both_species"</span>,
  force_new_file <span class='op'>=</span> <span class='cn'>TRUE</span>,
  specificity_quantiles <span class='op'>=</span> <span class='cn'>TRUE</span>,
  numberOfBins <span class='op'>=</span> <span class='fl'>40</span>,
  dendrograms <span class='op'>=</span> <span class='cn'>TRUE</span>,
  return_ctd <span class='op'>=</span> <span class='cn'>FALSE</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>exp</th>
      <td><p>Numerical matrix with row for each gene and column for each cell.
Row names are gene symbols. Column names are cell IDs which can be
cross referenced against the annot data frame.</p></td>
    </tr>
    <tr>
      <th>annotLevels</th>
      <td><p>List with arrays of strings containing the cell type
names associated with each column in <code>exp</code>.</p></td>
    </tr>
    <tr>
      <th>groupName</th>
      <td><p>A human readable name for referring to the dataset being</p></td>
    </tr>
    <tr>
      <th>no_cores</th>
      <td><p>Number of cores that should be used to speedup the
computation.
<em>NOTE</em>: Use <code>no_cores=1</code> when using this package in windows system.</p></td>
    </tr>
    <tr>
      <th>savePath</th>
      <td><p>Directory where the CTD file should be saved.</p></td>
    </tr>
    <tr>
      <th>file_prefix</th>
      <td><p>Prefix to add to saved CTD file name.</p></td>
    </tr>
    <tr>
      <th>as_sparse</th>
      <td><p>Convert <code>exp</code> to a sparse <code>Matrix</code>.</p></td>
    </tr>
    <tr>
      <th>as_DelayedArray</th>
      <td><p>Convert <code>exp</code> to <code>DelayedArray</code>.</p></td>
    </tr>
    <tr>
      <th>normSpec</th>
      <td><p>Boolean indicating whether specificity data should be
transformed to a normal distribution by cell type, giving equivalent scores
across all cell types.</p></td>
    </tr>
    <tr>
      <th>convert_orths</th>
      <td><p>If <code>input_species!=output_species</code> and
<code>convert_orths=TRUE</code>, will drop genes without
1:1 <code>output_species</code> orthologs and then convert <code>exp</code> gene names
to those of <code>output_species</code>.</p></td>
    </tr>
    <tr>
      <th>input_species</th>
      <td><p>The species that the <code>exp</code> dataset comes from.</p></td>
    </tr>
    <tr>
      <th>output_species</th>
      <td><p>Species to convert <code>exp</code> to
(Default: "human").</p></td>
    </tr>
    <tr>
      <th>non121_strategy</th>
      <td><p>How to handle genes that don't have
1:1 mappings between <code>input_species</code>:<code>output_species</code>.
Options include:<br /></p><ul>
<li><p><code>"drop_both_species" or "dbs" or 1</code> : <br />
Drop genes that have duplicate
mappings in either the <code>input_species</code> or <code>output_species</code> <br />
(<em>DEFAULT</em>).</p></li>
<li><p><code>"drop_input_species" or "dis" or 2</code> : <br />
Only drop genes that have duplicate
mappings in the <code>input_species</code>.</p></li>
<li><p><code>"drop_output_species" or "dos" or 3</code> : <br />
Only drop genes that have duplicate
mappings in the <code>output_species</code>.</p></li>
<li><p><code>"keep_both_species" or "kbs" or 4</code> : <br />
Keep all genes regardless of whether
they have duplicate mappings in either species.</p></li>
<li><p><code>"keep_popular" or "kp" or 5</code> : <br />
Return only the most "popular" interspecies ortholog mappings.
 This procedure tends to yield a greater number of returned genes
 but at the cost of many of them not being true biological 1:1 orthologs.</p></li>
<li><p><code>"sum","mean","median","min" or "max"</code> : <br />
 When <code>gene_df</code> is a matrix and <code>gene_output="rownames"</code>,
  these options will aggregate many-to-one gene mappings
  (<code>input_species</code>-to-<code>output_species</code>)
  after dropping any duplicate genes in the <code>output_species</code>.</p></li>
</ul></td>
    </tr>
    <tr>
      <th>force_new_file</th>
      <td><p>If a file of the same name as the one
being created already exists, overwrite it.</p></td>
    </tr>
    <tr>
      <th>specificity_quantiles</th>
      <td><p>Compute specificity quantiles.
Recommended to set to <code>TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>numberOfBins</th>
      <td><p>Number of quantile 'bins' to use (40 is recommended)</p></td>
    </tr>
    <tr>
      <th>dendrograms</th>
      <td><p>Add dendrogram plots</p></td>
    </tr>
    <tr>
      <th>return_ctd</th>
      <td><p>Return the CTD object in a list along with the file name,
instead of just the file name.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Print messages.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments passed to
<a href='https://rdrr.io/pkg/orthogene/man/convert_orthologs.html'>convert_orthologs</a>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>File names for the saved CellTypeData (CTD) files.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Load the single cell data</span>
<span class='va'>cortex_mrna</span> <span class='op'>&lt;-</span> <span class='fu'>ewceData</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ewceData/man/cortex_mrna.html'>cortex_mrna</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'><span class='co'># Use only a subset to keep the example quick</span>
<span class='va'>expData</span> <span class='op'>&lt;-</span> <span class='va'>cortex_mrna</span><span class='op'>$</span><span class='va'>exp</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>100</span>, <span class='op'>]</span>
<span class='va'>l1</span> <span class='op'>&lt;-</span> <span class='va'>cortex_mrna</span><span class='op'>$</span><span class='va'>annot</span><span class='op'>$</span><span class='va'>level1class</span>
<span class='va'>l2</span> <span class='op'>&lt;-</span> <span class='va'>cortex_mrna</span><span class='op'>$</span><span class='va'>annot</span><span class='op'>$</span><span class='va'>level2class</span>
<span class='va'>annotLevels</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>l1 <span class='op'>=</span> <span class='va'>l1</span>, l2 <span class='op'>=</span> <span class='va'>l2</span><span class='op'>)</span>
<span class='va'>fNames_ALLCELLS</span> <span class='op'>&lt;-</span> <span class='fu'>EWCE</span><span class='fu'>::</span><span class='fu'>generate_celltype_data</span><span class='op'>(</span>
    exp <span class='op'>=</span> <span class='va'>expData</span>,
    annotLevels <span class='op'>=</span> <span class='va'>annotLevels</span>,
    groupName <span class='op'>=</span> <span class='st'>"allKImouse"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>+  1  core(s) assigned as workers ( 63  reserved).</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>+ Calculating normalized mean expression.</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>+ Calculating normalized specificity.</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Loading required namespace: ggdendro</span></div><div class='output co'>#&gt; <span class='message'>+ Saving results ==&gt;  /tmp/Rtmp5DLuZU/ctd_allKImouse.rda</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


