<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Multiple EWCE results from multiple studies — merged_ewce • EWCE</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Multiple EWCE results from multiple studies — merged_ewce" />
<meta property="og:description" content="merged_ewce combines enrichment results from multiple studies
targetting the same scientific problem" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/conditional.html">Conditional enrichment examples</a>
    </li>
    <li>
      <a href="../articles/create_CTD.html">Creating CellTypeDatasets</a>
    </li>
    <li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
    <li>
      <a href="../articles/transcriptomes.html">Applications to transcriptomes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/NathanSkene/EWCE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Multiple EWCE results from multiple studies</h1>
    <small class="dont-index">Source: <a href='https://github.com/NathanSkene/EWCE/blob/master/R/merged_ewce.r'><code>R/merged_ewce.r</code></a></small>
    <div class="hidden name"><code>merged_ewce.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>merged_ewce</code> combines enrichment results from multiple studies
targetting the same scientific problem</p>
    </div>

    <pre class="usage"><span class='fu'>merged_ewce</span><span class='op'>(</span><span class='va'>results</span>, reps <span class='op'>=</span> <span class='fl'>100</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>results</th>
      <td><p>a list of EWCE results generated using
<a href='add_res_to_merging_list.html'>add_res_to_merging_list</a>.</p></td>
    </tr>
    <tr>
      <th>reps</th>
      <td><p>Number of random gene lists to generate (Default=100 but should
be &gt;=10,000 for publication-quality results).</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>dataframe in which each row gives the statistics (p-value, fold
change and number of standard deviations from the mean) associated with the
enrichment of the stated cell type in the gene list.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Load the single cell data</span>
<span class='va'>ctd</span> <span class='op'>&lt;-</span> <span class='fu'>ewceData</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ewceData/man/ctd.html'>ctd</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'>
<span class='co'># Use 3 bootstrap lists for speed, for publishable analysis use &gt;10000</span>
<span class='va'>reps</span> <span class='op'>&lt;-</span> <span class='fl'>3</span>
<span class='co'># Use 5 up/down regulated genes (thresh) for speed, default is 250</span>
<span class='va'>thresh</span> <span class='op'>&lt;-</span> <span class='fl'>5</span>

<span class='co'># Load the data</span>
<span class='va'>tt_alzh_BA36</span> <span class='op'>&lt;-</span> <span class='fu'>ewceData</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ewceData/man/tt_alzh_BA36.html'>tt_alzh_BA36</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'><span class='va'>tt_alzh_BA44</span> <span class='op'>&lt;-</span> <span class='fu'>ewceData</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ewceData/man/tt_alzh_BA44.html'>tt_alzh_BA44</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?ewceData and browseVignettes('ewceData') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'>
<span class='co'># Run EWCE analysis</span>
<span class='va'>tt_results_36</span> <span class='op'>&lt;-</span> <span class='fu'>EWCE</span><span class='fu'>::</span><span class='fu'><a href='ewce_expression_data.html'>ewce_expression_data</a></span><span class='op'>(</span>
    sct_data <span class='op'>=</span> <span class='va'>ctd</span>,
    tt <span class='op'>=</span> <span class='va'>tt_alzh_BA36</span>,
    thresh <span class='op'>=</span> <span class='va'>thresh</span>,
    annotLevel <span class='op'>=</span> <span class='fl'>1</span>,
    reps <span class='op'>=</span> <span class='va'>reps</span>,
    ttSpecies <span class='op'>=</span> <span class='st'>"human"</span>,
    sctSpecies <span class='op'>=</span> <span class='st'>"mouse"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Returning 10,854 unique genes from the user-supplied bg.</span></div><div class='output co'>#&gt; <span class='message'>Standardising CellTypeDataset</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Checking gene list inputs.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: human</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for human</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 19,129 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 19,129 genes from human.</span></div><div class='output co'>#&gt; <span class='message'>Standardising sct_data.</span></div><div class='output co'>#&gt; <span class='message'>Converting gene list input to standardised human genes.</span></div><div class='output co'>#&gt; <span class='message'>Running without gene size control.</span></div><div class='output co'>#&gt; <span class='message'>6 hit genes remain after filtering.</span></div><div class='output co'>#&gt; <span class='message'>Computing summed proportions.</span></div><div class='output co'>#&gt; <span class='message'>Testing for enrichment in 7 cell types...</span></div><div class='output co'>#&gt; <span class='message'>Sorting results by p-value.</span></div><div class='output co'>#&gt; <span class='message'>Computing BH-corrected q-values.</span></div><div class='output co'>#&gt; <span class='message'>1 significant cell type enrichment results @ q&lt;0.05 : </span></div><div class='output co'>#&gt; <span class='message'>          CellType annotLevel p fold_change sd_from_mean q</span>
#&gt; <span class='message'>1 oligodendrocytes          1 0    1.945157     3.480483 0</span></div><div class='output co'>#&gt; <span class='message'>Returning 10,854 unique genes from the user-supplied bg.</span></div><div class='output co'>#&gt; <span class='message'>Standardising CellTypeDataset</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Checking gene list inputs.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: human</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for human</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 19,129 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 19,129 genes from human.</span></div><div class='output co'>#&gt; <span class='message'>Standardising sct_data.</span></div><div class='output co'>#&gt; <span class='message'>Converting gene list input to standardised human genes.</span></div><div class='output co'>#&gt; <span class='message'>Running without gene size control.</span></div><div class='output co'>#&gt; <span class='message'>5 hit genes remain after filtering.</span></div><div class='output co'>#&gt; <span class='message'>Computing summed proportions.</span></div><div class='output co'>#&gt; <span class='message'>Testing for enrichment in 7 cell types...</span></div><div class='output co'>#&gt; <span class='message'>Sorting results by p-value.</span></div><div class='output co'>#&gt; <span class='message'>Computing BH-corrected q-values.</span></div><div class='output co'>#&gt; <span class='message'>2 significant cell type enrichment results @ q&lt;0.05 : </span></div><div class='output co'>#&gt; <span class='message'>      CellType annotLevel p fold_change sd_from_mean q</span>
#&gt; <span class='message'>1    microglia          1 0    1.422840    0.8918925 0</span>
#&gt; <span class='message'>2 pyramidal SS          1 0    1.146459    0.7165786 0</span></div><div class='input'><span class='va'>tt_results_44</span> <span class='op'>&lt;-</span> <span class='fu'>EWCE</span><span class='fu'>::</span><span class='fu'><a href='ewce_expression_data.html'>ewce_expression_data</a></span><span class='op'>(</span>
    sct_data <span class='op'>=</span> <span class='va'>ctd</span>,
    tt <span class='op'>=</span> <span class='va'>tt_alzh_BA44</span>,
    thresh <span class='op'>=</span> <span class='va'>thresh</span>,
    annotLevel <span class='op'>=</span> <span class='fl'>1</span>,
    reps <span class='op'>=</span> <span class='va'>reps</span>,
    ttSpecies <span class='op'>=</span> <span class='st'>"human"</span>,
    sctSpecies <span class='op'>=</span> <span class='st'>"mouse"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Returning 10,854 unique genes from the user-supplied bg.</span></div><div class='output co'>#&gt; <span class='message'>Standardising CellTypeDataset</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Checking gene list inputs.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: human</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for human</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 19,129 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 19,129 genes from human.</span></div><div class='output co'>#&gt; <span class='message'>Standardising sct_data.</span></div><div class='output co'>#&gt; <span class='message'>Converting gene list input to standardised human genes.</span></div><div class='output co'>#&gt; <span class='message'>Running without gene size control.</span></div><div class='output co'>#&gt; <span class='message'>6 hit genes remain after filtering.</span></div><div class='output co'>#&gt; <span class='message'>Computing summed proportions.</span></div><div class='output co'>#&gt; <span class='message'>Testing for enrichment in 7 cell types...</span></div><div class='output co'>#&gt; <span class='message'>Sorting results by p-value.</span></div><div class='output co'>#&gt; <span class='message'>Computing BH-corrected q-values.</span></div><div class='output co'>#&gt; <span class='message'>2 significant cell type enrichment results @ q&lt;0.05 : </span></div><div class='output co'>#&gt; <span class='message'>           CellType annotLevel p fold_change sd_from_mean q</span>
#&gt; <span class='message'>1  oligodendrocytes          1 0    1.438299     4.677587 0</span>
#&gt; <span class='message'>2 endothelial-mural          1 0    1.923386     3.394031 0</span></div><div class='output co'>#&gt; <span class='message'>Returning 10,854 unique genes from the user-supplied bg.</span></div><div class='output co'>#&gt; <span class='message'>Standardising CellTypeDataset</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Converting to sparse matrix.</span></div><div class='output co'>#&gt; <span class='message'>Checking gene list inputs.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: human</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for human</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 19,129 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 19,129 genes from human.</span></div><div class='output co'>#&gt; <span class='message'>Standardising sct_data.</span></div><div class='output co'>#&gt; <span class='message'>Converting gene list input to standardised human genes.</span></div><div class='output co'>#&gt; <span class='message'>Running without gene size control.</span></div><div class='output co'>#&gt; <span class='message'>5 hit genes remain after filtering.</span></div><div class='output co'>#&gt; <span class='message'>Computing summed proportions.</span></div><div class='output co'>#&gt; <span class='message'>Testing for enrichment in 7 cell types...</span></div><div class='output co'>#&gt; <span class='message'>Sorting results by p-value.</span></div><div class='output co'>#&gt; <span class='message'>Computing BH-corrected q-values.</span></div><div class='output co'>#&gt; <span class='message'>1 significant cell type enrichment results @ q&lt;0.05 : </span></div><div class='output co'>#&gt; <span class='message'>       CellType annotLevel p fold_change sd_from_mean q</span>
#&gt; <span class='message'>1 pyramidal CA1          1 0    1.359822     3.275729 0</span></div><div class='input'>
<span class='co'># Fill a list with the results</span>
<span class='va'>results</span> <span class='op'>&lt;-</span> <span class='fu'>EWCE</span><span class='fu'>::</span><span class='fu'><a href='add_res_to_merging_list.html'>add_res_to_merging_list</a></span><span class='op'>(</span><span class='va'>tt_results_36</span><span class='op'>)</span>
<span class='va'>results</span> <span class='op'>&lt;-</span> <span class='fu'>EWCE</span><span class='fu'>::</span><span class='fu'><a href='add_res_to_merging_list.html'>add_res_to_merging_list</a></span><span class='op'>(</span><span class='va'>tt_results_44</span>, <span class='va'>results</span><span class='op'>)</span>

<span class='co'># Perform the merged analysis</span>
<span class='co'># For publication reps should be higher</span>
<span class='va'>merged_res</span> <span class='op'>&lt;-</span> <span class='fu'>EWCE</span><span class='fu'>::</span><span class='fu'>merged_ewce</span><span class='op'>(</span>
    results <span class='op'>=</span> <span class='va'>results</span>,
    reps <span class='op'>=</span> <span class='fl'>2</span>
<span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span><span class='va'>merged_res</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                                   CellType       p        fc sd_from_mean
#&gt; astrocytes_ependymal  astrocytes_ependymal 0.10910 1.1678985   1.16198959
#&gt; endothelial-mural        endothelial-mural 0.10710 1.1688657   1.25405010
#&gt; interneurons                  interneurons 0.67155 0.7853842  -0.69648688
#&gt; microglia                        microglia 0.55440 0.7672514  -0.53308113
#&gt; oligodendrocytes          oligodendrocytes 0.00000 1.6584841   6.20503534
#&gt; pyramidal CA1                pyramidal CA1 0.66670 0.8605966  -0.90599247
#&gt; pyramidal SS                  pyramidal SS 0.77715 0.8408472  -0.67325296
#&gt; astrocytes_ependymal1 astrocytes_ependymal 0.33665 1.0835837   0.44723149
#&gt; endothelial-mural1       endothelial-mural 0.55350 0.7635425  -0.66420002
#&gt; interneurons1                 interneurons 1.00000 0.8301179  -1.47510472
#&gt; microglia1                       microglia 0.22160 1.0609015   0.23482404
#&gt; oligodendrocytes1         oligodendrocytes 0.33665 1.3615614   0.88132354
#&gt; pyramidal CA11               pyramidal CA1 0.00000 1.1758751   2.24622798
#&gt; pyramidal SS1                 pyramidal SS 0.33490 1.0068599   0.03247843
#&gt;                       Direction
#&gt; astrocytes_ependymal         Up
#&gt; endothelial-mural            Up
#&gt; interneurons                 Up
#&gt; microglia                    Up
#&gt; oligodendrocytes             Up
#&gt; pyramidal CA1                Up
#&gt; pyramidal SS                 Up
#&gt; astrocytes_ependymal1      Down
#&gt; endothelial-mural1         Down
#&gt; interneurons1              Down
#&gt; microglia1                 Down
#&gt; oligodendrocytes1          Down
#&gt; pyramidal CA11             Down
#&gt; pyramidal SS1              Down</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


