<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting started • EWCE</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting started">
<meta property="og:description" content="EWCE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/NathanSkene/EWCE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting started</h1>
                        <h4 data-toc-skip class="author"><h4>
Authors: Alan Murphy, Brian Schilder, and Nathan Skene
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
Updated: Jun-21-2022
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/NathanSkene/EWCE/blob/HEAD/vignettes/EWCE.Rmd" class="external-link"><code>vignettes/EWCE.Rmd</code></a></small>
      <div class="hidden name"><code>EWCE.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <code>EWCE</code> R package is designed to facilitate expression
weighted cell type enrichment analysis as described in our <em>Frontiers
in Neuroscience</em> paper<span class="citation"><sup>1</sup></span>.
<code>EWCE</code> can be applied to any gene list.</p>
<p>Using <code>EWCE</code> essentially involves two steps:</p>
<ol style="list-style-type: decimal">
<li>Prepare a single-cell reference; i.e. CellTypeDataset (CTD).
Alternatively, you can use one of the pre-generated CTDs we provide via
the package <code>ewceData</code> (which comes with
<code>EWCE</code>).<br>
</li>
<li>Run cell type enrichment on a gene list using the
<code>bootstrap_enrichment_test</code> function.</li>
</ol>
<p><strong>NOTE</strong>: This documentation is for the development
version of <code>EWCE</code>. See <a href="https://bioconductor.org/packages/release/bioc/html/EWCE.html" class="external-link">Bioconductor</a>
for documentation on the current release version.</p>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/NathanSkene/EWCE" class="external-link">EWCE</a></span><span class="op">)</span> </code></pre></div>
<pre><code><span class="co">## Loading required package: RNOmni</span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span>

<span class="co">#### Package name ####</span>
<span class="va">pkg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">tolower</a></span><span class="op">(</span><span class="st">"EWCE"</span><span class="op">)</span>
<span class="co">#### Username of DockerHub account ####</span>
<span class="va">docker_user</span> <span class="op">&lt;-</span> <span class="st">"neurogenomicslab"</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="run-cell-type-enrichment-tests">Run cell-type enrichment tests<a class="anchor" aria-label="anchor" href="#run-cell-type-enrichment-tests"></a>
</h2>
<div class="section level3">
<h3 id="prepare-input-data">1. Prepare input data<a class="anchor" aria-label="anchor" href="#prepare-input-data"></a>
</h3>
<div class="section level4">
<h4 id="celltypedataset">CellTypeDataset<a class="anchor" aria-label="anchor" href="#celltypedataset"></a>
</h4>
<p>Load a CTD previously generated from mouse cortex and hypothalamus
single-cell RNA-seq data (from the Karolinska Institute).</p>
<div class="section level5">
<h5 id="ctd-levels">CTD levels<a class="anchor" aria-label="anchor" href="#ctd-levels"></a>
</h5>
<p>Each level of a CTD corresponds to increasingly refined
cell-type/-subtype annotations. For example, in the CTD
<code><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ewceData::ctd()</a></code> level 1 includes the cell-type
“interneurons”, while level 2 breaks these this group into 16 different
interneuron subtypes (“Int…”).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## snapshotDate(): 2022-06-15</span></code></pre>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
</div>
<div class="section level5">
<h5 id="plot-ctd-mean_exp">Plot CTD mean_exp<a class="anchor" aria-label="anchor" href="#plot-ctd-mean_exp"></a>
</h5>
<p>Plot the expression of four markers genes across all cell types in
the CTD.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plt_exp</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_ctd.html">plot_ctd</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,
                        level <span class="op">=</span> <span class="fl">1</span>,
                        genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,
                        metric <span class="op">=</span> <span class="st">"mean_exp"</span><span class="op">)</span></code></pre></div>
<p><img src="EWCE_files/figure-html/unnamed-chunk-2-1.png" width="672"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plt_spec</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_ctd.html">plot_ctd</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,
                         level <span class="op">=</span> <span class="fl">2</span>,
                         genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,
                         metric <span class="op">=</span> <span class="st">"specificity"</span><span class="op">)</span></code></pre></div>
<p><img src="EWCE_files/figure-html/unnamed-chunk-3-1.png" width="864"></p>
</div>
</div>
<div class="section level4">
<h4 id="gene-list">Gene list<a class="anchor" aria-label="anchor" href="#gene-list"></a>
</h4>
<p>Gene lists input into <em>EWCE</em> can comes from any source
(e.g. GWAS, candidate genes, pathways).</p>
<p>Here, we provide an example gene list of Alzheimer’s disease-related
nominated from a GWAS.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/example_genelist.html" class="external-link">example_genelist</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  [1] "APOE"     "BIN1"     "CLU"      "ABCA7"    "CR1"      "PICALM"  </span>
<span class="co">##  [7] "MS4A6A"   "CD33"     "MS4A4E"   "CD2AP"    "EOGA1"    "INPP5D"  </span>
<span class="co">## [13] "MEF2C"    "HLA-DRB5" "ZCWPW1"   "NME8"     "PTK2B"    "CELF1"   </span>
<span class="co">## [19] "SORL1"    "FERMT2"   "SLC24A4"  "CASS4"</span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="run-cell-type-enrichment-tests-1">2. Run cell type enrichment tests<a class="anchor" aria-label="anchor" href="#run-cell-type-enrichment-tests-1"></a>
</h3>
<p>We now run the cell type enrichment tests on the gene list. Since the
CTD is from mouse data (and is annotated using mouse genes) we specify
the argument <code>sctSpecies="mouse"</code>.
<code>bootstrap_enrichment_test</code> will automaticlaly convert the
mouse genes to human genes.</p>
<p>Since the gene list came from GWAS in humans, we set
<code>genelistSpecies="human"</code>.</p>
<p><em>Note</em>: We set the seed at the top of this vignette to ensure
reproducibility in the bootstrap sampling function.</p>
<div class="section level5">
<h5 id="hyperparameters">Hyperparameters<a class="anchor" aria-label="anchor" href="#hyperparameters"></a>
</h5>
<p><em>Note</em>: We use 100 repetitions here for the purposes of a
quick example, but in practice you would want to use
<code>reps=10000</code> for publishable results.</p>
</div>
<div class="section level5">
<h5 id="parallelisation">Parallelisation<a class="anchor" aria-label="anchor" href="#parallelisation"></a>
</h5>
<p>You can now speed up the bootstrapping process by parallelising
across multiple cores with the parameter <code>no_cores</code>
(<code>=1</code> by default).</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">reps</span> <span class="op">&lt;-</span> <span class="fl">100</span>
<span class="va">annotLevel</span> <span class="op">&lt;-</span> <span class="fl">1</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">full_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,
                                                sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,
                                                genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,
                                                hits <span class="op">=</span> <span class="va">hits</span>, 
                                                reps <span class="op">=</span> <span class="va">reps</span>,
                                                annotLevel <span class="op">=</span> <span class="va">annotLevel</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## 1 core(s) assigned as workers (1reserved).</span></code></pre>
<pre><code><span class="co">## Generating gene background for mouse x human ==&gt; human</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in homologene.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Gene table with 21,207 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 21,207 genes from mouse.</span></code></pre>
<pre><code><span class="co">## --</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in homologene.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## --</span></code></pre>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## data.frame format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from Gene.Symbol.</span></code></pre>
<pre><code><span class="co">## 21,207 genes extracted.</span></code></pre>
<pre><code><span class="co">## Converting mouse ==&gt; human orthologs using: homologene</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in homologene.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in homologene.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Checking for genes without orthologs in human.</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Extracting genes from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Checking for genes without 1:1 orthologs.</span></code></pre>
<pre><code><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene (many:1).</span></code></pre>
<pre><code><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene (1:many).</span></code></pre>
<pre><code><span class="co">## Filtering gene_df with gene_map</span></code></pre>
<pre><code><span class="co">## Adding input_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## Adding ortholog_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## Total genes dropped after convert_orthologs :</span>
<span class="co">##    4,725 / 21,207 (22%)</span></code></pre>
<pre><code><span class="co">## Total genes remaining after convert_orthologs :</span>
<span class="co">##    16,482 / 21,207 (78%)</span></code></pre>
<pre><code><span class="co">## --</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in homologene.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## --</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 intersect background genes used.</span></code></pre>
<pre><code><span class="co">## Standardising CellTypeDataset</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## Checking gene list inputs.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in homologene.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Standardising sct_data.</span></code></pre>
<pre><code><span class="co">## Converting gene list input to standardised human genes.</span></code></pre>
<pre><code><span class="co">## Running without gene size control.</span></code></pre>
<pre><code><span class="co">## 17 hit genes remain after filtering.</span></code></pre>
<pre><code><span class="co">## Computing summed proportions.</span></code></pre>
<pre><code><span class="co">## Testing for enrichment in 7 cell types...</span></code></pre>
<pre><code><span class="co">## Sorting results by p-value.</span></code></pre>
<pre><code><span class="co">## Computing BH-corrected q-values.</span></code></pre>
<pre><code><span class="co">## 1 significant cell type enrichment results @ q&lt;0.05 :</span></code></pre>
<pre><code><span class="co">##    CellType annotLevel p fold_change sd_from_mean q</span>
<span class="co">## 1 microglia          1 0    2.003754     3.822969 0</span></code></pre>
<p>The main table of results is stored in
<code>full_results$results</code>.</p>
<p>In this case, microglia were the only cell type that was
significantly enriched in the Alzheimer’s disease gene list.</p>
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">full_results</span><span class="op">$</span><span class="va">results</span><span class="op">)</span></code></pre></div>
<table class="table">
<colgroup>
<col width="23%">
<col width="23%">
<col width="12%">
<col width="5%">
<col width="13%">
<col width="14%">
<col width="6%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">CellType</th>
<th align="right">annotLevel</th>
<th align="right">p</th>
<th align="right">fold_change</th>
<th align="right">sd_from_mean</th>
<th align="right">q</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">microglia</td>
<td align="left">microglia</td>
<td align="right">1</td>
<td align="right">0.00</td>
<td align="right">2.0037539</td>
<td align="right">3.8229690</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">astrocytes_ependymal</td>
<td align="left">astrocytes_ependymal</td>
<td align="right">1</td>
<td align="right">0.11</td>
<td align="right">1.3594176</td>
<td align="right">1.4291523</td>
<td align="right">0.385</td>
</tr>
<tr class="odd">
<td align="left">oligodendrocytes</td>
<td align="left">oligodendrocytes</td>
<td align="right">1</td>
<td align="right">0.78</td>
<td align="right">0.7903958</td>
<td align="right">-0.8909301</td>
<td align="right">1.000</td>
</tr>
<tr class="even">
<td align="left">endothelial_mural</td>
<td align="left">endothelial_mural</td>
<td align="right">1</td>
<td align="right">0.83</td>
<td align="right">0.7587306</td>
<td align="right">-0.9521828</td>
<td align="right">1.000</td>
</tr>
<tr class="odd">
<td align="left">pyramidal_SS</td>
<td align="left">pyramidal_SS</td>
<td align="right">1</td>
<td align="right">0.84</td>
<td align="right">0.8338200</td>
<td align="right">-0.9271986</td>
<td align="right">1.000</td>
</tr>
<tr class="even">
<td align="left">pyramidal_CA1</td>
<td align="left">pyramidal_CA1</td>
<td align="right">1</td>
<td align="right">0.90</td>
<td align="right">0.7882024</td>
<td align="right">-1.1989117</td>
<td align="right">1.000</td>
</tr>
<tr class="odd">
<td align="left">interneurons</td>
<td align="left">interneurons</td>
<td align="right">1</td>
<td align="right">1.00</td>
<td align="right">0.3868205</td>
<td align="right">-3.1123590</td>
<td align="right">1.000</td>
</tr>
</tbody>
</table>
<p>The results can be visualised using another function, which shows for
each cell type, the number of standard deviations from the mean the
level of expression was found to be in the target gene list, relative to
the bootstrapped mean.</p>
<p>The dendrogram at the top shows how the cell types are hierarchically
clustered by transcriptional similarity.</p>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">full_results</span><span class="op">$</span><span class="va">results</span>,
                           mtc_method <span class="op">=</span> <span class="st">"BH"</span>,
                           ctd <span class="op">=</span> <span class="va">ctd</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Loading required namespace: cowplot</span></code></pre>
<pre><code><span class="co">## Loading required namespace: gridExtra</span></code></pre>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span></code></pre>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">withDendro</span><span class="op">)</span></code></pre></div>
<p><img src="EWCE_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="docker">Docker<a class="anchor" aria-label="anchor" href="#docker"></a>
</h2>
<p>ewce is now available via <a href="https://hub.docker.com/repository/docker/neurogenomicslab/ewce" class="external-link">DockerHub</a>
as a containerised environment with Rstudio and all necessary
dependencies pre-installed.</p>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
</div>
<div class="section level3">
<h3 id="method-1-via-docker">Method 1: via Docker<a class="anchor" aria-label="anchor" href="#method-1-via-docker"></a>
</h3>
<p>First, <a href="https://docs.docker.com/get-docker/" class="external-link">install
Docker</a> if you have not already.</p>
<p>Create an image of the <a href="https://www.docker.com/" class="external-link">Docker</a>
container in command line:</p>
<pre><code>docker pull neurogenomicslab/ewce</code></pre>
<p>Once the image has been created, you can launch it with:</p>
<pre><code>docker run \
  -d \
  -e ROOT=true \
  -e PASSWORD=bioc \
  -v ~/Desktop:/Desktop \
  -v /Volumes:/Volumes \
  -p 8787:8787 \
  neurogenomicslab/ewce</code></pre>
<ul>
<li>The <code>-d</code> ensures the container will run in “detached”
mode, which means it will persist even after you’ve closed your command
line session.<br>
</li>
<li>Optionally, you can also install the <a href="https://www.docker.com/products/docker-desktop" class="external-link">Docker Desktop</a>
to easily manage your containers.<br>
</li>
<li>You can set the password to whatever you like by changing the
<code>-e PASSWORD=...</code> flag.<br>
</li>
<li>The username will be <em>“rstudio”</em> by default.</li>
</ul>
</div>
<div class="section level3">
<h3 id="method-2-via-singularity">Method 2: via Singularity<a class="anchor" aria-label="anchor" href="#method-2-via-singularity"></a>
</h3>
<p>If you are using a system that does not allow Docker (as is the case
for many institutional computing clusters), you can instead <a href="https://sylabs.io/guides/2.6/user-guide/singularity_and_docker.html" class="external-link">install
Docker images via Singularity</a>.</p>
<pre><code>singularity pull docker://neurogenomicslab/ewce</code></pre>
</div>
<div class="section level3">
<h3 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h3>
<p>Finally, launch the containerised Rstudio by entering the following
URL in any web browser: <em><a href="http://localhost:8787/" class="external-link uri">http://localhost:8787/</a></em></p>
<p>Login using the credentials set during the Installation steps.</p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.2.0 (2022-04-22)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 20.04.4 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] ewceData_1.5.0      ExperimentHub_2.5.0 AnnotationHub_3.5.0</span>
<span class="co">## [4] BiocFileCache_2.5.0 dbplyr_2.2.0        BiocGenerics_0.43.0</span>
<span class="co">## [7] EWCE_1.5.3          RNOmni_1.0.0        BiocStyle_2.25.0   </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] colorspace_2.0-3              ggtree_3.5.0                 </span>
<span class="co">##   [3] ggsignif_0.6.3                ellipsis_0.3.2               </span>
<span class="co">##   [5] rprojroot_2.0.3               XVector_0.37.0               </span>
<span class="co">##   [7] GenomicRanges_1.49.0          fs_1.5.2                     </span>
<span class="co">##   [9] aplot_0.1.6                   farver_2.1.0                 </span>
<span class="co">##  [11] ggpubr_0.4.0                  bit64_4.0.5                  </span>
<span class="co">##  [13] interactiveDisplayBase_1.35.0 AnnotationDbi_1.59.1         </span>
<span class="co">##  [15] fansi_1.0.3                   orthogene_1.3.1              </span>
<span class="co">##  [17] codetools_0.2-18              cachem_1.0.6                 </span>
<span class="co">##  [19] knitr_1.39                    jsonlite_1.8.0               </span>
<span class="co">##  [21] broom_0.8.0                   png_0.1-7                    </span>
<span class="co">##  [23] shiny_1.7.1                   BiocManager_1.30.18          </span>
<span class="co">##  [25] compiler_4.2.0                httr_1.4.3                   </span>
<span class="co">##  [27] backports_1.4.1               lazyeval_0.2.2               </span>
<span class="co">##  [29] assertthat_0.2.1              Matrix_1.4-1                 </span>
<span class="co">##  [31] fastmap_1.1.0                 limma_3.53.3                 </span>
<span class="co">##  [33] cli_3.3.0                     later_1.3.0                  </span>
<span class="co">##  [35] htmltools_0.5.2               tools_4.2.0                  </span>
<span class="co">##  [37] gtable_0.3.0                  glue_1.6.2                   </span>
<span class="co">##  [39] GenomeInfoDbData_1.2.8        reshape2_1.4.4               </span>
<span class="co">##  [41] dplyr_1.0.9                   rappdirs_0.3.3               </span>
<span class="co">##  [43] Rcpp_1.0.8.3                  carData_3.0-5                </span>
<span class="co">##  [45] Biobase_2.57.1                jquerylib_0.1.4              </span>
<span class="co">##  [47] pkgdown_2.0.4.9000            vctrs_0.4.1                  </span>
<span class="co">##  [49] Biostrings_2.65.1             nlme_3.1-158                 </span>
<span class="co">##  [51] ape_5.6-2                     babelgene_22.3               </span>
<span class="co">##  [53] xfun_0.31                     stringr_1.4.0                </span>
<span class="co">##  [55] mime_0.12                     lifecycle_1.0.1              </span>
<span class="co">##  [57] rstatix_0.7.0                 zlibbioc_1.43.0              </span>
<span class="co">##  [59] scales_1.2.0                  ragg_1.2.2                   </span>
<span class="co">##  [61] promises_1.2.0.1              MatrixGenerics_1.9.0         </span>
<span class="co">##  [63] parallel_4.2.0                SummarizedExperiment_1.27.1  </span>
<span class="co">##  [65] gprofiler2_0.2.1              SingleCellExperiment_1.19.0  </span>
<span class="co">##  [67] yaml_2.3.5                    curl_4.3.2                   </span>
<span class="co">##  [69] gridExtra_2.3                 memoise_2.0.1                </span>
<span class="co">##  [71] ggfun_0.0.6                   ggplot2_3.3.6                </span>
<span class="co">##  [73] yulab.utils_0.0.4             sass_0.4.1                   </span>
<span class="co">##  [75] stringi_1.7.6                 RSQLite_2.2.14               </span>
<span class="co">##  [77] highr_0.9                     BiocVersion_3.16.0           </span>
<span class="co">##  [79] S4Vectors_0.35.1              desc_1.4.1                   </span>
<span class="co">##  [81] tidytree_0.3.9                filelock_1.0.2               </span>
<span class="co">##  [83] BiocParallel_1.31.8           GenomeInfoDb_1.33.3          </span>
<span class="co">##  [85] rlang_1.0.2                   pkgconfig_2.0.3              </span>
<span class="co">##  [87] systemfonts_1.0.4             bitops_1.0-7                 </span>
<span class="co">##  [89] matrixStats_0.62.0            evaluate_0.15                </span>
<span class="co">##  [91] lattice_0.20-45               purrr_0.3.4                  </span>
<span class="co">##  [93] labeling_0.4.2                htmlwidgets_1.5.4            </span>
<span class="co">##  [95] treeio_1.21.0                 patchwork_1.1.1              </span>
<span class="co">##  [97] cowplot_1.1.1                 bit_4.0.4                    </span>
<span class="co">##  [99] tidyselect_1.1.2              plyr_1.8.7                   </span>
<span class="co">## [101] magrittr_2.0.3                bookdown_0.27                </span>
<span class="co">## [103] R6_2.5.1                      IRanges_2.31.0               </span>
<span class="co">## [105] generics_0.1.2                DelayedArray_0.23.0          </span>
<span class="co">## [107] DBI_1.1.3                     pillar_1.7.0                 </span>
<span class="co">## [109] KEGGREST_1.37.2               abind_1.4-5                  </span>
<span class="co">## [111] RCurl_1.98-1.7                tibble_3.1.7                 </span>
<span class="co">## [113] homologene_1.4.68.19.3.27     crayon_1.5.1                 </span>
<span class="co">## [115] car_3.1-0                     utf8_1.2.2                   </span>
<span class="co">## [117] plotly_4.10.0                 rmarkdown_2.14               </span>
<span class="co">## [119] grid_4.2.0                    data.table_1.14.2            </span>
<span class="co">## [121] blob_1.2.3                    digest_0.6.29                </span>
<span class="co">## [123] xtable_1.8-4                  HGNChelper_0.8.1             </span>
<span class="co">## [125] tidyr_1.2.0                   httpuv_1.6.5                 </span>
<span class="co">## [127] gridGraphics_0.5-1            textshaping_0.3.6            </span>
<span class="co">## [129] stats4_4.2.0                  munsell_0.5.0                </span>
<span class="co">## [131] viridisLite_0.4.0             ggplotify_0.1.0              </span>
<span class="co">## [133] bslib_0.3.1</span></code></pre>
</details>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-skene_2016" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Skene, N. &amp; Grant, S. Identification of
vulnerable cell types in major brain disorders using single cell
transcriptomes and expression weighted cell type enrichment.
<em>Frontiers in Neuroscience</em> (2016). doi:<a href="https://doi.org/10.3389/fnins.2016.00016" class="external-link">10.3389/fnins.2016.00016</a>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
