<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Extended examples • EWCE</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Extended examples">
<meta property="og:description" content="EWCE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.3.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/NathanSkene/EWCE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Extended examples</h1>
                        <h4 data-toc-skip class="author"><h4>
Authors: Alan Murphy, Brian Schilder, and Nathan Skene
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
Updated: Dec-23-2021
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/NathanSkene/EWCE/blob/HEAD/vignettes/extended.Rmd" class="external-link"><code>vignettes/extended.Rmd</code></a></small>
      <div class="hidden name"><code>extended.Rmd</code></div>

    </div>

    
    
<div class="section level1">
<h1 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/NathanSkene/EWCE" class="external-link">EWCE</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Loading required package: RNOmni</span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level1">
<h1 id="run-cell-type-enrichment-tests">Run cell-type enrichment tests<a class="anchor" aria-label="anchor" href="#run-cell-type-enrichment-tests"></a>
</h1>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In the following vignette, we provide more a more in-depth version of the examples provided in the <a href="https://nathanskene.github.io/EWCE/articles/EWCE.html" class="external-link">Getting started vignette</a>.</p>
</div>
<div class="section level2">
<h2 id="prepare-input-data">Prepare input data<a class="anchor" aria-label="anchor" href="#prepare-input-data"></a>
</h2>
<div class="section level3">
<h3 id="celltypedataset">CellTypeDataset<a class="anchor" aria-label="anchor" href="#celltypedataset"></a>
</h3>
<p>For this example we use a subset of the genes from the merged dataset generated in the <a href="#create-a-celltypedataset">Create a CellTypeDataset</a> section below, which is accessed using <code><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ewceData::ctd()</a></code>.</p>
<div class="section level4">
<h4 id="ctd-levels">CTD levels<a class="anchor" aria-label="anchor" href="#ctd-levels"></a>
</h4>
<p>Each level of a CTD corresponds to increasingly refined cell-type/-subtype annotations. For example, in the CTD <code><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ewceData::ctd()</a></code> level 1 includes the cell-type “interneurons”, while level 2 breaks these this group into 16 different interneuron subtypes (“Int…”).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Load merged cortex and hypothalamus dataset generated by Karolinska institute</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># i.e. ctd_MergedKI</span></code></pre></div>
<pre><code><span class="co">## snapshotDate(): 2021-11-24</span></code></pre>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_ctd.html">plot_ctd</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,
                      level <span class="op">=</span> <span class="fl">1</span>,
                      genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,
                      metric <span class="op">=</span> <span class="st">"mean_exp"</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-1-1.png" width="700"></p>
</div>
</div>
<div class="section level3">
<h3 id="gene-lists">Gene lists<a class="anchor" aria-label="anchor" href="#gene-lists"></a>
</h3>
<p>For the first demonstration of <code>EWCE</code> we will test for whether genes that are genetically associated with Alzheimer’s disease are enriched in any particular cell type.</p>
<p>This example gene list is stored within the <code>ewceData</code> package:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/example_genelist.html" class="external-link">example_genelist</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  [1] "APOE"     "BIN1"     "CLU"      "ABCA7"    "CR1"      "PICALM"  </span>
<span class="co">##  [7] "MS4A6A"   "CD33"     "MS4A4E"   "CD2AP"    "EOGA1"    "INPP5D"  </span>
<span class="co">## [13] "MEF2C"    "HLA-DRB5" "ZCWPW1"   "NME8"     "PTK2B"    "CELF1"   </span>
<span class="co">## [19] "SORL1"    "FERMT2"   "SLC24A4"  "CASS4"</span></code></pre>
<div class="section level4">
<h4 id="gene-formats-and-species">Gene formats and species<a class="anchor" aria-label="anchor" href="#gene-formats-and-species"></a>
</h4>
<p>All gene IDs are assumed by the package to be provided in gene symbol format (rather than Ensembl/Entrez). Symbols can be provided as any species-specific gene symbols supported by the package <code>orthogene</code>, though the <code>genelistSpecies</code> argument will need to be set appropriately.</p>
<p>Likewise, the single-cell dataset can be from any species, but the <code>sctSpecies</code> argument must be set accordingly.</p>
<p>The example gene list here stores the human genes associated with human disease, and hence are HGNC symbols.</p>
<p>The next step is to determine the most suitable background set. This can be user-supplied, but by default the background is all 1:1 ortholog genes shared by <code>genelistSpecies</code> and <code>sctSpecies</code> that are also present in <code>sct_data</code>.</p>
</div>
<div class="section level4">
<h4 id="notes-on-orthogene">Notes on <code>orthogene</code><a class="anchor" aria-label="anchor" href="#notes-on-orthogene"></a>
</h4>
<p><code>orthogene</code> substantially improves upon previous ortholog translations that used the static <code><a href="https://rdrr.io/pkg/ewceData/man/mouse_to_human_homologs.html" class="external-link">ewceData::mouse_to_human_homologs()</a></code> file as the former is updated using the <a href="https://www.ncbi.nlm.nih.gov/homologene" class="external-link">Homologene</a> database periodically.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp</span> <span class="op">&lt;-</span> <span class="va">ctd</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">mean_exp</span>

<span class="co">#### Old conversion method ####</span>
<span class="va">m2h</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/mouse_to_human_homologs.html" class="external-link">mouse_to_human_homologs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp_old</span> <span class="op">&lt;-</span> <span class="va">exp</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">m2h</span><span class="op">$</span><span class="va">MGI.symbol</span>,<span class="op">]</span>

<span class="co">#### New conversion method (used by EWCE internally) ####</span>
<span class="va">exp_new</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/orthogene/man/convert_orthologs.html" class="external-link">convert_orthologs</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp</span>,
                                        input_species <span class="op">=</span> <span class="st">"mouse"</span>, 
                                        output_species <span class="op">=</span> <span class="st">"human"</span>, 
                                        method <span class="op">=</span> <span class="st">"homologene"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## Dense matrix format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from rownames.</span></code></pre>
<pre><code><span class="co">## 15,259 genes extracted.</span></code></pre>
<pre><code><span class="co">## Converting mouse ==&gt; human orthologs using: homologene</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Checking for genes without orthologs in human.</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 13,416 genes extracted.</span></code></pre>
<pre><code><span class="co">## Extracting genes from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## 13,416 genes extracted.</span></code></pre>
<pre><code><span class="co">## Checking for genes without 1:1 orthologs.</span></code></pre>
<pre><code><span class="co">## Dropping 46 genes that have multiple input_gene per ortholog_gene.</span></code></pre>
<pre><code><span class="co">## Dropping 56 genes that have multiple ortholog_gene per input_gene.</span></code></pre>
<pre><code><span class="co">## Filtering gene_df with gene_map</span></code></pre>
<pre><code><span class="co">## Setting ortholog_gene to rownames.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## Total genes dropped after convert_orthologs :</span>
<span class="co">##    2,016 / 15,259 (13%)</span></code></pre>
<pre><code><span class="co">## Total genes remaining after convert_orthologs :</span>
<span class="co">##    13,243 / 15,259 (87%)</span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#### Report ####</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"The new method retains "</span>,
        <span class="fu"><a href="https://rdrr.io/r/base/formatc.html" class="external-link">formatC</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">exp_new</span><span class="op">)</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">exp_old</span><span class="op">)</span>, big.mark <span class="op">=</span> <span class="st">","</span><span class="op">)</span>,
        <span class="st">" more genes than the old method."</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## The new method retains 918 more genes than the old method.</span></code></pre>
<p><code>orthogene</code> is also used internally to standardise gene lists supplied to <code>EWCE</code> functions, such as <code>EWCE::bootstrap_enrichment_test(hits = &lt;gene_list&gt;)</code>.</p>
<p>Not only can it map these gene lists across species, but it can also map them within species. For example, if you provide a list of Ensembl IDs, it will automatically convert them to standardised HGNC gene symbols so they’re compatible with the similarly standardised CellTypeDataset.</p>
</div>
</div>
<div class="section level3">
<h3 id="setting-analysis-parameters">Setting analysis parameters<a class="anchor" aria-label="anchor" href="#setting-analysis-parameters"></a>
</h3>
<p>We now need to set the parameters for the analysis. For a publishable analysis we would want to generate over 10,000 random lists and determine their expression levels, but for computational speed let us only use <code>reps=100</code>. We want to analyse level 1 annotations so set level to 1.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Use 100 bootstrap lists for speed, for publishable analysis use &gt;=10000</span>
<span class="va">reps</span> <span class="op">&lt;-</span> <span class="fl">100</span> 
<span class="co"># Use level 1 annotations (i.e. Interneurons)</span>
<span class="va">annotLevel</span> <span class="op">&lt;-</span> <span class="fl">1</span> </code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="enrichment-tests">Enrichment tests<a class="anchor" aria-label="anchor" href="#enrichment-tests"></a>
</h2>
<div class="section level3">
<h3 id="default-tests">Default tests<a class="anchor" aria-label="anchor" href="#default-tests"></a>
</h3>
<p>We have now loaded the SCT data, prepared the gene lists and set the parameters. We run the model as follows.</p>
<p><strong>Note</strong>: We set the seed at the top of this vignette to ensure reproducibility in the bootstrap sampling function.</p>
<div class="section level4">
<h4 id="parallelisation">Parallelisation<a class="anchor" aria-label="anchor" href="#parallelisation"></a>
</h4>
<p>You can now speed up the bootstrapping process by parallelising across multiple cores with the parameter <code>no_cores</code> (<code>=1</code> by default).</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Bootstrap significance test, no control for transcript length and GC content </span>
<span class="va">full_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,
                                                sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,
                                                genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,
                                                hits <span class="op">=</span> <span class="va">hits</span>, 
                                                reps <span class="op">=</span> <span class="va">reps</span>,
                                                annotLevel <span class="op">=</span> <span class="va">annotLevel</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## 1 core(s) assigned as workers (1reserved).</span></code></pre>
<pre><code><span class="co">## Generating gene background for mouse x human ==&gt; human</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: mmusculus</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Gene table with 21,207 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 21,207 genes from mmusculus.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## data.frame format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from Gene.Symbol.</span></code></pre>
<pre><code><span class="co">## 21,207 genes extracted.</span></code></pre>
<pre><code><span class="co">## Converting mmusculus ==&gt; hsapiens orthologs using: homologene</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Checking for genes without orthologs in hsapiens.</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Extracting genes from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Checking for genes without 1:1 orthologs.</span></code></pre>
<pre><code><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene.</span></code></pre>
<pre><code><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene.</span></code></pre>
<pre><code><span class="co">## Filtering gene_df with gene_map</span></code></pre>
<pre><code><span class="co">## Adding input_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## Adding ortholog_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## Total genes dropped after convert_orthologs :</span>
<span class="co">##    4,725 / 21,207 (22%)</span></code></pre>
<pre><code><span class="co">## Total genes remaining after convert_orthologs :</span>
<span class="co">##    16,482 / 21,207 (78%)</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 intersect background genes used.</span></code></pre>
<pre><code><span class="co">## Standardising CellTypeDataset</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## Checking gene list inputs.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Standardising sct_data.</span></code></pre>
<pre><code><span class="co">## Converting gene list input to standardised human genes.</span></code></pre>
<pre><code><span class="co">## Running without gene size control.</span></code></pre>
<pre><code><span class="co">## 17 hit genes remain after filtering.</span></code></pre>
<pre><code><span class="co">## Computing summed proportions.</span></code></pre>
<pre><code><span class="co">## Testing for enrichment in 7 cell types...</span></code></pre>
<pre><code><span class="co">## Sorting results by p-value.</span></code></pre>
<pre><code><span class="co">## Computing BH-corrected q-values.</span></code></pre>
<pre><code><span class="co">## 1 significant cell type enrichment results @ q&lt;0.05 :</span></code></pre>
<pre><code><span class="co">##    CellType annotLevel p fold_change sd_from_mean q</span>
<span class="co">## 1 microglia          1 0    2.003754     3.822969 0</span></code></pre>
<p>A note on both the background and target gene lists, other common gene list objects can be used as inputs such as <code>BiocSet::BiocSet</code> and <code>GSEABase::GeneSet</code>. Below is an example of how to format each for the target gene list (<code>hits</code>):</p>
<div class="sourceCode" id="cb148"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"BiocSet"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"BiocSet"</span><span class="op">)</span>
<span class="op">}</span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"GSEABase"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"GSEABase"</span><span class="op">)</span>
<span class="op">}</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BiocSet</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GSEABase</span><span class="op">)</span>

<span class="co"># Save both approaches as hits which will be passed to bootstrap_enrichment_test</span>
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Inpp5d"</span>,<span class="st">"Cd2ap"</span>,<span class="st">"Nme8"</span>,
          <span class="st">"Cass4"</span>,<span class="st">"Mef2c"</span>,<span class="st">"Zcwpw1"</span>,<span class="st">"Bin1"</span>,
          <span class="st">"Clu"</span>,<span class="st">"Celf1"</span>,<span class="st">"Abca7"</span>,<span class="st">"Slc24a4"</span>,
          <span class="st">"Ptk2b"</span>,<span class="st">"Picalm"</span>,<span class="st">"Fermt2"</span>,<span class="st">"Sorl1"</span><span class="op">)</span>

<span class="co">#BiocSet::BiocSet, BiocSet_target contains the gene list target</span>
<span class="va">BiocSet_target</span> <span class="op">&lt;-</span> <span class="fu">BiocSet</span><span class="fu">::</span><span class="fu">BiocSet</span><span class="op">(</span>set1 <span class="op">=</span> <span class="va">genes</span><span class="op">)</span> 
<span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu">BiocSet</span><span class="fu">::</span><span class="fu">es_element</span><span class="op">(</span><span class="va">BiocSet_target</span><span class="op">)</span><span class="op">)</span>

<span class="co">#GSEABase::GeneSet, GeneSet_target contains the gene list target</span>
<span class="va">GeneSet_target</span> <span class="op">&lt;-</span> <span class="fu">GSEABase</span><span class="fu">::</span><span class="fu">GeneSet</span><span class="op">(</span><span class="va">genes</span><span class="op">)</span>
<span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu">GSEABase</span><span class="fu">::</span><span class="fu">geneIds</span><span class="op">(</span><span class="va">GeneSet_target</span><span class="op">)</span> </code></pre></div>
<p>The main table of results is stored in <code>full_results$results</code>. We can see the most significant results using:</p>
<div class="sourceCode" id="cb149"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">full_results</span><span class="op">$</span><span class="va">results</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">CellType</th>
<th align="right">annotLevel</th>
<th align="right">p</th>
<th align="right">fold_change</th>
<th align="right">sd_from_mean</th>
<th align="right">q</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">microglia</td>
<td align="left">microglia</td>
<td align="right">1</td>
<td align="right">0.00</td>
<td align="right">2.0037539</td>
<td align="right">3.8229690</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">astrocytes_ependymal</td>
<td align="left">astrocytes_ependymal</td>
<td align="right">1</td>
<td align="right">0.11</td>
<td align="right">1.3594176</td>
<td align="right">1.4291523</td>
<td align="right">0.385</td>
</tr>
<tr class="odd">
<td align="left">oligodendrocytes</td>
<td align="left">oligodendrocytes</td>
<td align="right">1</td>
<td align="right">0.78</td>
<td align="right">0.7903958</td>
<td align="right">-0.8909301</td>
<td align="right">1.000</td>
</tr>
<tr class="even">
<td align="left">endothelial_mural</td>
<td align="left">endothelial_mural</td>
<td align="right">1</td>
<td align="right">0.83</td>
<td align="right">0.7587306</td>
<td align="right">-0.9521828</td>
<td align="right">1.000</td>
</tr>
<tr class="odd">
<td align="left">pyramidal_SS</td>
<td align="left">pyramidal_SS</td>
<td align="right">1</td>
<td align="right">0.84</td>
<td align="right">0.8338200</td>
<td align="right">-0.9271986</td>
<td align="right">1.000</td>
</tr>
<tr class="even">
<td align="left">pyramidal_CA1</td>
<td align="left">pyramidal_CA1</td>
<td align="right">1</td>
<td align="right">0.90</td>
<td align="right">0.7882024</td>
<td align="right">-1.1989117</td>
<td align="right">1.000</td>
</tr>
<tr class="odd">
<td align="left">interneurons</td>
<td align="left">interneurons</td>
<td align="right">1</td>
<td align="right">1.00</td>
<td align="right">0.3868205</td>
<td align="right">-3.1123590</td>
<td align="right">1.000</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="plot-results">Plot results<a class="anchor" aria-label="anchor" href="#plot-results"></a>
</h4>
<p>The results can be visualised using another function, which shows for each cell type, the number of standard deviations from the mean the level of expression was found to be in the target gene list, relative to the bootstrapped mean:</p>
<div class="sourceCode" id="cb150"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">full_results</span><span class="op">$</span><span class="va">results</span>,
                             mtc_method <span class="op">=</span><span class="st">"BH"</span>,
                             ctd <span class="op">=</span> <span class="va">ctd</span><span class="op">)</span> <span class="co"># optional (unless you want dendrogram)</span></code></pre></div>
<pre><code><span class="co">## Loading required namespace: cowplot</span></code></pre>
<pre><code><span class="co">## Loading required namespace: gridExtra</span></code></pre>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span></code></pre>
<div class="sourceCode" id="cb154"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># print(plot_list$plain)</span></code></pre></div>
<p>For publications it can be useful to plot a dendrogram alongside the plot. This can be done by including the cell type data as an additional argument. The dendrogram should automatically align with the graph ticks (thanks to <a href="https://github.com/ovrhuman" class="external-link">Robert Gordon-Smith</a>, Mres Molecular and Cellular Biosciences, Imperial College London for this solution):</p>
<div class="sourceCode" id="cb155"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">withDendro</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>If you want to view the characteristics of enrichment for each gene within the list then the <code>generate_bootstrap_plots</code> function should be used. This saves the plots into the BootstrapPlots folder. This takes the results of a bootstrapping analysis so as to only generate plots for significant enrichments. The <code>listFileName</code> argument is used to give the generated graphs a particular file name. The <code>savePath</code> argument is used here to save the files to a temporary directory, this can be updated to your preferred location. The file path where it was saved is returned so the temporary directory can be located if used.</p>
<div class="sourceCode" id="cb156"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bt_plot_location</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/generate_bootstrap_plots.html">generate_bootstrap_plots</a></span><span class="op">(</span>
  sct_data <span class="op">=</span> <span class="va">ctd</span>,
  hits <span class="op">=</span> <span class="va">hits</span>,
  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,
  genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,
  reps <span class="op">=</span> <span class="va">reps</span>,
  annotLevel <span class="op">=</span> <span class="va">annotLevel</span>,
  full_results <span class="op">=</span> <span class="va">full_results</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="control-for-transcript-length-and-gc-content">Control for transcript length and GC-content<a class="anchor" aria-label="anchor" href="#control-for-transcript-length-and-gc-content"></a>
</h3>
<p>When analysing genes found through genetic association studies it is important to consider biases which might be introduced as a result of transcript length and GC-content. The package can control for these by selecting the bootstrap lists such that the <em>i<sup>th</sup></em> gene in the random list has properties similar to the<em>i<sup>th</sup></em> gene in the target list. To enable the algorithm to do this it needs to be passed the gene lists as HGNC symbols rather than MGI.</p>
<p>The bootstrapping function then takes different arguments:</p>
<div class="sourceCode" id="cb157"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Bootstrap significance test controlling for transcript length and GC content</span>
<span class="va">cont_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span>
  sct_data <span class="op">=</span> <span class="va">ctd</span>,
  hits <span class="op">=</span> <span class="va">hits</span>,
  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,
  genelistSpecies <span class="op">=</span> <span class="st">"human"</span>, 
  reps <span class="op">=</span> <span class="va">reps</span>,
  annotLevel <span class="op">=</span> <span class="va">annotLevel</span>,
  geneSizeControl <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## 1 core(s) assigned as workers (1reserved).</span></code></pre>
<pre><code><span class="co">## Generating gene background for mouse x human ==&gt; human</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: mmusculus</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Gene table with 21,207 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 21,207 genes from mmusculus.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## data.frame format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from Gene.Symbol.</span></code></pre>
<pre><code><span class="co">## 21,207 genes extracted.</span></code></pre>
<pre><code><span class="co">## Converting mmusculus ==&gt; hsapiens orthologs using: homologene</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Checking for genes without orthologs in hsapiens.</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Extracting genes from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Checking for genes without 1:1 orthologs.</span></code></pre>
<pre><code><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene.</span></code></pre>
<pre><code><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene.</span></code></pre>
<pre><code><span class="co">## Filtering gene_df with gene_map</span></code></pre>
<pre><code><span class="co">## Adding input_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## Adding ortholog_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## Total genes dropped after convert_orthologs :</span>
<span class="co">##    4,725 / 21,207 (22%)</span></code></pre>
<pre><code><span class="co">## Total genes remaining after convert_orthologs :</span>
<span class="co">##    16,482 / 21,207 (78%)</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 intersect background genes used.</span></code></pre>
<pre><code><span class="co">## Standardising CellTypeDataset</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## Checking gene list inputs.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Standardising sct_data.</span></code></pre>
<pre><code><span class="co">## Converting gene list input to standardised human genes.</span></code></pre>
<pre><code><span class="co">## Running with gene size control.</span></code></pre>
<pre><code><span class="co">## Warning: genelistSpecies not provided. Setting to 'human' by default.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: gprofiler</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Gene table with 60,568 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 60,568 genes from human.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: gprofiler</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Gene table with 60,568 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 60,568 genes from human.</span></code></pre>
<pre><code><span class="co">## 60,568 human Ensembl IDs and 39,358 human Gene Symbols imported.</span></code></pre>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<pre><code><span class="co">## Controlled bootstrapping network generated.</span></code></pre>
<pre><code><span class="co">## 20 hit genes remain after filtering.</span></code></pre>
<pre><code><span class="co">## Computing summed proportions.</span></code></pre>
<pre><code><span class="co">## Testing for enrichment in 7 cell types...</span></code></pre>
<pre><code><span class="co">## Sorting results by p-value.</span></code></pre>
<pre><code><span class="co">## Computing BH-corrected q-values.</span></code></pre>
<pre><code><span class="co">## 2 significant cell type enrichment results @ q&lt;0.05 :</span></code></pre>
<pre><code><span class="co">##               CellType annotLevel p fold_change sd_from_mean q</span>
<span class="co">## 1            microglia          1 0    3.534540     6.116934 0</span>
<span class="co">## 2 astrocytes_ependymal          1 0    2.384203     3.928435 0</span></code></pre>
<div class="section level4">
<h4 id="plot-results-1">Plot results<a class="anchor" aria-label="anchor" href="#plot-results-1"></a>
</h4>
<p>We plot these results using <code>ewce_plot</code>:</p>
<div class="sourceCode" id="cb277"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">cont_results</span><span class="op">$</span><span class="va">results</span>,
                             mtc_method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p>This shows that the controlled method generates enrichments that are generally comparable to the standard method.</p>
</div>
</div>
<div class="section level3">
<h3 id="test-different-ctd-levels">Test different CTD levels<a class="anchor" aria-label="anchor" href="#test-different-ctd-levels"></a>
</h3>
<p>Both the analyses shown above were run on level 1 annotations. It is possible to test on the level 2 cell type level annotations by changing one of the arguments.</p>
<div class="sourceCode" id="cb278"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Bootstrap significance test controlling for transcript length and GC content</span>
<span class="va">cont_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,
                                                hits <span class="op">=</span> <span class="va">hits</span>, 
                                                sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,
                                                genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,
                                                reps <span class="op">=</span> <span class="va">reps</span>,
                                                annotLevel <span class="op">=</span> <span class="fl">2</span>,
                                                geneSizeControl <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## 1 core(s) assigned as workers (1reserved).</span></code></pre>
<pre><code><span class="co">## Generating gene background for mouse x human ==&gt; human</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: mmusculus</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Gene table with 21,207 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 21,207 genes from mmusculus.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## data.frame format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from Gene.Symbol.</span></code></pre>
<pre><code><span class="co">## 21,207 genes extracted.</span></code></pre>
<pre><code><span class="co">## Converting mmusculus ==&gt; hsapiens orthologs using: homologene</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Checking for genes without orthologs in hsapiens.</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Extracting genes from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Checking for genes without 1:1 orthologs.</span></code></pre>
<pre><code><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene.</span></code></pre>
<pre><code><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene.</span></code></pre>
<pre><code><span class="co">## Filtering gene_df with gene_map</span></code></pre>
<pre><code><span class="co">## Adding input_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## Adding ortholog_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## Total genes dropped after convert_orthologs :</span>
<span class="co">##    4,725 / 21,207 (22%)</span></code></pre>
<pre><code><span class="co">## Total genes remaining after convert_orthologs :</span>
<span class="co">##    16,482 / 21,207 (78%)</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 intersect background genes used.</span></code></pre>
<pre><code><span class="co">## Standardising CellTypeDataset</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## Checking gene list inputs.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Standardising sct_data.</span></code></pre>
<pre><code><span class="co">## Converting gene list input to standardised human genes.</span></code></pre>
<pre><code><span class="co">## Running with gene size control.</span></code></pre>
<pre><code><span class="co">## Warning: genelistSpecies not provided. Setting to 'human' by default.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: gprofiler</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Gene table with 60,568 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 60,568 genes from human.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: gprofiler</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Gene table with 60,568 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 60,568 genes from human.</span></code></pre>
<pre><code><span class="co">## 60,568 human Ensembl IDs and 39,358 human Gene Symbols imported.</span></code></pre>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<pre><code><span class="co">## Controlled bootstrapping network generated.</span></code></pre>
<pre><code><span class="co">## 20 hit genes remain after filtering.</span></code></pre>
<pre><code><span class="co">## Computing summed proportions.</span></code></pre>
<pre><code><span class="co">## Testing for enrichment in 48 cell types...</span></code></pre>
<pre><code><span class="co">## Sorting results by p-value.</span></code></pre>
<pre><code><span class="co">## Computing BH-corrected q-values.</span></code></pre>
<pre><code><span class="co">## 5 significant cell type enrichment results @ q&lt;0.05 :</span></code></pre>
<pre><code><span class="co">##   CellType annotLevel p fold_change sd_from_mean q</span>
<span class="co">## 1     Pvm1          2 0    4.560488     5.718893 0</span>
<span class="co">## 2     Pvm2          2 0    4.394139     5.582515 0</span>
<span class="co">## 3     Mgl2          2 0    4.806329     5.569659 0</span>
<span class="co">## 4     Mgl1          2 0    4.292407     4.929137 0</span>
<span class="co">## 5   Astro1          2 0    2.821808     3.484294 0</span></code></pre>
<div class="section level4">
<h4 id="plot-results-2">Plot results<a class="anchor" aria-label="anchor" href="#plot-results-2"></a>
</h4>
<div class="sourceCode" id="cb398"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">cont_results</span><span class="op">$</span><span class="va">results</span>,
                             mtc_method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>With the cell subtype analysis each microglial subtype was enriched and correspondingly we see here that the microglial cell type is enriched.</p>
</div>
</div>
<div class="section level3">
<h3 id="plot-results-from-multiple-sets-of-enrichment-results">Plot results from multiple sets of enrichment results<a class="anchor" aria-label="anchor" href="#plot-results-from-multiple-sets-of-enrichment-results"></a>
</h3>
<p>It is often useful to plot results from multiple gene list analyses together. The <code>ewce_plot</code> function allows multiple enrichment analyses to be performed together. To achieve this the results data frames are just appended onto each other, with an additional <code>list</code> column added detailing which analysis they relate to.</p>
<p>To demonstrate this we need to first generate a second analysis so let us sample thirty random genes, and run the bootstrapping analysis on it.</p>
<div class="sourceCode" id="cb399"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#### Generate random gene list ####</span>
<span class="va">human.bg</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/mouse_to_human_homologs.html" class="external-link">mouse_to_human_homologs</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">HGNC.symbol</span></code></pre></div>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<div class="sourceCode" id="cb402"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gene.list.2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">human.bg</span>,size <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>

<span class="va">second_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,
                                                  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,
                                                  hits <span class="op">=</span> <span class="va">gene.list.2</span>, 
                                                  reps <span class="op">=</span> <span class="va">reps</span>,
                                                  annotLevel <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## 1 core(s) assigned as workers (1reserved).</span></code></pre>
<pre><code><span class="co">## Warning: genelistSpecies not provided. Setting to 'human' by default.</span></code></pre>
<pre><code><span class="co">## Generating gene background for mouse x human ==&gt; human</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: mmusculus</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Gene table with 21,207 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 21,207 genes from mmusculus.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## data.frame format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from Gene.Symbol.</span></code></pre>
<pre><code><span class="co">## 21,207 genes extracted.</span></code></pre>
<pre><code><span class="co">## Converting mmusculus ==&gt; hsapiens orthologs using: homologene</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Checking for genes without orthologs in hsapiens.</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Extracting genes from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Checking for genes without 1:1 orthologs.</span></code></pre>
<pre><code><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene.</span></code></pre>
<pre><code><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene.</span></code></pre>
<pre><code><span class="co">## Filtering gene_df with gene_map</span></code></pre>
<pre><code><span class="co">## Adding input_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## Adding ortholog_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## Total genes dropped after convert_orthologs :</span>
<span class="co">##    4,725 / 21,207 (22%)</span></code></pre>
<pre><code><span class="co">## Total genes remaining after convert_orthologs :</span>
<span class="co">##    16,482 / 21,207 (78%)</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 intersect background genes used.</span></code></pre>
<pre><code><span class="co">## Standardising CellTypeDataset</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## Checking gene list inputs.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Standardising sct_data.</span></code></pre>
<pre><code><span class="co">## Converting gene list input to standardised human genes.</span></code></pre>
<pre><code><span class="co">## Running without gene size control.</span></code></pre>
<pre><code><span class="co">## 25 hit genes remain after filtering.</span></code></pre>
<pre><code><span class="co">## Computing summed proportions.</span></code></pre>
<pre><code><span class="co">## Testing for enrichment in 7 cell types...</span></code></pre>
<pre><code><span class="co">## Sorting results by p-value.</span></code></pre>
<pre><code><span class="co">## Computing BH-corrected q-values.</span></code></pre>
<pre><code><span class="co">## 1 significant cell type enrichment results @ q&lt;0.05 :</span></code></pre>
<pre><code><span class="co">##               CellType annotLevel p fold_change sd_from_mean q</span>
<span class="co">## 1 astrocytes_ependymal          1 0    1.478319     2.759059 0</span></code></pre>
<div class="sourceCode" id="cb502"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">full_res2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">full_results</span><span class="op">$</span><span class="va">results</span>,
                       list<span class="op">=</span><span class="st">"Alzheimers"</span><span class="op">)</span>
<span class="va">rando_res2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">second_results</span><span class="op">$</span><span class="va">results</span>,
                        list<span class="op">=</span><span class="st">"Random"</span><span class="op">)</span>
<span class="va">merged_results</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">full_res2</span>, <span class="va">rando_res2</span><span class="op">)</span></code></pre></div>
<div class="section level4">
<h4 id="plot-results-3">Plot results<a class="anchor" aria-label="anchor" href="#plot-results-3"></a>
</h4>
<p>Here we apply multiple-testing correction (BH = Benjamini-Hochberg) to guard against positive results due to chance.</p>
<div class="sourceCode" id="cb503"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">merged_results</span>,
                             mtc_method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p>As expected, the second randomly generated gene list shows no significant enrichment results.</p>
</div>
</div>
</div>
</div>
<div class="section level1">
<h1 id="create-a-celltypedataset">Create a CellTypeDataset<a class="anchor" aria-label="anchor" href="#create-a-celltypedataset"></a>
</h1>
<div class="section level2">
<h2 id="introduction-1">Introduction<a class="anchor" aria-label="anchor" href="#introduction-1"></a>
</h2>
<p><code>EWCE</code> was originally designed to work with the single-cell cortical transcriptome data from the <a href="http://linnarssonlab.org/" class="external-link">Linnarsson Lab</a><span class="citation"><sup>1</sup></span> (available <a href="http://linnarssonlab.org/cortex/" class="external-link">here</a>).</p>
<p>Using this package it is possible to read in any single-cell transcriptome data, provided that you have a cell by gene expression matrix (with each cell as a separate column) and a separate annotation dataframe, with a row for each cell.</p>
<p>The <code>EWCE</code> process involves testing for whether the genes in a target list have higher levels of expression in a given cell type than can reasonably be expected by chance. The probability distribution for this is estimated by randomly generating gene lists of equal length from a set of background genes.</p>
<p><code>EWCE</code> can be applied to any gene list. In the original paper we reported its application to genetic and transcriptomic datasets..<span class="citation"><sup>2</sup></span></p>
<p>Note that throughout this vignette we use the terms ‘cell type’ and ‘cell subtype’ to refer to two levels of annotation of what a cell type is. This is described in further detail in our paper<span class="citation"><sup>2</sup></span>, but relates to the two levels of annotation provided in the Linnarsson dataset.<span class="citation"><sup>1</sup></span> In this dataset a cell is described as having a cell type (i.e. ‘Interneuron’) and subcell type (i.e. ‘Int11’ a.k.a Interneuron type 11).</p>
<p>Here we provide an example of converting a scRNA-seq dataset (<a href="https://www.science.org/doi/10.1126/science.aaa1934" class="external-link">Zeisel 2015</a>) into a CellTypeDataset (CTD) so it can used with <em>EWCE</em>.</p>
</div>
<div class="section level2">
<h2 id="loading-datasets">Loading datasets<a class="anchor" aria-label="anchor" href="#loading-datasets"></a>
</h2>
<p>The first step for all analyses is to load the single-cell transcriptome (SCT) data. For the purposes of this example we will use the dataset described in: &gt; <a href="https://www.science.org/doi/10.1126/science.aaa1934" class="external-link">Cell types in the mouse cortex and hippocampus revealed by single-cell RNA-seq</a>, Science, 2015</p>
<p>This is stored and can be loaded as follows from <code>ewceData</code> package.</p>
<p><em>EWCE</em> can generate CTD from two different data input types: 1. A gene x cell expression matrix (can be dense matrix, sparse matrix, data.frame, or DelayedArray).<br>
2. A <code>SingleCellExperiment</code> (SCE) or <code>SummarizedExperiment</code> object.</p>
<p>To check the data, we can quickly plot the distribution of expression of a given gene across all the cell types.</p>
<div class="sourceCode" id="cb504"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># example datasets held in ewceData package</span>
<span class="va">cortex_mrna</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/cortex_mrna.html" class="external-link">cortex_mrna</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<div class="sourceCode" id="cb507"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gene</span> <span class="op">&lt;-</span> <span class="st">"Necab1"</span>
<span class="va">cellExpDist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Expression<span class="op">=</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">[</span><span class="va">gene</span>,<span class="op">]</span>,
                          Celltype<span class="op">=</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span><span class="op">[</span>
                            <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span>,<span class="op">]</span><span class="op">$</span><span class="va">level1class</span><span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">cellExpDist</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Celltype</span>, y<span class="op">=</span><span class="va">Expression</span><span class="op">)</span>, outlier.alpha <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<p>To note for the analysis of the <code>cortex_mrna</code> dataset, I will proved the alternative for an SCE object under each segment of code if there is any difference.</p>
</div>
<div class="section level2">
<h2 id="convert-single-cell-formats">Convert single-cell formats<a class="anchor" aria-label="anchor" href="#convert-single-cell-formats"></a>
</h2>
<p>If you would like to try running the functions on an SCE object instead, we can convert <code>cortex_mrna</code> with the package <a href="https://github.com/bschilder/scKirby" class="external-link"><code>scKirby</code></a>. <code>scKirby</code> automatically infers the input file type and converts it to SCE format by default.</p>
<div class="sourceCode" id="cb508"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"SingleCellExperiment"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"SingleCellExperiment"</span><span class="op">)</span><span class="op">}</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span> 
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"scKirby"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"bschilder/scKirby"</span><span class="op">)</span><span class="op">}</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scKirby</span><span class="op">)</span>

<span class="co"># Make SCE object</span>
<span class="va">cortex_mrna</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/cortex_mrna.html" class="external-link">cortex_mrna</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">cortex_mrna_sce</span> <span class="op">&lt;-</span> <span class="fu">scKirby</span><span class="fu">::</span><span class="fu">ingest_data</span><span class="op">(</span><span class="va">cortex_mrna</span>, save_output <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="co"># Now to plot the SCE dataset:</span>
<span class="va">gene</span><span class="op">=</span><span class="st">"Necab1"</span>
<span class="va">cellExpDist_sce</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>e<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">gene</span>,<span class="op">]</span>,
                             l1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span><span class="op">[</span>
                               <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
                             <span class="op">]</span><span class="op">$</span><span class="va">level1class</span><span class="op">)</span>
                             
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">cellExpDist_sce</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">l1</span>,y<span class="op">=</span><span class="va">e</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Cell type"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Unique Molecule Count"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> </code></pre></div>
</div>
<div class="section level2">
<h2 id="correct-gene-symbols">Correct gene symbols<a class="anchor" aria-label="anchor" href="#correct-gene-symbols"></a>
</h2>
<p>It is very common for publicly available transcriptome datasets to use incorrect gene symbols (often some gene names will have been mangled by opening in Excel).</p>
<p>A function is provided to correct these where out of date aliases were used and give a warning if appears possible that excel has mangled some of the gene names. This function can be used with a downloaded reference file like the <a href="%22http://www.informatics.jax.org/downloads/reports/MRK_List2.rpt%22" class="external-link">MRK_List2 file</a> from MGI which lists known synonyms for MGI gene symbols. If no file path is passed, there is a loaded reference dataset of the MRK_List2 which will be used: <code><a href="https://rdrr.io/pkg/ewceData/man/mgi_synonym_data.html" class="external-link">mgi_synonym_data()</a></code>. We recommend running this function on all input datasets.</p>
<div class="sourceCode" id="cb509"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span> <span class="op">=</span> <span class="fu"><a href="../reference/fix_bad_mgi_symbols.html">fix_bad_mgi_symbols</a></span><span class="op">(</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span></code></pre></div>
<p>For the SCE version, pass the whole SCE object:</p>
<div class="sourceCode" id="cb510"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Note that data in SCE object must be in 'counts' assay</span>
<span class="co">#The fix_bad_hgnc_symbols function can similarly take an SCE object as an input</span>
<span class="va">cortex_mrna_sce</span> <span class="op">=</span> <span class="fu"><a href="../reference/fix_bad_mgi_symbols.html">fix_bad_mgi_symbols</a></span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="drop-genes">Drop genes<a class="anchor" aria-label="anchor" href="#drop-genes"></a>
</h2>
<p>The vignette takes a while to go through if you use all genes. So let’s restrict to 1000 random genes.</p>
<div class="sourceCode" id="cb511"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">nKeep</span> <span class="op">&lt;-</span> <span class="fl">1000</span>
<span class="va">must_keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span> 
<span class="va">keep_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">must_keep</span>,<span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span>,<span class="fl">997</span><span class="op">)</span><span class="op">)</span>
<span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span> <span class="op">&lt;-</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">[</span><span class="va">keep_genes</span>,<span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 1000 3005</span></code></pre>
<div class="sourceCode" id="cb513"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#### SCE ####</span>
<span class="va">nKeep</span> <span class="op">=</span> <span class="fl">1000</span>
<span class="va">must_keep</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span> 
<span class="va">keep_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">must_keep</span>,<span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">rowRanges</a></span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span><span class="op">)</span>,<span class="fl">997</span><span class="op">)</span><span class="op">)</span>
<span class="va">cortex_mrna_sce</span> <span class="op">=</span> <span class="va">cortex_mrna_sce</span><span class="op">[</span><span class="va">keep_genes</span>,<span class="op">]</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="normalization-optional">Normalization [Optional]<a class="anchor" aria-label="anchor" href="#normalization-optional"></a>
</h2>
<p>A different number of reads is found across each cell. We suggest using <a href="https://cran.r-project.org/web/packages/sctransform/index.html" class="external-link"><code>scTransform</code></a> to normalise for differences due to cell size, then linearly scale. Note that this might be slow.</p>
<div class="sourceCode" id="cb514"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp_scT_normed</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/sct_normalize.html">sct_normalize</a></span><span class="op">(</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span> </code></pre></div>
<div class="section level3">
<h3 id="singlecellexperiment">SingleCellExperiment<a class="anchor" aria-label="anchor" href="#singlecellexperiment"></a>
</h3>
<div class="sourceCode" id="cb515"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#### SCE ####</span>
<span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span><span class="op">$</span><span class="va">normcounts</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/sct_normalize.html">sct_normalize</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span></code></pre></div>
<p>Note that this is optional (and was not used in the original EWCE publication) so by all means ignore this <code>scTransform</code> step.</p>
</div>
</div>
<div class="section level2">
<h2 id="calculate-specificity-matrices">Calculate specificity matrices<a class="anchor" aria-label="anchor" href="#calculate-specificity-matrices"></a>
</h2>
<p>Next,<code>drop_uninformative_genes</code> drops uninformative genes in order to reduce compute time and noise in subsequent steps. It achieves this through several steps, each of which are optional: 1. <strong>Drop non-1:1 orthologs</strong>: Removes genes that don’t have 1:1 orthologs with the output_species (“human” by default). 2. <strong>Drop non-varying genes</strong>: Removes genes that don’t vary across cells based on variance deciles. 3. <strong>Drop non-differentially expressed genes (DEGs)</strong>: Removes genes that are not significantly differentially expressed across cell-types. Multiple DEG methods are currently available (see <code><a href="../reference/drop_uninformative_genes.html">?drop_uninformative_genes</a></code> for up-to-date info).</p>
<div class="section level3">
<h3 id="parallelisation-1">Parallelisation<a class="anchor" aria-label="anchor" href="#parallelisation-1"></a>
</h3>
<p>You can now speed up the DGE process by parallelising across multiple cores with the parameter <code>no_cores</code> (<code>=1</code> by default).</p>
<div class="sourceCode" id="cb516"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Generate cell type data for just the cortex/hippocampus data  </span>
<span class="va">exp_CortexOnly_DROPPED</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/drop_uninformative_genes.html">drop_uninformative_genes</a></span><span class="op">(</span>
  exp <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span>, 
  input_species <span class="op">=</span> <span class="st">"mouse"</span>,
  output_species <span class="op">=</span> <span class="st">"human"</span>,
  level2annot <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span> </code></pre></div>
<pre><code><span class="co">## Check 1000Check 3005</span></code></pre>
<pre><code><span class="co">## 1 core(s) assigned as workers (1reserved).</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## Checking for non-expressed genes.</span></code></pre>
<pre><code><span class="co">## Checking for cells with no expressed genes.</span></code></pre>
<pre><code><span class="co">## DGE:: Limma...</span></code></pre>
<pre><code><span class="co">## 228 / 1,000 genes dropped @ DGE adj_pval_thresh &lt; 1e-05</span></code></pre>
<pre><code><span class="co">## Time difference of 1.055907 secs</span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="generate-celltypedataset">Generate CellTypeDataset<a class="anchor" aria-label="anchor" href="#generate-celltypedataset"></a>
</h2>
<p>Now we generate the CellTypeDataset with the</p>
<p>By default the function returns the path where the resulting file was saved. If you want to return the CTD itself as well, set <code>return_ctd=TRUE</code> and it will be returned in a list (along with the save path).</p>
<div class="section level3">
<h3 id="parallelisation-2">Parallelisation<a class="anchor" aria-label="anchor" href="#parallelisation-2"></a>
</h3>
<p>You can now speed up <code>generate_celltype_data</code> by parallelising across multiple cores with the parameter <code>no_cores</code> (<code>=1</code> by default).</p>
<div class="sourceCode" id="cb525"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">annotLevels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>level1class<span class="op">=</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level1class</span>,
                    level2class<span class="op">=</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span>

<span class="va">fNames_CortexOnly</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/generate_celltype_data.html">generate_celltype_data</a></span><span class="op">(</span>
  exp <span class="op">=</span> <span class="va">exp_CortexOnly_DROPPED</span>,
  annotLevels <span class="op">=</span> <span class="va">annotLevels</span>,
  groupName <span class="op">=</span> <span class="st">"kiCortexOnly"</span><span class="op">)</span> </code></pre></div>
<pre><code><span class="co">## 1 core(s) assigned as workers (1reserved).</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## + Calculating normalized mean expression.</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## + Calculating normalized specificity.</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## Loading required namespace: ggdendro</span></code></pre>
<pre><code><span class="co">## + Saving results ==&gt;  /tmp/RtmpY0ncgc/ctd_kiCortexOnly.rda</span></code></pre>
<div class="sourceCode" id="cb534"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctd_CortexOnly</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/load_rdata.html">load_rdata</a></span><span class="op">(</span><span class="va">fNames_CortexOnly</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="singlecellexperiment-1">SingleCellExperiment<a class="anchor" aria-label="anchor" href="#singlecellexperiment-1"></a>
</h3>
<div class="sourceCode" id="cb535"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#### SCE ####</span>
<span class="co"># Generate cell type data for just the cortex/hippocampus data</span>
<span class="va">exp_CortexOnly_DROPPED</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/drop_uninformative_genes.html">drop_uninformative_genes</a></span><span class="op">(</span>exp<span class="op">=</span><span class="va">cortex_mrna_sce</span>,
                                                   drop_nonhuman_genes <span class="op">=</span> <span class="cn">T</span>,
                                                   input_species <span class="op">=</span> <span class="st">"mouse"</span>,
                                                   level2annot<span class="op">=</span><span class="va">cortex_mrna_sce</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span>

<span class="va">annotLevels</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>level1class<span class="op">=</span><span class="va">cortex_mrna_sce</span><span class="op">$</span><span class="va">level1class</span>,
                   level2class<span class="op">=</span><span class="va">cortex_mrna_sce</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span>

<span class="va">fNames_CortexOnly</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_celltype_data.html">generate_celltype_data</a></span><span class="op">(</span>exp<span class="op">=</span><span class="va">exp_CortexOnly_DROPPED</span>,
                                           annotLevels<span class="op">=</span><span class="va">annotLevels</span>,
                                           groupName<span class="op">=</span><span class="st">"kiCortexOnly"</span>,
                                           savePath<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> </code></pre></div>
<p>To note on the SCE approach, these are the only differences for handling the analysis of an SCE object. All downstream analysis on <code>cortex_mrna</code> would be the same. The same approach can be used for other datasets in the vignette if converted such as <code>hypothalamus_mrna</code> or your own SCE object.</p>
</div>
</div>
</div>
<div class="section level1">
<h1 id="merge-two-single-cell-datasets">Merge two single-cell datasets<a class="anchor" aria-label="anchor" href="#merge-two-single-cell-datasets"></a>
</h1>
<p>Often it is useful to merge two single-cell datasets. For instance, there are separate files for the cortex and hypothalamus datasets generated by the Karolinska institute. A subset of the resulting merged dataset is loaded using <code><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd()</a></code> but it is worth understanding the approach (outlined below) if you want to repeat this for other single-cell datasets.</p>
<p>The dataset is available from the <a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE74nnn/GSE74672/suppl/GSE74672_expressed_mols_with_classes.xlsx.gz">GEO</a> but is also available in the <code>ewceData</code> package: <code><a href="https://rdrr.io/pkg/ewceData/man/hypothalamus_mrna.html" class="external-link">hypothalamus_mrna()</a></code>. See the <code>ewceData</code> for details of how this file was preprocessed after first being downloaded from GEO, unzipped and read into R.</p>
<p>We can now merge the hypothalamus dataset with the cortex dataset and then calculate specificity:</p>
<div class="section level2">
<h2 id="load-hypothalamus-dataset">Load hypothalamus dataset<a class="anchor" aria-label="anchor" href="#load-hypothalamus-dataset"></a>
</h2>
<div class="sourceCode" id="cb536"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hypothalamus_mrna</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/hypothalamus_mrna.html" class="external-link">hypothalamus_mrna</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
</div>
<div class="section level2">
<h2 id="fix-bad-mgi-symbols">Fix bad MGI symbols<a class="anchor" aria-label="anchor" href="#fix-bad-mgi-symbols"></a>
</h2>
<div class="sourceCode" id="cb539"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hypothalamus_mrna</span><span class="op">$</span><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/fix_bad_mgi_symbols.html">fix_bad_mgi_symbols</a></span><span class="op">(</span>exp <span class="op">=</span> <span class="va">hypothalamus_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<pre><code><span class="co">## 1808 rows do not have proper MGI symbols</span></code></pre>
<pre><code><span class="co">## Rims1_loc1, Rims1_loc2, Fam178b_loc2, Fam178b_loc1, 2900092D14Rik, 4930487H11Rik, Spag16_loc1, Spag16_loc2, Pnkd_loc2, Pnkd_loc1, Ccdc108, 1700019O17Rik, Myeov2, Ren2, Gpr52, 1700015E13Rik, Olfr426, Hmga2-ps1, B020014A21Rik, Gm5512_loc1</span></code></pre>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<pre><code><span class="co">## 0 poorly annotated genes are replicates of existing genes. These are:</span></code></pre>
<pre><code><span class="co">## </span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## 0 rows should have been corrected by checking synonyms.</span></code></pre>
<pre><code><span class="co">## 1808 rows STILL do not have proper MGI symbols.</span></code></pre>
<div class="sourceCode" id="cb551"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/fix_bad_mgi_symbols.html">fix_bad_mgi_symbols</a></span><span class="op">(</span>exp <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<pre><code><span class="co">## 80 rows do not have proper MGI symbols</span></code></pre>
<pre><code><span class="co">## C430002E04Rik, Gm21637_loc1, Hbb-b2, Rims1_loc2, Gm21947, Gm6367_loc1, Gm10670_loc1, Slc14a2_loc1, Gm13298_loc2, Gm19522, Lilrb4, Gm11194, AI597468, Gltscr2, Gm14431_loc4, Ndnl2, Gm6642, Tmem35, Gm3264_loc2, Gm17821</span></code></pre>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<pre><code><span class="co">## 0 poorly annotated genes are replicates of existing genes. These are:</span></code></pre>
<pre><code><span class="co">## </span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## 36 rows should have been corrected by checking synonyms.</span></code></pre>
<pre><code><span class="co">## 44 rows STILL do not have proper MGI symbols.</span></code></pre>
</div>
<div class="section level2">
<h2 id="merge-ctd">Merge CTD<a class="anchor" aria-label="anchor" href="#merge-ctd"></a>
</h2>
<p>Merge the CTDs - again this can be run with SCE or other ranged SE objects as input.</p>
<div class="sourceCode" id="cb563"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">merged_KI</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/merge_two_expfiles.html">merge_two_expfiles</a></span><span class="op">(</span>exp1 <span class="op">=</span> <span class="va">hypothalamus_mrna</span><span class="op">$</span><span class="va">exp</span>, 
                                      exp2 <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span>,
                                      annot1 <span class="op">=</span> <span class="va">hypothalamus_mrna</span><span class="op">$</span><span class="va">annot</span>, 
                                      annot2 <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span>,
                                      name1 <span class="op">=</span> <span class="st">"Hypothalamus (KI)"</span>, 
                                      name2 <span class="op">=</span> <span class="st">"Cortex/Hippo (KI)"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## 23,317 non-overlapping gene(s) removed from exp1.</span></code></pre>
<pre><code><span class="co">## 0 non-overlapping gene(s) removed from exp2.</span></code></pre>
<pre><code><span class="co">## 1,000 intersecting genes remain.</span></code></pre>
<pre><code><span class="co">## Converting to data.frame</span>
<span class="co">## Converting to data.frame</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span></code></pre>
</div>
<div class="section level2">
<h2 id="drop-uninformative-genes">Drop uninformative genes<a class="anchor" aria-label="anchor" href="#drop-uninformative-genes"></a>
</h2>
<p>Drop genes which don’t vary significantly between cell types.</p>
<div class="sourceCode" id="cb569"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp_merged_DROPPED</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/drop_uninformative_genes.html">drop_uninformative_genes</a></span><span class="op">(</span>
  exp <span class="op">=</span> <span class="va">merged_KI</span><span class="op">$</span><span class="va">exp</span>, 
  drop_nonhuman_genes <span class="op">=</span> <span class="cn">TRUE</span>, 
  input_species <span class="op">=</span> <span class="st">"mouse"</span>,
  level2annot <span class="op">=</span> <span class="va">merged_KI</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Check 1000Check 3777</span></code></pre>
<pre><code><span class="co">## 1 core(s) assigned as workers (1reserved).</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## Checking for non-expressed genes.</span></code></pre>
<pre><code><span class="co">## Checking for cells with no expressed genes.</span></code></pre>
<pre><code><span class="co">## DGE:: Limma...</span></code></pre>
<pre><code><span class="co">## 1 / 1,000 genes dropped @ DGE adj_pval_thresh &lt; 1e-05</span></code></pre>
<pre><code><span class="co">## Time difference of 0.7758758 secs</span></code></pre>
</div>
<div class="section level2">
<h2 id="generate-celltypedataset-1">Generate CellTypeDataset<a class="anchor" aria-label="anchor" href="#generate-celltypedataset-1"></a>
</h2>
<div class="sourceCode" id="cb578"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">annotLevels</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>level1class<span class="op">=</span><span class="va">merged_KI</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level1class</span>,
                   level2class<span class="op">=</span><span class="va">merged_KI</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span>

<span class="co"># Update file path to where you want the cell type data files to save on disk</span>
<span class="va">fNames_MergedKI</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/generate_celltype_data.html">generate_celltype_data</a></span><span class="op">(</span>exp <span class="op">=</span> <span class="va">exp_merged_DROPPED</span>,
                                                annotLevels <span class="op">=</span> <span class="va">annotLevels</span>,
                                                groupName <span class="op">=</span> <span class="st">"MergedKI"</span>,
                                                savePath <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> </code></pre></div>
<pre><code><span class="co">## 1 core(s) assigned as workers (1reserved).</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## + Calculating normalized mean expression.</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## + Calculating normalized specificity.</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## + Saving results ==&gt;  /tmp/RtmpY0ncgc/ctd_MergedKI.rda</span></code></pre>
<div class="sourceCode" id="cb586"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctd_MergedKI</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/load_rdata.html">load_rdata</a></span><span class="op">(</span><span class="va">fNames_MergedKI</span><span class="op">)</span> </code></pre></div>
</div>
<div class="section level2">
<h2 id="understanding-specificity-matrices">Understanding specificity matrices<a class="anchor" aria-label="anchor" href="#understanding-specificity-matrices"></a>
</h2>
<p>While not required for further analyses it helps to understand what the outputs of this function are.</p>
<p>Note firstly, that it is a list such that <code>ctd[[1]]</code> contains data relating to level 1 annotations and <code>ctd[[2]]</code> relates to level 2 annotations.</p>
<p>Using the <code>ggplot2</code> package to visualise the data, let us examine the expression of a few genes. If you have not already done so you will need to first install the ggplot2 package with <code>install.packages("ggplot2")</code>.</p>
<p>For this example we use a subset of the genes from the merged dataset generated above, which is accessed using <code><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ewceData::ctd()</a></code>. We recommend that you use the code above to regenerate this though and drop the first command from the below section.</p>
<div class="sourceCode" id="cb587"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## You can also import the pre-merged cortex and hypothalamus dataset</span>
<span class="co"># generated by Karolinska institute. `ewceData::ctd()` is the same file as </span>
<span class="co"># ctd_MergedKI above.</span>
<span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span> </code></pre></div>
<div class="sourceCode" id="cb588"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="va">ctd_MergedKI</span> 

<span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_ctd.html">plot_ctd</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,
                      level <span class="op">=</span> <span class="fl">1</span>,
                      genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,
                      metric <span class="op">=</span> <span class="st">"mean_exp"</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-26-1.png" width="672"></p>
<p>This graph shows the average expression of three genes: <em>Apoe, Gfap</em> and <em>Gapdh</em>. While there are substantial differences in which cell types express these genes, the dominant effect seen here is the overall expression level of the data. For the purposes of this analysis though, we are not interested in overall expression level and only wish to know about the proportion of a genes expression which is found in a particular cell type. We can study this instead using the following code which examines the data frame <code>ctd[[1]]$specificity</code>:</p>
<div class="sourceCode" id="cb589"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_ctd.html">plot_ctd</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,
                      level <span class="op">=</span> <span class="fl">1</span>,
                      genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,
                      metric <span class="op">=</span> <span class="st">"specificity"</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-27-1.png" width="672"></p>
<p>We can now see in this graph that <em>Gfap</em> is the most specific to a cell type (Type 1 Astrocytes) of any of those three genes, with over 60% of its expression found in that cell type.</p>
<p>It can also be seen that the majority of expression of Gapdh is in neurons but because their are a greater number of neuronal subtypes, the total expression proportion appears lower. We can examine expression across level 2 cell type level annotations by looking at <code>ctd[[2]]$specificity</code>:</p>
<div class="sourceCode" id="cb590"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_ctd.html">plot_ctd</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,
                      level <span class="op">=</span> <span class="fl">2</span>,
                      genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,
                      metric <span class="op">=</span> <span class="st">"specificity"</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-28-1.png" width="672"></p>
</div>
</div>
<div class="section level1">
<h1 id="run-conditional-cell-type-enrichment-tests">Run conditional cell-type enrichment tests<a class="anchor" aria-label="anchor" href="#run-conditional-cell-type-enrichment-tests"></a>
</h1>
<div class="section level2">
<h2 id="prepare-data">Prepare data<a class="anchor" aria-label="anchor" href="#prepare-data"></a>
</h2>
<div class="sourceCode" id="cb591"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<div class="sourceCode" id="cb594"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/example_genelist.html" class="external-link">example_genelist</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span>
<span class="co">## loading from cache</span></code></pre>
<div class="sourceCode" id="cb596"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## NOTE: rep=100 for demo purposes only. </span>
<span class="co">## Use &gt;=10,000 for publication-quality results.</span>
<span class="va">reps</span> <span class="op">&lt;-</span> <span class="fl">100</span> 
<span class="va">annotLevel</span> <span class="op">&lt;-</span> <span class="fl">1</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="controlling-for-expression-in-another-cell-type">Controlling for expression in another cell type<a class="anchor" aria-label="anchor" href="#controlling-for-expression-in-another-cell-type"></a>
</h2>
<p>In a followup paper we found that an enrichment detected for Schizophrenia in Somatosensory Pyramidal neurons could be explained by accounting for expression in Hippocampal CA1 pyramidal neurons. These results are described here:</p>
<p><a href="https://www.nature.com/articles/s41588-018-0129-5" class="external-link">Skene, et al. Genetic identification of brain cell types underlying schizophrenia.Nature Genetics, 2018.</a></p>
<p>Those results were generated using an alternative enrichment method designed for use with GWAS Summary Statistics rather than gene sets. The same sort of approach can be extended to EWCE as well, and we have implemented it within this package. When testing for enrichment the other gene sets that are sampled are selected to have equivalent specificity in the controlled cell type.</p>
<p>We demonstrate it’s use below to test whether the enrichment in astrocytes is still present after controlling for the enrichment within microglia.</p>
<div class="section level3">
<h3 id="parallelisation-3">Parallelisation<a class="anchor" aria-label="anchor" href="#parallelisation-3"></a>
</h3>
<p>You can now speed up the bootstrapping process by parallelising across multiple cores with the parameter <code>no_cores</code> (<code>=1</code> by default).</p>
<div class="sourceCode" id="cb597"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">unconditional_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span>
  sct_data <span class="op">=</span> <span class="va">ctd</span>,
  hits <span class="op">=</span> <span class="va">hits</span>,
  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,
  genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,
  reps <span class="op">=</span> <span class="va">reps</span>,
  annotLevel <span class="op">=</span> <span class="va">annotLevel</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## 1 core(s) assigned as workers (1reserved).</span></code></pre>
<pre><code><span class="co">## Generating gene background for mouse x human ==&gt; human</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: mmusculus</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Gene table with 21,207 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 21,207 genes from mmusculus.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## data.frame format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from Gene.Symbol.</span></code></pre>
<pre><code><span class="co">## 21,207 genes extracted.</span></code></pre>
<pre><code><span class="co">## Converting mmusculus ==&gt; hsapiens orthologs using: homologene</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Checking for genes without orthologs in hsapiens.</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Extracting genes from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Checking for genes without 1:1 orthologs.</span></code></pre>
<pre><code><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene.</span></code></pre>
<pre><code><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene.</span></code></pre>
<pre><code><span class="co">## Filtering gene_df with gene_map</span></code></pre>
<pre><code><span class="co">## Adding input_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## Adding ortholog_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## Total genes dropped after convert_orthologs :</span>
<span class="co">##    4,725 / 21,207 (22%)</span></code></pre>
<pre><code><span class="co">## Total genes remaining after convert_orthologs :</span>
<span class="co">##    16,482 / 21,207 (78%)</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 intersect background genes used.</span></code></pre>
<pre><code><span class="co">## Standardising CellTypeDataset</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## Checking gene list inputs.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Standardising sct_data.</span></code></pre>
<pre><code><span class="co">## Converting gene list input to standardised human genes.</span></code></pre>
<pre><code><span class="co">## Running without gene size control.</span></code></pre>
<pre><code><span class="co">## 17 hit genes remain after filtering.</span></code></pre>
<pre><code><span class="co">## Computing summed proportions.</span></code></pre>
<pre><code><span class="co">## Testing for enrichment in 7 cell types...</span></code></pre>
<pre><code><span class="co">## Sorting results by p-value.</span></code></pre>
<pre><code><span class="co">## Computing BH-corrected q-values.</span></code></pre>
<pre><code><span class="co">## 1 significant cell type enrichment results @ q&lt;0.05 :</span></code></pre>
<pre><code><span class="co">##    CellType annotLevel p fold_change sd_from_mean q</span>
<span class="co">## 1 microglia          1 0    2.072508     3.667557 0</span></code></pre>
<div class="sourceCode" id="cb696"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">conditional_results_micro</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span> <span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span>
  sct_data <span class="op">=</span> <span class="va">ctd</span>,
  hits <span class="op">=</span> <span class="va">hits</span>,
  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,
  genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,
  reps <span class="op">=</span> <span class="va">reps</span>,
  annotLevel <span class="op">=</span> <span class="va">annotLevel</span>,
  controlledCT <span class="op">=</span> <span class="st">"microglia"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## 1 core(s) assigned as workers (1reserved).</span></code></pre>
<pre><code><span class="co">## Generating gene background for mouse x human ==&gt; human</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: mmusculus</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Gene table with 21,207 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 21,207 genes from mmusculus.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## data.frame format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from Gene.Symbol.</span></code></pre>
<pre><code><span class="co">## 21,207 genes extracted.</span></code></pre>
<pre><code><span class="co">## Converting mmusculus ==&gt; hsapiens orthologs using: homologene</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Checking for genes without orthologs in hsapiens.</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Extracting genes from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Checking for genes without 1:1 orthologs.</span></code></pre>
<pre><code><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene.</span></code></pre>
<pre><code><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene.</span></code></pre>
<pre><code><span class="co">## Filtering gene_df with gene_map</span></code></pre>
<pre><code><span class="co">## Adding input_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## Adding ortholog_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## Total genes dropped after convert_orthologs :</span>
<span class="co">##    4,725 / 21,207 (22%)</span></code></pre>
<pre><code><span class="co">## Total genes remaining after convert_orthologs :</span>
<span class="co">##    16,482 / 21,207 (78%)</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 intersect background genes used.</span></code></pre>
<pre><code><span class="co">## Standardising CellTypeDataset</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## Checking gene list inputs.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Standardising sct_data.</span></code></pre>
<pre><code><span class="co">## Converting gene list input to standardised human genes.</span></code></pre>
<pre><code><span class="co">## Running without gene size control.</span></code></pre>
<pre><code><span class="co">## 17 hit genes remain after filtering.</span></code></pre>
<pre><code><span class="co">## Computing summed proportions.</span></code></pre>
<pre><code><span class="co">## Generating controlled bootstrap gene sets.</span></code></pre>
<pre><code><span class="co">## Testing for enrichment in 7 cell types...</span></code></pre>
<pre><code><span class="co">## Sorting results by p-value.</span></code></pre>
<pre><code><span class="co">## Computing BH-corrected q-values.</span></code></pre>
<pre><code><span class="co">## 0 significant cell type enrichment results @ q&lt;0.05 :</span></code></pre>
<div class="sourceCode" id="cb795"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">conditional_results_astro</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span>
  sct_data <span class="op">=</span> <span class="va">ctd</span>,
  hits <span class="op">=</span> <span class="va">hits</span>,
  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,
  genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,
  reps <span class="op">=</span> <span class="va">reps</span>,
  annotLevel <span class="op">=</span> <span class="va">annotLevel</span>,
  controlledCT <span class="op">=</span> <span class="st">"astrocytes_ependymal"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## 1 core(s) assigned as workers (1reserved).</span></code></pre>
<pre><code><span class="co">## Generating gene background for mouse x human ==&gt; human</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: mmusculus</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Gene table with 21,207 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 21,207 genes from mmusculus.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## data.frame format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from Gene.Symbol.</span></code></pre>
<pre><code><span class="co">## 21,207 genes extracted.</span></code></pre>
<pre><code><span class="co">## Converting mmusculus ==&gt; hsapiens orthologs using: homologene</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Checking for genes without orthologs in hsapiens.</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Extracting genes from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Checking for genes without 1:1 orthologs.</span></code></pre>
<pre><code><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene.</span></code></pre>
<pre><code><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene.</span></code></pre>
<pre><code><span class="co">## Filtering gene_df with gene_map</span></code></pre>
<pre><code><span class="co">## Adding input_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## Adding ortholog_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## Total genes dropped after convert_orthologs :</span>
<span class="co">##    4,725 / 21,207 (22%)</span></code></pre>
<pre><code><span class="co">## Total genes remaining after convert_orthologs :</span>
<span class="co">##    16,482 / 21,207 (78%)</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 intersect background genes used.</span></code></pre>
<pre><code><span class="co">## Standardising CellTypeDataset</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## Checking gene list inputs.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Standardising sct_data.</span></code></pre>
<pre><code><span class="co">## Converting gene list input to standardised human genes.</span></code></pre>
<pre><code><span class="co">## Running without gene size control.</span></code></pre>
<pre><code><span class="co">## 17 hit genes remain after filtering.</span></code></pre>
<pre><code><span class="co">## Computing summed proportions.</span></code></pre>
<pre><code><span class="co">## Generating controlled bootstrap gene sets.</span></code></pre>
<pre><code><span class="co">## Testing for enrichment in 7 cell types...</span></code></pre>
<pre><code><span class="co">## Sorting results by p-value.</span></code></pre>
<pre><code><span class="co">## Computing BH-corrected q-values.</span></code></pre>
<pre><code><span class="co">## 0 significant cell type enrichment results @ q&lt;0.05 :</span></code></pre>
</div>
<div class="section level3">
<h3 id="merge-and-plot-results">Merge and plot results<a class="anchor" aria-label="anchor" href="#merge-and-plot-results"></a>
</h3>
<div class="sourceCode" id="cb894"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">merged_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">unconditional_results</span><span class="op">$</span><span class="va">results</span>,
             list<span class="op">=</span><span class="st">"Unconditional Enrichment"</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">conditional_results_micro</span><span class="op">$</span><span class="va">results</span>,
             list<span class="op">=</span><span class="st">"Conditional Enrichment (Microglia controlled)"</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">conditional_results_astro</span><span class="op">$</span><span class="va">results</span>,
             list<span class="op">=</span><span class="st">"Conditional Enrichment (Astrocyte controlled)"</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">merged_results</span>,
                             mtc_method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-31-1.png" width="700"></p>
<p>When controlling for astrocytes the enrichment in astrocytes is totally abolished as expected, and vica versa. The enrichment in microglia remains strongly significant however after controlling for astrocytes., suggesting that this enrichment is independent of that in astrocytes.</p>
</div>
</div>
<div class="section level2">
<h2 id="gene-set-enrichment-analysis-controlling-for-cell-type-expression">Gene set enrichment analysis controlling for cell type expression<a class="anchor" aria-label="anchor" href="#gene-set-enrichment-analysis-controlling-for-cell-type-expression"></a>
</h2>
<p>Traditionally the standard analysis run on all gene sets was the GO enrichment analysis. Once you have established that a given gene list is enriched for a given cell type, it becomes questionable whether a GO enrichment is driven purely by the underlying cell type enrichment. For instance, it is well established that genes associated with schizophrenia are enriched for the human post-synaptic density genes, however, it has also been shown that schizophrenia is enriched for specificity in CA1 pyramidal neurons (which highly express hPSD genes). These two enrichments can be disassociated using the following analysis:</p>
<pre><code><span class="co"># set seed for bootstrap reproducibility </span>

<span class="va">mouse_to_human_homologs</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/mouse_to_human_homologs.html" class="external-link">mouse_to_human_homologs</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">m2h</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">mouse_to_human_homologs</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HGNC.symbol"</span>,<span class="st">"MGI.symbol"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>

<span class="va">schiz_genes</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/schiz_genes.html" class="external-link">schiz_genes</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">id_genes</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/id_genes.html" class="external-link">id_genes</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">mouse.hits.schiz</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">m2h</span><span class="op">[</span><span class="va">m2h</span><span class="op">$</span><span class="va">HGNC.symbol</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">schiz_genes</span>,<span class="st">"MGI.symbol"</span><span class="op">]</span><span class="op">)</span>
<span class="va">mouse.hits.id</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">m2h</span><span class="op">[</span><span class="va">m2h</span><span class="op">$</span><span class="va">HGNC.symbol</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">id_genes</span>,<span class="st">"MGI.symbol"</span><span class="op">]</span><span class="op">)</span>
<span class="va">mouse.bg</span>  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">m2h</span><span class="op">$</span><span class="va">MGI.symbol</span><span class="op">)</span>

<span class="va">hpsd_genes</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/hpsd_genes.html" class="external-link">hpsd_genes</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">mouse.hpsd</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">m2h</span><span class="op">[</span><span class="va">m2h</span><span class="op">$</span><span class="va">HGNC.symbol</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">hpsd_genes</span>,<span class="st">"MGI.symbol"</span><span class="op">]</span><span class="op">)</span>

<span class="va">rbfox_genes</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/rbfox_genes.html" class="external-link">rbfox_genes</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">res_hpsd_schiz</span> <span class="op">=</span> <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.schiz</span>, 
                                                functional_genes <span class="op">=</span> <span class="va">mouse.hpsd</span>, 
                                                bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, 
                                                sct_data <span class="op">=</span> <span class="va">ctd</span>, 
                                                annotLevel <span class="op">=</span> <span class="fl">1</span>, 
                                                reps<span class="op">=</span><span class="va">reps</span>, 
                                                controlledCT<span class="op">=</span><span class="st">"pyramidal CA1"</span><span class="op">)</span>
<span class="va">res_rbfox_schiz</span> <span class="op">=</span> 
  <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.schiz</span>, 
                                  functional_genes <span class="op">=</span> <span class="va">rbfox_genes</span>, 
                                  bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, 
                                  sct_data <span class="op">=</span> <span class="va">ctd</span>, 
                                  annotLevel <span class="op">=</span> <span class="fl">1</span>, 
                                  reps<span class="op">=</span><span class="va">reps</span>, 
                                  controlledCT<span class="op">=</span><span class="st">"pyramidal CA1"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_hpsd_schiz</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_rbfox_schiz</span><span class="op">)</span>

<span class="va">res_hpsd_id</span> <span class="op">=</span> 
  <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.id</span>, 
                                  functional_genes <span class="op">=</span> <span class="va">mouse.hpsd</span>, 
                                  bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, 
                                  sct_data <span class="op">=</span> <span class="va">ctd</span>, 
                                  annotLevel <span class="op">=</span> <span class="fl">1</span>, 
                                  reps<span class="op">=</span><span class="va">reps</span>, 
                                  controlledCT <span class="op">=</span> <span class="st">"pyramidal SS"</span><span class="op">)</span>
<span class="va">res_rbfox_id</span> <span class="op">=</span> <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.id</span>, 
                                              functional_genes <span class="op">=</span> <span class="va">rbfox_genes</span>, 
                                              bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, 
                                              sct_data <span class="op">=</span> <span class="va">ctd</span>, 
                                              annotLevel <span class="op">=</span> <span class="fl">1</span>, 
                                              reps<span class="op">=</span><span class="va">reps</span>, 
                                              controlledCT<span class="op">=</span><span class="st">"pyramidal SS"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_hpsd_id</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_rbfox_id</span><span class="op">)</span></code></pre>
<p>The analysis also tests for enrichment of Rbfox binding genes in the schizophrenia susceptibility genes, as well as both hPSD and Rbfox genes in Intellectual Disability genes. All of the enrichments are confirmed as still being present after controlling for the associated cell type, apart from the enrichment of PSD genes in Schizophrenia which falls from borderline to non-significant.</p>
</div>
<div class="section level2">
<h2 id="controlling-for-multiple-cell-types">Controlling for multiple cell types<a class="anchor" aria-label="anchor" href="#controlling-for-multiple-cell-types"></a>
</h2>
<pre><code><span class="va">controlledCTs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pyramidal CA1"</span>,<span class="st">"pyramidal SS"</span>,<span class="st">"interneurons"</span><span class="op">)</span>

<span class="va">res_hpsd_schiz</span> <span class="op">=</span> 
  <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.schiz</span>, 
                                  functional_genes <span class="op">=</span> <span class="va">mouse.hpsd</span>, 
                                  bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, 
                                  sct_data <span class="op">=</span> <span class="va">ctd</span>, 
                                  annotLevel <span class="op">=</span> <span class="fl">1</span>, 
                                  reps<span class="op">=</span><span class="va">reps</span>, 
                                  controlledCT<span class="op">=</span><span class="va">controlledCTs</span><span class="op">)</span>
<span class="va">res_rbfox_schiz</span> <span class="op">=</span> 
  <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.schiz</span>, 
                                  functional_genes <span class="op">=</span> <span class="va">rbfox_genes</span>, 
                                  bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, sct_data <span class="op">=</span> <span class="va">ctd</span>, 
                                  annotLevel <span class="op">=</span> <span class="fl">1</span>, reps<span class="op">=</span><span class="va">reps</span>, 
                                  controlledCT<span class="op">=</span><span class="va">controlledCTs</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_hpsd_schiz</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_rbfox_schiz</span><span class="op">)</span>

<span class="va">res_hpsd_id</span> <span class="op">=</span> 
  <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.id</span>, 
                                  functional_genes <span class="op">=</span> <span class="va">mouse.hpsd</span>, 
                                  bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, 
                                  sct_data <span class="op">=</span> <span class="va">ctd</span>, 
                                  annotLevel <span class="op">=</span> <span class="fl">1</span>, 
                                  reps<span class="op">=</span><span class="va">reps</span>, 
                                  controlledCT<span class="op">=</span><span class="va">controlledCTs</span><span class="op">)</span>
<span class="va">res_rbfox_id</span> <span class="op">=</span> <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.id</span>, 
                                              functional_genes <span class="op">=</span> <span class="va">rbfox_genes</span>, 
                                              bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, 
                                              sct_data <span class="op">=</span> <span class="va">ctd</span>, 
                                              annotLevel <span class="op">=</span> <span class="fl">1</span>, 
                                              reps<span class="op">=</span><span class="va">reps</span>, 
                                              controlledCT<span class="op">=</span><span class="va">controlledCTs</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_hpsd_id</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_rbfox_id</span><span class="op">)</span></code></pre>
</div>
</div>
<div class="section level1">
<h1 id="apply-to-transcriptomic-data">Apply to transcriptomic data<a class="anchor" aria-label="anchor" href="#apply-to-transcriptomic-data"></a>
</h1>
<div class="section level2">
<h2 id="prepare-data-1">Prepare data<a class="anchor" aria-label="anchor" href="#prepare-data-1"></a>
</h2>
<div class="sourceCode" id="cb897"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<div class="sourceCode" id="cb900"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tt_alzh</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/tt_alzh.html" class="external-link">tt_alzh</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span>
<span class="co">## loading from cache</span></code></pre>
<div class="sourceCode" id="cb902"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## NOTE: rep=100 for demo purposes only. </span>
<span class="co">## Use &gt;=10,000 for publication-quality results.</span>
<span class="va">reps</span> <span class="op">&lt;-</span> <span class="fl">100</span> </code></pre></div>
</div>
<div class="section level2">
<h2 id="analysing-single-transcriptome-study">Analysing single transcriptome study<a class="anchor" aria-label="anchor" href="#analysing-single-transcriptome-study"></a>
</h2>
<p>For the prior analyses the gene lists were not associated with any numeric values or directionality. The methodology for extending this form of analysis to transcriptomic studies simply involves thresholding the most upregulated and downregulated genes.</p>
<p>To demonstrate this we have an example dataset <code>tt_alzh</code>. This data frame was generated using <code>limma</code> from a set of post-mortem tissue samples from Brodmann area 46 which were described in a paper by the Haroutunian lab<span class="citation"><sup>3</sup></span>.</p>
<p>The first step is to load the data, obtain the MGI ids, sort the rows by t-statistic and then select the most up/down-regulated genes. The package then has a function <code>ewce_expression_data</code> which thresholds and selects the gene sets, and calls the EWCE function.</p>
<p>Below we show the function call using the default settings, but if desired different threshold values can be used, or alternative columns used to sort the table.</p>
<div class="sourceCode" id="cb903"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># ewce_expression_data calls bootstrap_enrichment_test so  </span>
<span class="va">tt_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_expression_data.html">ewce_expression_data</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,
                                          tt <span class="op">=</span> <span class="va">tt_alzh</span>,
                                          annotLevel <span class="op">=</span> <span class="fl">1</span>,
                                          ttSpecies <span class="op">=</span> <span class="st">"human"</span>,
                                          sctSpecies <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: genelistSpecies not provided. Setting to 'human' by default.</span></code></pre>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## character format detected.</span></code></pre>
<pre><code><span class="co">## Converting to data.frame</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 15,259 genes extracted.</span></code></pre>
<pre><code><span class="co">## Converting mouse ==&gt; human orthologs using: gprofiler</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: mmusculus</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Checking for genes without orthologs in human.</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 16,021 genes extracted.</span></code></pre>
<pre><code><span class="co">## Extracting genes from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## 16,021 genes extracted.</span></code></pre>
<pre><code><span class="co">## Dropping 2,769 NAs of all kinds from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## Checking for genes without 1:1 orthologs.</span></code></pre>
<pre><code><span class="co">## Dropping 478 genes that have multiple input_gene per ortholog_gene.</span></code></pre>
<pre><code><span class="co">## Dropping 299 genes that have multiple ortholog_gene per input_gene.</span></code></pre>
<pre><code><span class="co">## Filtering gene_df with gene_map</span></code></pre>
<pre><code><span class="co">## Returning gene_map as dictionary</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## Total genes dropped after convert_orthologs :</span>
<span class="co">##    3,025 / 15,259 (20%)</span></code></pre>
<pre><code><span class="co">## Total genes remaining after convert_orthologs :</span>
<span class="co">##    12,234 / 15,259 (80%)</span></code></pre>
<pre><code><span class="co">## Generating gene background for mouse x human ==&gt; human</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: mmusculus</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Gene table with 21,207 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 21,207 genes from mmusculus.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## data.frame format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from Gene.Symbol.</span></code></pre>
<pre><code><span class="co">## 21,207 genes extracted.</span></code></pre>
<pre><code><span class="co">## Converting mmusculus ==&gt; hsapiens orthologs using: homologene</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mmusculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Checking for genes without orthologs in hsapiens.</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Extracting genes from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## 17,355 genes extracted.</span></code></pre>
<pre><code><span class="co">## Checking for genes without 1:1 orthologs.</span></code></pre>
<pre><code><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene.</span></code></pre>
<pre><code><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene.</span></code></pre>
<pre><code><span class="co">## Filtering gene_df with gene_map</span></code></pre>
<pre><code><span class="co">## Adding input_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## Adding ortholog_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## Total genes dropped after convert_orthologs :</span>
<span class="co">##    4,725 / 21,207 (22%)</span></code></pre>
<pre><code><span class="co">## Total genes remaining after convert_orthologs :</span>
<span class="co">##    16,482 / 21,207 (78%)</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 16,482 intersect background genes used.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Using intersect between background gene lists: 16,482 genes.</span></code></pre>
<pre><code><span class="co">## Standardising sct_data.</span></code></pre>
<pre><code><span class="co">## Converting to sparse matrix.</span>
<span class="co">## Converting to sparse matrix.</span></code></pre>
<pre><code><span class="co">## Using 1st column of tt as gene column: HGNC.symbol</span></code></pre>
<pre><code><span class="co">## 1 core(s) assigned as workers (1reserved).</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Standardising CellTypeDataset</span></code></pre>
<pre><code><span class="co">## Checking gene list inputs.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Standardising sct_data.</span></code></pre>
<pre><code><span class="co">## Converting gene list input to standardised human genes.</span></code></pre>
<pre><code><span class="co">## Running without gene size control.</span></code></pre>
<pre><code><span class="co">## 225 hit genes remain after filtering.</span></code></pre>
<pre><code><span class="co">## Computing summed proportions.</span></code></pre>
<pre><code><span class="co">## Testing for enrichment in 7 cell types...</span></code></pre>
<pre><code><span class="co">## Sorting results by p-value.</span></code></pre>
<pre><code><span class="co">## Computing BH-corrected q-values.</span></code></pre>
<pre><code><span class="co">## 3 significant cell type enrichment results @ q&lt;0.05 :</span></code></pre>
<pre><code><span class="co">##               CellType annotLevel p fold_change sd_from_mean q</span>
<span class="co">## 1    endothelial_mural          1 0    1.258281     4.547613 0</span>
<span class="co">## 2 astrocytes_ependymal          1 0    1.199796     3.529581 0</span>
<span class="co">## 3            microglia          1 0    1.207604     2.958473 0</span></code></pre>
<pre><code><span class="co">## 1 core(s) assigned as workers (1reserved).</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Standardising CellTypeDataset</span></code></pre>
<pre><code><span class="co">## Checking gene list inputs.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Standardising sct_data.</span></code></pre>
<pre><code><span class="co">## Converting gene list input to standardised human genes.</span></code></pre>
<pre><code><span class="co">## Running without gene size control.</span></code></pre>
<pre><code><span class="co">## 233 hit genes remain after filtering.</span></code></pre>
<pre><code><span class="co">## Computing summed proportions.</span></code></pre>
<pre><code><span class="co">## Testing for enrichment in 7 cell types...</span></code></pre>
<pre><code><span class="co">## Sorting results by p-value.</span></code></pre>
<pre><code><span class="co">## Computing BH-corrected q-values.</span></code></pre>
<pre><code><span class="co">## 3 significant cell type enrichment results @ q&lt;0.05 :</span></code></pre>
<pre><code><span class="co">##        CellType annotLevel p fold_change sd_from_mean q</span>
<span class="co">## 1  pyramidal_SS          1 0    1.294264     7.251418 0</span>
<span class="co">## 2 pyramidal_CA1          1 0    1.209827     4.568601 0</span>
<span class="co">## 3  interneurons          1 0    1.210311     3.667389 0</span></code></pre>
<div class="section level3">
<h3 id="plot-results-4">Plot results<a class="anchor" aria-label="anchor" href="#plot-results-4"></a>
</h3>
<p>The results of this analysis can be plotted using the <code>ewce_plot</code> function.</p>
<div class="sourceCode" id="cb1081"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span><span class="va">tt_results</span><span class="op">$</span><span class="va">joint_results</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-34-1.png" width="672"></p>
<p>As was reported in our paper, neuronal genes are found to be downregulated while glial genes are upregulated.</p>
</div>
</div>
<div class="section level2">
<h2 id="generating-bootstrap-plots-for-transcriptomes">Generating bootstrap plots for transcriptomes<a class="anchor" aria-label="anchor" href="#generating-bootstrap-plots-for-transcriptomes"></a>
</h2>
<p>A common request is to explain which differentially expressed genes are associated with a cell type…</p>
<div class="sourceCode" id="cb1082"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">full_result_path</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/generate_bootstrap_plots_for_transcriptome.html">generate_bootstrap_plots_for_transcriptome</a></span><span class="op">(</span>
  sct_data <span class="op">=</span> <span class="va">ctd</span>,
  tt <span class="op">=</span> <span class="va">tt_alzh</span>,
  annotLevel <span class="op">=</span> <span class="fl">1</span>,
  full_results <span class="op">=</span> <span class="va">tt_results</span>,
  listFileName <span class="op">=</span> <span class="st">"examples"</span>,
  reps <span class="op">=</span> <span class="va">reps</span>,
  ttSpecies <span class="op">=</span> <span class="st">"human"</span>,
  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,
  sig_only <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="merging-multiple-transcriptome-studies">Merging multiple transcriptome studies<a class="anchor" aria-label="anchor" href="#merging-multiple-transcriptome-studies"></a>
</h2>
<p>Where multiple transcriptomic studies have been performed with the same purpose, i.e. seeking differential expression in dlPFC of post-mortem schizophrenics, it is common to want to determine whether they exhibit any shared signal. EWCE can be used to merge the results of multiple studies.</p>
<p>To demonstrate this we use a two further Alzheimer’s transcriptome datasets coming from Brodmann areas 36 and 44: these area stored in <code>tt_alzh_BA36</code> and <code>tt_alzh_BA44</code>. The first step is to run EWCE on each of these individually and store the output into one list.</p>
<div class="section level3">
<h3 id="load-data">Load data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h3>
<div class="sourceCode" id="cb1083"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tt_alzh_BA36</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/tt_alzh_BA36.html" class="external-link">tt_alzh_BA36</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">tt_alzh_BA44</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/tt_alzh_BA44.html" class="external-link">tt_alzh_BA44</a></span><span class="op">(</span><span class="op">)</span> </code></pre></div>
</div>
<div class="section level3">
<h3 id="run-ewce-analysis">Run EWCE analysis<a class="anchor" aria-label="anchor" href="#run-ewce-analysis"></a>
</h3>
<div class="sourceCode" id="cb1084"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tt_results_36</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_expression_data.html">ewce_expression_data</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,
                                             tt <span class="op">=</span> <span class="va">tt_alzh_BA36</span>,
                                             annotLevel <span class="op">=</span> <span class="fl">1</span>,
                                             ttSpecies <span class="op">=</span> <span class="st">"human"</span>,
                                             sctSpecies <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span>
<span class="va">tt_results_44</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_expression_data.html">ewce_expression_data</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,
                                             tt <span class="op">=</span> <span class="va">tt_alzh_BA44</span>,
                                             annotLevel <span class="op">=</span> <span class="fl">1</span>,
                                             ttSpecies <span class="op">=</span> <span class="st">"human"</span>,
                                             sctSpecies <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span>

<span class="co"># Fill a list with the results</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/add_res_to_merging_list.html">add_res_to_merging_list</a></span><span class="op">(</span><span class="va">tt_results</span><span class="op">)</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/add_res_to_merging_list.html">add_res_to_merging_list</a></span><span class="op">(</span><span class="va">tt_results_36</span>,<span class="va">results</span><span class="op">)</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/add_res_to_merging_list.html">add_res_to_merging_list</a></span><span class="op">(</span><span class="va">tt_results_44</span>,<span class="va">results</span><span class="op">)</span>

<span class="co"># Perform the merged analysis</span>
<span class="co"># For publication reps should be higher</span>
<span class="va">merged_res</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/merged_ewce.html">merged_ewce</a></span><span class="op">(</span>results <span class="op">=</span> <span class="va">results</span>,
                                reps <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> 
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">merged_res</span><span class="op">)</span></code></pre></div>
<p>The results can then be plotted as normal using the <code>ewce_plot</code> function.</p>
<div class="sourceCode" id="cb1085"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span><span class="va">merged_res</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></code></pre></div>
<p>The merged results from all three Alzheimer’s brain regions are found to be remarkably similar, as was reported in our paper.</p>
</div>
</div>
</div>
<div class="section level1">
<h1 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h1>
<details><div class="sourceCode" id="cb1086"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R Under development (unstable) (2021-12-16 r81386)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] ewceData_1.3.0      ExperimentHub_2.3.4 AnnotationHub_3.3.7</span>
<span class="co">##  [4] BiocFileCache_2.3.3 dbplyr_2.1.1        BiocGenerics_0.41.2</span>
<span class="co">##  [7] ggplot2_3.3.5       EWCE_1.3.1          RNOmni_1.0.0       </span>
<span class="co">## [10] BiocStyle_2.23.1   </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] colorspace_2.0-2              ggsignif_0.6.3               </span>
<span class="co">##   [3] ellipsis_0.3.2                rprojroot_2.0.2              </span>
<span class="co">##   [5] XVector_0.35.0                ggdendro_0.1.22              </span>
<span class="co">##   [7] GenomicRanges_1.47.5          fs_1.5.2                     </span>
<span class="co">##   [9] farver_2.1.0                  ggpubr_0.4.0                 </span>
<span class="co">##  [11] bit64_4.0.5                   interactiveDisplayBase_1.33.0</span>
<span class="co">##  [13] AnnotationDbi_1.57.1          fansi_0.5.0                  </span>
<span class="co">##  [15] orthogene_1.1.1               cachem_1.0.6                 </span>
<span class="co">##  [17] knitr_1.37                    jsonlite_1.7.2               </span>
<span class="co">##  [19] broom_0.7.10                  png_0.1-7                    </span>
<span class="co">##  [21] shiny_1.7.1                   BiocManager_1.30.16          </span>
<span class="co">##  [23] compiler_4.2.0                httr_1.4.2                   </span>
<span class="co">##  [25] backports_1.4.1               lazyeval_0.2.2               </span>
<span class="co">##  [27] assertthat_0.2.1              Matrix_1.4-0                 </span>
<span class="co">##  [29] fastmap_1.1.0                 limma_3.51.2                 </span>
<span class="co">##  [31] later_1.3.0                   htmltools_0.5.2              </span>
<span class="co">##  [33] tools_4.2.0                   gtable_0.3.0                 </span>
<span class="co">##  [35] glue_1.6.0                    GenomeInfoDbData_1.2.7       </span>
<span class="co">##  [37] reshape2_1.4.4                dplyr_1.0.7                  </span>
<span class="co">##  [39] rappdirs_0.3.3                Rcpp_1.0.7                   </span>
<span class="co">##  [41] carData_3.0-4                 Biobase_2.55.0               </span>
<span class="co">##  [43] jquerylib_0.1.4               pkgdown_2.0.1.9000           </span>
<span class="co">##  [45] vctrs_0.3.8                   Biostrings_2.63.0            </span>
<span class="co">##  [47] babelgene_21.4                xfun_0.29                    </span>
<span class="co">##  [49] stringr_1.4.0                 mime_0.12                    </span>
<span class="co">##  [51] lifecycle_1.0.1               rstatix_0.7.0                </span>
<span class="co">##  [53] MASS_7.3-54                   zlibbioc_1.41.0              </span>
<span class="co">##  [55] scales_1.1.1                  ragg_1.2.1                   </span>
<span class="co">##  [57] promises_1.2.0.1              MatrixGenerics_1.7.0         </span>
<span class="co">##  [59] parallel_4.2.0                SummarizedExperiment_1.25.3  </span>
<span class="co">##  [61] gprofiler2_0.2.1              SingleCellExperiment_1.17.2  </span>
<span class="co">##  [63] yaml_2.2.1                    curl_4.3.2                   </span>
<span class="co">##  [65] gridExtra_2.3                 memoise_2.0.1                </span>
<span class="co">##  [67] sass_0.4.0                    stringi_1.7.6                </span>
<span class="co">##  [69] RSQLite_2.2.9                 highr_0.9                    </span>
<span class="co">##  [71] BiocVersion_3.15.0            S4Vectors_0.33.8             </span>
<span class="co">##  [73] desc_1.4.0                    filelock_1.0.2               </span>
<span class="co">##  [75] BiocParallel_1.29.6           GenomeInfoDb_1.31.1          </span>
<span class="co">##  [77] rlang_0.4.12                  pkgconfig_2.0.3              </span>
<span class="co">##  [79] systemfonts_1.0.3             bitops_1.0-7                 </span>
<span class="co">##  [81] matrixStats_0.61.0            evaluate_0.14                </span>
<span class="co">##  [83] lattice_0.20-45               purrr_0.3.4                  </span>
<span class="co">##  [85] labeling_0.4.2                patchwork_1.1.1              </span>
<span class="co">##  [87] htmlwidgets_1.5.4             cowplot_1.1.1                </span>
<span class="co">##  [89] bit_4.0.4                     tidyselect_1.1.1             </span>
<span class="co">##  [91] plyr_1.8.6                    magrittr_2.0.1               </span>
<span class="co">##  [93] bookdown_0.24                 R6_2.5.1                     </span>
<span class="co">##  [95] IRanges_2.29.1                generics_0.1.1               </span>
<span class="co">##  [97] DelayedArray_0.21.2           DBI_1.1.2                    </span>
<span class="co">##  [99] withr_2.4.3                   pillar_1.6.4                 </span>
<span class="co">## [101] KEGGREST_1.35.0               abind_1.4-5                  </span>
<span class="co">## [103] RCurl_1.98-1.5                tibble_3.1.6                 </span>
<span class="co">## [105] homologene_1.4.68.19.3.27     crayon_1.4.2                 </span>
<span class="co">## [107] car_3.0-12                    utf8_1.2.2                   </span>
<span class="co">## [109] plotly_4.10.0                 rmarkdown_2.11               </span>
<span class="co">## [111] grid_4.2.0                    data.table_1.14.2            </span>
<span class="co">## [113] blob_1.2.2                    digest_0.6.29                </span>
<span class="co">## [115] xtable_1.8-4                  HGNChelper_0.8.1             </span>
<span class="co">## [117] tidyr_1.1.4                   httpuv_1.6.4                 </span>
<span class="co">## [119] textshaping_0.3.6             stats4_4.2.0                 </span>
<span class="co">## [121] munsell_0.5.0                 viridisLite_0.4.0            </span>
<span class="co">## [123] bslib_0.3.1</span></code></pre>
</details><hr>
</div>
<div class="section level1 unnumbered">
<h1 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h1>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-zeisel2015cell" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Zeisel, A. <em>et al.</em> Cell types in the mouse cortex and hippocampus revealed by single-cell RNA-seq. <em>Science</em> <strong>347,</strong> 1138–1142 (2015).</div>
</div>
<div id="ref-skene_2016" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Skene, N. &amp; Grant, S. Identification of vulnerable cell types in major brain disorders using single cell transcriptomes and expression weighted cell type enrichment. <em>Frontiers in Neuroscience</em> (2016). doi:<a href="https://doi.org/10.3389/fnins.2016.00016" class="external-link">10.3389/fnins.2016.00016</a>
</div>
</div>
<div id="ref-haroutunian2009transcriptional" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">Haroutunian, V., Katsel, P. &amp; Schmeidler, J. Transcriptional vulnerability of brain regions in alzheimer’s disease and dementia. <em>Neurobiology of aging</em> <strong>30,</strong> 561–573 (2009).</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
