<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Extended examples • EWCE</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Extended examples">
<meta property="og:description" content="EWCE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.11.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/extended.html">Extended examples</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/NathanSkene/EWCE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Extended examples</h1>
                        <h4 data-toc-skip class="author"><h4>
<br>
Authors: Alan Murphy, Brian Schilder, and Nathan Skene<br>
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
<br>
Updated: Oct-31-2023<br>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/NathanSkene/EWCE/blob/HEAD/vignettes/extended.Rmd" class="external-link"><code>vignettes/extended.Rmd</code></a></small>
      <div class="hidden name"><code>extended.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/NathanSkene/EWCE" class="external-link">EWCE</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: RNOmni</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span></span></code></pre></div>
<p><strong>NOTE</strong>: This documentation is for the development
version of <code>EWCE</code>. See <a href="https://bioconductor.org/packages/release/bioc/html/EWCE.html" class="external-link">Bioconductor</a>
for documentation on the current release version.</p>
</div>
<div class="section level2">
<h2 id="run-cell-type-enrichment-tests">Run cell-type enrichment tests<a class="anchor" aria-label="anchor" href="#run-cell-type-enrichment-tests"></a>
</h2>
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p>In the following vignette, we provide more a more in-depth version of
the examples provided in the <a href="https://nathanskene.github.io/EWCE/articles/EWCE.html" class="external-link">Getting
started vignette</a>.</p>
</div>
<div class="section level3">
<h3 id="prepare-input-data">Prepare input data<a class="anchor" aria-label="anchor" href="#prepare-input-data"></a>
</h3>
<div class="section level4">
<h4 id="celltypedataset">CellTypeDataset<a class="anchor" aria-label="anchor" href="#celltypedataset"></a>
</h4>
<p>For this example we use a subset of the genes from the merged dataset
generated in the <a href="#create-a-celltypedataset">Create a
CellTypeDataset</a> section below, which is accessed using
<code><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ewceData::ctd()</a></code>.</p>
<div class="section level5">
<h5 id="ctd-levels">CTD levels<a class="anchor" aria-label="anchor" href="#ctd-levels"></a>
</h5>
<p>Each level of a CTD corresponds to increasingly refined
cell-type/-subtype annotations. For example, in the CTD
<code><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ewceData::ctd()</a></code> level 1 includes the cell-type
“interneurons”, while level 2 breaks these this group into 16 different
interneuron subtypes (“Int…”).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load merged cortex and hypothalamus dataset generated by Karolinska institute</span></span>
<span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># i.e. ctd_MergedKI</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_ctd.html">plot_ctd</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>                        level <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                        genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,</span>
<span>                        metric <span class="op">=</span> <span class="st">"mean_exp"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-1-1.png" width="700"></p>
<p><strong>Note</strong> - You can load <code><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ewceData::ctd()</a></code>
offline by passing <code>localhub = TRUE</code>. This will work off a
previously cached version of the reference dataset from
<code>ExperimentHub</code>.</p>
</div>
</div>
<div class="section level4">
<h4 id="gene-lists">Gene lists<a class="anchor" aria-label="anchor" href="#gene-lists"></a>
</h4>
<p>For the first demonstration of <code>EWCE</code> we will test for
whether genes that are genetically associated with Alzheimer’s disease
are enriched in any particular cell type.</p>
<p>This example gene list is stored within the <code>ewceData</code>
package:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/example_genelist.html" class="external-link">example_genelist</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "APOE"     "BIN1"     "CLU"      "ABCA7"    "CR1"      "PICALM"  </span></span>
<span><span class="co">##  [7] "MS4A6A"   "CD33"     "MS4A4E"   "CD2AP"    "EOGA1"    "INPP5D"  </span></span>
<span><span class="co">## [13] "MEF2C"    "HLA-DRB5" "ZCWPW1"   "NME8"     "PTK2B"    "CELF1"   </span></span>
<span><span class="co">## [19] "SORL1"    "FERMT2"   "SLC24A4"  "CASS4"</span></span></code></pre>
<p><strong>Note</strong> - You can load
<code><a href="https://rdrr.io/pkg/ewceData/man/example_genelist.html" class="external-link">ewceData::example_genelist()</a></code> offline by passing
<code>localhub = TRUE</code>. This will work off a previously cached
version of the reference dataset from <code>ExperimentHub</code>.</p>
<div class="section level5">
<h5 id="gene-formats-and-species">Gene formats and species<a class="anchor" aria-label="anchor" href="#gene-formats-and-species"></a>
</h5>
<p>All gene IDs are assumed by the package to be provided in gene symbol
format (rather than Ensembl/Entrez). Symbols can be provided as any
species-specific gene symbols supported by the package
<code>orthogene</code>, though the <code>genelistSpecies</code> argument
will need to be set appropriately.</p>
<p>Likewise, the single-cell dataset can be from any species, but the
<code>sctSpecies</code> argument must be set accordingly.</p>
<p>The example gene list here stores the human genes associated with
human disease, and hence are HGNC symbols.</p>
<p>The next step is to determine the most suitable background set. This
can be user-supplied, but by default the background is all 1:1 ortholog
genes shared by <code>genelistSpecies</code> and <code>sctSpecies</code>
that are also present in <code>sct_data</code>.</p>
</div>
<div class="section level5">
<h5 id="notes-on-orthogene">Notes on <code>orthogene</code><a class="anchor" aria-label="anchor" href="#notes-on-orthogene"></a>
</h5>
<p><code>orthogene</code> substantially improves upon previous ortholog
translations that used the static
<code><a href="https://rdrr.io/pkg/ewceData/man/mouse_to_human_homologs.html" class="external-link">ewceData::mouse_to_human_homologs()</a></code> file as the former is
updated using the <a href="https://www.ncbi.nlm.nih.gov/homologene" class="external-link">Homologene</a> database
periodically.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp</span> <span class="op">&lt;-</span> <span class="va">ctd</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">mean_exp</span></span>
<span></span>
<span><span class="co">#### Old conversion method ####</span></span>
<span><span class="co">#if running offline pass localhub = TRUE</span></span>
<span><span class="va">m2h</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/mouse_to_human_homologs.html" class="external-link">mouse_to_human_homologs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp_old</span> <span class="op">&lt;-</span> <span class="va">exp</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">m2h</span><span class="op">$</span><span class="va">MGI.symbol</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co">#### New conversion method (used by EWCE internally) ####</span></span>
<span><span class="va">exp_new</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/orthogene/man/convert_orthologs.html" class="external-link">convert_orthologs</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp</span>,</span>
<span>                                        input_species <span class="op">=</span> <span class="st">"mouse"</span>, </span>
<span>                                        output_species <span class="op">=</span> <span class="st">"human"</span>, </span>
<span>                                        method <span class="op">=</span> <span class="st">"homologene"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Preparing gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## Dense matrix format detected.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from rownames.</span></span></code></pre>
<pre><code><span><span class="co">## 15,259 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Converting mouse ==&gt; human orthologs using: homologene</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without orthologs in human.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from input_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 13,416 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from ortholog_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 13,416 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without 1:1 orthologs.</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 46 genes that have multiple input_gene per ortholog_gene (many:1).</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 56 genes that have multiple ortholog_gene per input_gene (1:many).</span></span></code></pre>
<pre><code><span><span class="co">## Filtering gene_df with gene_map</span></span></code></pre>
<pre><code><span><span class="co">## Setting ortholog_gene to rownames.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## Total genes dropped after convert_orthologs :</span></span>
<span><span class="co">##    2,016 / 15,259 (13%)</span></span></code></pre>
<pre><code><span><span class="co">## Total genes remaining after convert_orthologs :</span></span>
<span><span class="co">##    13,243 / 15,259 (87%)</span></span></code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#### Report ####</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"The new method retains "</span>,</span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/formatc.html" class="external-link">formatC</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">exp_new</span><span class="op">)</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">exp_old</span><span class="op">)</span>, big.mark <span class="op">=</span> <span class="st">","</span><span class="op">)</span>,</span>
<span>        <span class="st">" more genes than the old method."</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## The new method retains 918 more genes than the old method.</span></span></code></pre>
<p><code>orthogene</code> is also used internally to standardise gene
lists supplied to <code>EWCE</code> functions, such as
<code>EWCE::bootstrap_enrichment_test(hits = &lt;gene_list&gt;)</code>.</p>
<p>Not only can it map these gene lists across species, but it can also
map them within species. For example, if you provide a list of Ensembl
IDs, it will automatically convert them to standardised HGNC gene
symbols so they’re compatible with the similarly standardised
CellTypeDataset.</p>
</div>
</div>
<div class="section level4">
<h4 id="setting-analysis-parameters">Setting analysis parameters<a class="anchor" aria-label="anchor" href="#setting-analysis-parameters"></a>
</h4>
<p>We now need to set the parameters for the analysis. For a publishable
analysis we would want to generate over 10,000 random lists and
determine their expression levels, but for computational speed let us
only use <code>reps=100</code>. We want to analyse level 1 annotations
so set level to 1.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Use 100 bootstrap lists for speed, for publishable analysis use &gt;=10000</span></span>
<span><span class="va">reps</span> <span class="op">&lt;-</span> <span class="fl">100</span> </span>
<span><span class="co"># Use level 1 annotations (i.e. Interneurons)</span></span>
<span><span class="va">annotLevel</span> <span class="op">&lt;-</span> <span class="fl">1</span> </span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="enrichment-tests">Enrichment tests<a class="anchor" aria-label="anchor" href="#enrichment-tests"></a>
</h3>
<div class="section level4">
<h4 id="default-tests">Default tests<a class="anchor" aria-label="anchor" href="#default-tests"></a>
</h4>
<p>We have now loaded the SCT data, prepared the gene lists and set the
parameters. We run the model as follows.</p>
<p><strong>Note</strong>: We set the seed at the top of this vignette to
ensure reproducibility in the bootstrap sampling function.</p>
<div class="section level5">
<h5 id="parallelisation">Parallelisation<a class="anchor" aria-label="anchor" href="#parallelisation"></a>
</h5>
<p>You can now speed up the bootstrapping process by parallelising
across multiple cores with the parameter <code>no_cores</code>
(<code>=1</code> by default).</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Bootstrap significance test, no control for transcript length and GC content </span></span>
<span><span class="va">full_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>                                                sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>                                                genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>                                                hits <span class="op">=</span> <span class="va">hits</span>, </span>
<span>                                                reps <span class="op">=</span> <span class="va">reps</span>,</span>
<span>                                                annotLevel <span class="op">=</span> <span class="va">annotLevel</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1 core(s) assigned as workers (1 reserved).</span></span></code></pre>
<pre><code><span><span class="co">## Generating gene background for mouse x human ==&gt; human</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 21,207 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 21,207 genes from mouse.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span>
<span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## Preparing gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## data.frame format detected.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from Gene.Symbol.</span></span></code></pre>
<pre><code><span><span class="co">## 21,207 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Converting mouse ==&gt; human orthologs using: homologene</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without orthologs in human.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from input_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from ortholog_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without 1:1 orthologs.</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene (many:1).</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene (1:many).</span></span></code></pre>
<pre><code><span><span class="co">## Filtering gene_df with gene_map</span></span></code></pre>
<pre><code><span><span class="co">## Adding input_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## Adding ortholog_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## Total genes dropped after convert_orthologs :</span></span>
<span><span class="co">##    4,725 / 21,207 (22%)</span></span></code></pre>
<pre><code><span><span class="co">## Total genes remaining after convert_orthologs :</span></span>
<span><span class="co">##    16,482 / 21,207 (78%)</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 intersect background genes used.</span></span></code></pre>
<pre><code><span><span class="co">## Standardising CellTypeDataset</span></span></code></pre>
<pre><code><span><span class="co">## Checking gene list inputs.</span></span></code></pre>
<pre><code><span><span class="co">## Running without gene size control.</span></span></code></pre>
<pre><code><span><span class="co">## 17 hit gene(s) remain after filtering.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene scores.</span></span></code></pre>
<pre><code><span><span class="co">## Using previously sampled genes.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene counts.</span></span></code></pre>
<pre><code><span><span class="co">## Testing for enrichment in 7 cell types...</span></span></code></pre>
<pre><code><span><span class="co">## Sorting results by p-value.</span></span></code></pre>
<pre><code><span><span class="co">## Computing BH-corrected q-values.</span></span></code></pre>
<pre><code><span><span class="co">## 1 significant cell type enrichment results @ q&lt;0.05 :</span></span></code></pre>
<pre><code><span><span class="co">##    CellType annotLevel p fold_change sd_from_mean q</span></span>
<span><span class="co">## 1 microglia          1 0    1.965915     3.938119 0</span></span></code></pre>
<p><strong>Note</strong> - You can run
<code><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test()</a></code> offline by passing
<code>localhub = TRUE</code>. This will work off a previously cached
version of the reference dataset from <code>ExperimentHub</code>.</p>
<p>A note on both the background and target gene lists, other common
gene list objects can be used as inputs such as
<code>BiocSet::BiocSet</code> and <code>GSEABase::GeneSet</code>. Below
is an example of how to format each for the target gene list
(<code>hits</code>):</p>
<div class="sourceCode" id="cb128"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"BiocSet"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"BiocSet"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"GSEABase"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"GSEABase"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BiocSet</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GSEABase</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Save both approaches as hits which will be passed to bootstrap_enrichment_test</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Inpp5d"</span>,<span class="st">"Cd2ap"</span>,<span class="st">"Nme8"</span>,</span>
<span>          <span class="st">"Cass4"</span>,<span class="st">"Mef2c"</span>,<span class="st">"Zcwpw1"</span>,<span class="st">"Bin1"</span>,</span>
<span>          <span class="st">"Clu"</span>,<span class="st">"Celf1"</span>,<span class="st">"Abca7"</span>,<span class="st">"Slc24a4"</span>,</span>
<span>          <span class="st">"Ptk2b"</span>,<span class="st">"Picalm"</span>,<span class="st">"Fermt2"</span>,<span class="st">"Sorl1"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#BiocSet::BiocSet, BiocSet_target contains the gene list target</span></span>
<span><span class="va">BiocSet_target</span> <span class="op">&lt;-</span> <span class="fu">BiocSet</span><span class="fu">::</span><span class="fu">BiocSet</span><span class="op">(</span>set1 <span class="op">=</span> <span class="va">genes</span><span class="op">)</span> </span>
<span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu">BiocSet</span><span class="fu">::</span><span class="fu">es_element</span><span class="op">(</span><span class="va">BiocSet_target</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#GSEABase::GeneSet, GeneSet_target contains the gene list target</span></span>
<span><span class="va">GeneSet_target</span> <span class="op">&lt;-</span> <span class="fu">GSEABase</span><span class="fu">::</span><span class="fu">GeneSet</span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span>
<span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu">GSEABase</span><span class="fu">::</span><span class="fu">geneIds</span><span class="op">(</span><span class="va">GeneSet_target</span><span class="op">)</span> </span></code></pre></div>
<p>The main table of results is stored in
<code>full_results$results</code>. We can see the most significant
results using:</p>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">full_results</span><span class="op">$</span><span class="va">results</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="23%">
<col width="23%">
<col width="12%">
<col width="5%">
<col width="13%">
<col width="14%">
<col width="6%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">CellType</th>
<th align="right">annotLevel</th>
<th align="right">p</th>
<th align="right">fold_change</th>
<th align="right">sd_from_mean</th>
<th align="right">q</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">microglia</td>
<td align="left">microglia</td>
<td align="right">1</td>
<td align="right">0.00</td>
<td align="right">1.9659148</td>
<td align="right">3.9381188</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">astrocytes_ependymal</td>
<td align="left">astrocytes_ependymal</td>
<td align="right">1</td>
<td align="right">0.13</td>
<td align="right">1.2624889</td>
<td align="right">1.1553910</td>
<td align="right">0.455</td>
</tr>
<tr class="odd">
<td align="left">pyramidal_SS</td>
<td align="left">pyramidal_SS</td>
<td align="right">1</td>
<td align="right">0.80</td>
<td align="right">0.8699242</td>
<td align="right">-0.8226268</td>
<td align="right">1.000</td>
</tr>
<tr class="even">
<td align="left">oligodendrocytes</td>
<td align="left">oligodendrocytes</td>
<td align="right">1</td>
<td align="right">0.87</td>
<td align="right">0.7631149</td>
<td align="right">-1.0861761</td>
<td align="right">1.000</td>
</tr>
<tr class="odd">
<td align="left">pyramidal_CA1</td>
<td align="left">pyramidal_CA1</td>
<td align="right">1</td>
<td align="right">0.89</td>
<td align="right">0.8202496</td>
<td align="right">-1.1738063</td>
<td align="right">1.000</td>
</tr>
<tr class="even">
<td align="left">endothelial_mural</td>
<td align="left">endothelial_mural</td>
<td align="right">1</td>
<td align="right">0.90</td>
<td align="right">0.7674534</td>
<td align="right">-1.1811797</td>
<td align="right">1.000</td>
</tr>
<tr class="odd">
<td align="left">interneurons</td>
<td align="left">interneurons</td>
<td align="right">1</td>
<td align="right">1.00</td>
<td align="right">0.4012954</td>
<td align="right">-3.4703413</td>
<td align="right">1.000</td>
</tr>
</tbody>
</table>
</div>
<div class="section level5">
<h5 id="plot-results">Plot results<a class="anchor" aria-label="anchor" href="#plot-results"></a>
</h5>
<p>The results can be visualised using another function, which shows for
each cell type, the number of standard deviations from the mean the
level of expression was found to be in the target gene list, relative to
the bootstrapped mean:</p>
<div class="sourceCode" id="cb130"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">full_results</span><span class="op">$</span><span class="va">results</span>,</span>
<span>                             mtc_method <span class="op">=</span><span class="st">"BH"</span>,</span>
<span>                               ctd <span class="op">=</span> <span class="va">ctd</span><span class="op">)</span> </span>
<span>  <span class="co"># print(plot_list$plain)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p>For publications it can be useful to plot a dendrogram alongside the
plot. This can be done by including the cell type data as an additional
argument. The dendrogram should automatically align with the graph ticks
(thanks to <a href="https://github.com/bobGSmith" class="external-link">Robert
Gordon-Smith</a> for this solution):</p>
<div class="sourceCode" id="cb131"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">withDendro</span><span class="op">)</span>  </span></code></pre></div>
<p>If you want to view the characteristics of enrichment for each gene
within the list then the <code>generate_bootstrap_plots</code> function
should be used. This saves the plots into the BootstrapPlots folder.
This takes the results of a bootstrapping analysis so as to only
generate plots for significant enrichments. The
<code>listFileName</code> argument is used to give the generated graphs
a particular file name. The <code>savePath</code> argument is used here
to save the files to a temporary directory, this can be updated to your
preferred location. The file path where it was saved is returned so the
temporary directory can be located if used.</p>
<div class="sourceCode" id="cb132"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bt_plot_location</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/generate_bootstrap_plots.html">generate_bootstrap_plots</a></span><span class="op">(</span></span>
<span>  sct_data <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>  hits <span class="op">=</span> <span class="va">hits</span>,</span>
<span>  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>  genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  reps <span class="op">=</span> <span class="va">reps</span>,</span>
<span>  annotLevel <span class="op">=</span> <span class="va">annotLevel</span>,</span>
<span>  full_results <span class="op">=</span> <span class="va">full_results</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="control-for-transcript-length-and-gc-content">Control for transcript length and GC-content<a class="anchor" aria-label="anchor" href="#control-for-transcript-length-and-gc-content"></a>
</h4>
<p>When analysing genes found through genetic association studies it is
important to consider biases which might be introduced as a result of
transcript length and GC-content. The package can control for these by
selecting the bootstrap lists such that the <em>i<sup>th</sup></em> gene
in the random list has properties similar to the<em>i<sup>th</sup></em>
gene in the target list. To enable the algorithm to do this it needs to
be passed the gene lists as HGNC symbols rather than MGI.</p>
<p>The bootstrapping function then takes different arguments:</p>
<div class="sourceCode" id="cb133"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Bootstrap significance test controlling for transcript length and GC content</span></span>
<span><span class="co">#if running offline pass localhub = TRUE</span></span>
<span><span class="va">cont_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span></span>
<span>  sct_data <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>  hits <span class="op">=</span> <span class="va">hits</span>,</span>
<span>  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>  genelistSpecies <span class="op">=</span> <span class="st">"human"</span>, </span>
<span>  reps <span class="op">=</span> <span class="va">reps</span>,</span>
<span>  annotLevel <span class="op">=</span> <span class="va">annotLevel</span>,</span>
<span>  geneSizeControl <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1 core(s) assigned as workers (1 reserved).</span></span></code></pre>
<pre><code><span><span class="co">## Generating gene background for mouse x human ==&gt; human</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 21,207 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 21,207 genes from mouse.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span>
<span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## Preparing gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## data.frame format detected.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from Gene.Symbol.</span></span></code></pre>
<pre><code><span><span class="co">## 21,207 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Converting mouse ==&gt; human orthologs using: homologene</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without orthologs in human.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from input_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from ortholog_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without 1:1 orthologs.</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene (many:1).</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene (1:many).</span></span></code></pre>
<pre><code><span><span class="co">## Filtering gene_df with gene_map</span></span></code></pre>
<pre><code><span><span class="co">## Adding input_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## Adding ortholog_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## Total genes dropped after convert_orthologs :</span></span>
<span><span class="co">##    4,725 / 21,207 (22%)</span></span></code></pre>
<pre><code><span><span class="co">## Total genes remaining after convert_orthologs :</span></span>
<span><span class="co">##    16,482 / 21,207 (78%)</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 intersect background genes used.</span></span></code></pre>
<pre><code><span><span class="co">## Standardising CellTypeDataset</span></span></code></pre>
<pre><code><span><span class="co">## Checking gene list inputs.</span></span></code></pre>
<pre><code><span><span class="co">## Running with gene size control.</span></span></code></pre>
<pre><code><span><span class="co">## Warning: sctSpecies_origin not provided. Setting to 'mouse' by default.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: gprofiler</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in gprofiler.</span></span></code></pre>
<pre><code><span><span class="co">## Using stored `gprofiler_orgs`.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: hsapiens</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 62,663 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 62,663 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## 62,663 human Ensembl IDs and 40,835 human Gene Symbols imported.</span></span></code></pre>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## Controlled bootstrapping network generated.</span></span></code></pre>
<pre><code><span><span class="co">## 17 hit gene(s) remain after filtering.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene scores.</span></span></code></pre>
<pre><code><span><span class="co">## Using previously sampled genes.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene counts.</span></span></code></pre>
<pre><code><span><span class="co">## Testing for enrichment in 7 cell types...</span></span></code></pre>
<pre><code><span><span class="co">## Sorting results by p-value.</span></span></code></pre>
<pre><code><span><span class="co">## Computing BH-corrected q-values.</span></span></code></pre>
<pre><code><span><span class="co">## 1 significant cell type enrichment results @ q&lt;0.05 :</span></span></code></pre>
<pre><code><span><span class="co">##    CellType annotLevel p fold_change sd_from_mean q</span></span>
<span><span class="co">## 1 microglia          1 0    1.982267     3.720942 0</span></span></code></pre>
<div class="section level5">
<h5 id="plot-results-1">Plot results<a class="anchor" aria-label="anchor" href="#plot-results-1"></a>
</h5>
<p>We plot these results using <code>ewce_plot</code>:</p>
<div class="sourceCode" id="cb228"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">cont_results</span><span class="op">$</span><span class="va">results</span>,</span>
<span>                             mtc_method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<p>This shows that the controlled method generates enrichments that are
generally comparable to the standard method.</p>
</div>
</div>
<div class="section level4">
<h4 id="test-different-ctd-levels">Test different CTD levels<a class="anchor" aria-label="anchor" href="#test-different-ctd-levels"></a>
</h4>
<p>Both the analyses shown above were run on level 1 annotations. It is
possible to test on the level 2 cell type level annotations by changing
one of the arguments.</p>
<div class="sourceCode" id="cb229"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Bootstrap significance test controlling for transcript length and GC content</span></span>
<span><span class="co">#if running offline pass localhub = TRUE</span></span>
<span><span class="va">cont_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>                                                hits <span class="op">=</span> <span class="va">hits</span>, </span>
<span>                                                sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>                                                genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>                                                reps <span class="op">=</span> <span class="va">reps</span>,</span>
<span>                                                annotLevel <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                                geneSizeControl <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1 core(s) assigned as workers (1 reserved).</span></span></code></pre>
<pre><code><span><span class="co">## Generating gene background for mouse x human ==&gt; human</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 21,207 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 21,207 genes from mouse.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span>
<span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## Preparing gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## data.frame format detected.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from Gene.Symbol.</span></span></code></pre>
<pre><code><span><span class="co">## 21,207 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Converting mouse ==&gt; human orthologs using: homologene</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without orthologs in human.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from input_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from ortholog_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without 1:1 orthologs.</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene (many:1).</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene (1:many).</span></span></code></pre>
<pre><code><span><span class="co">## Filtering gene_df with gene_map</span></span></code></pre>
<pre><code><span><span class="co">## Adding input_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## Adding ortholog_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## Total genes dropped after convert_orthologs :</span></span>
<span><span class="co">##    4,725 / 21,207 (22%)</span></span></code></pre>
<pre><code><span><span class="co">## Total genes remaining after convert_orthologs :</span></span>
<span><span class="co">##    16,482 / 21,207 (78%)</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 intersect background genes used.</span></span></code></pre>
<pre><code><span><span class="co">## Standardising CellTypeDataset</span></span></code></pre>
<pre><code><span><span class="co">## Checking gene list inputs.</span></span></code></pre>
<pre><code><span><span class="co">## Running with gene size control.</span></span></code></pre>
<pre><code><span><span class="co">## Warning: sctSpecies_origin not provided. Setting to 'mouse' by default.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: gprofiler</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in gprofiler.</span></span></code></pre>
<pre><code><span><span class="co">## Using stored `gprofiler_orgs`.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: hsapiens</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 62,663 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 62,663 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## 62,663 human Ensembl IDs and 40,835 human Gene Symbols imported.</span></span></code></pre>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## Controlled bootstrapping network generated.</span></span></code></pre>
<pre><code><span><span class="co">## 17 hit gene(s) remain after filtering.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene scores.</span></span></code></pre>
<pre><code><span><span class="co">## Using previously sampled genes.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene counts.</span></span></code></pre>
<pre><code><span><span class="co">## Testing for enrichment in 48 cell types...</span></span></code></pre>
<pre><code><span><span class="co">## Sorting results by p-value.</span></span></code></pre>
<pre><code><span><span class="co">## Computing BH-corrected q-values.</span></span></code></pre>
<pre><code><span><span class="co">## 2 significant cell type enrichment results @ q&lt;0.05 :</span></span></code></pre>
<pre><code><span><span class="co">##   CellType annotLevel p fold_change sd_from_mean q</span></span>
<span><span class="co">## 1     Pvm2          2 0    2.456313     3.199867 0</span></span>
<span><span class="co">## 2     Mgl1          2 0    2.378474     3.070365 0</span></span></code></pre>
<div class="section level5">
<h5 id="plot-results-2">Plot results<a class="anchor" aria-label="anchor" href="#plot-results-2"></a>
</h5>
<div class="sourceCode" id="cb324"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">cont_results</span><span class="op">$</span><span class="va">results</span>,</span>
<span>                             mtc_method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>With the cell subtype analysis each microglial subtype was enriched
and correspondingly we see here that the microglial cell type is
enriched.</p>
</div>
</div>
<div class="section level4">
<h4 id="plot-results-from-multiple-sets-of-enrichment-results">Plot results from multiple sets of enrichment results<a class="anchor" aria-label="anchor" href="#plot-results-from-multiple-sets-of-enrichment-results"></a>
</h4>
<p>It is often useful to plot results from multiple gene list analyses
together. The <code>ewce_plot</code> function allows multiple enrichment
analyses to be performed together. To achieve this the results data
frames are just appended onto each other, with an additional
<code>list</code> column added detailing which analysis they relate
to.</p>
<p>To demonstrate this we need to first generate a second analysis so
let us sample thirty random genes, and run the bootstrapping analysis on
it.</p>
<div class="sourceCode" id="cb325"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#### Generate random gene list ####</span></span>
<span><span class="co">#if running offline pass localhub = TRUE</span></span>
<span><span class="va">human.bg</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/mouse_to_human_homologs.html" class="external-link">mouse_to_human_homologs</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">HGNC.symbol</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb328"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene.list.2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">human.bg</span>,size <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#if running offline pass localhub = TRUE</span></span>
<span><span class="va">second_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>                                                  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>                                                  hits <span class="op">=</span> <span class="va">gene.list.2</span>, </span>
<span>                                                  reps <span class="op">=</span> <span class="va">reps</span>,</span>
<span>                                                  annotLevel <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1 core(s) assigned as workers (1 reserved).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: genelistSpecies not provided. Setting to 'human' by default.</span></span></code></pre>
<pre><code><span><span class="co">## Generating gene background for mouse x human ==&gt; human</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 21,207 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 21,207 genes from mouse.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span>
<span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## Preparing gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## data.frame format detected.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from Gene.Symbol.</span></span></code></pre>
<pre><code><span><span class="co">## 21,207 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Converting mouse ==&gt; human orthologs using: homologene</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without orthologs in human.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from input_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from ortholog_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without 1:1 orthologs.</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene (many:1).</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene (1:many).</span></span></code></pre>
<pre><code><span><span class="co">## Filtering gene_df with gene_map</span></span></code></pre>
<pre><code><span><span class="co">## Adding input_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## Adding ortholog_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## Total genes dropped after convert_orthologs :</span></span>
<span><span class="co">##    4,725 / 21,207 (22%)</span></span></code></pre>
<pre><code><span><span class="co">## Total genes remaining after convert_orthologs :</span></span>
<span><span class="co">##    16,482 / 21,207 (78%)</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 intersect background genes used.</span></span></code></pre>
<pre><code><span><span class="co">## Standardising CellTypeDataset</span></span></code></pre>
<pre><code><span><span class="co">## Checking gene list inputs.</span></span></code></pre>
<pre><code><span><span class="co">## Running without gene size control.</span></span></code></pre>
<pre><code><span><span class="co">## 18 hit gene(s) remain after filtering.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene scores.</span></span></code></pre>
<pre><code><span><span class="co">## Using previously sampled genes.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene counts.</span></span></code></pre>
<pre><code><span><span class="co">## Testing for enrichment in 7 cell types...</span></span></code></pre>
<pre><code><span><span class="co">## Sorting results by p-value.</span></span></code></pre>
<pre><code><span><span class="co">## Computing BH-corrected q-values.</span></span></code></pre>
<pre><code><span><span class="co">## 0 significant cell type enrichment results @ q&lt;0.05 :</span></span></code></pre>
<div class="sourceCode" id="cb410"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">full_res2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">full_results</span><span class="op">$</span><span class="va">results</span>,</span>
<span>                       list<span class="op">=</span><span class="st">"Alzheimers"</span><span class="op">)</span></span>
<span><span class="va">rando_res2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">second_results</span><span class="op">$</span><span class="va">results</span>,</span>
<span>                        list<span class="op">=</span><span class="st">"Random"</span><span class="op">)</span></span>
<span><span class="va">merged_results</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">full_res2</span>, <span class="va">rando_res2</span><span class="op">)</span></span></code></pre></div>
<div class="section level5">
<h5 id="plot-results-3">Plot results<a class="anchor" aria-label="anchor" href="#plot-results-3"></a>
</h5>
<p>Here we apply multiple-testing correction (BH = Benjamini-Hochberg)
to guard against positive results due to chance.</p>
<div class="sourceCode" id="cb411"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">merged_results</span>,</span>
<span>                             mtc_method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<p>As expected, the second randomly generated gene list shows no
significant enrichment results.</p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="create-a-celltypedataset">Create a CellTypeDataset<a class="anchor" aria-label="anchor" href="#create-a-celltypedataset"></a>
</h2>
<div class="section level3">
<h3 id="introduction-1">Introduction<a class="anchor" aria-label="anchor" href="#introduction-1"></a>
</h3>
<p><code>EWCE</code> was originally designed to work with the
single-cell cortical transcriptome data from the <a href="http://linnarssonlab.org/" class="external-link">Linnarsson Lab</a><span class="citation"><sup>1</sup></span> (available <a href="http://linnarssonlab.org/cortex/" class="external-link">here</a>).</p>
<p>Using this package it is possible to read in any single-cell
transcriptome data, provided that you have a cell by gene expression
matrix (with each cell as a separate column) and a separate annotation
dataframe, with a row for each cell.</p>
<p>The <code>EWCE</code> process involves testing for whether the genes
in a target list have higher levels of expression in a given cell type
than can reasonably be expected by chance. The probability distribution
for this is estimated by randomly generating gene lists of equal length
from a set of background genes.</p>
<p><code>EWCE</code> can be applied to any gene list. In the original
paper we reported its application to genetic and transcriptomic
datasets<span class="citation"><sup>2</sup></span>..</p>
<p>Note that throughout this vignette we use the terms ‘cell type’ and
‘cell subtype’ to refer to two levels of annotation of what a cell type
is. This is described in further detail in our paper<span class="citation"><sup>2</sup></span>, but relates to the two levels of
annotation provided in the Linnarsson dataset<span class="citation"><sup>1</sup></span>. In this dataset a cell is
described as having a cell type (i.e. ‘Interneuron’) and subcell type
(i.e. ‘Int11’ a.k.a Interneuron type 11).</p>
<p>Here we provide an example of converting a scRNA-seq dataset (<a href="https://www.science.org/doi/10.1126/science.aaa1934" class="external-link">Zeisel
2015</a>) into a CellTypeDataset (CTD) so it can used with
<em>EWCE</em>.</p>
</div>
<div class="section level3">
<h3 id="loading-datasets">Loading datasets<a class="anchor" aria-label="anchor" href="#loading-datasets"></a>
</h3>
<p>The first step for all analyses is to load the single-cell
transcriptome (SCT) data. For the purposes of this example we will use
the dataset described in: &gt; <a href="https://www.science.org/doi/10.1126/science.aaa1934" class="external-link">Cell types in
the mouse cortex and hippocampus revealed by single-cell RNA-seq</a>,
Science, 2015</p>
<p>This is stored and can be loaded as follows from
<code>ewceData</code> package.</p>
<p><em>EWCE</em> can generate CTD from two different data input types:
1. A gene x cell expression matrix (can be dense matrix, sparse matrix,
data.frame, or DelayedArray).<br>
2. A <code>SingleCellExperiment</code> (SCE) or
<code>SummarizedExperiment</code> object.</p>
<p>To check the data, we can quickly plot the distribution of expression
of a given gene across all the cell types.</p>
<div class="sourceCode" id="cb412"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># example datasets held in ewceData package</span></span>
<span><span class="co">#if running offline pass localhub = TRUE</span></span>
<span><span class="va">cortex_mrna</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/cortex_mrna.html" class="external-link">cortex_mrna</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb415"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="st">"Necab1"</span></span>
<span><span class="va">cellExpDist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Expression<span class="op">=</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">[</span><span class="va">gene</span>,<span class="op">]</span>,</span>
<span>                          Celltype<span class="op">=</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span><span class="op">[</span></span>
<span>                            <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span>,<span class="op">]</span><span class="op">$</span><span class="va">level1class</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">cellExpDist</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Celltype</span>, y<span class="op">=</span><span class="va">Expression</span><span class="op">)</span>, outlier.alpha <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p>To note for the analysis of the <code>cortex_mrna</code> dataset, I
will proved the alternative for an SCE object under each segment of code
if there is any difference.</p>
</div>
<div class="section level3">
<h3 id="convert-single-cell-formats">Convert single-cell formats<a class="anchor" aria-label="anchor" href="#convert-single-cell-formats"></a>
</h3>
<p>If you would like to try running the functions on an SCE object
instead, we can convert <code>cortex_mrna</code> with the package <a href="https://github.com/neurogenomics/scKirby" class="external-link"><code>scKirby</code></a>.
<code>scKirby</code> automatically infers the input file type and
converts it to SCE format by default.</p>
<div class="sourceCode" id="cb416"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"SingleCellExperiment"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"SingleCellExperiment"</span><span class="op">)</span><span class="op">}</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span> </span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"scKirby"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"bschilder/scKirby"</span><span class="op">)</span><span class="op">}</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scKirby</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Make SCE object</span></span>
<span><span class="va">cortex_mrna</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/cortex_mrna.html" class="external-link">cortex_mrna</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">cortex_mrna_sce</span> <span class="op">&lt;-</span> <span class="fu">scKirby</span><span class="fu">::</span><span class="fu">ingest_data</span><span class="op">(</span><span class="va">cortex_mrna</span>, save_output <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Now to plot the SCE dataset:</span></span>
<span><span class="va">gene</span><span class="op">=</span><span class="st">"Necab1"</span></span>
<span><span class="va">cellExpDist_sce</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>e<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">gene</span>,<span class="op">]</span>,</span>
<span>                             l1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span><span class="op">[</span></span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                             <span class="op">]</span><span class="op">$</span><span class="va">level1class</span><span class="op">)</span></span>
<span>                             </span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">cellExpDist_sce</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">l1</span>,y<span class="op">=</span><span class="va">e</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Cell type"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Unique Molecule Count"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
</div>
<div class="section level3">
<h3 id="correct-gene-symbols">Correct gene symbols<a class="anchor" aria-label="anchor" href="#correct-gene-symbols"></a>
</h3>
<p>It is very common for publicly available transcriptome datasets to
use incorrect gene symbols (often some gene names will have been mangled
by opening in Excel).</p>
<p>A function is provided to correct these where out of date aliases
were used and give a warning if appears possible that excel has mangled
some of the gene names. This function can be used with a downloaded
reference file like the <a href="http://www.informatics.jax.org/downloads/reports/MRK_List2.rpt" class="external-link">MRK_List2
file</a> from MGI which lists known synonyms for MGI gene symbols. If no
file path is passed, there is a loaded reference dataset of the
MRK_List2 which will be used: <code><a href="https://rdrr.io/pkg/ewceData/man/mgi_synonym_data.html" class="external-link">ewceData::mgi_synonym_data()</a></code>.
We recommend running this function on all input datasets.</p>
<div class="sourceCode" id="cb417"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#if running offline pass localhub = TRUE</span></span>
<span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span> <span class="op">=</span> <span class="fu"><a href="../reference/fix_bad_mgi_symbols.html">fix_bad_mgi_symbols</a></span><span class="op">(</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span></span></code></pre></div>
<p><strong>Note</strong> - You can run
<code><a href="../reference/fix_bad_mgi_symbols.html">fix_bad_mgi_symbols()</a></code> offline by passing
<code>localhub = TRUE</code>. This will work off a previously cached
version of the reference dataset from <code>ExperimentHub</code>.</p>
<p>For the SCE version, pass the whole SCE object:</p>
<div class="sourceCode" id="cb418"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Note that data in SCE object must be in 'counts' assay</span></span>
<span><span class="co">#The fix_bad_hgnc_symbols function can similarly take an SCE object as an input</span></span>
<span><span class="va">cortex_mrna_sce</span> <span class="op">=</span> <span class="fu"><a href="../reference/fix_bad_mgi_symbols.html">fix_bad_mgi_symbols</a></span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span></span></code></pre></div>
<p><strong>Note</strong> - You can run
<code><a href="../reference/ewce_expression_data.html">ewce_expression_data()</a></code> offline by passing
<code>localhub = TRUE</code>. This will work off a previously cached
version of the reference dataset from <code>ExperimentHub</code>.</p>
</div>
<div class="section level3">
<h3 id="drop-genes">Drop genes<a class="anchor" aria-label="anchor" href="#drop-genes"></a>
</h3>
<p>The vignette takes a while to go through if you use all genes. So
let’s restrict to 1000 random genes.</p>
<div class="sourceCode" id="cb419"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nKeep</span> <span class="op">&lt;-</span> <span class="fl">1000</span></span>
<span><span class="va">must_keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span> </span>
<span><span class="va">keep_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">must_keep</span>,<span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span>,<span class="fl">997</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span> <span class="op">&lt;-</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">[</span><span class="va">keep_genes</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1000 3005</span></span></code></pre>
<div class="sourceCode" id="cb421"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#### SCE ####</span></span>
<span><span class="va">nKeep</span> <span class="op">=</span> <span class="fl">1000</span></span>
<span><span class="va">must_keep</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span> </span>
<span><span class="va">keep_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">must_keep</span>,<span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span><span class="op">)</span>,<span class="fl">997</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">cortex_mrna_sce</span> <span class="op">=</span> <span class="va">cortex_mrna_sce</span><span class="op">[</span><span class="va">keep_genes</span>,<span class="op">]</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="normalization-optional">Normalization [Optional]<a class="anchor" aria-label="anchor" href="#normalization-optional"></a>
</h3>
<p>A different number of reads is found across each cell. We suggest
using <a href="https://github.com/satijalab/sctransform" class="external-link"><code>scTransform</code></a>
to normalise for differences due to cell size, then linearly scale. Note
that this might be slow.</p>
<div class="sourceCode" id="cb422"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp_scT_normed</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/sct_normalize.html">sct_normalize</a></span><span class="op">(</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span> </span></code></pre></div>
<div class="section level4">
<h4 id="singlecellexperiment">SingleCellExperiment<a class="anchor" aria-label="anchor" href="#singlecellexperiment"></a>
</h4>
<div class="sourceCode" id="cb423"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#### SCE ####</span></span>
<span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span><span class="op">$</span><span class="va">normcounts</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/sct_normalize.html">sct_normalize</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">cortex_mrna_sce</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span></span></code></pre></div>
<p>Note that this is optional (and was not used in the original EWCE
publication) so by all means ignore this <code>scTransform</code>
step.</p>
</div>
</div>
<div class="section level3">
<h3 id="calculate-specificity-matrices">Calculate specificity matrices<a class="anchor" aria-label="anchor" href="#calculate-specificity-matrices"></a>
</h3>
<p>Next,<code>drop_uninformative_genes</code> drops uninformative genes
in order to reduce compute time and noise in subsequent steps. It
achieves this through several steps, each of which are optional: 1.
<strong>Drop non-1:1 orthologs</strong>: Removes genes that don’t have
1:1 orthologs with the output_species (“human” by default). 2.
<strong>Drop non-varying genes</strong>: Removes genes that don’t vary
across cells based on variance deciles. 3. <strong>Drop
non-differentially expressed genes (DEGs)</strong>: Removes genes that
are not significantly differentially expressed across cell-types.
Multiple DEG methods are currently available (see
<code><a href="../reference/drop_uninformative_genes.html">?drop_uninformative_genes</a></code> for up-to-date info).</p>
<div class="section level4">
<h4 id="parallelisation-1">Parallelisation<a class="anchor" aria-label="anchor" href="#parallelisation-1"></a>
</h4>
<p>You can now speed up the DGE process by parallelising across multiple
cores with the parameter <code>no_cores</code> (<code>=1</code> by
default).</p>
<div class="sourceCode" id="cb424"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Generate cell type data for just the cortex/hippocampus data  </span></span>
<span><span class="va">exp_CortexOnly_DROPPED</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/drop_uninformative_genes.html">drop_uninformative_genes</a></span><span class="op">(</span></span>
<span>  exp <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span>, </span>
<span>  input_species <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>  output_species <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  level2annot <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">## Check 1000Check 3005</span></span></code></pre>
<pre><code><span><span class="co">## Warning: sctSpecies_origin not provided. Setting to 'mouse' by default.</span></span></code></pre>
<pre><code><span><span class="co">## 1 core(s) assigned as workers (1 reserved).</span></span></code></pre>
<pre><code><span><span class="co">## Converting to sparse matrix.</span></span></code></pre>
<pre><code><span><span class="co">## Checking for non-expressed genes.</span></span></code></pre>
<pre><code><span><span class="co">## Checking for cells with no expressed genes.</span></span></code></pre>
<pre><code><span><span class="co">## DGE:: Limma...</span></span></code></pre>
<pre><code><span><span class="co">## 219 / 1,000 genes dropped @ DGE adj_pval_thresh &lt; 1e-05</span></span></code></pre>
<pre><code><span><span class="co">## Time difference of 0.4254897 secs</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="generate-celltypedataset">Generate CellTypeDataset<a class="anchor" aria-label="anchor" href="#generate-celltypedataset"></a>
</h3>
<p>Now we generate the CellTypeDataset with the</p>
<p>By default the function returns the path where the resulting file was
saved. If you want to return the CTD itself as well, set
<code>return_ctd=TRUE</code> and it will be returned in a list (along
with the save path).</p>
<div class="section level4">
<h4 id="parallelisation-2">Parallelisation<a class="anchor" aria-label="anchor" href="#parallelisation-2"></a>
</h4>
<p>You can now speed up <code>generate_celltype_data</code> by
parallelising across multiple cores with the parameter
<code>no_cores</code> (<code>=1</code> by default).</p>
<div class="sourceCode" id="cb434"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotLevels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>level1class<span class="op">=</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level1class</span>,</span>
<span>                    level2class<span class="op">=</span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fNames_CortexOnly</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/generate_celltype_data.html">generate_celltype_data</a></span><span class="op">(</span></span>
<span>  exp <span class="op">=</span> <span class="va">exp_CortexOnly_DROPPED</span>,</span>
<span>  annotLevels <span class="op">=</span> <span class="va">annotLevels</span>,</span>
<span>  groupName <span class="op">=</span> <span class="st">"kiCortexOnly"</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">## 1 core(s) assigned as workers (1 reserved).</span></span></code></pre>
<pre><code><span><span class="co">## Converting to sparse matrix.</span></span></code></pre>
<pre><code><span><span class="co">## + Calculating normalized mean expression.</span></span></code></pre>
<pre><code><span><span class="co">## Converting to sparse matrix.</span></span>
<span><span class="co">## Converting to sparse matrix.</span></span></code></pre>
<pre><code><span><span class="co">## + Calculating normalized specificity.</span></span></code></pre>
<pre><code><span><span class="co">## Converting to sparse matrix.</span></span>
<span><span class="co">## Converting to sparse matrix.</span></span>
<span><span class="co">## Converting to sparse matrix.</span></span>
<span><span class="co">## Converting to sparse matrix.</span></span></code></pre>
<pre><code><span><span class="co">## Loading required namespace: ggdendro</span></span></code></pre>
<pre><code><span><span class="co">## + Saving results ==&gt;  /tmp/RtmpYkIP22/ctd_kiCortexOnly.rda</span></span></code></pre>
<div class="sourceCode" id="cb443"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ctd_CortexOnly</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/load_rdata.html">load_rdata</a></span><span class="op">(</span><span class="va">fNames_CortexOnly</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="singlecellexperiment-1">SingleCellExperiment<a class="anchor" aria-label="anchor" href="#singlecellexperiment-1"></a>
</h4>
<div class="sourceCode" id="cb444"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#### SCE ####</span></span>
<span><span class="co"># Generate cell type data for just the cortex/hippocampus data</span></span>
<span><span class="va">exp_CortexOnly_DROPPED</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/drop_uninformative_genes.html">drop_uninformative_genes</a></span><span class="op">(</span>exp<span class="op">=</span><span class="va">cortex_mrna_sce</span>,</span>
<span>                                                   drop_nonhuman_genes <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                                                   input_species <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>                                                   level2annot<span class="op">=</span><span class="va">cortex_mrna_sce</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span></span>
<span></span>
<span><span class="va">annotLevels</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>level1class<span class="op">=</span><span class="va">cortex_mrna_sce</span><span class="op">$</span><span class="va">level1class</span>,</span>
<span>                   level2class<span class="op">=</span><span class="va">cortex_mrna_sce</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fNames_CortexOnly</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_celltype_data.html">generate_celltype_data</a></span><span class="op">(</span>exp<span class="op">=</span><span class="va">exp_CortexOnly_DROPPED</span>,</span>
<span>                                           annotLevels<span class="op">=</span><span class="va">annotLevels</span>,</span>
<span>                                           groupName<span class="op">=</span><span class="st">"kiCortexOnly"</span>,</span>
<span>                                           savePath<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
<p>To note on the SCE approach, these are the only differences for
handling the analysis of an SCE object. All downstream analysis on
<code>cortex_mrna</code> would be the same. The same approach can be
used for other datasets in the vignette if converted such as
<code>hypothalamus_mrna</code> or your own SCE object.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="merge-two-single-cell-datasets">Merge two single-cell datasets<a class="anchor" aria-label="anchor" href="#merge-two-single-cell-datasets"></a>
</h2>
<p>Often it is useful to merge two single-cell datasets. For instance,
there are separate files for the cortex and hypothalamus datasets
generated by the Karolinska institute. A subset of the resulting merged
dataset is loaded using <code>ctd()</code> but it is worth understanding
the approach (outlined below) if you want to repeat this for other
single-cell datasets.</p>
<p>The dataset is available from the <a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE74nnn/GSE74672/suppl/GSE74672_expressed_mols_with_classes.xlsx.gz">GEO</a>
but is also available in the <code>ewceData</code> package:
<code>hypothalamus_mrna()</code>. See the <code>ewceData</code> for
details of how this file was preprocessed after first being downloaded
from GEO, unzipped and read into R.</p>
<p>We can now merge the hypothalamus dataset with the cortex dataset and
then calculate specificity:</p>
<div class="section level3">
<h3 id="load-hypothalamus-dataset">Load hypothalamus dataset<a class="anchor" aria-label="anchor" href="#load-hypothalamus-dataset"></a>
</h3>
<div class="sourceCode" id="cb445"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#if running offline pass localhub = TRUE</span></span>
<span><span class="va">hypothalamus_mrna</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/hypothalamus_mrna.html" class="external-link">hypothalamus_mrna</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="fix-bad-mgi-symbols">Fix bad MGI symbols<a class="anchor" aria-label="anchor" href="#fix-bad-mgi-symbols"></a>
</h3>
<div class="sourceCode" id="cb448"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#if running offline pass localhub = TRUE</span></span>
<span><span class="va">hypothalamus_mrna</span><span class="op">$</span><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/fix_bad_mgi_symbols.html">fix_bad_mgi_symbols</a></span><span class="op">(</span>exp <span class="op">=</span> <span class="va">hypothalamus_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## 1808 rows do not have proper MGI symbols</span></span></code></pre>
<pre><code><span><span class="co">## Rims1_loc1, Rims1_loc2, Fam178b_loc2, Fam178b_loc1, 2900092D14Rik, 4930487H11Rik, Spag16_loc1, Spag16_loc2, Pnkd_loc2, Pnkd_loc1, Ccdc108, 1700019O17Rik, Myeov2, Ren2, Gpr52, 1700015E13Rik, Olfr426, Hmga2-ps1, B020014A21Rik, Gm5512_loc1</span></span></code></pre>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## 0 poorly annotated genes are replicates of existing genes. These are:</span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## Converting to sparse matrix.</span></span></code></pre>
<pre><code><span><span class="co">## 0 rows should have been corrected by checking synonyms.</span></span></code></pre>
<pre><code><span><span class="co">## 1808 rows STILL do not have proper MGI symbols.</span></span></code></pre>
<div class="sourceCode" id="cb460"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/fix_bad_mgi_symbols.html">fix_bad_mgi_symbols</a></span><span class="op">(</span>exp <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## 89 rows do not have proper MGI symbols</span></span></code></pre>
<pre><code><span><span class="co">## Srp54b_loc1, Gpr124, 4930528A17Rik, Dpcd, Snora19_loc2, Ccdc104, A630089N07Rik, 4930404H11Rik, Gm17746, Duxbl2_loc1, Pnkd_loc1, 4930542D17Rik, Vmn2r47_loc4, Cml1, Gyltl1b, Prune, C920025E04Rik, 4931429I11Rik, Gm10697_loc1, 5430417L22Rik</span></span></code></pre>
<pre><code><span><span class="co">## Warning in EWCE::fix_bad_mgi_symbols(exp = cortex_mrna$exp): Possible presence</span></span>
<span><span class="co">## of excel corrupted date-like genes: Sepp1</span></span></code></pre>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## 0 poorly annotated genes are replicates of existing genes. These are:</span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## Converting to sparse matrix.</span></span></code></pre>
<pre><code><span><span class="co">## 41 rows should have been corrected by checking synonyms.</span></span></code></pre>
<pre><code><span><span class="co">## 48 rows STILL do not have proper MGI symbols.</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="merge-ctd">Merge CTD<a class="anchor" aria-label="anchor" href="#merge-ctd"></a>
</h3>
<p>Merge the CTDs - again this can be run with SCE or other ranged SE
objects as input.</p>
<div class="sourceCode" id="cb473"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">merged_KI</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/merge_two_expfiles.html">merge_two_expfiles</a></span><span class="op">(</span>exp1 <span class="op">=</span> <span class="va">hypothalamus_mrna</span><span class="op">$</span><span class="va">exp</span>, </span>
<span>                                      exp2 <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">exp</span>,</span>
<span>                                      annot1 <span class="op">=</span> <span class="va">hypothalamus_mrna</span><span class="op">$</span><span class="va">annot</span>, </span>
<span>                                      annot2 <span class="op">=</span> <span class="va">cortex_mrna</span><span class="op">$</span><span class="va">annot</span>,</span>
<span>                                      name1 <span class="op">=</span> <span class="st">"Hypothalamus (KI)"</span>, </span>
<span>                                      name2 <span class="op">=</span> <span class="st">"Cortex/Hippo (KI)"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 23,317 non-overlapping gene(s) removed from exp1.</span></span></code></pre>
<pre><code><span><span class="co">## 0 non-overlapping gene(s) removed from exp2.</span></span></code></pre>
<pre><code><span><span class="co">## 1,000 intersecting genes remain.</span></span></code></pre>
<pre><code><span><span class="co">## Converting to data.frame</span></span>
<span><span class="co">## Converting to data.frame</span></span></code></pre>
<pre><code><span><span class="co">## Converting to sparse matrix.</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="drop-uninformative-genes">Drop uninformative genes<a class="anchor" aria-label="anchor" href="#drop-uninformative-genes"></a>
</h3>
<p>Drop genes which don’t vary significantly between cell types.</p>
<div class="sourceCode" id="cb479"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp_merged_DROPPED</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/drop_uninformative_genes.html">drop_uninformative_genes</a></span><span class="op">(</span></span>
<span>  exp <span class="op">=</span> <span class="va">merged_KI</span><span class="op">$</span><span class="va">exp</span>, </span>
<span>  drop_nonhuman_genes <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>  input_species <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>  level2annot <span class="op">=</span> <span class="va">merged_KI</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Check 1000Check 3777</span></span></code></pre>
<pre><code><span><span class="co">## Warning: sctSpecies_origin not provided. Setting to 'mouse' by default.</span></span></code></pre>
<pre><code><span><span class="co">## 1 core(s) assigned as workers (1 reserved).</span></span></code></pre>
<pre><code><span><span class="co">## Converting to sparse matrix.</span></span></code></pre>
<pre><code><span><span class="co">## Checking for non-expressed genes.</span></span></code></pre>
<pre><code><span><span class="co">## Checking for cells with no expressed genes.</span></span></code></pre>
<pre><code><span><span class="co">## DGE:: Limma...</span></span></code></pre>
<pre><code><span><span class="co">## 1 / 1,000 genes dropped @ DGE adj_pval_thresh &lt; 1e-05</span></span></code></pre>
<pre><code><span><span class="co">## Time difference of 0.9404855 secs</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="generate-celltypedataset-1">Generate CellTypeDataset<a class="anchor" aria-label="anchor" href="#generate-celltypedataset-1"></a>
</h3>
<div class="sourceCode" id="cb489"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotLevels</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>level1class<span class="op">=</span><span class="va">merged_KI</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level1class</span>,</span>
<span>                   level2class<span class="op">=</span><span class="va">merged_KI</span><span class="op">$</span><span class="va">annot</span><span class="op">$</span><span class="va">level2class</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Update file path to where you want the cell type data files to save on disk</span></span>
<span><span class="va">fNames_MergedKI</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/generate_celltype_data.html">generate_celltype_data</a></span><span class="op">(</span>exp <span class="op">=</span> <span class="va">exp_merged_DROPPED</span>,</span>
<span>                                                annotLevels <span class="op">=</span> <span class="va">annotLevels</span>,</span>
<span>                                                groupName <span class="op">=</span> <span class="st">"MergedKI"</span>,</span>
<span>                                                savePath <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">## 1 core(s) assigned as workers (1 reserved).</span></span></code></pre>
<pre><code><span><span class="co">## Converting to sparse matrix.</span></span></code></pre>
<pre><code><span><span class="co">## + Calculating normalized mean expression.</span></span></code></pre>
<pre><code><span><span class="co">## Converting to sparse matrix.</span></span>
<span><span class="co">## Converting to sparse matrix.</span></span></code></pre>
<pre><code><span><span class="co">## + Calculating normalized specificity.</span></span></code></pre>
<pre><code><span><span class="co">## Converting to sparse matrix.</span></span>
<span><span class="co">## Converting to sparse matrix.</span></span>
<span><span class="co">## Converting to sparse matrix.</span></span>
<span><span class="co">## Converting to sparse matrix.</span></span></code></pre>
<pre><code><span><span class="co">## + Saving results ==&gt;  /tmp/RtmpYkIP22/ctd_MergedKI.rda</span></span></code></pre>
<div class="sourceCode" id="cb497"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ctd_MergedKI</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/load_rdata.html">load_rdata</a></span><span class="op">(</span><span class="va">fNames_MergedKI</span><span class="op">)</span> </span></code></pre></div>
</div>
<div class="section level3">
<h3 id="understanding-specificity-matrices">Understanding specificity matrices<a class="anchor" aria-label="anchor" href="#understanding-specificity-matrices"></a>
</h3>
<p>While not required for further analyses it helps to understand what
the outputs of this function are.</p>
<p>Note firstly, that it is a list such that <code>ctd[[1]]</code>
contains data relating to level 1 annotations and <code>ctd[[2]]</code>
relates to level 2 annotations.</p>
<p>Using the <code>ggplot2</code> package to visualise the data, let us
examine the expression of a few genes. If you have not already done so
you will need to first install the ggplot2 package with
<code>install.packages("ggplot2")</code>.</p>
<p>For this example we use a subset of the genes from the merged dataset
generated above, which is accessed using <code><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ewceData::ctd()</a></code>.
We recommend that you use the code above to regenerate this though and
drop the first command from the below section.</p>
<div class="sourceCode" id="cb498"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## You can also import the pre-merged cortex and hypothalamus dataset</span></span>
<span><span class="co"># generated by Karolinska institute. `ewceData::ctd()` is the same file as </span></span>
<span><span class="co"># ctd_MergedKI above.</span></span>
<span><span class="co">#if running offline pass localhub = TRUE</span></span>
<span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
<div class="sourceCode" id="cb499"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="va">ctd_MergedKI</span> </span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_ctd.html">plot_ctd</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>                      level <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                      genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,</span>
<span>                      metric <span class="op">=</span> <span class="st">"mean_exp"</span><span class="op">)</span> </span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-25-1.png" width="672"></p>
<p>This graph shows the average expression of three genes: <em>Apoe,
Gfap</em> and <em>Gapdh</em>. While there are substantial differences in
which cell types express these genes, the dominant effect seen here is
the overall expression level of the data. For the purposes of this
analysis though, we are not interested in overall expression level and
only wish to know about the proportion of a genes expression which is
found in a particular cell type. We can study this instead using the
following code which examines the data frame
<code>ctd[[1]]$specificity</code>:</p>
<div class="sourceCode" id="cb500"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_ctd.html">plot_ctd</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>                      level <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                      genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,</span>
<span>                      metric <span class="op">=</span> <span class="st">"specificity"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-26-1.png" width="672"></p>
<p>We can now see in this graph that <em>Gfap</em> is the most specific
to a cell type (Type 1 Astrocytes) of any of those three genes, with
over 60% of its expression found in that cell type.</p>
<p>It can also be seen that the majority of expression of Gapdh is in
neurons but because their are a greater number of neuronal subtypes, the
total expression proportion appears lower. We can examine expression
across level 2 cell type level annotations by looking at
<code>ctd[[2]]$specificity</code>:</p>
<div class="sourceCode" id="cb501"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_ctd.html">plot_ctd</a></span><span class="op">(</span>ctd <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>                      level <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                      genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Apoe"</span>,<span class="st">"Gfap"</span>,<span class="st">"Gapdh"</span><span class="op">)</span>,</span>
<span>                      metric <span class="op">=</span> <span class="st">"specificity"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-27-1.png" width="672"></p>
</div>
</div>
<div class="section level2">
<h2 id="run-conditional-cell-type-enrichment-tests">Run conditional cell-type enrichment tests<a class="anchor" aria-label="anchor" href="#run-conditional-cell-type-enrichment-tests"></a>
</h2>
<div class="section level3">
<h3 id="prepare-data">Prepare data<a class="anchor" aria-label="anchor" href="#prepare-data"></a>
</h3>
<div class="sourceCode" id="cb502"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#if running offline pass localhub = TRUE</span></span>
<span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb505"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/example_genelist.html" class="external-link">example_genelist</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span>
<span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb507"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## NOTE: rep=100 for demo purposes only. </span></span>
<span><span class="co">## Use &gt;=10,000 for publication-quality results.</span></span>
<span><span class="va">reps</span> <span class="op">&lt;-</span> <span class="fl">100</span> </span>
<span><span class="va">annotLevel</span> <span class="op">&lt;-</span> <span class="fl">1</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="controlling-for-expression-in-another-cell-type">Controlling for expression in another cell type<a class="anchor" aria-label="anchor" href="#controlling-for-expression-in-another-cell-type"></a>
</h3>
<p>In a followup paper we found that an enrichment detected for
Schizophrenia in Somatosensory Pyramidal neurons could be explained by
accounting for expression in Hippocampal CA1 pyramidal neurons. These
results are described here:</p>
<p><a href="https://www.nature.com/articles/s41588-018-0129-5" class="external-link">Skene, et
al. Genetic identification of brain cell types underlying
schizophrenia.Nature Genetics, 2018.</a></p>
<p>Those results were generated using an alternative enrichment method
designed for use with GWAS Summary Statistics rather than gene sets. The
same sort of approach can be extended to EWCE as well, and we have
implemented it within this package. When testing for enrichment the
other gene sets that are sampled are selected to have equivalent
specificity in the controlled cell type.</p>
<p>We demonstrate it’s use below to test whether the enrichment in
astrocytes is still present after controlling for the enrichment within
microglia.</p>
<div class="section level4">
<h4 id="parallelisation-3">Parallelisation<a class="anchor" aria-label="anchor" href="#parallelisation-3"></a>
</h4>
<p>You can now speed up the bootstrapping process by parallelising
across multiple cores with the parameter <code>no_cores</code>
(<code>=1</code> by default).</p>
<div class="sourceCode" id="cb508"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#if running offline pass localhub = TRUE</span></span>
<span><span class="va">unconditional_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span></span>
<span>  sct_data <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>  hits <span class="op">=</span> <span class="va">hits</span>,</span>
<span>  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>  genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  reps <span class="op">=</span> <span class="va">reps</span>,</span>
<span>  annotLevel <span class="op">=</span> <span class="va">annotLevel</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1 core(s) assigned as workers (1 reserved).</span></span></code></pre>
<pre><code><span><span class="co">## Generating gene background for mouse x human ==&gt; human</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 21,207 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 21,207 genes from mouse.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span>
<span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## Preparing gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## data.frame format detected.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from Gene.Symbol.</span></span></code></pre>
<pre><code><span><span class="co">## 21,207 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Converting mouse ==&gt; human orthologs using: homologene</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without orthologs in human.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from input_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from ortholog_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without 1:1 orthologs.</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene (many:1).</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene (1:many).</span></span></code></pre>
<pre><code><span><span class="co">## Filtering gene_df with gene_map</span></span></code></pre>
<pre><code><span><span class="co">## Adding input_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## Adding ortholog_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## Total genes dropped after convert_orthologs :</span></span>
<span><span class="co">##    4,725 / 21,207 (22%)</span></span></code></pre>
<pre><code><span><span class="co">## Total genes remaining after convert_orthologs :</span></span>
<span><span class="co">##    16,482 / 21,207 (78%)</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 intersect background genes used.</span></span></code></pre>
<pre><code><span><span class="co">## Standardising CellTypeDataset</span></span></code></pre>
<pre><code><span><span class="co">## Checking gene list inputs.</span></span></code></pre>
<pre><code><span><span class="co">## Running without gene size control.</span></span></code></pre>
<pre><code><span><span class="co">## 17 hit gene(s) remain after filtering.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene scores.</span></span></code></pre>
<pre><code><span><span class="co">## Using previously sampled genes.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene counts.</span></span></code></pre>
<pre><code><span><span class="co">## Testing for enrichment in 7 cell types...</span></span></code></pre>
<pre><code><span><span class="co">## Sorting results by p-value.</span></span></code></pre>
<pre><code><span><span class="co">## Computing BH-corrected q-values.</span></span></code></pre>
<pre><code><span><span class="co">## 1 significant cell type enrichment results @ q&lt;0.05 :</span></span></code></pre>
<pre><code><span><span class="co">##    CellType annotLevel p fold_change sd_from_mean q</span></span>
<span><span class="co">## 1 microglia          1 0    2.065383     4.074891 0</span></span></code></pre>
<div class="sourceCode" id="cb590"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">conditional_results_micro</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span> <span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span></span>
<span>  sct_data <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>  hits <span class="op">=</span> <span class="va">hits</span>,</span>
<span>  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>  genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  reps <span class="op">=</span> <span class="va">reps</span>,</span>
<span>  annotLevel <span class="op">=</span> <span class="va">annotLevel</span>,</span>
<span>  controlledCT <span class="op">=</span> <span class="st">"microglia"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1 core(s) assigned as workers (1 reserved).</span></span></code></pre>
<pre><code><span><span class="co">## Generating gene background for mouse x human ==&gt; human</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 21,207 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 21,207 genes from mouse.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span>
<span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## Preparing gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## data.frame format detected.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from Gene.Symbol.</span></span></code></pre>
<pre><code><span><span class="co">## 21,207 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Converting mouse ==&gt; human orthologs using: homologene</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without orthologs in human.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from input_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from ortholog_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without 1:1 orthologs.</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene (many:1).</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene (1:many).</span></span></code></pre>
<pre><code><span><span class="co">## Filtering gene_df with gene_map</span></span></code></pre>
<pre><code><span><span class="co">## Adding input_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## Adding ortholog_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## Total genes dropped after convert_orthologs :</span></span>
<span><span class="co">##    4,725 / 21,207 (22%)</span></span></code></pre>
<pre><code><span><span class="co">## Total genes remaining after convert_orthologs :</span></span>
<span><span class="co">##    16,482 / 21,207 (78%)</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 intersect background genes used.</span></span></code></pre>
<pre><code><span><span class="co">## Standardising CellTypeDataset</span></span></code></pre>
<pre><code><span><span class="co">## Checking gene list inputs.</span></span></code></pre>
<pre><code><span><span class="co">## Running without gene size control.</span></span></code></pre>
<pre><code><span><span class="co">## 17 hit gene(s) remain after filtering.</span></span></code></pre>
<pre><code><span><span class="co">## Generating controlled bootstrap gene sets.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene scores.</span></span></code></pre>
<pre><code><span><span class="co">## Using previously sampled genes.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene counts.</span></span></code></pre>
<pre><code><span><span class="co">## Testing for enrichment in 7 cell types...</span></span></code></pre>
<pre><code><span><span class="co">## Sorting results by p-value.</span></span></code></pre>
<pre><code><span><span class="co">## Computing BH-corrected q-values.</span></span></code></pre>
<pre><code><span><span class="co">## 1 significant cell type enrichment results @ q&lt;0.05 :</span></span></code></pre>
<pre><code><span><span class="co">##               CellType annotLevel p fold_change sd_from_mean q</span></span>
<span><span class="co">## 1 astrocytes_ependymal          1 0    1.546605     2.355024 0</span></span></code></pre>
<div class="sourceCode" id="cb673"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">conditional_results_astro</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/bootstrap_enrichment_test.html">bootstrap_enrichment_test</a></span><span class="op">(</span></span>
<span>  sct_data <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>  hits <span class="op">=</span> <span class="va">hits</span>,</span>
<span>  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>  genelistSpecies <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  reps <span class="op">=</span> <span class="va">reps</span>,</span>
<span>  annotLevel <span class="op">=</span> <span class="va">annotLevel</span>,</span>
<span>  controlledCT <span class="op">=</span> <span class="st">"astrocytes_ependymal"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1 core(s) assigned as workers (1 reserved).</span></span></code></pre>
<pre><code><span><span class="co">## Generating gene background for mouse x human ==&gt; human</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 21,207 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 21,207 genes from mouse.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span>
<span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## Preparing gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## data.frame format detected.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from Gene.Symbol.</span></span></code></pre>
<pre><code><span><span class="co">## 21,207 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Converting mouse ==&gt; human orthologs using: homologene</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without orthologs in human.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from input_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from ortholog_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without 1:1 orthologs.</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene (many:1).</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene (1:many).</span></span></code></pre>
<pre><code><span><span class="co">## Filtering gene_df with gene_map</span></span></code></pre>
<pre><code><span><span class="co">## Adding input_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## Adding ortholog_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## Total genes dropped after convert_orthologs :</span></span>
<span><span class="co">##    4,725 / 21,207 (22%)</span></span></code></pre>
<pre><code><span><span class="co">## Total genes remaining after convert_orthologs :</span></span>
<span><span class="co">##    16,482 / 21,207 (78%)</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 intersect background genes used.</span></span></code></pre>
<pre><code><span><span class="co">## Standardising CellTypeDataset</span></span></code></pre>
<pre><code><span><span class="co">## Checking gene list inputs.</span></span></code></pre>
<pre><code><span><span class="co">## Running without gene size control.</span></span></code></pre>
<pre><code><span><span class="co">## 17 hit gene(s) remain after filtering.</span></span></code></pre>
<pre><code><span><span class="co">## Generating controlled bootstrap gene sets.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene scores.</span></span></code></pre>
<pre><code><span><span class="co">## Using previously sampled genes.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene counts.</span></span></code></pre>
<pre><code><span><span class="co">## Testing for enrichment in 7 cell types...</span></span></code></pre>
<pre><code><span><span class="co">## Sorting results by p-value.</span></span></code></pre>
<pre><code><span><span class="co">## Computing BH-corrected q-values.</span></span></code></pre>
<pre><code><span><span class="co">## 0 significant cell type enrichment results @ q&lt;0.05 :</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="merge-and-plot-results">Merge and plot results<a class="anchor" aria-label="anchor" href="#merge-and-plot-results"></a>
</h4>
<div class="sourceCode" id="cb755"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">merged_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">unconditional_results</span><span class="op">$</span><span class="va">results</span>,</span>
<span>             list<span class="op">=</span><span class="st">"Unconditional Enrichment"</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">conditional_results_micro</span><span class="op">$</span><span class="va">results</span>,</span>
<span>             list<span class="op">=</span><span class="st">"Conditional Enrichment (Microglia controlled)"</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">conditional_results_astro</span><span class="op">$</span><span class="va">results</span>,</span>
<span>             list<span class="op">=</span><span class="st">"Conditional Enrichment (Astrocyte controlled)"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="op">{</span></span>
<span> <span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span>total_res <span class="op">=</span> <span class="va">merged_results</span>,</span>
<span>                             mtc_method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span> </span>
<span> <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-30-1.png" width="700"></p>
<p>When controlling for astrocytes the enrichment in astrocytes is
totally abolished as expected, and vica versa. The enrichment in
microglia remains strongly significant however after controlling for
astrocytes., suggesting that this enrichment is independent of that in
astrocytes.</p>
</div>
</div>
<div class="section level3">
<h3 id="gene-set-enrichment-analysis-controlling-for-cell-type-expression">Gene set enrichment analysis controlling for cell type
expression<a class="anchor" aria-label="anchor" href="#gene-set-enrichment-analysis-controlling-for-cell-type-expression"></a>
</h3>
<p>Traditionally the standard analysis run on all gene sets was the GO
enrichment analysis. Once you have established that a given gene list is
enriched for a given cell type, it becomes questionable whether a GO
enrichment is driven purely by the underlying cell type enrichment. For
instance, it is well established that genes associated with
schizophrenia are enriched for the human post-synaptic density genes,
however, it has also been shown that schizophrenia is enriched for
specificity in CA1 pyramidal neurons (which highly express hPSD genes).
These two enrichments can be disassociated using the following
analysis:</p>
<div class="sourceCode" id="cb756"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set seed for bootstrap reproducibility </span></span>
<span></span>
<span><span class="va">mouse_to_human_homologs</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/mouse_to_human_homologs.html" class="external-link">mouse_to_human_homologs</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">m2h</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">mouse_to_human_homologs</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HGNC.symbol"</span>,<span class="st">"MGI.symbol"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">schiz_genes</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/schiz_genes.html" class="external-link">schiz_genes</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">id_genes</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/id_genes.html" class="external-link">id_genes</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">mouse.hits.schiz</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">m2h</span><span class="op">[</span><span class="va">m2h</span><span class="op">$</span><span class="va">HGNC.symbol</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">schiz_genes</span>,<span class="st">"MGI.symbol"</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">mouse.hits.id</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">m2h</span><span class="op">[</span><span class="va">m2h</span><span class="op">$</span><span class="va">HGNC.symbol</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">id_genes</span>,<span class="st">"MGI.symbol"</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">mouse.bg</span>  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">m2h</span><span class="op">$</span><span class="va">MGI.symbol</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hpsd_genes</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/hpsd_genes.html" class="external-link">hpsd_genes</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">mouse.hpsd</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">m2h</span><span class="op">[</span><span class="va">m2h</span><span class="op">$</span><span class="va">HGNC.symbol</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">hpsd_genes</span>,<span class="st">"MGI.symbol"</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">rbfox_genes</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/rbfox_genes.html" class="external-link">rbfox_genes</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res_hpsd_schiz</span> <span class="op">=</span> <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.schiz</span>, </span>
<span>                                                functional_genes <span class="op">=</span> <span class="va">mouse.hpsd</span>, </span>
<span>                                                bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, </span>
<span>                                                sct_data <span class="op">=</span> <span class="va">ctd</span>, </span>
<span>                                                annotLevel <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                                reps<span class="op">=</span><span class="va">reps</span>, </span>
<span>                                                controlledCT<span class="op">=</span><span class="st">"pyramidal CA1"</span><span class="op">)</span></span>
<span><span class="va">res_rbfox_schiz</span> <span class="op">=</span> </span>
<span>  <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.schiz</span>, </span>
<span>                                  functional_genes <span class="op">=</span> <span class="va">rbfox_genes</span>, </span>
<span>                                  bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, </span>
<span>                                  sct_data <span class="op">=</span> <span class="va">ctd</span>, </span>
<span>                                  annotLevel <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                  reps<span class="op">=</span><span class="va">reps</span>, </span>
<span>                                  controlledCT<span class="op">=</span><span class="st">"pyramidal CA1"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_hpsd_schiz</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_rbfox_schiz</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res_hpsd_id</span> <span class="op">=</span> </span>
<span>  <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.id</span>, </span>
<span>                                  functional_genes <span class="op">=</span> <span class="va">mouse.hpsd</span>, </span>
<span>                                  bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, </span>
<span>                                  sct_data <span class="op">=</span> <span class="va">ctd</span>, </span>
<span>                                  annotLevel <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                  reps<span class="op">=</span><span class="va">reps</span>, </span>
<span>                                  controlledCT <span class="op">=</span> <span class="st">"pyramidal SS"</span><span class="op">)</span></span>
<span><span class="va">res_rbfox_id</span> <span class="op">=</span> <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.id</span>, </span>
<span>                                              functional_genes <span class="op">=</span> <span class="va">rbfox_genes</span>, </span>
<span>                                              bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, </span>
<span>                                              sct_data <span class="op">=</span> <span class="va">ctd</span>, </span>
<span>                                              annotLevel <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                              reps<span class="op">=</span><span class="va">reps</span>, </span>
<span>                                              controlledCT<span class="op">=</span><span class="st">"pyramidal SS"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_hpsd_id</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_rbfox_id</span><span class="op">)</span></span></code></pre></div>
<p>The analysis also tests for enrichment of Rbfox binding genes in the
schizophrenia susceptibility genes, as well as both hPSD and Rbfox genes
in Intellectual Disability genes. All of the enrichments are confirmed
as still being present after controlling for the associated cell type,
apart from the enrichment of PSD genes in Schizophrenia which falls from
borderline to non-significant.</p>
</div>
<div class="section level3">
<h3 id="controlling-for-multiple-cell-types">Controlling for multiple cell types<a class="anchor" aria-label="anchor" href="#controlling-for-multiple-cell-types"></a>
</h3>
<div class="sourceCode" id="cb757"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">controlledCTs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pyramidal CA1"</span>,<span class="st">"pyramidal SS"</span>,<span class="st">"interneurons"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res_hpsd_schiz</span> <span class="op">=</span> </span>
<span>  <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.schiz</span>, </span>
<span>                                  functional_genes <span class="op">=</span> <span class="va">mouse.hpsd</span>, </span>
<span>                                  bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, </span>
<span>                                  sct_data <span class="op">=</span> <span class="va">ctd</span>, </span>
<span>                                  annotLevel <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                  reps<span class="op">=</span><span class="va">reps</span>, </span>
<span>                                  controlledCT<span class="op">=</span><span class="va">controlledCTs</span><span class="op">)</span></span>
<span><span class="va">res_rbfox_schiz</span> <span class="op">=</span> </span>
<span>  <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.schiz</span>, </span>
<span>                                  functional_genes <span class="op">=</span> <span class="va">rbfox_genes</span>, </span>
<span>                                  bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, sct_data <span class="op">=</span> <span class="va">ctd</span>, </span>
<span>                                  annotLevel <span class="op">=</span> <span class="fl">1</span>, reps<span class="op">=</span><span class="va">reps</span>, </span>
<span>                                  controlledCT<span class="op">=</span><span class="va">controlledCTs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_hpsd_schiz</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_rbfox_schiz</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res_hpsd_id</span> <span class="op">=</span> </span>
<span>  <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.id</span>, </span>
<span>                                  functional_genes <span class="op">=</span> <span class="va">mouse.hpsd</span>, </span>
<span>                                  bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, </span>
<span>                                  sct_data <span class="op">=</span> <span class="va">ctd</span>, </span>
<span>                                  annotLevel <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                  reps<span class="op">=</span><span class="va">reps</span>, </span>
<span>                                  controlledCT<span class="op">=</span><span class="va">controlledCTs</span><span class="op">)</span></span>
<span><span class="va">res_rbfox_id</span> <span class="op">=</span> <span class="fu"><a href="../reference/controlled_geneset_enrichment.html">controlled_geneset_enrichment</a></span><span class="op">(</span>disease_genes<span class="op">=</span><span class="va">mouse.hits.id</span>, </span>
<span>                                              functional_genes <span class="op">=</span> <span class="va">rbfox_genes</span>, </span>
<span>                                              bg_genes <span class="op">=</span> <span class="va">mouse.bg</span>, </span>
<span>                                              sct_data <span class="op">=</span> <span class="va">ctd</span>, </span>
<span>                                              annotLevel <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                              reps<span class="op">=</span><span class="va">reps</span>, </span>
<span>                                              controlledCT<span class="op">=</span><span class="va">controlledCTs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_hpsd_id</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res_rbfox_id</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="apply-to-transcriptomic-data">Apply to transcriptomic data<a class="anchor" aria-label="anchor" href="#apply-to-transcriptomic-data"></a>
</h2>
<div class="section level3">
<h3 id="prepare-data-1">Prepare data<a class="anchor" aria-label="anchor" href="#prepare-data-1"></a>
</h3>
<div class="sourceCode" id="cb758"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#if running offline pass localhub = TRUE</span></span>
<span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/ctd.html" class="external-link">ctd</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb761"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tt_alzh</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/tt_alzh.html" class="external-link">tt_alzh</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?ewceData and browseVignettes('ewceData') for documentation</span></span>
<span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb763"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## NOTE: rep=100 for demo purposes only. </span></span>
<span><span class="co">## Use &gt;=10,000 for publication-quality results.</span></span>
<span><span class="va">reps</span> <span class="op">&lt;-</span> <span class="fl">100</span> </span></code></pre></div>
</div>
<div class="section level3">
<h3 id="analysing-single-transcriptome-study">Analysing single transcriptome study<a class="anchor" aria-label="anchor" href="#analysing-single-transcriptome-study"></a>
</h3>
<p>For the prior analyses the gene lists were not associated with any
numeric values or directionality. The methodology for extending this
form of analysis to transcriptomic studies simply involves thresholding
the most upregulated and downregulated genes.</p>
<p>To demonstrate this we have an example dataset <code>tt_alzh</code>.
This data frame was generated using <code>limma</code> from a set of
post-mortem tissue samples from Brodmann area 46 which were described in
a paper by the Haroutunian lab<span class="citation"><sup>3</sup></span>.</p>
<p>The first step is to load the data, obtain the MGI ids, sort the rows
by t-statistic and then select the most up/down-regulated genes. The
package then has a function <code>ewce_expression_data</code> which
thresholds and selects the gene sets, and calls the EWCE function.</p>
<p>Below we show the function call using the default settings, but if
desired different threshold values can be used, or alternative columns
used to sort the table.</p>
<div class="sourceCode" id="cb764"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ewce_expression_data calls bootstrap_enrichment_test so  </span></span>
<span><span class="va">tt_results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_expression_data.html">ewce_expression_data</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>                                          tt <span class="op">=</span> <span class="va">tt_alzh</span>,</span>
<span>                                          annotLevel <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                          ttSpecies <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>                                          sctSpecies <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: genelistSpecies not provided. Setting to 'human' by default.</span></span></code></pre>
<pre><code><span><span class="co">## Warning: sctSpecies_origin not provided. Setting to 'mouse' by default.</span></span>
<span><span class="co">## Warning: sctSpecies_origin not provided. Setting to 'mouse' by default.</span></span></code></pre>
<pre><code><span><span class="co">## Preparing gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## character format detected.</span></span></code></pre>
<pre><code><span><span class="co">## Converting to data.frame</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from input_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 15,259 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Converting mouse ==&gt; human orthologs using: homologene</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without orthologs in human.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from input_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 13,416 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from ortholog_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 13,416 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without 1:1 orthologs.</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 46 genes that have multiple input_gene per ortholog_gene (many:1).</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 56 genes that have multiple ortholog_gene per input_gene (1:many).</span></span></code></pre>
<pre><code><span><span class="co">## Filtering gene_df with gene_map</span></span></code></pre>
<pre><code><span><span class="co">## Returning gene_map as dictionary</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## Total genes dropped after convert_orthologs :</span></span>
<span><span class="co">##    2,016 / 15,259 (13%)</span></span></code></pre>
<pre><code><span><span class="co">## Total genes remaining after convert_orthologs :</span></span>
<span><span class="co">##    13,243 / 15,259 (87%)</span></span></code></pre>
<pre><code><span><span class="co">## Generating gene background for mouse x human ==&gt; human</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 21,207 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 21,207 genes from mouse.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span>
<span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## Preparing gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## data.frame format detected.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from Gene.Symbol.</span></span></code></pre>
<pre><code><span><span class="co">## 21,207 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Converting mouse ==&gt; human orthologs using: homologene</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: mouse</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for mouse</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 10090</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without orthologs in human.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from input_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Extracting genes from ortholog_gene.</span></span></code></pre>
<pre><code><span><span class="co">## 17,355 genes extracted.</span></span></code></pre>
<pre><code><span><span class="co">## Checking for genes without 1:1 orthologs.</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 131 genes that have multiple input_gene per ortholog_gene (many:1).</span></span></code></pre>
<pre><code><span><span class="co">## Dropping 498 genes that have multiple ortholog_gene per input_gene (1:many).</span></span></code></pre>
<pre><code><span><span class="co">## Filtering gene_df with gene_map</span></span></code></pre>
<pre><code><span><span class="co">## Adding input_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## Adding ortholog_gene col to gene_df.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## Total genes dropped after convert_orthologs :</span></span>
<span><span class="co">##    4,725 / 21,207 (22%)</span></span></code></pre>
<pre><code><span><span class="co">## Total genes remaining after convert_orthologs :</span></span>
<span><span class="co">##    16,482 / 21,207 (78%)</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## Gathering ortholog reports.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## --</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## =========== REPORT SUMMARY ===========</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) target_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 19,129 / 19,129 (100%) reference_species genes remain after ortholog conversion.</span></span></code></pre>
<pre><code><span><span class="co">## 16,482 intersect background genes used.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all genes using: homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Retrieving all organisms available in homologene.</span></span></code></pre>
<pre><code><span><span class="co">## Mapping species name: human</span></span></code></pre>
<pre><code><span><span class="co">## Common name mapping found for human</span></span></code></pre>
<pre><code><span><span class="co">## 1 organism identified from search: 9606</span></span></code></pre>
<pre><code><span><span class="co">## Gene table with 19,129 rows retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## Returning all 19,129 genes from human.</span></span></code></pre>
<pre><code><span><span class="co">## Returning 19,129 unique genes from entire human genome.</span></span></code></pre>
<pre><code><span><span class="co">## Using intersect between background gene lists: 16,482 genes.</span></span></code></pre>
<pre><code><span><span class="co">## Standardising sct_data.</span></span></code></pre>
<pre><code><span><span class="co">## Using 1st column of tt as gene column: HGNC.symbol</span></span></code></pre>
<pre><code><span><span class="co">## 1 core(s) assigned as workers (1 reserved).</span></span></code></pre>
<pre><code><span><span class="co">## Standardising CellTypeDataset</span></span></code></pre>
<pre><code><span><span class="co">## Checking gene list inputs.</span></span></code></pre>
<pre><code><span><span class="co">## Running without gene size control.</span></span></code></pre>
<pre><code><span><span class="co">## 224 hit gene(s) remain after filtering.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene scores.</span></span></code></pre>
<pre><code><span><span class="co">## Using previously sampled genes.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene counts.</span></span></code></pre>
<pre><code><span><span class="co">## Testing for enrichment in 7 cell types...</span></span></code></pre>
<pre><code><span><span class="co">## Sorting results by p-value.</span></span></code></pre>
<pre><code><span><span class="co">## Computing BH-corrected q-values.</span></span></code></pre>
<pre><code><span><span class="co">## 3 significant cell type enrichment results @ q&lt;0.05 :</span></span></code></pre>
<pre><code><span><span class="co">##               CellType annotLevel    p fold_change sd_from_mean          q</span></span>
<span><span class="co">## 1    endothelial_mural          1 0.00    1.246976     4.548717 0.00000000</span></span>
<span><span class="co">## 2 astrocytes_ependymal          1 0.00    1.210210     3.535254 0.00000000</span></span>
<span><span class="co">## 3            microglia          1 0.01    1.210084     2.813829 0.02333333</span></span></code></pre>
<pre><code><span><span class="co">## 1 core(s) assigned as workers (1 reserved).</span></span></code></pre>
<pre><code><span><span class="co">## Standardising CellTypeDataset</span></span></code></pre>
<pre><code><span><span class="co">## Checking gene list inputs.</span></span></code></pre>
<pre><code><span><span class="co">## Running without gene size control.</span></span></code></pre>
<pre><code><span><span class="co">## 232 hit gene(s) remain after filtering.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene scores.</span></span></code></pre>
<pre><code><span><span class="co">## Using previously sampled genes.</span></span></code></pre>
<pre><code><span><span class="co">## Computing gene counts.</span></span></code></pre>
<pre><code><span><span class="co">## Testing for enrichment in 7 cell types...</span></span></code></pre>
<pre><code><span><span class="co">## Sorting results by p-value.</span></span></code></pre>
<pre><code><span><span class="co">## Computing BH-corrected q-values.</span></span></code></pre>
<pre><code><span><span class="co">## 3 significant cell type enrichment results @ q&lt;0.05 :</span></span></code></pre>
<pre><code><span><span class="co">##        CellType annotLevel p fold_change sd_from_mean q</span></span>
<span><span class="co">## 1  pyramidal_SS          1 0    1.293795     6.501417 0</span></span>
<span><span class="co">## 2 pyramidal_CA1          1 0    1.221309     4.726326 0</span></span>
<span><span class="co">## 3  interneurons          1 0    1.216514     3.743645 0</span></span></code></pre>
<p><strong>Note</strong> - You can run
<code><a href="../reference/ewce_expression_data.html">ewce_expression_data()</a></code> offline by passing
<code>localhub = TRUE</code>. This will work off a previously cached
version of the reference dataset from <code>ExperimentHub</code>.</p>
<div class="section level4">
<h4 id="plot-results-4">Plot results<a class="anchor" aria-label="anchor" href="#plot-results-4"></a>
</h4>
<p>The results of this analysis can be plotted using the
<code>ewce_plot</code> function.</p>
<div class="sourceCode" id="cb899"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span><span class="va">tt_results</span><span class="op">$</span><span class="va">joint_results</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p><img src="extended_files/figure-html/unnamed-chunk-33-1.png" width="672"></p>
<p>As was reported in our paper, neuronal genes are found to be
downregulated while glial genes are upregulated.</p>
</div>
</div>
<div class="section level3">
<h3 id="generating-bootstrap-plots-for-transcriptomes">Generating bootstrap plots for transcriptomes<a class="anchor" aria-label="anchor" href="#generating-bootstrap-plots-for-transcriptomes"></a>
</h3>
<p>A common request is to explain which differentially expressed genes
are associated with a cell type…</p>
<div class="sourceCode" id="cb900"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">full_result_path</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/generate_bootstrap_plots_for_transcriptome.html">generate_bootstrap_plots_for_transcriptome</a></span><span class="op">(</span></span>
<span>  sct_data <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>  tt <span class="op">=</span> <span class="va">tt_alzh</span>,</span>
<span>  annotLevel <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  full_results <span class="op">=</span> <span class="va">tt_results</span>,</span>
<span>  listFileName <span class="op">=</span> <span class="st">"examples"</span>,</span>
<span>  reps <span class="op">=</span> <span class="va">reps</span>,</span>
<span>  ttSpecies <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  sctSpecies <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>  sig_only <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="merging-multiple-transcriptome-studies">Merging multiple transcriptome studies<a class="anchor" aria-label="anchor" href="#merging-multiple-transcriptome-studies"></a>
</h3>
<p>Where multiple transcriptomic studies have been performed with the
same purpose, i.e. seeking differential expression in dlPFC of
post-mortem schizophrenics, it is common to want to determine whether
they exhibit any shared signal. EWCE can be used to merge the results of
multiple studies.</p>
<p>To demonstrate this we use a two further Alzheimer’s transcriptome
datasets coming from Brodmann areas 36 and 44: these area stored in
<code>tt_alzh_BA36</code> and <code>tt_alzh_BA44</code>. The first step
is to run EWCE on each of these individually and store the output into
one list.</p>
<div class="section level4">
<h4 id="load-data">Load data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h4>
<div class="sourceCode" id="cb901"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#if running offline pass localhub = TRUE</span></span>
<span><span class="va">tt_alzh_BA36</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/tt_alzh_BA36.html" class="external-link">tt_alzh_BA36</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">tt_alzh_BA44</span> <span class="op">&lt;-</span> <span class="fu">ewceData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ewceData/man/tt_alzh_BA44.html" class="external-link">tt_alzh_BA44</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
</div>
<div class="section level4">
<h4 id="run-ewce-analysis">Run EWCE analysis<a class="anchor" aria-label="anchor" href="#run-ewce-analysis"></a>
</h4>
<div class="sourceCode" id="cb902"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tt_results_36</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_expression_data.html">ewce_expression_data</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>                                             tt <span class="op">=</span> <span class="va">tt_alzh_BA36</span>,</span>
<span>                                             annotLevel <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                             ttSpecies <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>                                             sctSpecies <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span></span>
<span><span class="va">tt_results_44</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_expression_data.html">ewce_expression_data</a></span><span class="op">(</span>sct_data <span class="op">=</span> <span class="va">ctd</span>,</span>
<span>                                             tt <span class="op">=</span> <span class="va">tt_alzh_BA44</span>,</span>
<span>                                             annotLevel <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                             ttSpecies <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>                                             sctSpecies <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Fill a list with the results</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/add_res_to_merging_list.html">add_res_to_merging_list</a></span><span class="op">(</span><span class="va">tt_results</span><span class="op">)</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/add_res_to_merging_list.html">add_res_to_merging_list</a></span><span class="op">(</span><span class="va">tt_results_36</span>,<span class="va">results</span><span class="op">)</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/add_res_to_merging_list.html">add_res_to_merging_list</a></span><span class="op">(</span><span class="va">tt_results_44</span>,<span class="va">results</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform the merged analysis</span></span>
<span><span class="co"># For publication reps should be higher</span></span>
<span><span class="va">merged_res</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/merged_ewce.html">merged_ewce</a></span><span class="op">(</span>results <span class="op">=</span> <span class="va">results</span>,</span>
<span>                                reps <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">merged_res</span><span class="op">)</span></span></code></pre></div>
<p>The results can then be plotted as normal using the
<code>ewce_plot</code> function.</p>
<div class="sourceCode" id="cb903"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">EWCE</span><span class="fu">::</span><span class="fu"><a href="../reference/ewce_plot.html">ewce_plot</a></span><span class="op">(</span><span class="va">merged_res</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot_list</span><span class="op">$</span><span class="va">plain</span><span class="op">)</span></span></code></pre></div>
<p>The merged results from all three Alzheimer’s brain regions are found
to be remarkably similar, as was reported in our paper.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb904"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] ewceData_1.10.0      ExperimentHub_2.10.0 AnnotationHub_3.10.0</span></span>
<span><span class="co">##  [4] BiocFileCache_2.10.1 dbplyr_2.4.0         BiocGenerics_0.48.0 </span></span>
<span><span class="co">##  [7] ggplot2_3.4.4        EWCE_1.11.1          RNOmni_1.0.1.2      </span></span>
<span><span class="co">## [10] BiocStyle_2.30.0    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] ggdendro_0.1.23               jsonlite_1.8.7               </span></span>
<span><span class="co">##   [3] magrittr_2.0.3                farver_2.1.1                 </span></span>
<span><span class="co">##   [5] rmarkdown_2.25                fs_1.6.3                     </span></span>
<span><span class="co">##   [7] zlibbioc_1.48.0               ragg_1.2.6                   </span></span>
<span><span class="co">##   [9] vctrs_0.6.4                   memoise_2.0.1                </span></span>
<span><span class="co">##  [11] RCurl_1.98-1.12               ggtree_3.10.0                </span></span>
<span><span class="co">##  [13] rstatix_0.7.2                 htmltools_0.5.6.1            </span></span>
<span><span class="co">##  [15] S4Arrays_1.2.0                curl_5.1.0                   </span></span>
<span><span class="co">##  [17] broom_1.0.5                   SparseArray_1.2.0            </span></span>
<span><span class="co">##  [19] gridGraphics_0.5-1            sass_0.4.7                   </span></span>
<span><span class="co">##  [21] bslib_0.5.1                   htmlwidgets_1.6.2            </span></span>
<span><span class="co">##  [23] desc_1.4.2                    plyr_1.8.9                   </span></span>
<span><span class="co">##  [25] HGNChelper_0.8.1              plotly_4.10.3                </span></span>
<span><span class="co">##  [27] cachem_1.0.8                  mime_0.12                    </span></span>
<span><span class="co">##  [29] lifecycle_1.0.3               pkgconfig_2.0.3              </span></span>
<span><span class="co">##  [31] Matrix_1.6-1.1                R6_2.5.1                     </span></span>
<span><span class="co">##  [33] fastmap_1.1.1                 GenomeInfoDbData_1.2.11      </span></span>
<span><span class="co">##  [35] MatrixGenerics_1.14.0         shiny_1.7.5.1                </span></span>
<span><span class="co">##  [37] digest_0.6.33                 aplot_0.2.2                  </span></span>
<span><span class="co">##  [39] colorspace_2.1-0              patchwork_1.1.3              </span></span>
<span><span class="co">##  [41] AnnotationDbi_1.64.0          S4Vectors_0.40.1             </span></span>
<span><span class="co">##  [43] grr_0.9.5                     rprojroot_2.0.3              </span></span>
<span><span class="co">##  [45] textshaping_0.3.7             GenomicRanges_1.54.1         </span></span>
<span><span class="co">##  [47] RSQLite_2.3.2                 ggpubr_0.6.0                 </span></span>
<span><span class="co">##  [49] labeling_0.4.3                filelock_1.0.2               </span></span>
<span><span class="co">##  [51] fansi_1.0.5                   httr_1.4.7                   </span></span>
<span><span class="co">##  [53] abind_1.4-5                   compiler_4.3.1               </span></span>
<span><span class="co">##  [55] withr_2.5.2                   bit64_4.0.5                  </span></span>
<span><span class="co">##  [57] backports_1.4.1               BiocParallel_1.36.0          </span></span>
<span><span class="co">##  [59] orthogene_1.8.0               carData_3.0-5                </span></span>
<span><span class="co">##  [61] DBI_1.1.3                     homologene_1.4.68.19.3.27    </span></span>
<span><span class="co">##  [63] highr_0.10                    ggsignif_0.6.4               </span></span>
<span><span class="co">##  [65] MASS_7.3-60                   rappdirs_0.3.3               </span></span>
<span><span class="co">##  [67] DelayedArray_0.28.0           tools_4.3.1                  </span></span>
<span><span class="co">##  [69] ape_5.7-1                     interactiveDisplayBase_1.40.0</span></span>
<span><span class="co">##  [71] httpuv_1.6.12                 glue_1.6.2                   </span></span>
<span><span class="co">##  [73] nlme_3.1-163                  promises_1.2.1               </span></span>
<span><span class="co">##  [75] grid_4.3.1                    reshape2_1.4.4               </span></span>
<span><span class="co">##  [77] generics_0.1.3                gtable_0.3.4                 </span></span>
<span><span class="co">##  [79] tidyr_1.3.0                   data.table_1.14.8            </span></span>
<span><span class="co">##  [81] car_3.1-2                     utf8_1.2.4                   </span></span>
<span><span class="co">##  [83] XVector_0.42.0                BiocVersion_3.18.0           </span></span>
<span><span class="co">##  [85] pillar_1.9.0                  stringr_1.5.0                </span></span>
<span><span class="co">##  [87] yulab.utils_0.1.0             babelgene_22.9               </span></span>
<span><span class="co">##  [89] limma_3.58.0                  later_1.3.1                  </span></span>
<span><span class="co">##  [91] dplyr_1.1.3                   treeio_1.26.0                </span></span>
<span><span class="co">##  [93] lattice_0.21-9                bit_4.0.5                    </span></span>
<span><span class="co">##  [95] tidyselect_1.2.0              SingleCellExperiment_1.24.0  </span></span>
<span><span class="co">##  [97] Biostrings_2.70.1             knitr_1.45                   </span></span>
<span><span class="co">##  [99] bookdown_0.36                 IRanges_2.36.0               </span></span>
<span><span class="co">## [101] SummarizedExperiment_1.32.0   stats4_4.3.1                 </span></span>
<span><span class="co">## [103] xfun_0.40                     Biobase_2.62.0               </span></span>
<span><span class="co">## [105] statmod_1.5.0                 matrixStats_1.0.0            </span></span>
<span><span class="co">## [107] stringi_1.7.12                lazyeval_0.2.2               </span></span>
<span><span class="co">## [109] ggfun_0.1.3                   yaml_2.3.7                   </span></span>
<span><span class="co">## [111] codetools_0.2-19              evaluate_0.22                </span></span>
<span><span class="co">## [113] tibble_3.2.1                  BiocManager_1.30.22          </span></span>
<span><span class="co">## [115] ggplotify_0.1.2               cli_3.6.1                    </span></span>
<span><span class="co">## [117] xtable_1.8-4                  systemfonts_1.0.5            </span></span>
<span><span class="co">## [119] munsell_0.5.0                 jquerylib_0.1.4              </span></span>
<span><span class="co">## [121] Rcpp_1.0.11                   GenomeInfoDb_1.38.0          </span></span>
<span><span class="co">## [123] gprofiler2_0.2.2              png_0.1-8                    </span></span>
<span><span class="co">## [125] parallel_4.3.1                ellipsis_0.3.2               </span></span>
<span><span class="co">## [127] pkgdown_2.0.7                 blob_1.2.4                   </span></span>
<span><span class="co">## [129] bitops_1.0-7                  viridisLite_0.4.2            </span></span>
<span><span class="co">## [131] tidytree_0.4.5                scales_1.2.1                 </span></span>
<span><span class="co">## [133] purrr_1.0.2                   crayon_1.5.2                 </span></span>
<span><span class="co">## [135] rlang_1.1.1                   KEGGREST_1.42.0</span></span></code></pre>
</details><hr>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-zeisel2015cell" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Zeisel, A. <em>et al.</em> Cell types in the
mouse cortex and hippocampus revealed by single-cell RNA-seq.
<em>Science</em> <strong>347,</strong> 1138–1142 (2015).</div>
</div>
<div id="ref-skene_2016" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Skene, N. &amp; Grant, S. Identification of
vulnerable cell types in major brain disorders using single cell
transcriptomes and expression weighted cell type enrichment.
<em>Frontiers in Neuroscience</em> (2016). doi:<a href="https://doi.org/10.3389/fnins.2016.00016" class="external-link">10.3389/fnins.2016.00016</a>
</div>
</div>
<div id="ref-haroutunian2009transcriptional" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">Haroutunian, V., Katsel, P. &amp; Schmeidler,
J. Transcriptional vulnerability of brain regions in alzheimer’s disease
and dementia. <em>Neurobiology of aging</em> <strong>30,</strong>
561–573 (2009).</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Alan Murphy, Brian Schilder, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
